<!doctype html>
<html class="no-js">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="graphein.rna" href="graphein.rna.html" /><link rel="prev" title="PPISP - Protein Protein Interaction Structure Prediction" href="../datasets/ppisp.html" />

    <meta name="generator" content="sphinx-4.1.2, furo 2021.08.17.beta43"/>
        <title>graphein.protein - Graphein 1.0.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?digest=910ac4ccc9d7c023151c34010d7b500fb9777a51" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?digest=16fb25fabf47304eee183a5e9af80b1ba98259b1" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  body[data-theme="dark"] {
    --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
  }
  @media (prefers-color-scheme: dark) {
    body:not([data-theme="light"]) {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
  }
</style>
    <link media="(prefers-color-scheme: dark)" rel="stylesheet" href="../_static/pygments_dark.css"></head>
  <body>
    
<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round">
      <path stroke="none" d="M0 0h24v24H0z" />
      <line x1="4" y1="6" x2="20" y2="6" />
      <line x1="10" y1="12" x2="20" y2="12" />
      <line x1="6" y1="18" x2="20" y2="18" />
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">Graphein 1.0.0 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand centered" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../_static/graphein.png" alt="Logo"/>
  </div>
  
  <span class="sidebar-brand-text">Graphein 1.0.0 documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder=Search name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/introduction.html">Graphein</a></li>
<li class="toctree-l1"><a class="reference internal" href="../license.html">License</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1 has-children"><a class="reference internal" href="../notebooks_index.html">Notebooks</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/residue_graphs.html">Graphein - Residue Graph Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/atom_graph_tutorial.html">Graphein - Atom Graph Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/alphafold_protein_graph_tutorial.html">Constructing Graphs from AlphaFold Protein Structure Database</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/protein_mesh_tutorial.html">Creating Protein Meshes in Graphein &amp; 3D Visualisation</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/ppi_tutorial.html">PPI Networks and Structural Interactomics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/grn_tutorial.html">Gene Regulatory Network Tutorial</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Datasets</span></p>
<ul>
<li class="toctree-l1 has-children"><a class="reference internal" href="../datasets.html">Datasets</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../datasets/pscdb.html">PSCDB - Protein Structural Change Database</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/pscdb_processing.html">PSCDB - Processing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/pscdb_baselines.html">PSCDB - Baselines</a></li>
<li class="toctree-l2"><a class="reference internal" href="../datasets/ppisp.html">PPISP - Protein Protein Interaction Structure Prediction</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul class="current">
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">graphein.protein</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.rna.html">graphein.rna</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.ppi.html">graphein.ppi</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.grn.html">graphein.grn</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.ml.html">graphein.ml</a></li>
</ul>

</div>
</div>
      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <article role="main">
        <div class="content-icon-container">
          <div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <div class="section" id="graphein-protein">
<h1>graphein.protein<a class="headerlink" href="#graphein-protein" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-graphein.protein.config">
<span id="config"></span><h2>Config<a class="headerlink" href="#module-graphein.protein.config" title="Permalink to this headline">¶</a></h2>
<p>Base Config object for use with Protein Graph Construction</p>
<dl class="py class">
<dt class="sig sig-object py" id="graphein.protein.config.GetContactsConfig">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">GetContactsConfig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_contacts_path</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pathlib.Path</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">PosixPath('/Users/arianjamasb/github/getcontacts')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contacts_dir</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pathlib.Path</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">PosixPath('/home/atj39/github/graphein/examples/contacts')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_dir</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pathlib.Path</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">PosixPath('/home/atj39/github/graphein/examples/pdbs')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'CA'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/config.html#GetContactsConfig"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.GetContactsConfig" title="Permalink to this definition">¶</a></dt>
<dd><p>Config object for parameters relating to running GetContacts. GetContacts is an optional dependency from which intramolecular interactions can be computed and used as edges in the graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>get_contacts_path</strong> (<em>pathlib.Path</em>) – Path to GetContacts installation</p></li>
<li><p><strong>contacts_dir</strong> (<em>pathlib.Path</em>) – Path to store output of GetContacts</p></li>
<li><p><strong>pdb_dir</strong> (<em>pathlib.Path</em>) – Path to PDB files to be used to compute intramolecular interactions.</p></li>
<li><p><strong>granularity</strong> (<em>str</em>) – Specifies the node types of the graph, defaults to “CA” for alpha-carbons as nodes. Other options are “CB” (beta-carbon), “atom” for all-atom graphs, and “centroid” for nodes positioned as residue centroids.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="graphein.protein.config.ProteinGraphConfig">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">ProteinGraphConfig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">*,</span> <span class="pre">granularity:</span> <span class="pre">Union[Literal['N',</span> <span class="pre">'CA',</span> <span class="pre">'C',</span> <span class="pre">'O',</span> <span class="pre">'CB',</span> <span class="pre">'OG',</span> <span class="pre">'CG',</span> <span class="pre">'CD1',</span> <span class="pre">'CD2',</span> <span class="pre">'CE1',</span> <span class="pre">'CE2',</span> <span class="pre">'CZ',</span> <span class="pre">'OD1',</span> <span class="pre">'ND2',</span> <span class="pre">'CG1',</span> <span class="pre">'CG2',</span> <span class="pre">'CD',</span> <span class="pre">'CE',</span> <span class="pre">'NZ',</span> <span class="pre">'OD2',</span> <span class="pre">'OE1',</span> <span class="pre">'NE2',</span> <span class="pre">'OE2',</span> <span class="pre">'OH',</span> <span class="pre">'NE',</span> <span class="pre">'NH1',</span> <span class="pre">'NH2',</span> <span class="pre">'OG1',</span> <span class="pre">'SD',</span> <span class="pre">'ND1',</span> <span class="pre">'SG',</span> <span class="pre">'NE1',</span> <span class="pre">'CE3',</span> <span class="pre">'CZ2',</span> <span class="pre">'CZ3',</span> <span class="pre">'CH2',</span> <span class="pre">'OXT'],</span> <span class="pre">Literal['atom',</span> <span class="pre">'centroids']]</span> <span class="pre">=</span> <span class="pre">'CA',</span> <span class="pre">keep_hets:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False,</span> <span class="pre">insertions:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False,</span> <span class="pre">pdb_dir:</span> <span class="pre">pathlib.Path</span> <span class="pre">=</span> <span class="pre">PosixPath('../examples/pdbs'),</span> <span class="pre">verbose:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False,</span> <span class="pre">exclude_waters:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">True,</span> <span class="pre">deprotonate:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False,</span> <span class="pre">protein_df_processing_functions:</span> <span class="pre">List[Callable]</span> <span class="pre">=</span> <span class="pre">None,</span> <span class="pre">edge_construction_functions:</span> <span class="pre">List[Union[Callable,</span> <span class="pre">str]]</span> <span class="pre">=</span> <span class="pre">[&lt;function</span> <span class="pre">add_peptide_bonds&gt;],</span> <span class="pre">node_metadata_functions:</span> <span class="pre">List[Union[Callable,</span> <span class="pre">str]]</span> <span class="pre">=</span> <span class="pre">[&lt;function</span> <span class="pre">meiler_embedding&gt;],</span> <span class="pre">edge_metadata_functions:</span> <span class="pre">List[Union[Callable,</span> <span class="pre">str]]</span> <span class="pre">=</span> <span class="pre">None,</span> <span class="pre">graph_metadata_functions:</span> <span class="pre">List[Callable]</span> <span class="pre">=</span> <span class="pre">None,</span> <span class="pre">get_contacts_config:</span> <span class="pre">graphein.protein.config.GetContactsConfig</span> <span class="pre">=</span> <span class="pre">None,</span> <span class="pre">dssp_config:</span> <span class="pre">graphein.protein.config.DSSPConfig</span> <span class="pre">=</span> <span class="pre">None</span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/config.html#ProteinGraphConfig"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.ProteinGraphConfig" title="Permalink to this definition">¶</a></dt>
<dd><p>Config Object for Protein Structure Graph Construction</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>granularity</strong> (<em>str</em><em> (</em><em>Union</em><em>[</em><em>graphein.protein.config.GraphAtoms</em><em>, </em><em>graphein.protein.config.GranularityOpts</em><em>]</em><em>)</em>) – Controls the granularity of the graph construction. “atom” builds an atomic-scale graph where
nodes are constituent atoms. Residue-level graphs can be build by specifying which constituent atom should
represent node positions (see graphein.protein.config.GraphAtoms). Additionally, “centroids” can be specified to
compute the centre of gravity for a given atom. Defaults to “CA” (alpha-Carbon)</p></li>
<li><p><strong>keep_hets</strong> (<em>bool</em>) – Controls whether or not heteroatoms are removed from the PDB file. These are typically modified
residules, bound ligands, crystallographic adjuvants, ions or water molecules.</p></li>
<li><p><strong>insertions</strong> (<em>bool</em>) – Controls whether or not insertions are allowed.</p></li>
<li><p><strong>pdb_dir</strong> (<em>pathlib.Path</em>) – Specifies path to download protein structures into.</p></li>
<li><p><strong>verbose</strong> (<em>bool</em>) – Specifies verbosity of graph creation process.</p></li>
<li><p><strong>exclude_waters</strong> – Specifies whether or not water molecules are excluded from the structure</p></li>
<li><p><strong>deprotonate</strong> (<em>bool</em>) – Specifies whether or not to remove H atoms from the graph.</p></li>
<li><p><strong>protein_df_processing_functions</strong> (<em>Optional</em><em>[</em><em>List</em><em>[</em><em>Callable</em><em>]</em><em>]</em>) – List of functions that take a pd.DataFrame and return a pd.DataFrame.
This allows users to define their own series of processing functions for the protein structure DataFrame and
override the default sequencing of processing steps provided by Graphein. We refer users to our low-level API
tutorial for more details.</p></li>
<li><p><strong>edge_construction_functions</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em>) – List of functions that take an nx.Graph and return an nx.Graph with desired
edges added. Prepared edge constructions can be found in graphein.protein.edges</p></li>
<li><p><strong>node_metadata_functions</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions that taken an nx.Graph</p></li>
<li><p><strong>edge_metadata_functions</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions that take an</p></li>
<li><p><strong>graph_metadata_functions</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions that take an nx.Graph and return an nx.Graph with added
graph-level features and metadata.</p></li>
<li><p><strong>get_contacts_config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>GetContactsConfig</em></a><em>, </em><em>optional</em>) – Config object containing parameters for running GetContacts for computing intramolecular
contact-based edges. Defaults to None.</p></li>
<li><p><strong>dssp_config</strong> (<em>DSSPConfig</em><em>, </em><em>optional</em>) – Config Object containing reference to executable. Defaults to None.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="graphein.protein.config.ProteinMeshConfig">
<em class="property"><span class="pre">class</span> </em><span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">ProteinMeshConfig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pymol_command_line_options</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'-cKq'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pymol_commands</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">['show</span> <span class="pre">surface']</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/config.html#ProteinMeshConfig"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.ProteinMeshConfig" title="Permalink to this definition">¶</a></dt>
<dd><p>Config object for parameters relating to ProteinMeshConfig Mesh construction with PyMol</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pymol_command_line_options</strong> (<em>str</em><em>, </em><em>optional</em>) – List of CLI args for running PyMol.
See: <a class="reference external" href="https://pymolwiki.org/index.php/Command_Line_Options">https://pymolwiki.org/index.php/Command_Line_Options</a>. Defaults to “-cKq”</p></li>
<li><p><strong>pymol_commands</strong> (<em>List</em><em>[</em><em>str</em><em>]</em><em>, </em><em>optional</em>) – List of Commands passed to PyMol in surface construction.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.graphs">
<span id="graphs"></span><h2>Graphs<a class="headerlink" href="#module-graphein.protein.graphs" title="Permalink to this headline">¶</a></h2>
<p>Functions for working with Protein Structure Graphs</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.add_nodes_to_graph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">add_nodes_to_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#add_nodes_to_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.add_nodes_to_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Add nodes into protein graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph with metadata to populate with nodes</p></li>
<li><p><strong>verbose</strong> (<em>bool</em>) – Controls verbosity of this step</p></li>
</ul>
</dd>
<dt class="field-even">Protein_df</dt>
<dd class="field-even"><p>DataFrame of protein structure containing nodes &amp; initial node metadata to add to the graph</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>nx.Graph with nodes added</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.assign_node_id_to_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">assign_node_id_to_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#assign_node_id_to_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.assign_node_id_to_dataframe" title="Permalink to this definition">¶</a></dt>
<dd><p>Assigns the node ID back to the pdb_df dataframe</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>protein_df</strong> (<em>pd.DataFrame</em>) – Structure Dataframe</p></li>
<li><p><strong>granularity</strong> (<em>str</em>) – Granularity of graph. Atom-level, resiidue (e.g. CA) or centroids</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Returns dataframe with added node_ids</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.calculate_centroid_positions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">calculate_centroid_positions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atoms</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#calculate_centroid_positions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.calculate_centroid_positions" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates position of sidechain centroids</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atoms</strong> (<em>pd.DataFrame</em>) – ATOM df of protein structure</p></li>
<li><p><strong>verbose</strong> (<em>bool</em>) – bool controlling verbosity</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>centroids (df)</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.compute_edges">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">compute_edges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">funcs</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Callable</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_contacts_config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><span class="pre">graphein.protein.config.GetContactsConfig</span></a><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#compute_edges"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.compute_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes edges for the protein structure graph. Will compute an pairwise distance matrix between nodes which is
added to the graph metadata to facilitate some edge computations.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph with nodes to add edges to</p></li>
<li><p><strong>funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em>) – List of edge construction functions</p></li>
<li><p><strong>get_contacts_config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>graphein.protein.config.GetContactsConfig</em></a>) – Config object for getcontacts if intramolecular edges are being used</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Graph with added edges</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.compute_rgroup_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">compute_rgroup_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#compute_rgroup_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.compute_rgroup_dataframe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the atoms that are in R-groups and not the backbone chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pdb_df</strong> (<em>pd.DataFrame</em>) – DataFrame to compute R group dataframe from</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Dataframe containing R-groups only (backbone atoms removed)</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.construct_graph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">construct_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.ProteinGraphConfig" title="graphein.protein.config.ProteinGraphConfig"><span class="pre">graphein.protein.config.ProteinGraphConfig</span></a><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_path</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_selection</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'all'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_processing_funcs</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Callable</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_construction_funcs</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Callable</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_annotation_funcs</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Callable</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_annotation_funcs</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Callable</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">graph_annotation_funcs</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Callable</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#construct_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.construct_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Constructs protein structure graph from a pdb_code or pdb_path. Users can provide a ProteinGraphConfig object.</p>
<p>However, config parameters can be overridden by passing arguments directly to the function.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinGraphConfig" title="graphein.protein.config.ProteinGraphConfig"><em>graphein.protein.config.ProteinGraphConfig</em></a><em>, </em><em>optional</em>) – ProteinGraphConfig object. If None, defaults to config in graphein.protein.config</p></li>
<li><p><strong>pdb_path</strong> (<em>str</em><em>, </em><em>optional</em>) – Path to pdb_file to build graph from</p></li>
<li><p><strong>pdb_code</strong> (<em>str</em><em>, </em><em>optional</em>) – 4-character PDB accession pdb_code to build graph from</p></li>
<li><p><strong>chain_selection</strong> (<em>str</em><em>, </em><em>optional</em>) – String of polypeptide chains to include in graph. E.g “ABDF” or “all”</p></li>
<li><p><strong>df_processing_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of dataframe processing functions</p></li>
<li><p><strong>edge_construction_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of edge construction functions</p></li>
<li><p><strong>edge_annotation_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of edge annotation functions</p></li>
<li><p><strong>node_annotation_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of node annotation functions</p></li>
<li><p><strong>graph_annotation_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em>) – List of graph annotation function</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Structure Graph</p>
</dd>
<dt class="field-odd">Type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.convert_structure_to_centroids">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">convert_structure_to_centroids</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#convert_structure_to_centroids"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.convert_structure_to_centroids" title="Permalink to this definition">¶</a></dt>
<dd><p>Overwrite existing (x, y, z) coordinates with centroids of the amino acids.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Pandas Dataframe config protein structure to convert into a dataframe of centroid positions</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.DataFrame with atoms/residues positiions converted into centroid positions</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.deprotonate_structure">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">deprotonate_structure</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#deprotonate_structure"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.deprotonate_structure" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove protons from PDB dataframe.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Atomic dataframe.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Atomic dataframe with all atom_name == “H” removed.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.filter_hetatms">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">filter_hetatms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">keep_hets</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#filter_hetatms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.filter_hetatms" title="Permalink to this definition">¶</a></dt>
<dd><p>Return hetatms of interest.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>df</strong> (<em>pd.DataFrame</em>) – Protein Structure dataframe to filter hetatoms from.</p></li>
<li><p><strong>keep_hets</strong> – List of hetero atom names to keep</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein structure dataframe with heteroatoms removed</p>
</dd>
</dl>
<p>:rtype pd.DataFrame</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.initialise_graph_with_metadata">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">initialise_graph_with_metadata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">raw_pdb_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_id</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#initialise_graph_with_metadata"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.initialise_graph_with_metadata" title="Permalink to this definition">¶</a></dt>
<dd><p>Initialises the nx Graph object with initial metadata</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>protein_df</strong> (<em>pd.DataFrame</em>) – Processed Dataframe of protein structure</p></li>
<li><p><strong>raw_pdb_df</strong> (<em>pd.DataFrame</em>) – Unprocessed dataframe of protein structure for comparison and traceability downstream</p></li>
<li><p><strong>pdb_id</strong> (<em>str</em>) – PDB Accession code</p></li>
<li><p><strong>granularity</strong> (<em>str</em>) – Granularity of the graph (eg “atom”, “CA”, “CB” etc or “centroid”)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Returns initial protein structure graph with metadata</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.process_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">process_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_df_processing_funcs</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Callable</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hetatom_df_processing_funcs</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Callable</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'centroids'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_selection</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'all'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">insertions</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">deprotonate</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">keep_hets</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#process_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.process_dataframe" title="Permalink to this definition">¶</a></dt>
<dd><p>Process ATOM and HETATM dataframes to produce singular dataframe used for graph construction</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>protein_df</strong> (<em>pd.DataFrame</em>) – Dataframe to process.
Should be the object returned from <cite>read_pdb_to_dataframe</cite>.</p></li>
<li><p><strong>atom_df_processing_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions to process dataframe. These must take in a dataframe and return a
dataframe. Defaults to None.</p></li>
<li><p><strong>hetatom_df_processing_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions to process dataframe. These must take in a dataframe and return a dataframe</p></li>
<li><p><strong>granularity</strong> (<em>str</em>) – The level of granualrity for the graph.
This determines the node definition.
Acceptable values include:
- “centroids”
- “atoms”
- any of the atom_names in the PDB file (e.g. “CA”, “CB”, “OG”, etc.)</p></li>
<li><p><strong>insertions</strong> – Whether or not to keep insertions.</p></li>
<li><p><strong>insertions</strong> – bool</p></li>
<li><p><strong>deprotonate</strong> (<em>bool</em>) – Whether or not to remove hydrogen atoms (i.e. deprotonation).</p></li>
<li><p><strong>keep_hets</strong> (<em>List</em><em>[</em><em>str</em><em>]</em>) – Hetatoms to keep. Defaults to an empty list.
To keep a hetatom, pass it inside a list of hetatom names to keep.</p></li>
<li><p><strong>verbose</strong> (<em>bool</em>) – Verbosity level.</p></li>
<li><p><strong>chain_selection</strong> (<em>str</em>) – Which protein chain to select. Defaults to “all”. Eg can use “ACF”
to select 3 chains (A, C &amp; F :)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A protein dataframe that can be consumed by
other graph construction functions.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.read_pdb_to_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">read_pdb_to_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_path</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'CA'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#read_pdb_to_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.read_pdb_to_dataframe" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads PDB file to PandasPDB object.</p>
<dl class="simple">
<dt>Returns <cite>atomic_df</cite>, which is a dataframe enumerating all atoms and their cartesian coordinates in 3D space. Also</dt><dd><p>contains associated metadata.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb_path</strong> (<em>str</em><em>, </em><em>optional</em>) – path to PDB file. Defaults to None.</p></li>
<li><p><strong>pdb_code</strong> (<em>str</em><em>, </em><em>optional</em>) – 4-character PDB accession. Defaults to None.</p></li>
<li><p><strong>verbose</strong> (<em>bool</em>) – print dataframe?</p></li>
<li><p><strong>granularity</strong> (<em>str</em>) – Specifies granularity of dataframe. See graphein.protein.config.ProteinGraphConfig for further
details.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Pd.DataFrame containing protein structure</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.remove_insertions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">remove_insertions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#remove_insertions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.remove_insertions" title="Permalink to this definition">¶</a></dt>
<dd><p>This function removes insertions from PDB dataframes</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Protein Structure dataframe to remove insertions from</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein structure dataframe with insertions removed</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.select_chains">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">select_chains</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_selection</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#select_chains"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.select_chains" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts relevant chains from protein_df</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>protein_df</strong> (<em>pd.DataFrame</em>) – pandas dataframe of PDB subsetted to relevant atoms (CA, CB)</p></li>
<li><p><strong>chain_selection</strong> (<em>str</em>) – Specifies chains that should be extracted from the larger complexed structure</p></li>
<li><p><strong>verbose</strong> (<em>bool</em>) – Print dataframe</p></li>
</ul>
</dd>
</dl>
<p>:return Protein structure dataframe containing only entries in the chain selection
:rtype: pd.DataFrame</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.subset_structure_to_atom_type">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">subset_structure_to_atom_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#subset_structure_to_atom_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.subset_structure_to_atom_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a subset of atomic dataframe that contains only certain atom names.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Protein Structure dataframe to subset</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Subsetted protein structure dataframe</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="edges">
<h2>Edges<a class="headerlink" href="#edges" title="Permalink to this headline">¶</a></h2>
<div class="section" id="module-graphein.protein.edges.distance">
<span id="distance"></span><h3>Distance<a class="headerlink" href="#module-graphein.protein.edges.distance" title="Permalink to this headline">¶</a></h3>
<p>Functions for computing biochemical edges of graphs.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_aromatic_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_aromatic_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_aromatic_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_aromatic_interactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Find all aromatic-aromatic interaction.
Criteria: phenyl ring centroids separated between 4.5A to 7A.
Phenyl rings are present on PHE, TRP, HIS and TYR.
Phenyl ring atoms on these amino acids are defined by the following
atoms:
- PHE: CG, CD, CE, CZ
- TRP: CD, CE, CH, CZ
- HIS: CG, CD, ND, NE, CE
- TYR: CG, CD, CE, CZ
Centroids of these atoms are taken by taking:</p>
<blockquote>
<div><p>(mean x), (mean y), (mean z)</p>
</div></blockquote>
<p>for each of the ring atoms.
Notes for future self/developers:
- Because of the requirement to pre-compute ring centroids, we do not</p>
<blockquote>
<div><p>use the functions written above (filter_dataframe, compute_distmat,
get_interacting_atoms), as they do not return centroid atom
euclidean coordinates.</p>
</div></blockquote>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_aromatic_sulphur_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_aromatic_sulphur_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_aromatic_sulphur_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_aromatic_sulphur_interactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Find all aromatic-sulphur interactions.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_cation_pi_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_cation_pi_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_cation_pi_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_cation_pi_interactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Add cation-pi interactions.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_delaunay_triangulation">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_delaunay_triangulation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_delaunay_triangulation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_delaunay_triangulation" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute the Delaunay triangulation of the protein structure.
This has been used in prior work. References:
- Harrison, R. W., Yu, X. &amp; Weber, I. T. Using triangulation to include</p>
<blockquote>
<div><p>target structure improves drug resistance prediction accuracy. in 1–1
(IEEE, 2013). doi:10.1109/ICCABS.2013.6629236</p>
</div></blockquote>
<ul class="simple">
<li><dl class="simple">
<dt>Yu, X., Weber, I. T. &amp; Harrison, R. W. Prediction of HIV drug</dt><dd><p>resistance from genotype with encoded three-dimensional protein
structure. BMC Genomics 15 Suppl 5, S1 (2014).</p>
</dd>
</dl>
</li>
</ul>
<p>Notes:
1. We do not use the add_interacting_resis function, because this</p>
<blockquote>
<div><p>interaction is computed on the CA atoms. Therefore, there is code
duplication. For now, I have chosen to leave this code duplication
in.</p>
</div></blockquote>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_distance_threshold">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_distance_threshold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">long_interaction_threshold</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">float</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">5.0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_distance_threshold"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_distance_threshold" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds edges to any nodes within a given distance of each other. Long interaction threshold is used
to specify minimum separation in sequence to add an edge between networkx nodes within the distance threshold
:param G: Protein Structure graph to add distance edges to
:param long_interaction_threshold: minimum distance in sequence for two nodes to be connected
:param threshold: Distance in angstroms, below which two nodes are connected
:return: Graph with distance-based edges added</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_disulfide_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_disulfide_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_disulfide_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_disulfide_interactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Find all disulfide interactions between CYS residues.</p>
<p>Criteria: sulfur atom pairs are within 2.2A of each other.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_hydrogen_bond_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_hydrogen_bond_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_hydrogen_bond_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_hydrogen_bond_interactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Add all hydrogen-bond interactions.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_hydrophobic_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_hydrophobic_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_hydrophobic_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_hydrophobic_interactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Find all hydrophobic interactions.</p>
<p>Performs searches between the following residues:
ALA, VAL, LEU, ILE, MET, PHE, TRP, PRO, TYR</p>
<p>Criteria: R-group residues are within 5A distance.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – </p></li>
<li><p><strong>rgroup_df</strong> (<em>pd.DataFrame</em><em>, </em><em>optional</em>) – </p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_interacting_resis">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_interacting_resis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interacting_atoms</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">numpy.array</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dataframe</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kind</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_interacting_resis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_interacting_resis" title="Permalink to this definition">¶</a></dt>
<dd><p>Add interacting residues to graph.</p>
<p>Returns a list of 2-tuples indicating the interacting residues based
on the interacting atoms. This is most typically called after the
get_interacting_atoms function above.</p>
<p>Also filters out the list such that the residues have to be at least
two apart.</p>
<p>### Parameters</p>
<ul class="simple">
<li><p>interacting_atoms:    (numpy array) result from get_interacting_atoms function.</p></li>
<li><dl class="simple">
<dt>dataframe:            (pandas dataframe) a pandas dataframe that</dt><dd><p>houses the euclidean locations of each atom.</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>kind:                 (list) the kind of interaction. Contains one</dt><dd><p>of :
- hydrophobic
- disulfide
- hbond
- ionic
- aromatic
- aromatic_sulphur
- cation_pi
- delaunay</p>
</dd>
</dl>
</li>
</ul>
<ul class="simple">
<li><dl class="simple">
<dt>filtered_interacting_resis: (set of tuples) the residues that are in</dt><dd><p>an interaction, with the interaction kind specified</p>
</dd>
</dl>
</li>
</ul>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_ionic_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_ionic_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.frame.DataFrame</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_ionic_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_ionic_interactions" title="Permalink to this definition">¶</a></dt>
<dd><p>Find all ionic interactions.</p>
<p>Criteria: ARG, LYS, HIS, ASP, and GLU residues are within 6A.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_k_nn_edges">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_k_nn_edges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">long_interaction_threshold</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">k</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mode</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'connectivity'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metric</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'minkowski'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">include_self</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_k_nn_edges"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_k_nn_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds edges to nodes based on K nearest neighbours. Long interaction threshold is used
to specify minimum separation in sequence to add an edge between networkx nodes within the distance threshold
:param G: Protein Structure graph to add distance edges to
:param long_interaction_threshold: minimum distance in sequence for two nodes to be connected
:param k: Number of neighbors for each sample.
:param mode: Type of returned matrix: ‘connectivity’ will return the connectivity matrix with ones and zeros,
and ‘distance’ will return the distances between neighbors according to the given metric.
:param metric: The distance metric used to calculate the k-Neighbors for each sample point.
The DistanceMetric class gives a list of available metrics.
The default distance is ‘euclidean’ (‘minkowski’ metric with the p param equal to 2.)
:param p: Power parameter for the Minkowski metric. When p = 1, this is equivalent to using manhattan_distance (l1),</p>
<blockquote>
<div><p>and euclidean_distance (l2) for p = 2. For arbitrary p, minkowski_distance (l_p) is used.</p>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>include_self</strong> – Whether or not to mark each sample as the first nearest neighbor to itself.</p>
</dd>
</dl>
<p>If ‘auto’, then True is used for mode=’connectivity’ and False for mode=’distance’.
:return: Graph with knn-based edges added</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_peptide_bonds">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_peptide_bonds</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_peptide_bonds"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_peptide_bonds" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds peptide backbone as edges to residues in each chain</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – networkx protein graph</p>
</dd>
</dl>
<p>:return G; networkx protein graph with added peptide bonds
:rtype: nx.Graph</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.compute_distmat">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">compute_distmat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#compute_distmat"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.compute_distmat" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute pairwise euclidean distances between every atom.</p>
<p>Design choice: passed in a DataFrame to enable easier testing on
dummy data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pdb_df</strong> (<em>pd.DataFrame</em>) – pd.Dataframe containing protein structure. Must contain columns [“x_coord”, “y_coord”, “z_coord”]</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.Dataframe of euclidean distance matrix</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.get_edges_by_bond_type">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">get_edges_by_bond_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bond_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span> </span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#get_edges_by_bond_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.get_edges_by_bond_type" title="Permalink to this definition">¶</a></dt>
<dd><p>Return edges of a particular bond type.</p>
<ul class="simple">
<li><p>bond_type: (str) one of the elements in the variable BOND_TYPES</p></li>
</ul>
<ul class="simple">
<li><p>resis: (list) a list of tuples, where each tuple is an edge.</p></li>
</ul>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.get_interacting_atoms">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">get_interacting_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">angstroms</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">float</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">distmat</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#get_interacting_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.get_interacting_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Find the atoms that are within a particular radius of one another.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.get_ring_atoms">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">get_ring_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataframe</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aa</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#get_ring_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.get_ring_atoms" title="Permalink to this definition">¶</a></dt>
<dd><p>Return ring atoms from a dataframe.</p>
<p>A helper function for add_aromatic_interactions.</p>
<p>Gets the ring atoms from the particular aromatic amino acid.</p>
<ul class="simple">
<li><p>dataframe: the dataframe containing the atom records.</p></li>
<li><p>aa: the amino acid of interest, passed in as 3-letter string.</p></li>
</ul>
<ul class="simple">
<li><dl class="simple">
<dt>dataframe: a filtered dataframe containing just those atoms from the</dt><dd><p>particular amino acid selected. e.g. equivalent to
selecting just the ring atoms from a particular amino
acid.</p>
</dd>
</dl>
</li>
</ul>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.get_ring_centroids">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">get_ring_centroids</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ring_atom_df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#get_ring_centroids"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.get_ring_centroids" title="Permalink to this definition">¶</a></dt>
<dd><p>Return aromatic ring centrods.</p>
<p>A helper function for add_aromatic_interactions.</p>
<p>Computes the ring centroids for each a particular amino acid’s ring
atoms.</p>
<p>Ring centroids are computed by taking the mean of the x, y, and z
coordinates.</p>
<ul class="simple">
<li><p>ring_atom_df: a dataframe computed using get_ring_atoms.</p></li>
<li><p>aa: the amino acid under study</p></li>
</ul>
<ul class="simple">
<li><dl class="simple">
<dt>centroid_df: a dataframe containing just the centroid coordinates of</dt><dd><p>the ring atoms of each residue.</p>
</dd>
</dl>
</li>
</ul>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.node_coords">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">node_coords</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#node_coords"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.node_coords" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the x, y, z coordinates of a node.
This is a helper function. Simplifies the code.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> – nx.Graph protein structure graph to extract coordinates from</p></li>
<li><p><strong>n</strong> – str node ID in graph to extract coordinates from</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Tuple of coordinates (x, y, z)</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.edges.intramolecular">
<span id="intramolecular"></span><h3>Intramolecular<a class="headerlink" href="#module-graphein.protein.edges.intramolecular" title="Permalink to this headline">¶</a></h3>
<p>Featurization functions for graph edges.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.add_contacts_edge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">add_contacts_edge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interaction_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#add_contacts_edge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.add_contacts_edge" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds specific interaction types to the protein graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – networkx protein graph</p></li>
<li><p><strong>interaction_type</strong> (<em>str</em>) – interaction type to be added</p></li>
</ul>
</dd>
<dt class="field-even">Return G</dt>
<dd class="field-even"><p>nx.Graph with specified interaction-based edges added.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.get_contacts_df">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">get_contacts_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">GetContactsConfig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pd.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#get_contacts_df"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.get_contacts_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads GetContact File and returns it as a pd.DataFrame</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>GetContactsConfig</em></a>) – GetContactsConfig object</p></li>
<li><p><strong>pdb_name</strong> (<em>str</em>) – Name of PDB file. Contacts files are name {pdb_name}_contacts.tsv</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>DataFrame of prased GetContacts output</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.hydrogen_bond">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">hydrogen_bond</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#hydrogen_bond"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.hydrogen_bond" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds hydrogen bonds to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add hydrogen bonds to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with hydrogen bonds added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.hydrophobic">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">hydrophobic</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#hydrophobic"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.hydrophobic" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds hydrophobic interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add hydrophobic interaction edges to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with hydrophobic interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.peptide_bonds">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">peptide_bonds</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#peptide_bonds"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.peptide_bonds" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds peptide backbone to residues in each chain</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph protein graph with added peptide bonds</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.pi_cation">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">pi_cation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#pi_cation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.pi_cation" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds pi-cation interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add pi-cation interactions to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with pi-pi_cation interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.pi_stacking">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">pi_stacking</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#pi_stacking"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.pi_stacking" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds pi-stacking interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add pi-stacking interactions to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with pi-stacking interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.read_contacts_file">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">read_contacts_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">GetContactsConfig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contacts_file</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pd.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#read_contacts_file"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.read_contacts_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Parses GetContacts file to an edgelist (pd.DataFrame)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>GetContactsConfig</em></a>) – GetContactsConfig object (graphein.protein.config.GetContactsConfig)</p></li>
<li><p><strong>contacts_file</strong> (<em>str</em>) – file name of contacts file</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Pandas Dataframe of edge list</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.run_get_contacts">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">run_get_contacts</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">GetContactsConfig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_id</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#run_get_contacts"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.run_get_contacts" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs GetContacts on a protein structure. If no file_name is provided, a PDB file is downloaded for the pdb_id</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>graphein.protein.config.GetContactsConfig</em></a>) – GetContactsConfig object containing GetContacts parameters</p></li>
<li><p><strong>pdb_id</strong> (<em>str</em><em>, </em><em>optional</em>) – 4-character PDB accession code</p></li>
<li><p><strong>file_name</strong> (<em>str</em><em>, </em><em>optional</em>) – PDB_name file to use, if annotations to be retrieved from the PDB</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.salt_bridge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">salt_bridge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#salt_bridge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.salt_bridge" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds salt bridges to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add salt bridges to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with salt bridges added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.t_stacking">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">t_stacking</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#t_stacking"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.t_stacking" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds t-stacking interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add t-stacking interactions to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with t-stacking interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.van_der_waals">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">van_der_waals</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#van_der_waals"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.van_der_waals" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds van der Waals interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add van der Waals interactions to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with van der Waals interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.edges.atomic">
<span id="atomic"></span><h3>Atomic<a class="headerlink" href="#module-graphein.protein.edges.atomic" title="Permalink to this headline">¶</a></h3>
<p>Functions for computing atomic structure of proteins</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.add_atomic_edges">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">add_atomic_edges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#add_atomic_edges"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.add_atomic_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes covalent edges based on atomic distances. Covalent radii add_atomic_edges assigned to each atom based on its bond assign_bond_states_to_dataframe
The distance matrix is then thresholded to entries less than this distance plus some tolerance to create and adjacency matrix.
This adjacency matrix is then parsed into an edge list and covalent edges added</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Atomic graph (nodes correspond to atoms) to populate with atomic bonds as edges</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Atomic graph with edges between bonded atoms added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.add_bond_order">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">add_bond_order</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#add_bond_order"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.add_bond_order" title="Permalink to this definition">¶</a></dt>
<dd><p>Assign bond orders to the covalent bond edges between atoms on the basis of bond length. Values are taken from:
Automatic Assignment of Chemical Connectivity to Organic Molecules in the CambridgeStructural Database.
Jon C. Baber and Edward E. Hodgkin*</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Atomic-level protein graph with covalent edges.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Atomic-level protein graph with covalent edges annotated with putative bond order.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>mx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.add_ring_status">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">add_ring_status</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#add_ring_status"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.add_ring_status" title="Permalink to this definition">¶</a></dt>
<dd><p>Identifies rings in the atomic graph. Assigns the edge attribute “RING” to edges in the ring. We do not distinguish
between aromatic and non aromatic rings.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Atom-level protein structure graph to add ring edge types to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Atom-level protein structure graph with added RING edge attribute</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.assign_bond_states_to_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">assign_bond_states_to_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#assign_bond_states_to_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.assign_bond_states_to_dataframe" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes a PandasPDB atom dataframe and assigns bond states to each atom based on:
Atomic Structures of all the Twenty Essential Amino Acids and a Tripeptide, with Bond Lengths as Sums of Atomic Covalent Radii
Heyrovska, 2008</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Pandas PDB dataframe</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Dataframe with added atom_bond_state column</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.assign_covalent_radii_to_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">assign_covalent_radii_to_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#assign_covalent_radii_to_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.assign_covalent_radii_to_dataframe" title="Permalink to this definition">¶</a></dt>
<dd><p>Assigns covalent radius to each atom based on its bond state. Using Values from :
Atomic Structures of all the Twenty Essential Amino Acids and a Tripeptide, with Bond Lengths as Sums of Atomic Covalent Radii
Heyrovska, 2008</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Pandas PDB dataframe with a bond_states_column</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Pandas PDB dataframe with added covalent_radius column</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.identify_bond_type_from_mapping">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">identify_bond_type_from_mapping</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">u</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">v</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">a</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span> </span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">query</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#identify_bond_type_from_mapping"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.identify_bond_type_from_mapping" title="Permalink to this definition">¶</a></dt>
<dd><p>Compares the bond length between two atoms in the graph, and the relevant experimental value by performing a lookup
against the watershed values in:
Automatic Assignment of Chemical Connectivity to Organic Molecules in the Cambridge
Structural Database. Jon C. Baber and Edward E. Hodgkin*</p>
<p>Bond orders are assigned in the order triple &lt; double &lt; single (e.g. if a bond is shorter than the triple bond
watershed (w_dt) then it is assigned as a triple bond. Similarly, if a bond is longer than this but shorter than the
double bond watershed, it is assigned double bond status.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph of atom-protein structure with atomic edges added</p></li>
<li><p><strong>u</strong> (<em>str</em>) – node 1 in edge</p></li>
<li><p><strong>v</strong> (<em>str</em>) – node 2 in edge</p></li>
<li><p><strong>a</strong> (<em>Dict</em><em>[</em><em>str</em><em>, </em><em>Any</em><em>]</em>) – edge data</p></li>
<li><p><strong>query</strong> (<em>str</em>) – “ELEMENTX-ELEMENTY” to perform lookup with (E.g. C-O, N-N)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Graph with atomic edge bond order assigned</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
</div>
<div class="section" id="features">
<h2>Features<a class="headerlink" href="#features" title="Permalink to this headline">¶</a></h2>
<div class="section" id="module-graphein.protein.features.nodes.aaindex">
<span id="node"></span><h3>Node<a class="headerlink" href="#module-graphein.protein.features.nodes.aaindex" title="Permalink to this headline">¶</a></h3>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.aaindex.aaindex1">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.aaindex.</span></span><span class="sig-name descname"><span class="pre">aaindex1</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">accession</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/aaindex.html#aaindex1"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.aaindex.aaindex1" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds AAIndex1 datavalues for a given accession as node features.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graphein to featurise</p></li>
<li><p><strong>accession</strong> (<em>str</em>) – AAIndex1 accession code for values to use</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Structure graph with AAindex1 node features added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.aaindex.fetch_AAIndex">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.aaindex.</span></span><span class="sig-name descname"><span class="pre">fetch_AAIndex</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">accession</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/aaindex.html#fetch_AAIndex"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.aaindex.fetch_AAIndex" title="Permalink to this definition">¶</a></dt>
<dd><p>Fetches AAindex1 dictionary from an accession code. The dictionary maps one-letter AA codes to float values</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>accession</strong> (<em>str</em>) – Aaindex1 accession code</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>tuple of record titel(str) and dictionary of AA:value mappings</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Tuple[str, Dict[str, float]]</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.nodes.amino_acid"></span><p>Featurization functions for amino acids.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.expasy_protein_scale">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">expasy_protein_scale</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_array</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">pandas.core.series.Series</span><span class="p"><span class="pre">,</span> </span><span class="pre">numpy.ndarray</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#expasy_protein_scale"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.expasy_protein_scale" title="Permalink to this definition">¶</a></dt>
<dd><p>Return amino acid features that come from the EXPASY protein scale.</p>
<p>Source: <a class="reference external" href="https://web.expasy.org/protscale/">https://web.expasy.org/protscale/</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n</strong> – Node in a NetworkX graph</p></li>
<li><p><strong>d</strong> – NetworkX node attributes.</p></li>
<li><p><strong>selection</strong> (<em>List</em><em>[</em><em>str</em><em>]</em><em>, </em><em>optional</em>) – List of columns to select. Viewable in graphein.protein.features.nodes.meiler_embeddings</p></li>
<li><p><strong>return_array</strong> (<em>bool</em>) – Bool indicating whether or not to return a np.ndarray of the features. Default is pd.Series</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.Series of amino acid features</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.Series</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.load_expasy_scales">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">load_expasy_scales</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#load_expasy_scales"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.load_expasy_scales" title="Permalink to this definition">¶</a></dt>
<dd><p>Load pre-downloaded EXPASY scales.</p>
<p>This helps with node featuarization.</p>
<p>The function is LRU-cached in memory for fast access
on each function call.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>pd.DataFrame containing expasy scales</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.load_meiler_embeddings">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">load_meiler_embeddings</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#load_meiler_embeddings"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.load_meiler_embeddings" title="Permalink to this definition">¶</a></dt>
<dd><p>Load pre-downloaded Meiler embeddings.</p>
<p>This helps with node featurization.</p>
<p>The function is LRU-cached in memory for fast access
on each function call.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>pd.DataFrame containing Meiler Embeddings from Meiler et al. 2001</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.meiler_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">meiler_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">numpy.array</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#meiler_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.meiler_embedding" title="Permalink to this definition">¶</a></dt>
<dd><p>Return amino acid features from reduced dimensional embeddings of amino acid physicochemical properties.</p>
<p>Source: <a class="reference external" href="https://link.springer.com/article/10.1007/s008940100038">https://link.springer.com/article/10.1007/s008940100038</a>
doi: <a class="reference external" href="https://doi.org/10.1007/s008940100038">https://doi.org/10.1007/s008940100038</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n</strong> – Node in a NetworkX graph</p></li>
<li><p><strong>d</strong> – NetworkX node attributes.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.Series of amino acid features</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.Series</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.nodes.dssp"></span><p>Featurization functions for graph nodes using DSSP predicted features.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.add_dssp_df">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">add_dssp_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">nx.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dssp_config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">DSSPConfig</span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">nx.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#add_dssp_df"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.add_dssp_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Construct DSSP dataframe and add as graph level variable to protein graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> – Input protein graph</p></li>
<li><p><strong>G</strong> – nx.Graph</p></li>
<li><p><strong>dssp_config</strong> (<em>DSSPConfig</em><em>, </em><em>optional</em>) – DSSPConfig object. Specifies which executable to run. Located in graphein.protein.config</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with DSSP dataframe added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.add_dssp_feature">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">add_dssp_feature</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#add_dssp_feature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.add_dssp_feature" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds add_dssp_feature specified amino acid feature as calculated
by DSSP to every node in a protein graph
:param G: Protein structure graph to add dssp feature to
:param feature: string specifying name of DSSP feature to add:
“chain”,
“resnum”,
“icode”,
“aa”,
“ss”,
“asa”,
“phi”,
“psi”,
“dssp_index”,
“NH_O_1_relidx”,
“NH_O_1_energy”,
“O_NH_1_relidx”,
“O_NH_1_energy”,
“NH_O_2_relidx”,
“NH_O_2_energy”,
“O_NH_2_relidx”,
“O_NH_2_energy”,</p>
<p>These names parse_dssp_df accessible in the DSSP_COLS list
:param G: Protein Graph to add features to
:type G: nx.Graph
:return: Protein structure graph with DSSP feature added to nodes
:rtype: nx.Graph</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.asa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">asa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#asa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.asa" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds ASA of each residue in protein graph as calculated by DSSP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with asa values added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.parse_dssp_df">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">parse_dssp_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dssp</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span> </span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#parse_dssp_df"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.parse_dssp_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Parse DSSP output to DataFrame</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>dssp</strong> (<em>Dict</em><em>[</em><em>str</em><em>, </em><em>Any</em><em>]</em>) – Dictionary containing DSSP output</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.Dataframe containing parsed DSSP output</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.phi">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">phi</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#phi"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.phi" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds phi-angles of each residue in protein graph as calculated by DSSP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with phi-angles values added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.process_dssp_df">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">process_dssp_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#process_dssp_df"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.process_dssp_df" title="Permalink to this definition">¶</a></dt>
<dd><p>Processes a DSSP DataFrame to make indexes align with node IDs</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – pd.DataFrame containing the parsed output from DSSP.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.DataFrame with node IDs</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.psi">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">psi</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#psi"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.psi" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds psi-angles of each residue in protein graph as calculated by DSSP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with psi-angles values added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.rsa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">rsa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#rsa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.rsa" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds RSA (relative solvent accessibility) of each residue in protein graph
as calculated by DSSP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with rsa values added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.secondary_structure">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">secondary_structure</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#secondary_structure"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.secondary_structure" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds secondary structure of each residue in protein graph
as calculated by DSSP in the form of a string</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with secondary structure added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.features.sequence.embeddings">
<span id="sequence"></span><h3>Sequence<a class="headerlink" href="#module-graphein.protein.features.sequence.embeddings" title="Permalink to this headline">¶</a></h3>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.embeddings.biovec_sequence_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">biovec_sequence_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#biovec_sequence_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.embeddings.biovec_sequence_embedding" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds BioVec sequence embedding feature to the graph. Computed over chains</p>
<p>Source: ProtVec: A Continuous Distributed Representation of Biological Sequences
Paper: <a class="reference external" href="http://arxiv.org/pdf/1503.05140v1.pdf">http://arxiv.org/pdf/1503.05140v1.pdf</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph protein structure graph with biovec embedding added. e.g. G.graph[“biovec_embedding_A”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.embeddings.compute_esm_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">compute_esm_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">representation</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'esm1b_t33_650M_UR50S'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_layer</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">33</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">np.ndarray</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#compute_esm_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.embeddings.compute_esm_embedding" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes sequence embedding using Pre-trained ESM model from FAIR</p>
<p>Sources:
Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences (2019)
Rives, Alexander and Meier, Joshua and Sercu, Tom and Goyal, Siddharth and Lin, Zeming and Liu, Jason and Guo,
Demi and Ott, Myle and Zitnick, C. Lawrence and Ma, Jerry and Fergus, Rob</p>
<p>Transformer protein language models are unsupervised structure learners 2020
Rao, Roshan M and Meier, Joshua and Sercu, Tom and Ovchinnikov, Sergey and Rives, Alexander</p>
<p>Pre-trained models:
Full Name layers params Dataset Embedding Dim Model URL
ESM-1b esm1b_t33_650M_UR50S 33 650M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt</a>
ESM1-main esm1_t34_670M_UR50S 34 670M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt</a>
esm1_t34_670M_UR50D 34 670M UR50/D 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt</a>
esm1_t34_670M_UR100 34 670M UR100 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt</a>
esm1_t12_85M_UR50S 12 85M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt</a>
esm1_t6_43M_UR50S 6 43M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sequence</strong> (<em>str</em>) – Protein sequence to embed (str)</p></li>
<li><p><strong>representation</strong> (<em>str</em>) – Type of embedding to extract. “residue: or “sequence”. Sequence-level embeddings are averaged residue embeddings</p></li>
<li><p><strong>model_name</strong> (<em>str</em>) – Name of pre-trained model to use</p></li>
<li><p><strong>output_layer</strong> (<em>int</em>) – integer indicating which layer the output should be taken from</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>embedding (np.ndarray)</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.embeddings.esm_residue_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">esm_residue_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'esm1b_t33_650M_UR50S'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_layer</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">33</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#esm_residue_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.embeddings.esm_residue_embedding" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes ESM residue embeddings from a protein sequence and adds the to the graph.</p>
<p>Sources:
Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences (2019)
Rives, Alexander and Meier, Joshua and Sercu, Tom and Goyal, Siddharth and Lin, Zeming and Liu, Jason and Guo,
Demi and Ott, Myle and Zitnick, C. Lawrence and Ma, Jerry and Fergus, Rob</p>
<p>Transformer protein language models are unsupervised structure learners 2020
Rao, Roshan M and Meier, Joshua and Sercu, Tom and Ovchinnikov, Sergey and Rives, Alexander</p>
<p>Pre-trained models:
Full Name layers params Dataset Embedding Dim Model URL
ESM-1b esm1b_t33_650M_UR50S 33 650M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt</a>
ESM1-main esm1_t34_670M_UR50S 34 670M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt</a>
esm1_t34_670M_UR50D 34 670M UR50/D 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt</a>
esm1_t34_670M_UR100 34 670M UR100 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt</a>
esm1_t12_85M_UR50S 12 85M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt</a>
esm1_t6_43M_UR50S 6 43M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add esm embedding to</p></li>
<li><p><strong>model_name</strong> (<em>str</em>) – Name of pre-trained model to use</p></li>
<li><p><strong>output_layer</strong> (<em>int</em>) – index of output layer in pre-trained model</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with esm embedding feature added to nodes</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.embeddings.esm_sequence_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">esm_sequence_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#esm_sequence_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.embeddings.esm_sequence_embedding" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes ESM sequence embedding feature over chains in a graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graphein protein structure graphein with esm embedding features added eg. G.graph[“esm_embedding_A”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.sequence.propy"></span><dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.aa_dipeptide_composition">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">aa_dipeptide_composition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#aa_dipeptide_composition"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.aa_dipeptide_composition" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the composition of AADs, dipeptide and 3-mers for a given protein sequence. Contains all composition</dt><dd><p>values of AADs, dipeptide and 3-mers (8420).</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with aa_dipeptide_composition feature added.
G.graph[“aa_dipeptide_composition_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.aa_spectrum">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">aa_spectrum</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#aa_spectrum"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.aa_spectrum" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the spectrum descriptors of 3-mers for a given protein. Contains the composition values of 8000 3-mers</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with aa_spectrum feature added. G.graph[“aa_spectrum_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.all_composition_descriptors">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">all_composition_descriptors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#all_composition_descriptors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.all_composition_descriptors" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate all composition descriptors based on seven different properties of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_descriptors feature added.
G.graph[“composition_descriptors_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.all_ctd_descriptors">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">all_ctd_descriptors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#all_ctd_descriptors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.all_ctd_descriptors" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate all CTD descriptors based seven different properties of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with ctd_descriptors feature added.
G.graph[“ctd_descriptors_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.all_distribution_descriptors">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">all_distribution_descriptors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#all_distribution_descriptors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.all_distribution_descriptors" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate all distribution descriptors based on seven different properties of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_descriptors feature added.
G.graph[“distribution_descriptors_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.all_transition_descriptors">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">all_transition_descriptors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#all_transition_descriptors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.all_transition_descriptors" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate all transition descriptors based on seven different properties of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_descriptors feature added.
G.graph[“transition_descriptors_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.amino_acid_composition">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">amino_acid_composition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#amino_acid_composition"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.amino_acid_composition" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the composition of Amino acids for a given protein sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>Optional</em><em>[</em><em>List</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with amino_acid_composition feature added. G.graph[“amino_acid_composition_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_all">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_all</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_all"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_all" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Compute Geary autocorrelation descriptors based on 8 properties of AADs. Result contains 30*8=240 Geary</dt><dd><p>autocorrelation descriptors based on the given properties(i.e., _AAPropert).</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_all feature added.
G.graph[“autocorrelation_geary_all_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_av_flexibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_av_flexibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_av_flexibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_av_flexibility" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on AvFlexibility. contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on AvFlexibility.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_av_flexibility feature added.
G.graph[“autocorrelation_geary_av_flexibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_free_energy">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_free_energy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_free_energy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_free_energy" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on FreeEnergy. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on FreeEnergy.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_free_energy feature added.
G.graph[“autocorrelation_geary_av_free_energy_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_hydrophobicity" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on hydrophobicity. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on hydrophobicity.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_hydrophobicity feature added.
G.graph[“autocorrelation_geary_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_mutability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_mutability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_mutability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_mutability" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on Mutability. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on mutability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_mutability feature added.
G.graph[“autocorrelation_geary_mutability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_polarizability" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on polarizability. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on polarizability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_polarizability feature added.
G.graph[“autocorrelation_geary_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_residue_asa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_residue_asa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_residue_asa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_residue_asa" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on ResidueASA. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on ResidueASA.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_residue_asa feature added.
G.graph[“autocorrelation_geary_residue_asa_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_residue_vol">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_residue_vol</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_residue_vol"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_residue_vol" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on ResidueVol. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on ResidueVol.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_residue_vol feature added.
G.graph[“autocorrelation_geary_residue_vol_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_steric">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_steric</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_steric"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_steric" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on Steric. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on Steric</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_steric feature added.
G.graph[“autocorrelation_geary_steric_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_all">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_all</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_all"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_all" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Compute Moran autocorrelation descriptors based on 8 properties of AADs. Result contains 30*8=240 Moran</dt><dd><p>autocorrelation descriptors based on the given properties(i.e., _AAPropert).</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_all feature added.
G.graph[“autocorrelation_moran_all_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_av_flexibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_av_flexibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_av_flexibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_av_flexibility" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on AvFlexibility. contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on AvFlexibility.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_av_flexibility feature added.
G.graph[“autocorrelation_moran_av_flexibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_free_energy">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_free_energy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_free_energy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_free_energy" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on FreeEnergy. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on FreeEnergy.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_free_energy feature added.
G.graph[“autocorrelation_moran_av_free_energy_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_hydrophobicity" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on hydrophobicity. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on hydrophobicity.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_hydrophobicity feature added.
G.graph[“autocorrelation_moran_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_mutability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_mutability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_mutability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_mutability" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on Mutability. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on mutability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_mutability feature added.
G.graph[“autocorrelation_moran_mutability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_polarizability" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on polarizability. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on polarizability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_polarizability feature added.
G.graph[“autocorrelation_moran_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_residue_asa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_residue_asa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_residue_asa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_residue_asa" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on ResidueASA. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on ResidueASA.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_residue_asa feature added.
G.graph[“autocorrelation_moran_residue_asa_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_residue_vol">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_residue_vol</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_residue_vol"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_residue_vol" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on ResidueVol. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on ResidueVol.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_residue_vol feature added.
G.graph[“autocorrelation_moran_residue_vol_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_steric">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_steric</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_steric"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_steric" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on Steric. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on Steric</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_steric feature added.
G.graph[“autocorrelation_moran_steric_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_all">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_all</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_all"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_all" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Compute NormalizedMoreauBroto autocorrelation descriptors based on 8 properties of AADs. Result contains 30*8=240</dt><dd><p>NormalizedMoreauBroto autocorrelation descriptors based on the given properties(i.e., _AAPropert).</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_all feature added.
G.graph[“autocorrelation_normalized_moreau_broto_all_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_av_flexibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_av_flexibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_av_flexibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_av_flexibility" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on AvFlexibility. contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on AvFlexibility.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_av_flexibility feature added.
G.graph[“autocorrelation_normalized_moreau_broto_av_flexibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_free_energy">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_free_energy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_free_energy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_free_energy" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on FreeEnergy. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on FreeEnergy.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_free_energy feature added.
G.graph[“autocorrelation_normalized_moreau_broto_av_free_energy_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_hydrophobicity" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on hydrophobicity. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on hydrophobicity.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_hydrophobicity feature added.
G.graph[“autocorrelation_normalized_moreau_broto_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_mutability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_mutability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_mutability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_mutability" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on Mutability. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on mutability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_mutability feature added.
G.graph[“autocorrelation_normalized_moreau_broto_mutability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_polarizability" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on polarizability. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on polarizability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_polarizability feature added.
G.graph[“autocorrelation_normalized_moreau_broto_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_residue_asa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_residue_asa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_residue_asa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_residue_asa" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on ResidueASA. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on ResidueASA.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_residue_asa feature added.
G.graph[“autocorrelation_normalized_moreau_broto_residue_asa_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_residue_vol">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_residue_vol</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_residue_vol"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_residue_vol" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on ResidueVol. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on ResidueVol.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_residue_vol feature added.
G.graph[“autocorrelation_normalized_moreau_broto_residue_vol_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_steric">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_steric</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_steric"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_steric" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on Steric. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on Steric</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_steric feature added.
G.graph[“autocorrelation_normalized_moreau_broto_steric_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_total">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_total</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_total"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_total" title="Permalink to this definition">¶</a></dt>
<dd><dl class="simple">
<dt>Compute all autocorrelation descriptors based on 8 properties of AADs. result contains 30*8*3=720 normalized Moreau</dt><dd><p>Broto, Moran, and Geary</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_total feature added.
G.graph[“autocorrelation_total_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_charge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_charge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_charge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate composition descriptors based on Charge of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_charge feature added.
G.graph[“composition_charge_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_descriptor">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_descriptor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAProperty</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">Any</span><span class="p"><span class="pre">,</span> </span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAPName</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_descriptor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_descriptor" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute composition descriptors.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>AAProperty</strong> (<em>Dict</em><em>[</em><em>Any</em><em>, </em><em>Any</em><em>]</em>) – contains classification of amino acids such as _Polarizability.</p></li>
<li><p><strong>AAPName</strong> (<em>str</em>) – used for indicating a AAP name.</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_{AAPName} feature added.
G.graph[“composition_{AAPName}_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_hydrophobicity" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate composition descriptors based on Hydrophobicity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_hydrophobicity feature added.
G.graph[“composition_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_normalized_vdwv">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_normalized_vdwv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_normalized_vdwv"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_normalized_vdwv" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate composition descriptors based on NormalizedVDWV of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_normalized_vdwv feature added.
G.graph[“composition_normalized_vdwv_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_polarity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_polarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_polarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_polarity" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate composition descriptors based on Polarity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_polarity feature added.
G.graph[“composition_polarity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_polarizability" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate composition descriptors based on Polarizability of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_polarizability feature added.
G.graph[“composition_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_secondary_str">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_secondary_str</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_secondary_str"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_secondary_str" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate composition descriptors based on SecondaryStr of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_secondary_str feature added.
G.graph[“composition_secondary_str_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_solvent_accessibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_solvent_accessibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_solvent_accessibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_solvent_accessibility" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate composition descriptors based on SolventAccessibility of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_solvent_accessibility feature added.
G.graph[“composition_solvent_accessibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.compute_propy_feature">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">compute_propy_feature</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">func</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Callable</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#compute_propy_feature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.compute_propy_feature" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes Propy Descriptors over chains in a Protein Graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph</p></li>
<li><p><strong>func</strong> (<em>Callable</em>) – ProPy wrapper function to compute</p></li>
<li><p><strong>feature_name</strong> (<em>str</em>) – Name of feature to index it in the nx.Graph object</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>str</em><em>]</em><em>, </em><em>optional</em>) – Type of aggregation to use when aggregating a feature over multiple chains. One of:
[“mean”, “man”, “sum”]. Defaults to None.</p></li>
</ul>
</dd>
<dt class="field-even">Return G</dt>
<dd class="field-even"><p>Returns protein Graph with features added. Features are accessible with G.graph[{feature_name}_{chain | aggegation_type}]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.dipeptide_composition">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">dipeptide_composition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#dipeptide_composition"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.dipeptide_composition" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the composition of dipeptidefor a given protein sequence. Contains composition of 400 dipeptides</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with dipeptide_composition feature added. G.graph[“dipeptide_composition_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_charge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_charge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_charge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate distribution descriptors based on Charge of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_charge feature added.
G.graph[“distribution_charge_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_descriptor">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_descriptor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAProperty</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">Any</span><span class="p"><span class="pre">,</span> </span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAPName</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_descriptor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_descriptor" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute distribution descriptors.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>AAProperty</strong> (<em>Dict</em><em>[</em><em>Any</em><em>, </em><em>Any</em><em>]</em>) – contains classifciation of amino acids such as _Polarizability.</p></li>
<li><p><strong>AAPName</strong> (<em>str</em>) – used for indicating a AAP name.</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_{AAPName} feature added.
G.graph[“distribution_{AAPName}_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_hydrophobicity" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate distribution descriptors based on Hydrophobicity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_hydrophobicity feature added.
G.graph[“distribution_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_normalized_vdwv">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_normalized_vdwv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_normalized_vdwv"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_normalized_vdwv" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate distribution descriptors based on NormalizedVDWV of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_normalized_vdwv feature added.
G.graph[“distribution_normalized_vdwv_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_polarity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_polarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_polarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_polarity" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate distribution descriptors based on Polarity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_polarity feature added.
G.graph[“distribution_polarity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_polarizability" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate distribution descriptors based on Polarizability of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_polarizability feature added.
G.graph[“distribution_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_secondary_str">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_secondary_str</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_secondary_str"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_secondary_str" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate distribution descriptors based on SecondaryStr of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_secondary_str feature added.
G.graph[“distribution_secondary_str_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_solvent_accessibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_solvent_accessibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_solvent_accessibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_solvent_accessibility" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate distribution descriptors based on SolventAccessibility of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_solvent_accessibility feature added.
G.graph[“distribution_solvent_accessibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.quasi_sequence_order">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">quasi_sequence_order</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxlag</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">float</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">0.1</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#quasi_sequence_order"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.quasi_sequence_order" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute quasi-sequence-order descriptors for a given protein.</p>
<dl class="simple">
<dt>Kuo-Chen Chou. Prediction of Protein Subcellar Locations by Incorporating Quasi-Sequence-Order Effect.</dt><dd><p>Biochemical and Biophysical Research Communications 2000, 278, 477-483.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>maxlag</strong> (<em>int</em>) – (int, optional (default: 30)) – the maximum lag and the length of the protein should be larger than
maxlag</p></li>
<li><p><strong>weight</strong> (<em>float</em>) – (float, optional (default: 0.1)) – a weight factor. Please see reference 1 for its choice.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.sequence_order_coupling_number_total">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">sequence_order_coupling_number_total</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxlag</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">30</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#sequence_order_coupling_number_total"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.sequence_order_coupling_number_total" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute the sequence order coupling numbers from 1 to maxlag for a given protein sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph</p></li>
<li><p><strong>(</strong><strong>default</strong> (<em>maxlag</em><em> (</em><em>int</em><em>, </em><em>optional</em>) – 30)) – the maximum lag and the length of the protein should be larger</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_charge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_charge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_charge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_charge" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate transition descriptors based on Charge of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_charge feature added.
G.graph[“transition_charge_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_descriptor">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_descriptor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAProperty</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">Any</span><span class="p"><span class="pre">,</span> </span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAPName</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_descriptor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_descriptor" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute transition descriptors.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>AAProperty</strong> (<em>Dict</em><em>[</em><em>Any</em><em>, </em><em>Any</em><em>]</em>) – contains classifciation of amino acids such as _Polarizability.</p></li>
<li><p><strong>AAPName</strong> (<em>str</em>) – used for indicating a AAP name.</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_{AAPName} feature added.
G.graph[“transition_{AAPName}_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_hydrophobicity" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate transition descriptors based on Hydrophobicity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_hydrophobicity feature added.
G.graph[“transition_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_normalized_vdwv">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_normalized_vdwv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_normalized_vdwv"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_normalized_vdwv" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate transition descriptors based on NormalizedVDWV of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_normalized_vdwv feature added.
G.graph[“transition_normalized_vdwv_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_polarity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_polarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_polarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_polarity" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate transition descriptors based on Polarity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_polarity feature added.
G.graph[“transition_polarity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_polarizability" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate transition descriptors based on Polarizability of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_polarizability feature added.
G.graph[“transition_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_secondary_str">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_secondary_str</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_secondary_str"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_secondary_str" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate transition descriptors based on SecondaryStr of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_secondary_str feature added.
G.graph[“transition_secondary_str_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_solvent_accessibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_solvent_accessibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_solvent_accessibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_solvent_accessibility" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate transition descriptors based on SolventAccessibility of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><em>str</em><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_solvent_accessibility feature added.
G.graph[“transition_solvent_accessibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.sequence.sequence"></span><p>Functions for graph-level featurization of the sequence of a protein. This submodule is focussed on physicochemical
proporties of the sequence.</p>
</div>
<div class="section" id="module-0">
<span id="sequence-utils"></span><h3>Sequence Utils<a class="headerlink" href="#module-0" title="Permalink to this headline">¶</a></h3>
<dl class="py function">
<dt class="sig sig-object py" id="id0">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">biovec_sequence_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#biovec_sequence_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#id0" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds BioVec sequence embedding feature to the graph. Computed over chains</p>
<p>Source: ProtVec: A Continuous Distributed Representation of Biological Sequences
Paper: <a class="reference external" href="http://arxiv.org/pdf/1503.05140v1.pdf">http://arxiv.org/pdf/1503.05140v1.pdf</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph protein structure graph with biovec embedding added. e.g. G.graph[“biovec_embedding_A”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="id1">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">compute_esm_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">representation</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'esm1b_t33_650M_UR50S'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_layer</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">33</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">np.ndarray</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#compute_esm_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#id1" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes sequence embedding using Pre-trained ESM model from FAIR</p>
<p>Sources:
Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences (2019)
Rives, Alexander and Meier, Joshua and Sercu, Tom and Goyal, Siddharth and Lin, Zeming and Liu, Jason and Guo,
Demi and Ott, Myle and Zitnick, C. Lawrence and Ma, Jerry and Fergus, Rob</p>
<p>Transformer protein language models are unsupervised structure learners 2020
Rao, Roshan M and Meier, Joshua and Sercu, Tom and Ovchinnikov, Sergey and Rives, Alexander</p>
<p>Pre-trained models:
Full Name layers params Dataset Embedding Dim Model URL
ESM-1b esm1b_t33_650M_UR50S 33 650M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt</a>
ESM1-main esm1_t34_670M_UR50S 34 670M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt</a>
esm1_t34_670M_UR50D 34 670M UR50/D 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt</a>
esm1_t34_670M_UR100 34 670M UR100 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt</a>
esm1_t12_85M_UR50S 12 85M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt</a>
esm1_t6_43M_UR50S 6 43M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sequence</strong> (<em>str</em>) – Protein sequence to embed (str)</p></li>
<li><p><strong>representation</strong> (<em>str</em>) – Type of embedding to extract. “residue: or “sequence”. Sequence-level embeddings are averaged residue embeddings</p></li>
<li><p><strong>model_name</strong> (<em>str</em>) – Name of pre-trained model to use</p></li>
<li><p><strong>output_layer</strong> (<em>int</em>) – integer indicating which layer the output should be taken from</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>embedding (np.ndarray)</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="id2">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">esm_residue_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'esm1b_t33_650M_UR50S'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_layer</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">int</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">33</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#esm_residue_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#id2" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes ESM residue embeddings from a protein sequence and adds the to the graph.</p>
<p>Sources:
Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences (2019)
Rives, Alexander and Meier, Joshua and Sercu, Tom and Goyal, Siddharth and Lin, Zeming and Liu, Jason and Guo,
Demi and Ott, Myle and Zitnick, C. Lawrence and Ma, Jerry and Fergus, Rob</p>
<p>Transformer protein language models are unsupervised structure learners 2020
Rao, Roshan M and Meier, Joshua and Sercu, Tom and Ovchinnikov, Sergey and Rives, Alexander</p>
<p>Pre-trained models:
Full Name layers params Dataset Embedding Dim Model URL
ESM-1b esm1b_t33_650M_UR50S 33 650M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt</a>
ESM1-main esm1_t34_670M_UR50S 34 670M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt</a>
esm1_t34_670M_UR50D 34 670M UR50/D 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt</a>
esm1_t34_670M_UR100 34 670M UR100 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt</a>
esm1_t12_85M_UR50S 12 85M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt</a>
esm1_t6_43M_UR50S 6 43M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add esm embedding to</p></li>
<li><p><strong>model_name</strong> (<em>str</em>) – Name of pre-trained model to use</p></li>
<li><p><strong>output_layer</strong> (<em>int</em>) – index of output layer in pre-trained model</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with esm embedding feature added to nodes</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="id3">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">esm_sequence_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#esm_sequence_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#id3" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes ESM sequence embedding feature over chains in a graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graphein protein structure graphein with esm embedding features added eg. G.graph[“esm_embedding_A”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.features.utils">
<span id="utils"></span><h3>Utils<a class="headerlink" href="#module-graphein.protein.features.utils" title="Permalink to this headline">¶</a></h3>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.utils.aggregate_graph_feature_over_chains">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.utils.</span></span><span class="sig-name descname"><span class="pre">aggregate_graph_feature_over_chains</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/utils.html#aggregate_graph_feature_over_chains"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.utils.aggregate_graph_feature_over_chains" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs aggregation of a feature over the chains. E.g. sums/averages/min/max molecular weights for each chain</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph of protein containing chain-specific features</p></li>
<li><p><strong>feature_name</strong> (<em>str</em>) – name of features to aggregate</p></li>
<li><p><strong>aggregation_type</strong> (<em>str</em>) – Type of aggregation to perform (min/max/sum/mean)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph of protein with a new aggregated feature G.graph[f”{feature_name}_{aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.utils.convert_graph_dict_feat_to_series">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.utils.</span></span><span class="sig-name descname"><span class="pre">convert_graph_dict_feat_to_series</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_name</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/utils.html#convert_graph_dict_feat_to_series"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.utils.convert_graph_dict_feat_to_series" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes in a graph and a graph-level feature_name. Converts this feature to a pd.Series.
This is useful as some features are output as dictionaries and we wish to standardise this.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph containing G.graph[f”{feature_name}”]: Dict[Any, Any]</p></li>
<li><p><strong>feature_name</strong> (<em>str</em>) – Name of feature to convert to dictionary</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph containing G.graph[f”{feature_name}”]: pd.Series</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
</div>
<div class="section" id="module-graphein.protein.meshes">
<span id="meshes"></span><h2>Meshes<a class="headerlink" href="#module-graphein.protein.meshes" title="Permalink to this headline">¶</a></h2>
<p>Functions to create protein meshes via pymol.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.check_for_pymol_installation">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">check_for_pymol_installation</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#check_for_pymol_installation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.check_for_pymol_installation" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks for presence of a pymol installation</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.configure_pymol_session">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">configure_pymol_session</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><span class="pre">graphein.protein.config.ProteinMeshConfig</span></a><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#configure_pymol_session"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.configure_pymol_session" title="Permalink to this definition">¶</a></dt>
<dd><p>Configures a PyMol session based on config.parse_pymol_commands. Uses default parameters -“cKq”.
See: <a class="reference external" href="https://pymolwiki.org/index.php/Command_Line_Options">https://pymolwiki.org/index.php/Command_Line_Options</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><em>graphein.protein.config.ProteinMeshConfig</em></a>) – ProteinMeshConfig to use</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.convert_verts_and_face_to_mesh">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">convert_verts_and_face_to_mesh</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">verts</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">torch.FloatTensor</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">faces</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">NamedTuple</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Meshes</span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#convert_verts_and_face_to_mesh"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.convert_verts_and_face_to_mesh" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts vertices and faces into a pytorch3d.structures Meshes object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>verts</strong> (<em>torch.FloatTensor</em>) – vertices</p></li>
<li><p><strong>faces</strong> (<em>NamedTuple</em>) – faces</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Meshes object</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pytorch3d.structures.Meshes</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.create_mesh">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">create_mesh</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_file</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">ProteinMeshConfig</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">torch.FloatTensor</span><span class="p"><span class="pre">,</span> </span><span class="pre">NamedTuple</span><span class="p"><span class="pre">,</span> </span><span class="pre">NamedTuple</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#create_mesh"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.create_mesh" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a PyTorch 3D mesh from a pdb_file or pdb code.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb_file</strong> (<em>str</em><em>, </em><em>optional</em>) – path to pdb pdb_file. Defaults to None</p></li>
<li><p><strong>pdb_code</strong> (<em>str</em><em>, </em><em>optional</em>) – 4-letter PDB accession code. Defaults to None.</p></li>
<li><p><strong>out_dir</strong> (<em>str</em><em>, </em><em>optional</em>) – output directory to store “.obj” file. Defaults to /tmp/</p></li>
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><em>graphein.protein.config.ProteinMeshConfig</em></a>) – ProteinMeshConfig config to use. Defaults to default config in graphein.protein.config</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>verts, faces, aux</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Tuple[torch.Tensor, torch.Tensor, torch.Tensor]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.get_obj_file">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">get_obj_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_file</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><span class="pre">graphein.protein.config.ProteinMeshConfig</span></a><span class="p"><span class="pre">]</span></span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#get_obj_file"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.get_obj_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs PyMol to compute surface/mesh for a given protein</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb_file</strong> (<em>str</em><em>, </em><em>optional</em>) – path to pdb_file to use</p></li>
<li><p><strong>pdb_code</strong> (<em>str</em><em>, </em><em>optional</em>) – 4-letter pdb accession code</p></li>
<li><p><strong>out_dir</strong> (<em>str</em><em>, </em><em>optional</em>) – path to output. Defaults to /tmp/</p></li>
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><em>graphein.protein.config.ProteinMeshConfig</em></a>) – ProteinMeshConfig containing pymol commands to run. Default is “show surface”</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>returns path to .obj file (str)</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.normalize_and_center_mesh_vertices">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">normalize_and_center_mesh_vertices</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">verts</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">torch.FloatTensor</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">torch.FloatTensor</span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#normalize_and_center_mesh_vertices"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.normalize_and_center_mesh_vertices" title="Permalink to this definition">¶</a></dt>
<dd><p>We scale normalize and center the target mesh to fit in a sphere of radius 1 centered at (0,0,0).
(scale, center) will be used to bring the predicted mesh to its original center and scale
Note that normalizing the target mesh, speeds up the optimization but is not necessary!</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>verts</strong> (<em>torch.FloatTensor</em>) – mesh vertices</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>normalized and centered vertices</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>torch.FloatTensor</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.parse_pymol_commands">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">parse_pymol_commands</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><span class="pre">graphein.protein.config.ProteinMeshConfig</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#parse_pymol_commands"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.parse_pymol_commands" title="Permalink to this definition">¶</a></dt>
<dd><p>Parses pymol commands from config. At the moment users can only supply a list of string commands.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><em>ProteinMeshConfig</em></a>) – ProteinMeshConfig c</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>list of pymol commands to run</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>List[str]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.run_pymol_commands">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">run_pymol_commands</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">commands</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#run_pymol_commands"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.run_pymol_commands" title="Permalink to this definition">¶</a></dt>
<dd><p>Runs Pymol Commands</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>commands</strong> (<em>List</em><em>[</em><em>str</em><em>]</em>) – List of commands to pass to PyMol</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.visualisation">
<span id="visualisation"></span><h2>Visualisation<a class="headerlink" href="#module-graphein.protein.visualisation" title="Permalink to this headline">¶</a></h2>
<p>Functions for plotting protein graphs and meshes</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.colour_edges">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">colour_edges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_map</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">matplotlib.colors.ListedColormap</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_by</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">'kind'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#colour_edges"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.colour_edges" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes edge colours based on the kind of bond/interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph to compute edge colours from</p></li>
<li><p><strong>colour_map</strong> (<em>matplotlib.colors.ListedColormap</em>) – Colourmap to use</p></li>
<li><p><strong>colour_by</strong> (<em>str</em>) – Edge attribute to colour by. Currently only “kind” is supported</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>List of edge colours</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>List[Tuple[float, float, float, float]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.colour_nodes">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">colour_nodes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_map</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">matplotlib.colors.ListedColormap</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_by</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">,</span> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#colour_nodes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.colour_nodes" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes node colours based on “degree”, “seq_position” or node attributes</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Graph to compute node colours for</p></li>
<li><p><strong>colour_map</strong> (<em>matplotlib.colors.ListedColormap</em>) – Colourmap to use.</p></li>
<li><p><strong>colour_by</strong> (<em>str</em>) – Manner in which to colour nodes. If not “degree” or “seq_position”, this must correspond to a node feature</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>List of node colours</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>List[Tuple[float, float, float, float]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.plot_pointcloud">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">plot_pointcloud</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mesh</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">Meshes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span> <span class="o"><span class="pre">=</span></span> <span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Axes3D</span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#plot_pointcloud"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.plot_pointcloud" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots pytorch3d Meshes object as pointcloud</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mesh</strong> (<em>pytorch3d.structures.meshes.Meshes</em>) – Meshes object to plot</p></li>
<li><p><strong>title</strong> (<em>str</em>) – Title of plot</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>returns Axes3D containing plot</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Axes3D</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.plot_protein_structure_graph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">plot_protein_structure_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">G:</span> <span class="pre">networkx.classes.graph.Graph</span></em>, <em class="sig-param"><span class="pre">angle:</span> <span class="pre">int</span> <span class="pre">=</span> <span class="pre">30</span></em>, <em class="sig-param"><span class="pre">plot_title:</span> <span class="pre">Optional[str]</span> <span class="pre">=</span> <span class="pre">None</span></em>, <em class="sig-param"><span class="pre">figsize:</span> <span class="pre">Tuple[int</span></em>, <em class="sig-param"><span class="pre">int]</span> <span class="pre">=</span> <span class="pre">(10</span></em>, <em class="sig-param"><span class="pre">7)</span></em>, <em class="sig-param"><span class="pre">node_alpha:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">0.7</span></em>, <em class="sig-param"><span class="pre">node_size_min:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">20.0</span></em>, <em class="sig-param"><span class="pre">node_size_multiplier:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">20.0</span></em>, <em class="sig-param"><span class="pre">label_node_ids:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">True</span></em>, <em class="sig-param"><span class="pre">node_colour_map=&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></em>, <em class="sig-param"><span class="pre">edge_color_map=&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></em>, <em class="sig-param"><span class="pre">colour_nodes_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'degree'</span></em>, <em class="sig-param"><span class="pre">colour_edges_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'kind'</span></em>, <em class="sig-param"><span class="pre">edge_alpha:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">0.5</span></em>, <em class="sig-param"><span class="pre">plot_style:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'ggplot'</span></em>, <em class="sig-param"><span class="pre">out_path:</span> <span class="pre">Optional[str]</span> <span class="pre">=</span> <span class="pre">None</span></em>, <em class="sig-param"><span class="pre">out_format:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'.png'</span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">mpl_toolkits.mplot3d.axes3d.Axes3D</span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#plot_protein_structure_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.plot_protein_structure_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots protein structure graph in Axes3D.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph Protein Structure graph to plot</p></li>
<li><p><strong>angle</strong> (<em>int</em>) – View angle</p></li>
<li><p><strong>plot_title</strong> (<em>str</em><em>, </em><em>optional</em>) – Title of plot. Defaults to None</p></li>
<li><p><strong>figsize</strong> (<em>Tuple</em><em>[</em><em>int</em><em>, </em><em>int</em><em>]</em>) – Size of figure, defaults to (10, 7)</p></li>
<li><p><strong>node_alpha</strong> (<em>float</em>) – Controls node transparency, defaults to 0.7</p></li>
<li><p><strong>node_size_min</strong> (<em>float</em>) – Specifies node minimum size, defaults to 20</p></li>
<li><p><strong>node_size_multiplier</strong> (<em>float</em>) – Scales node size by a constant. Node sizes reflect degree.</p></li>
<li><p><strong>label_node_ids</strong> (<em>bool</em>) – bool indicating whether or not to plot node_id labels</p></li>
<li><p><strong>node_colour_map</strong> (<em>plt.cm</em>) – colour map to use for nodes</p></li>
<li><p><strong>edge_color_map</strong> (<em>plt.cm</em>) – colour map to use for edges</p></li>
<li><p><strong>colour_nodes_by</strong> (<em>str</em>) – Specifies how to colour nodes. “degree”, “seq_position” or a node feature</p></li>
<li><p><strong>colour_edges_by</strong> – Specifies how to colour edges. Currently only “kind” is supported</p></li>
<li><p><strong>edge_alpha</strong> – Controls edge transparency</p></li>
<li><p><strong>plot_style</strong> – matplotlib style sheet to use</p></li>
<li><p><strong>out_path</strong> – If not none, writes plot to this location</p></li>
<li><p><strong>out_format</strong> – Fileformat to use for plot</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.plotly_protein_structure_graph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">plotly_protein_structure_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">G:</span> <span class="pre">networkx.classes.graph.Graph</span></em>, <em class="sig-param"><span class="pre">plot_title:</span> <span class="pre">Optional[str]</span> <span class="pre">=</span> <span class="pre">None</span></em>, <em class="sig-param"><span class="pre">figsize:</span> <span class="pre">Tuple[int</span></em>, <em class="sig-param"><span class="pre">int]</span> <span class="pre">=</span> <span class="pre">(620</span></em>, <em class="sig-param"><span class="pre">650)</span></em>, <em class="sig-param"><span class="pre">node_alpha:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">0.7</span></em>, <em class="sig-param"><span class="pre">node_size_min:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">20.0</span></em>, <em class="sig-param"><span class="pre">node_size_multiplier:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">20.0</span></em>, <em class="sig-param"><span class="pre">label_node_ids:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">True</span></em>, <em class="sig-param"><span class="pre">node_colour_map=&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></em>, <em class="sig-param"><span class="pre">edge_color_map=&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></em>, <em class="sig-param"><span class="pre">colour_nodes_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'degree'</span></em>, <em class="sig-param"><span class="pre">colour_edges_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'type'</span></em>, <em class="sig-param"><span class="pre">edge_alpha:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">0.5</span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objs._figure.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#plotly_protein_structure_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.plotly_protein_structure_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots protein structure graph using plotly.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph Protein Structure graph to plot</p></li>
<li><p><strong>plot_title</strong> (<em>str</em><em>, </em><em>optional</em>) – Title of plot, defaults to None</p></li>
<li><p><strong>figsize</strong> (<em>Tuple</em><em>[</em><em>int</em><em>, </em><em>int</em><em>]</em>) – Size of figure, defaults to (620, 650)</p></li>
<li><p><strong>node_alpha</strong> (<em>float</em>) – Controls node transparency, defaults to 0.7</p></li>
<li><p><strong>node_size_min</strong> (<em>float</em>) – Specifies node minimum size</p></li>
<li><p><strong>node_size_multiplier</strong> (<em>float</em>) – Scales node size by a constant. Node sizes reflect degree.</p></li>
<li><p><strong>label_node_ids</strong> (<em>bool</em>) – bool indicating whether or not to plot node_id labels</p></li>
<li><p><strong>node_colour_map</strong> (<em>plt.cm</em>) – colour map to use for nodes</p></li>
<li><p><strong>edge_color_map</strong> (<em>plt.cm</em>) – colour map to use for edges</p></li>
<li><p><strong>colour_nodes_by</strong> (<em>str</em>) – Specifies how to colour nodes. “degree”, “seq_position” or a node feature</p></li>
<li><p><strong>colour_edges_by</strong> (<em>str</em>) – Specifies how to colour edges. Currently only “kind” is supported</p></li>
<li><p><strong>edge_alpha</strong> (<em>float</em>) – Controls edge transparency</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Plotly Graph Objects plot</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>go.Figure</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="id4">
<h2>Utils<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-graphein.protein.utils"></span><p>Provides utility functions for use across Graphein</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.download_pdb">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">download_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#download_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.download_pdb" title="Permalink to this definition">¶</a></dt>
<dd><p>Download PDB structure from PDB</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pdb_code</strong> (<em>str</em>) – 4 character PDB accession code</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>returns filepath to downloaded structure</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.filter_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">filter_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataframe</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">pandas.core.frame.DataFrame</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">by_column</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">list_of_values</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">Any</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boolean</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">bool</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pandas.core.frame.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#filter_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.filter_dataframe" title="Permalink to this definition">¶</a></dt>
<dd><p>Filter function for dataframe.
Filters the [dataframe] such that the [by_column] values have to be
in the [list_of_values] list if boolean == True, or not in the list
if boolean == False</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataframe</strong> (<em>pd.DataFrame</em>) – pd.DataFrame to filter</p></li>
<li><p><strong>by_column</strong> (<em>str</em>) – str denoting by_column of dataframe to filter</p></li>
<li><p><strong>list_of_values</strong> (<em>List</em><em>[</em><em>Any</em><em>]</em>) – List of values to filter with</p></li>
<li><p><strong>boolean</strong> (<em>bool</em>) – indicates whether to keep or exclude matching list_of_values. True -&gt; in list, false -&gt; not in list</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Filtered dataframe</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.get_protein_name_from_filename">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">get_protein_name_from_filename</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_path</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#get_protein_name_from_filename"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.get_protein_name_from_filename" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts a filename from a pdb_path</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pdb_path</strong> (<em>str</em>) – Path to extract filename from</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>file name</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.three_to_one_with_mods">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">three_to_one_with_mods</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">res</span></span><span class="p"><span class="pre">:</span></span> <span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">str</span></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#three_to_one_with_mods"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.three_to_one_with_mods" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts three letter AA codes into 1 letter. Allows for modified residues.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>res</strong> (<em>str</em>) – Three letter residue code str:</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>1-letter residue code</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>
</div>
</div>

      </article>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="graphein.rna.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">graphein.rna</div>
              </div>
              <svg><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="../datasets/ppisp.html">
              <svg><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">PPISP - Protein Protein Interaction Structure Prediction</div>
                
              </div>
            </a>
        </div>

        <div class="related-information">
              Copyright &#169; 2020, Arian Jamasb
            |
            Built with <a href="https://www.sphinx-doc.org/">Sphinx</a>
              and
              <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
              <a href="https://github.com/pradyunsg/furo">Furo theme</a>.
            |
            <a class="muted-link" href="../_sources/modules/graphein.protein.rst.txt"
               rel="nofollow">
              Show Source
            </a>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            Contents
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">graphein.protein</a><ul>
<li><a class="reference internal" href="#module-graphein.protein.config">Config</a></li>
<li><a class="reference internal" href="#module-graphein.protein.graphs">Graphs</a></li>
<li><a class="reference internal" href="#edges">Edges</a><ul>
<li><a class="reference internal" href="#module-graphein.protein.edges.distance">Distance</a></li>
<li><a class="reference internal" href="#module-graphein.protein.edges.intramolecular">Intramolecular</a></li>
<li><a class="reference internal" href="#module-graphein.protein.edges.atomic">Atomic</a></li>
</ul>
</li>
<li><a class="reference internal" href="#features">Features</a><ul>
<li><a class="reference internal" href="#module-graphein.protein.features.nodes.aaindex">Node</a></li>
<li><a class="reference internal" href="#module-graphein.protein.features.sequence.embeddings">Sequence</a></li>
<li><a class="reference internal" href="#module-0">Sequence Utils</a></li>
<li><a class="reference internal" href="#module-graphein.protein.features.utils">Utils</a></li>
</ul>
</li>
<li><a class="reference internal" href="#module-graphein.protein.meshes">Meshes</a></li>
<li><a class="reference internal" href="#module-graphein.protein.visualisation">Visualisation</a></li>
<li><a class="reference internal" href="#id4">Utils</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/scripts/main.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>