<!doctype html>
<html class="no-js">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark">
        <script async src="https://www.googletagmanager.com/gtag/js?id=G-ZKD1FQDEYH"></script>
        <script>
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            
            gtag('config', 'G-ZKD1FQDEYH');
            
        </script>
    
<meta property="og:title" content="graphein.protein" />
  
<meta property="og:type" content="website" />
  
<meta property="og:url" content="https://graphein.ai/modules/graphein.protein.html" />
  
<meta property="og:description" content="Config: Base Config object for use with Protein Graph Construction. Graphs: Functions for working with Protein Structure Graphs. Edges: Distance: Functions for computing biochemical edges of graphs..." />
  
<meta property="og:image" content="https://graphein.ai/_static/graphein.png" />
  
<meta property="og:image:alt" content="graphein.protein" />
  <link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="graphein.rna" href="graphein.rna.html" /><link rel="prev" title="PPISP - Protein Protein Interaction Structure Prediction" href="../datasets/ppisp.html" />

    <meta name="generator" content="sphinx-4.4.0, furo 2022.02.14.1"/>
        <title>graphein.protein - Graphein 1.2.1</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?digest=a8f031f701c6af50c9919b48f173a5785af06fac" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/tabs.css" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?digest=25ceb02ed1c46dc30f2321ff83e92799f69dfdb9" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  body[data-theme="dark"] {
    --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
  }
  @media (prefers-color-scheme: dark) {
    body:not([data-theme="light"]) {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
  }
</style></head>
  <body>
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    
<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">Graphein 1.2.1</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand centered" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../_static/graphein.png" alt="Logo"/>
  </div>
  
  <span class="sidebar-brand-text">Graphein 1.2.1</span>
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder=Search name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/introduction.html">Graphein</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/usage.html">Usage</a></li>
<li class="toctree-l1"><a class="reference internal" href="../license.html">License</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1 has-children"><a class="reference internal" href="../protein_notebooks.html">Protein</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/residue_graphs.html">Graphein - Residue Graph Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/atom_graph_tutorial.html">Graphein - Atom Graph Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/alphafold_protein_graph_tutorial.html">Constructing Graphs from AlphaFold Protein Structure Database</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/subgraphing_tutorial.html">Constructing Subgraphs in Graphein</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/protein_mesh_tutorial.html">Creating Protein Meshes in Graphein &amp; 3D Visualisation</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/protein_graph_analytics.html">Protein Graph Analytics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/interactive_plotly_example.html">Interactive Visualisation with Plotly</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../rna_notebooks.html">RNA Graphs</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/rna_graph_tutorial.html">Graphein RNA Graph Construction Tutorial</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../ppi_notebooks.html">PPI Networks</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" role="switch" type="checkbox"/><label for="toctree-checkbox-3"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/ppi_tutorial.html">PPI Networks and Structural Interactomics</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../grn_notebooks.html">Gene Regulatory Networks</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" role="switch" type="checkbox"/><label for="toctree-checkbox-4"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/grn_tutorial.html">Gene Regulatory Network Tutorial</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Machine Learning</span></p>
<ul>
<li class="toctree-l1 has-children"><a class="reference internal" href="../datasets.html">Datasets</a><input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" role="switch" type="checkbox"/><label for="toctree-checkbox-5"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../datasets/pscdb.html">PSCDB - Protein Structural Change Database</a></li>
<li class="toctree-l2"><a class="reference internal" href="../notebooks/pscdb_baselines.html">PSCDB - Baselines</a></li>
<li class="toctree-l2"><a class="reference internal" href="../datasets/ppisp.html">PPISP - Protein Protein Interaction Structure Prediction</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul class="current">
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">graphein.protein</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.rna.html">graphein.rna</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.ppi.html">graphein.ppi</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.grn.html">graphein.grn</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.ml.html">graphein.ml</a></li>
<li class="toctree-l1"><a class="reference internal" href="graphein.utils.html">graphein.utils</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Contributing</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../contributing/contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributing/code_of_conduct.html">Contributor Covenant Code of Conduct</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contributing/contributors.html">Contributors</a></li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <div class="content-icon-container"><div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <div class="section" id="graphein-protein">
<h1>graphein.protein<a class="headerlink" href="#graphein-protein" title="Permalink to this headline">#</a></h1>
<div class="section" id="module-graphein.protein.config">
<span id="config"></span><h2>Config<a class="headerlink" href="#module-graphein.protein.config" title="Permalink to this headline">#</a></h2>
<p>Base Config object for use with Protein Graph Construction.</p>
<dl class="py class">
<dt class="sig sig-object py" id="graphein.protein.config.GetContactsConfig">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">GetContactsConfig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_contacts_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/pathlib.html#pathlib.Path" title="(in Python v3.10)"><span class="pre">pathlib.Path</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">PosixPath('/Users/arianjamasb/github/getcontacts')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">contacts_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/pathlib.html#pathlib.Path" title="(in Python v3.10)"><span class="pre">pathlib.Path</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">PosixPath('/Users/arianjamasb/github/graphein/examples/contacts')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/pathlib.html#pathlib.Path" title="(in Python v3.10)"><span class="pre">pathlib.Path</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">PosixPath('/Users/arianjamasb/github/graphein/examples/pdbs')</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'CA'</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/config.html#GetContactsConfig"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.GetContactsConfig" title="Permalink to this definition">#</a></dt>
<dd><p>Config object for parameters relating to running <code class="docutils literal notranslate"><span class="pre">GetContacts</span></code>.
<code class="docutils literal notranslate"><span class="pre">GetContacts</span></code> is an optional dependency from which intramolecular interactions can be computed and used as edges in the graph.</p>
<p>More information about <code class="docutils literal notranslate"><span class="pre">GetContacts</span></code> can be found at <a class="reference external" href="https://getcontacts.github.io/">https://getcontacts.github.io/</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>get_contacts_path</strong> (<a class="reference external" href="https://docs.python.org/3/library/pathlib.html#pathlib.Path" title="(in Python v3.10)"><em>pathlib.Path</em></a>) – Path to <code class="docutils literal notranslate"><span class="pre">GetContacts</span></code> installation</p></li>
<li><p><strong>contacts_dir</strong> (<a class="reference external" href="https://docs.python.org/3/library/pathlib.html#pathlib.Path" title="(in Python v3.10)"><em>pathlib.Path</em></a>) – Path to store output of <code class="docutils literal notranslate"><span class="pre">GetContacts</span></code></p></li>
<li><p><strong>pdb_dir</strong> (<a class="reference external" href="https://docs.python.org/3/library/pathlib.html#pathlib.Path" title="(in Python v3.10)"><em>pathlib.Path</em></a>) – Path to PDB files to be used to compute intramolecular interactions.</p></li>
<li><p><strong>granularity</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Specifies the node types of the graph, defaults to <code class="docutils literal notranslate"><span class="pre">"CA"</span></code> for alpha-carbons as nodes. Other options are <code class="docutils literal notranslate"><span class="pre">"CB"</span></code> (beta-carbon), <code class="docutils literal notranslate"><span class="pre">"atom"</span></code> for all-atom graphs, and <code class="docutils literal notranslate"><span class="pre">"centroid"</span></code> for nodes positioned as residue centroids.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.config.GranularityOpts">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">GranularityOpts</span></span><a class="headerlink" href="#graphein.protein.config.GranularityOpts" title="Permalink to this definition">#</a></dt>
<dd><p>Allowable granularity options for nodes in the graph.</p>
<p>alias of <a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Literal" title="(in Python v3.10)"><code class="xref py py-obj docutils literal notranslate"><span class="pre">Literal</span></code></a>[‘atom’, ‘centroids’]</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.config.GraphAtoms">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">GraphAtoms</span></span><a class="headerlink" href="#graphein.protein.config.GraphAtoms" title="Permalink to this definition">#</a></dt>
<dd><p>Allowable atom types for nodes in the graph.</p>
<p>alias of <a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Literal" title="(in Python v3.10)"><code class="xref py py-obj docutils literal notranslate"><span class="pre">Literal</span></code></a>[‘N’, ‘CA’, ‘C’, ‘O’, ‘CB’, ‘OG’, ‘CG’, ‘CD1’, ‘CD2’, ‘CE1’, ‘CE2’, ‘CZ’, ‘OD1’, ‘ND2’, ‘CG1’, ‘CG2’, ‘CD’, ‘CE’, ‘NZ’, ‘OD2’, ‘OE1’, ‘NE2’, ‘OE2’, ‘OH’, ‘NE’, ‘NH1’, ‘NH2’, ‘OG1’, ‘SD’, ‘ND1’, ‘SG’, ‘NE1’, ‘CE3’, ‘CZ2’, ‘CZ3’, ‘CH2’, ‘OXT’]</p>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="graphein.protein.config.PartialMatchOperator">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">PartialMatchOperator</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regex_paths</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">types</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/config.html#PartialMatchOperator"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.PartialMatchOperator" title="Permalink to this definition">#</a></dt>
<dd><p>Custom operator for deepdiff comparison. This operator compares whether the two partials are equal.</p>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="graphein.protein.config.PathMatchOperator">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">PathMatchOperator</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">regex_paths</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">types</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/config.html#PathMatchOperator"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.PathMatchOperator" title="Permalink to this definition">#</a></dt>
<dd><p>Custom operator for deepdiff comparison. This operator compares whether the two pathlib Paths are equal.</p>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="graphein.protein.config.ProteinGraphConfig">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">ProteinGraphConfig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="pre">*,</span> <span class="pre">granularity:</span> <span class="pre">typing.Union[typing.Literal['N',</span> <span class="pre">'CA',</span> <span class="pre">'C',</span> <span class="pre">'O',</span> <span class="pre">'CB',</span> <span class="pre">'OG',</span> <span class="pre">'CG',</span> <span class="pre">'CD1',</span> <span class="pre">'CD2',</span> <span class="pre">'CE1',</span> <span class="pre">'CE2',</span> <span class="pre">'CZ',</span> <span class="pre">'OD1',</span> <span class="pre">'ND2',</span> <span class="pre">'CG1',</span> <span class="pre">'CG2',</span> <span class="pre">'CD',</span> <span class="pre">'CE',</span> <span class="pre">'NZ',</span> <span class="pre">'OD2',</span> <span class="pre">'OE1',</span> <span class="pre">'NE2',</span> <span class="pre">'OE2',</span> <span class="pre">'OH',</span> <span class="pre">'NE',</span> <span class="pre">'NH1',</span> <span class="pre">'NH2',</span> <span class="pre">'OG1',</span> <span class="pre">'SD',</span> <span class="pre">'ND1',</span> <span class="pre">'SG',</span> <span class="pre">'NE1',</span> <span class="pre">'CE3',</span> <span class="pre">'CZ2',</span> <span class="pre">'CZ3',</span> <span class="pre">'CH2',</span> <span class="pre">'OXT'],</span> <span class="pre">typing.Literal['atom',</span> <span class="pre">'centroids']]</span> <span class="pre">=</span> <span class="pre">'CA',</span> <span class="pre">keep_hets:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False,</span> <span class="pre">insertions:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False,</span> <span class="pre">pdb_dir:</span> <span class="pre">pathlib.Path</span> <span class="pre">=</span> <span class="pre">PosixPath('../examples/pdbs'),</span> <span class="pre">verbose:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False,</span> <span class="pre">exclude_waters:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">True,</span> <span class="pre">deprotonate:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False,</span> <span class="pre">protein_df_processing_functions:</span> <span class="pre">typing.List[typing.Callable]</span> <span class="pre">=</span> <span class="pre">None,</span> <span class="pre">edge_construction_functions:</span> <span class="pre">typing.List[typing.Union[typing.Callable,</span> <span class="pre">str]]</span> <span class="pre">=</span> <span class="pre">[&lt;function</span> <span class="pre">add_peptide_bonds&gt;],</span> <span class="pre">node_metadata_functions:</span> <span class="pre">typing.List[typing.Union[typing.Callable,</span> <span class="pre">str]]</span> <span class="pre">=</span> <span class="pre">[&lt;function</span> <span class="pre">meiler_embedding&gt;],</span> <span class="pre">edge_metadata_functions:</span> <span class="pre">typing.List[typing.Union[typing.Callable,</span> <span class="pre">str]]</span> <span class="pre">=</span> <span class="pre">None,</span> <span class="pre">graph_metadata_functions:</span> <span class="pre">typing.List[typing.Callable]</span> <span class="pre">=</span> <span class="pre">None,</span> <span class="pre">get_contacts_config:</span> <span class="pre">graphein.protein.config.GetContactsConfig</span> <span class="pre">=</span> <span class="pre">None,</span> <span class="pre">dssp_config:</span> <span class="pre">graphein.protein.config.DSSPConfig</span> <span class="pre">=</span> <span class="pre">None</span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/config.html#ProteinGraphConfig"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.ProteinGraphConfig" title="Permalink to this definition">#</a></dt>
<dd><p>Config Object for Protein Structure Graph Construction.</p>
<p>If you encounter a problematic structure, perusing <a class="reference external" href="https://www.umass.edu/microbio/chime/pe_beta/pe/protexpl/badpdbs.htm">https://www.umass.edu/microbio/chime/pe_beta/pe/protexpl/badpdbs.htm</a> may provide some additional insight.
PDBs are notoriously troublesome and this is an excellent overview.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>granularity</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em> (</em><em>Union</em><em>[</em><em>graphein.protein.config.GraphAtoms</em><em>, </em><em>graphein.protein.config.GranularityOpts</em><em>]</em><em>)</em>) – Controls the granularity of the graph construction. <code class="docutils literal notranslate"><span class="pre">"atom"</span></code> builds an atomic-scale graph where
nodes are constituent atoms. Residue-level graphs can be build by specifying which constituent atom should
represent node positions (see <a class="reference internal" href="#graphein.protein.config.GraphAtoms" title="graphein.protein.config.GraphAtoms"><code class="xref py py-const docutils literal notranslate"><span class="pre">GraphAtoms</span></code></a>). Additionally, <code class="docutils literal notranslate"><span class="pre">"centroids"</span></code> can be specified to
compute the centre of gravity for a given atom (Specified in <a class="reference internal" href="#graphein.protein.config.GranularityOpts" title="graphein.protein.config.GranularityOpts"><code class="xref py py-const docutils literal notranslate"><span class="pre">GranularityOpts</span></code></a>).
Defaults to <code class="docutils literal notranslate"><span class="pre">"CA"</span></code> (alpha-Carbon).</p></li>
<li><p><strong>keep_hets</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – <p>Controls whether or not heteroatoms are removed from the PDB file. These are typically modified
residues, bound ligands, crystallographic adjuvants, ions or water molecules.</p>
<p>For more information, see: <a class="reference external" href="https://proteopedia.org/wiki/index.php/Hetero_atoms">https://proteopedia.org/wiki/index.php/Hetero_atoms</a></p>
</p></li>
<li><p><strong>insertions</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Controls whether or not insertions are allowed.</p></li>
<li><p><strong>pdb_dir</strong> (<a class="reference external" href="https://docs.python.org/3/library/pathlib.html#pathlib.Path" title="(in Python v3.10)"><em>pathlib.Path</em></a>) – Specifies path to download protein structures into.</p></li>
<li><p><strong>verbose</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Specifies verbosity of graph creation process.</p></li>
<li><p><strong>exclude_waters</strong> – Specifies whether or not water molecules are excluded from the structure</p></li>
<li><p><strong>deprotonate</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Specifies whether or not to remove <code class="docutils literal notranslate"><span class="pre">H</span></code> atoms from the graph.</p></li>
<li><p><strong>protein_df_processing_functions</strong> (<em>Optional</em><em>[</em><em>List</em><em>[</em><em>Callable</em><em>]</em><em>]</em>) – List of functions that take a <code class="docutils literal notranslate"><span class="pre">pd.DataFrame</span></code> and return a <code class="docutils literal notranslate"><span class="pre">pd.DataFrame</span></code>.
This allows users to define their own series of processing functions for the protein structure DataFrame and
override the default sequencing of processing steps provided by Graphein. We refer users to our low-level API
tutorial for more details.</p></li>
<li><p><strong>edge_construction_functions</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em>) – List of functions that take an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> and return an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> with desired
edges added. Prepared edge constructions can be found in <span class="xref std std-ref">graphein.protein.edges</span></p></li>
<li><p><strong>node_metadata_functions</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions that take an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code></p></li>
<li><p><strong>edge_metadata_functions</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions that take an</p></li>
<li><p><strong>graph_metadata_functions</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions that take an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> and return an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> with added
graph-level features and metadata.</p></li>
<li><p><strong>get_contacts_config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>GetContactsConfig</em></a><em>, </em><em>optional</em>) – Config object containing parameters for running <code class="docutils literal notranslate"><span class="pre">GetContacts</span></code> for computing intramolecular
contact-based edges. Defaults to None.</p></li>
<li><p><strong>dssp_config</strong> (<em>DSSPConfig</em><em>, </em><em>optional</em>) – Config Object containing reference to <code class="docutils literal notranslate"><span class="pre">DSSP</span></code> executable. Defaults to None.
<strong>NB</strong> DSSP must be installed. See installation instructions: <a class="reference external" href="https://graphein.ai/getting_started/installation.html#optional-dependencies">https://graphein.ai/getting_started/installation.html#optional-dependencies</a></p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py class">
<dt class="sig sig-object py" id="graphein.protein.config.ProteinMeshConfig">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">ProteinMeshConfig</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="o"><span class="pre">*</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pymol_command_line_options</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'-cKq'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pymol_commands</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">['show</span> <span class="pre">surface']</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/config.html#ProteinMeshConfig"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.ProteinMeshConfig" title="Permalink to this definition">#</a></dt>
<dd><p>Config object for parameters relating to Protein Mesh construction with <code class="docutils literal notranslate"><span class="pre">PyMol</span></code></p>
<p><strong>NB</strong> PyMol must be installed. See: <a class="reference external" href="https://graphein.ai/getting_started/installation.html#optional-dependencies">https://graphein.ai/getting_started/installation.html#optional-dependencies</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pymol_command_line_options</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – List of CLI args for running PyMol.
See: <a class="reference external" href="https://pymolwiki.org/index.php/Command_Line_Options">https://pymolwiki.org/index.php/Command_Line_Options</a>. Defaults to <code class="docutils literal notranslate"><span class="pre">"-cKq"</span></code> ()</p></li>
<li><p><strong>pymol_commands</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – List of Commands passed to PyMol in surface construction.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.config.partial_functions_equal">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.config.</span></span><span class="sig-name descname"><span class="pre">partial_functions_equal</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">func1</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">functools.partial</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">func2</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">functools.partial</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/config.html#partial_functions_equal"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.config.partial_functions_equal" title="Permalink to this definition">#</a></dt>
<dd><p>Determine whether two partial functions are equal.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>func1</strong> (<em>partial</em>) – Partial function to check</p></li>
<li><p><strong>func2</strong> (<em>partial</em>) – Partial function to check</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Whether the two functions are equal</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)">bool</a></p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.graphs">
<span id="graphs"></span><h2>Graphs<a class="headerlink" href="#module-graphein.protein.graphs" title="Permalink to this headline">#</a></h2>
<p>Functions for working with Protein Structure Graphs.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.add_nodes_to_graph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">add_nodes_to_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#add_nodes_to_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.add_nodes_to_graph" title="Permalink to this definition">#</a></dt>
<dd><p>Add nodes into protein graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> with metadata to populate with nodes.</p></li>
<li><p><strong>verbose</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Controls verbosity of this step.</p></li>
</ul>
</dd>
<dt class="field-even">Protein_df</dt>
<dd class="field-even"><p>DataFrame of protein structure containing nodes &amp; initial node metadata to add to the graph.</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>nx.Graph with nodes added.</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.assign_node_id_to_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">assign_node_id_to_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#assign_node_id_to_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.assign_node_id_to_dataframe" title="Permalink to this definition">#</a></dt>
<dd><p>Assigns the node ID back to the <code class="docutils literal notranslate"><span class="pre">pdb_df</span></code> dataframe</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>protein_df</strong> (<em>pd.DataFrame</em>) – Structure Dataframe</p></li>
<li><p><strong>granularity</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Granularity of graph. Atom-level, residue (e.g. <code class="docutils literal notranslate"><span class="pre">CA</span></code>) or <code class="docutils literal notranslate"><span class="pre">centroids</span></code>
See: <code class="xref py py-const docutils literal notranslate"><span class="pre">GRAPH_ATOMS</span></code> and <code class="xref py py-const docutils literal notranslate"><span class="pre">GRANULARITY_OPTS</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Returns dataframe with added <code class="docutils literal notranslate"><span class="pre">node_ids</span></code></p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.calculate_centroid_positions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">calculate_centroid_positions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">atoms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#calculate_centroid_positions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.calculate_centroid_positions" title="Permalink to this definition">#</a></dt>
<dd><p>Calculates position of sidechain centroids.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>atoms</strong> (<em>pd.DataFrame</em>) – ATOM df of protein structure.</p></li>
<li><p><strong>verbose</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – bool controlling verbosity.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>centroids (df).</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.compute_edges">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">compute_edges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">funcs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">get_contacts_config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><span class="pre">graphein.protein.config.GetContactsConfig</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#compute_edges"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.compute_edges" title="Permalink to this definition">#</a></dt>
<dd><p>Computes edges for the protein structure graph. Will compute an pairwise distance matrix between nodes which is
added to the graph metadata to facilitate some edge computations.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph with nodes to add edges to.</p></li>
<li><p><strong>funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em>) – List of edge construction functions.</p></li>
<li><p><strong>get_contacts_config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>graphein.protein.config.GetContactsConfig</em></a>) – Config object for <code class="docutils literal notranslate"><span class="pre">GetContacts</span></code> if intramolecular edges are being used.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Graph with added edges.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.construct_graph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">construct_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.ProteinGraphConfig" title="graphein.protein.config.ProteinGraphConfig"><span class="pre">graphein.protein.config.ProteinGraphConfig</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_selection</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'all'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">df_processing_funcs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_construction_funcs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_annotation_funcs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_annotation_funcs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">graph_annotation_funcs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#construct_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.construct_graph" title="Permalink to this definition">#</a></dt>
<dd><p>Constructs protein structure graph from a <code class="docutils literal notranslate"><span class="pre">pdb_code</span></code> or <code class="docutils literal notranslate"><span class="pre">pdb_path</span></code>. Users can provide a <a class="reference internal" href="#graphein.protein.config.ProteinGraphConfig" title="graphein.protein.config.ProteinGraphConfig"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinGraphConfig</span></code></a> object to specify construction parameters.</p>
<p>However, config parameters can be overridden by passing arguments directly to the function.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinGraphConfig" title="graphein.protein.config.ProteinGraphConfig"><em>graphein.protein.config.ProteinGraphConfig</em></a><em>, </em><em>optional</em>) – <a class="reference internal" href="#graphein.protein.config.ProteinGraphConfig" title="graphein.protein.config.ProteinGraphConfig"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinGraphConfig</span></code></a> object. If None, defaults to config in <code class="docutils literal notranslate"><span class="pre">graphein.protein.config</span></code>.</p></li>
<li><p><strong>pdb_path</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – Path to <code class="docutils literal notranslate"><span class="pre">pdb_file</span></code> to build graph from. Default is <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>pdb_code</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – 4-character PDB accession pdb_code to build graph from. Default is <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>chain_selection</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – String of polypeptide chains to include in graph. E.g <code class="docutils literal notranslate"><span class="pre">"ABDF"</span></code> or <code class="docutils literal notranslate"><span class="pre">"all"</span></code>. Default is <code class="docutils literal notranslate"><span class="pre">"all"</span></code>.</p></li>
<li><p><strong>df_processing_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of dataframe processing functions. Default is <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>edge_construction_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of edge construction functions. Default is <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>edge_annotation_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of edge annotation functions. Default is <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>node_annotation_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of node annotation functions. Default is <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>graph_annotation_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em>) – List of graph annotation function. Default is <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Structure Graph</p>
</dd>
<dt class="field-odd">Type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.convert_structure_to_centroids">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">convert_structure_to_centroids</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#convert_structure_to_centroids"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.convert_structure_to_centroids" title="Permalink to this definition">#</a></dt>
<dd><p>Overwrite existing <code class="docutils literal notranslate"><span class="pre">(x,</span> <span class="pre">y,</span> <span class="pre">z)</span></code> coordinates with centroids of the amino acids.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Pandas Dataframe protein structure to convert into a dataframe of centroid positions.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.DataFrame with atoms/residues positions converted into centroid positions.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.deprotonate_structure">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">deprotonate_structure</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#deprotonate_structure"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.deprotonate_structure" title="Permalink to this definition">#</a></dt>
<dd><p>Remove protons from PDB dataframe.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Atomic dataframe.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Atomic dataframe with all <code class="docutils literal notranslate"><span class="pre">atom_name</span> <span class="pre">==</span> <span class="pre">"H"</span></code> removed.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.filter_hetatms">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">filter_hetatms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">keep_hets</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#filter_hetatms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.filter_hetatms" title="Permalink to this definition">#</a></dt>
<dd><p>Return hetatms of interest.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>df</strong> (<em>pd.DataFrame</em>) – Protein Structure dataframe to filter hetatoms from.</p></li>
<li><p><strong>keep_hets</strong> – List of hetero atom names to keep.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein structure dataframe with heteroatoms removed</p>
</dd>
</dl>
<p>:rtype pd.DataFrame</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.initialise_graph_with_metadata">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">initialise_graph_with_metadata</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">raw_pdb_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#initialise_graph_with_metadata"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.initialise_graph_with_metadata" title="Permalink to this definition">#</a></dt>
<dd><p>Initializes the nx Graph object with initial metadata.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>protein_df</strong> (<em>pd.DataFrame</em>) – Processed Dataframe of protein structure.</p></li>
<li><p><strong>raw_pdb_df</strong> (<em>pd.DataFrame</em>) – Unprocessed dataframe of protein structure for comparison and traceability downstream.</p></li>
<li><p><strong>pdb_id</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – PDB Accession code.</p></li>
<li><p><strong>granularity</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Granularity of the graph (eg <code class="docutils literal notranslate"><span class="pre">"atom"</span></code>, <code class="docutils literal notranslate"><span class="pre">"CA"</span></code>, <code class="docutils literal notranslate"><span class="pre">"CB"</span></code> etc or <code class="docutils literal notranslate"><span class="pre">"centroid"</span></code>).
See: <code class="xref py py-const docutils literal notranslate"><span class="pre">GRAPH_ATOMS</span></code> and <code class="xref py py-const docutils literal notranslate"><span class="pre">GRANULARITY_OPTS</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Returns initial protein structure graph with metadata.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.process_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">process_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_df_processing_funcs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">hetatom_df_processing_funcs</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'centroids'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_selection</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'all'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">insertions</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">deprotonate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">keep_hets</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">[]</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#process_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.process_dataframe" title="Permalink to this definition">#</a></dt>
<dd><p>Process ATOM and HETATM dataframes to produce singular dataframe used for graph construction.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>protein_df</strong> (<em>pd.DataFrame</em>) – Dataframe to process.
Should be the object returned from <a class="reference internal" href="#graphein.protein.graphs.read_pdb_to_dataframe" title="graphein.protein.graphs.read_pdb_to_dataframe"><code class="xref py py-func docutils literal notranslate"><span class="pre">read_pdb_to_dataframe()</span></code></a>.</p></li>
<li><p><strong>atom_df_processing_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions to process dataframe. These must take in a dataframe and return a
dataframe. Defaults to None.</p></li>
<li><p><strong>hetatom_df_processing_funcs</strong> (<em>List</em><em>[</em><em>Callable</em><em>]</em><em>, </em><em>optional</em>) – List of functions to process the hetatpm dataframe. These must take in a dataframe and return a dataframe</p></li>
<li><p><strong>granularity</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – The level of granularity for the graph. This determines the node definition.
Acceptable values include: <code class="docutils literal notranslate"><span class="pre">"centroids"</span></code>, <code class="docutils literal notranslate"><span class="pre">"atoms"</span></code>,
any of the atom_names in the PDB file (e.g. <code class="docutils literal notranslate"><span class="pre">"CA"</span></code>, <code class="docutils literal notranslate"><span class="pre">"CB"</span></code>, <code class="docutils literal notranslate"><span class="pre">"OG"</span></code>, etc.).
See: <code class="xref py py-const docutils literal notranslate"><span class="pre">GRAPH_ATOMS</span></code> and <code class="xref py py-const docutils literal notranslate"><span class="pre">GRANULARITY_OPTS</span></code>.</p></li>
<li><p><strong>insertions</strong> – Whether or not to keep insertions.</p></li>
<li><p><strong>insertions</strong> – bool</p></li>
<li><p><strong>deprotonate</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether or not to remove hydrogen atoms (i.e. deprotonation).</p></li>
<li><p><strong>keep_hets</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – Hetatoms to keep. Defaults to an empty list.
To keep a hetatom, pass it inside a list of hetatom names to keep.</p></li>
<li><p><strong>verbose</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Verbosity level.</p></li>
<li><p><strong>chain_selection</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Which protein chain to select. Defaults to <code class="docutils literal notranslate"><span class="pre">"all"</span></code>. Eg can use <code class="docutils literal notranslate"><span class="pre">"ACF"</span></code>
to select 3 chains (<code class="docutils literal notranslate"><span class="pre">A</span></code>, <code class="docutils literal notranslate"><span class="pre">C</span></code> &amp; <code class="docutils literal notranslate"><span class="pre">F</span></code>)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>A protein dataframe that can be consumed by
other graph construction functions.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.read_pdb_to_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">read_pdb_to_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'CA'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#read_pdb_to_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.read_pdb_to_dataframe" title="Permalink to this definition">#</a></dt>
<dd><p>Reads PDB file to <code class="docutils literal notranslate"><span class="pre">PandasPDB</span></code> object.</p>
<p>Returns <code class="docutils literal notranslate"><span class="pre">atomic_df</span></code>, which is a dataframe enumerating all atoms and their cartesian coordinates in 3D space. Also
contains associated metadata from the PDB file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb_path</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – path to PDB file. Defaults to None.</p></li>
<li><p><strong>pdb_code</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – 4-character PDB accession. Defaults to None.</p></li>
<li><p><strong>verbose</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – print dataframe?</p></li>
<li><p><strong>granularity</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Specifies granularity of dataframe. See <a class="reference internal" href="#graphein.protein.config.ProteinGraphConfig" title="graphein.protein.config.ProteinGraphConfig"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinGraphConfig</span></code></a> for further
details.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">pd.DataFrame</span></code> containing protein structure</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.remove_insertions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">remove_insertions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">keep</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'first'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#remove_insertions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.remove_insertions" title="Permalink to this definition">#</a></dt>
<dd><p>This function removes insertions from PDB dataframes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>df</strong> (<em>pd.DataFrame</em>) – Protein Structure dataframe to remove insertions from.</p></li>
<li><p><strong>keep</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Specifies which insertion to keep. Options are <code class="docutils literal notranslate"><span class="pre">"first"</span></code> or <code class="docutils literal notranslate"><span class="pre">"last"</span></code>.
Default is <code class="docutils literal notranslate"><span class="pre">"first"</span></code></p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein structure dataframe with insertions removed</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.select_chains">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">select_chains</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">protein_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">chain_selection</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">verbose</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#select_chains"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.select_chains" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts relevant chains from <code class="docutils literal notranslate"><span class="pre">protein_df</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>protein_df</strong> (<em>pd.DataFrame</em>) – pandas dataframe of PDB subsetted to relevant atoms (<code class="docutils literal notranslate"><span class="pre">CA</span></code>, <code class="docutils literal notranslate"><span class="pre">CB</span></code>).</p></li>
<li><p><strong>chain_selection</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Specifies chains that should be extracted from the larger complexed structure.</p></li>
<li><p><strong>verbose</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Print dataframe?</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein structure dataframe containing only entries in the chain selection.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.graphs.subset_structure_to_atom_type">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.graphs.</span></span><span class="sig-name descname"><span class="pre">subset_structure_to_atom_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">granularity</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/graphs.html#subset_structure_to_atom_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.graphs.subset_structure_to_atom_type" title="Permalink to this definition">#</a></dt>
<dd><p>Return a subset of atomic dataframe that contains only certain atom names.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Protein Structure dataframe to subset.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Subsetted protein structure dataframe.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="edges">
<h2>Edges<a class="headerlink" href="#edges" title="Permalink to this headline">#</a></h2>
<div class="section" id="module-graphein.protein.edges.distance">
<span id="distance"></span><h3>Distance<a class="headerlink" href="#module-graphein.protein.edges.distance" title="Permalink to this headline">#</a></h3>
<p>Functions for computing biochemical edges of graphs.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_aromatic_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_aromatic_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_aromatic_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_aromatic_interactions" title="Permalink to this definition">#</a></dt>
<dd><p>Find all aromatic-aromatic interaction.</p>
<p>Criteria: phenyl ring centroids separated between 4.5A to 7A.
Phenyl rings are present on <code class="docutils literal notranslate"><span class="pre">PHE,</span> <span class="pre">TRP,</span> <span class="pre">HIS,</span> <span class="pre">TYR</span></code> (<a class="reference internal" href="#graphein.protein.resi_atoms.AROMATIC_RESIS" title="graphein.protein.resi_atoms.AROMATIC_RESIS"><code class="xref py py-const docutils literal notranslate"><span class="pre">AROMATIC_RESIS</span></code></a>).
Phenyl ring atoms on these amino acids are defined by the following
atoms:
- PHE: CG, CD, CE, CZ
- TRP: CD, CE, CH, CZ
- HIS: CG, CD, ND, NE, CE
- TYR: CG, CD, CE, CZ
Centroids of these atoms are taken by taking:</p>
<blockquote>
<div><p>(mean x), (mean y), (mean z)</p>
</div></blockquote>
<p>for each of the ring atoms.
Notes for future self/developers:
- Because of the requirement to pre-compute ring centroids, we do not</p>
<blockquote>
<div><p>use the functions written above (filter_dataframe, compute_distmat,
get_interacting_atoms), as they do not return centroid atom
euclidean coordinates.</p>
</div></blockquote>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_aromatic_sulphur_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_aromatic_sulphur_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_aromatic_sulphur_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_aromatic_sulphur_interactions" title="Permalink to this definition">#</a></dt>
<dd><p>Find all aromatic-sulphur interactions.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_cation_pi_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_cation_pi_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_cation_pi_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_cation_pi_interactions" title="Permalink to this definition">#</a></dt>
<dd><p>Add cation-pi interactions.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_delaunay_triangulation">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_delaunay_triangulation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">allowable_nodes</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_delaunay_triangulation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_delaunay_triangulation" title="Permalink to this definition">#</a></dt>
<dd><p>Compute the Delaunay triangulation of the protein structure.</p>
<p>This has been used in prior work. References:</p>
<blockquote>
<div><p>Harrison, R. W., Yu, X. &amp; Weber, I. T. Using triangulation to include
target structure improves drug resistance prediction accuracy. in 1–1
(IEEE, 2013). doi:10.1109/ICCABS.2013.6629236</p>
<p>Yu, X., Weber, I. T. &amp; Harrison, R. W. Prediction of HIV drug
resistance from genotype with encoded three-dimensional protein
structure. BMC Genomics 15 Suppl 5, S1 (2014).</p>
</div></blockquote>
<p>Notes:
1. We do not use the add_interacting_resis function, because this</p>
<blockquote>
<div><p>interaction is computed on the <code class="docutils literal notranslate"><span class="pre">CA</span></code> atoms. Therefore, there is code
duplication. For now, I have chosen to leave this code duplication
in.</p>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – The networkx graph to add the triangulation to.</p></li>
<li><p><strong>allowable_nodes</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – The nodes to include in the triangulation. If <code class="docutils literal notranslate"><span class="pre">None</span></code> (default), no filtering is done.
This parameter is used to filter out nodes that are not desired in the triangulation.
Eg if you wanted to construct a delaunay triangulation of the CA atoms of an atomic graph.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_distance_threshold">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_distance_threshold</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">long_interaction_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5.0</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_distance_threshold"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_distance_threshold" title="Permalink to this definition">#</a></dt>
<dd><p>Adds edges to any nodes within a given distance of each other. Long interaction threshold is used
to specify minimum separation in sequence to add an edge between networkx nodes within the distance threshold</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Structure graph to add distance edges to</p></li>
<li><p><strong>long_interaction_threshold</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – minimum distance in sequence for two nodes to be connected</p></li>
<li><p><strong>threshold</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Distance in angstroms, below which two nodes are connected</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Graph with distance-based edges added</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_disulfide_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_disulfide_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_disulfide_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_disulfide_interactions" title="Permalink to this definition">#</a></dt>
<dd><p>Find all disulfide interactions between CYS residues (<a class="reference internal" href="#graphein.protein.resi_atoms.DISULFIDE_RESIS" title="graphein.protein.resi_atoms.DISULFIDE_RESIS"><code class="xref py py-const docutils literal notranslate"><span class="pre">DISULFIDE_RESIS</span></code></a>, <a class="reference internal" href="#graphein.protein.resi_atoms.DISULFIDE_ATOMS" title="graphein.protein.resi_atoms.DISULFIDE_ATOMS"><code class="xref py py-const docutils literal notranslate"><span class="pre">DISULFIDE_ATOMS</span></code></a>).</p>
<p>Criteria: sulfur atom pairs are within 2.2A of each other.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – networkx protein graph</p></li>
<li><p><strong>rgroup_df</strong> (<em>pd.DataFrame</em><em>, </em><em>optional</em>) – pd.DataFrame containing rgroup data, defaults to None, which retrieves the df from the provided nx graph.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_hydrogen_bond_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_hydrogen_bond_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_hydrogen_bond_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_hydrogen_bond_interactions" title="Permalink to this definition">#</a></dt>
<dd><p>Add all hydrogen-bond interactions.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_hydrophobic_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_hydrophobic_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_hydrophobic_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_hydrophobic_interactions" title="Permalink to this definition">#</a></dt>
<dd><p>Find all hydrophobic interactions.</p>
<p>Performs searches between the following residues:
<code class="docutils literal notranslate"><span class="pre">[ALA,</span> <span class="pre">VAL,</span> <span class="pre">LEU,</span> <span class="pre">ILE,</span> <span class="pre">MET,</span> <span class="pre">PHE,</span> <span class="pre">TRP,</span> <span class="pre">PRO,</span> <span class="pre">TYR]</span></code> (<a class="reference internal" href="#graphein.protein.resi_atoms.HYDROPHOBIC_RESIS" title="graphein.protein.resi_atoms.HYDROPHOBIC_RESIS"><code class="xref py py-const docutils literal notranslate"><span class="pre">HYDROPHOBIC_RESIS</span></code></a>).</p>
<p>Criteria: R-group residues are within 5A distance.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add hydrophobic interactions to.</p></li>
<li><p><strong>rgroup_df</strong> (<em>pd.DataFrame</em><em>, </em><em>optional</em>) – Optional dataframe of R-group atoms.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_interacting_resis">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_interacting_resis</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interacting_atoms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.22)"><span class="pre">numpy.ndarray</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">kind</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_interacting_resis"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_interacting_resis" title="Permalink to this definition">#</a></dt>
<dd><p>Add interacting residues to graph.</p>
<p>Returns a list of 2-tuples indicating the interacting residues based
on the interacting atoms. This is most typically called after the
get_interacting_atoms function above.</p>
<p>Also filters out the list such that the residues have to be at least
two apart.</p>
<p>### Parameters</p>
<ul class="simple">
<li><p>interacting_atoms:    (numpy array) result from get_interacting_atoms function.</p></li>
<li><dl class="simple">
<dt>dataframe:            (pandas dataframe) a pandas dataframe that</dt><dd><p>houses the euclidean locations of each atom.</p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>kind:                 (list) the kind of interaction. Contains one</dt><dd><p>of :
- hydrophobic
- disulfide
- hbond
- ionic
- aromatic
- aromatic_sulphur
- cation_pi
- delaunay</p>
</dd>
</dl>
</li>
</ul>
<ul class="simple">
<li><dl class="simple">
<dt>filtered_interacting_resis: (set of tuples) the residues that are in</dt><dd><p>an interaction, with the interaction kind specified</p>
</dd>
</dl>
</li>
</ul>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_ionic_interactions">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_ionic_interactions</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rgroup_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_ionic_interactions"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_ionic_interactions" title="Permalink to this definition">#</a></dt>
<dd><p>Find all ionic interactions.</p>
<p>Criteria: <code class="docutils literal notranslate"><span class="pre">[ARG,</span> <span class="pre">LYS,</span> <span class="pre">HIS,</span> <span class="pre">ASP,</span> <span class="pre">and</span> <span class="pre">GLU]</span></code> (<a class="reference internal" href="#graphein.protein.resi_atoms.IONIC_RESIS" title="graphein.protein.resi_atoms.IONIC_RESIS"><code class="xref py py-const docutils literal notranslate"><span class="pre">IONIC_RESIS</span></code></a>) residues are within 6A.
We also check for opposing charges (<a class="reference internal" href="#graphein.protein.resi_atoms.POS_AA" title="graphein.protein.resi_atoms.POS_AA"><code class="xref py py-const docutils literal notranslate"><span class="pre">POS_AA</span></code></a>, <a class="reference internal" href="#graphein.protein.resi_atoms.NEG_AA" title="graphein.protein.resi_atoms.NEG_AA"><code class="xref py py-const docutils literal notranslate"><span class="pre">NEG_AA</span></code></a>)</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_k_nn_edges">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_k_nn_edges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">long_interaction_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">k</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mode</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'connectivity'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">metric</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'minkowski'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">include_self</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_k_nn_edges"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_k_nn_edges" title="Permalink to this definition">#</a></dt>
<dd><p>Adds edges to nodes based on K nearest neighbours. Long interaction threshold is used
to specify minimum separation in sequence to add an edge between networkx nodes within the distance threshold</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Structure graph to add distance edges to</p></li>
<li><p><strong>long_interaction_threshold</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – minimum distance in sequence for two nodes to be connected</p></li>
<li><p><strong>k</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – Number of neighbors for each sample.</p></li>
<li><p><strong>mode</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Type of returned matrix: <code class="docutils literal notranslate"><span class="pre">"connectivity"</span></code> will return the connectivity matrix with ones and zeros,
and <code class="docutils literal notranslate"><span class="pre">"distance"</span></code> will return the distances between neighbors according to the given metric.</p></li>
<li><p><strong>metric</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – The distance metric used to calculate the k-Neighbors for each sample point.
The DistanceMetric class gives a list of available metrics.
The default distance is <code class="docutils literal notranslate"><span class="pre">"euclidean"</span></code> (<code class="docutils literal notranslate"><span class="pre">"minkowski"</span></code> metric with the <code class="docutils literal notranslate"><span class="pre">p</span></code> param equal to <code class="docutils literal notranslate"><span class="pre">2</span></code>).</p></li>
<li><p><strong>p</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – Power parameter for the Minkowski metric. When <code class="docutils literal notranslate"><span class="pre">p</span> <span class="pre">=</span> <span class="pre">1</span></code>, this is equivalent to using <code class="docutils literal notranslate"><span class="pre">manhattan_distance</span></code> (l1),
and <code class="docutils literal notranslate"><span class="pre">euclidean_distance</span></code> (l2) for <code class="docutils literal notranslate"><span class="pre">p</span> <span class="pre">=</span> <span class="pre">2</span></code>. For arbitrary <code class="docutils literal notranslate"><span class="pre">p</span></code>, <code class="docutils literal notranslate"><span class="pre">minkowski_distance</span></code> (l_p) is used. Default is <code class="docutils literal notranslate"><span class="pre">2</span></code> (euclidean).</p></li>
<li><p><strong>include_self</strong> (<em>Union</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – Whether or not to mark each sample as the first nearest neighbor to itself.
If <code class="docutils literal notranslate"><span class="pre">"auto"</span></code>, then <code class="docutils literal notranslate"><span class="pre">True</span></code> is used for <code class="docutils literal notranslate"><span class="pre">mode="connectivity"</span></code> and <code class="docutils literal notranslate"><span class="pre">False</span></code> for <code class="docutils literal notranslate"><span class="pre">mode="distance"</span></code>. Default is <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Graph with knn-based edges added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.add_peptide_bonds">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">add_peptide_bonds</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#add_peptide_bonds"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.add_peptide_bonds" title="Permalink to this definition">#</a></dt>
<dd><p>Adds peptide backbone as edges to residues in each chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – networkx protein graph.</p>
</dd>
<dt class="field-even">Return G</dt>
<dd class="field-even"><p>networkx protein graph with added peptide bonds.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.compute_distmat">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">compute_distmat</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#compute_distmat"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.compute_distmat" title="Permalink to this definition">#</a></dt>
<dd><p>Compute pairwise euclidean distances between every atom.</p>
<p>Design choice: passed in a DataFrame to enable easier testing on
dummy data.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pdb_df</strong> (<em>pd.DataFrame</em>) – pd.Dataframe containing protein structure. Must contain columns [“x_coord”, “y_coord”, “z_coord”]</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.Dataframe of euclidean distance matrix</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.get_edges_by_bond_type">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">get_edges_by_bond_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bond_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.10)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#get_edges_by_bond_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.get_edges_by_bond_type" title="Permalink to this definition">#</a></dt>
<dd><p>Return edges of a particular bond type.</p>
<ul class="simple">
<li><p>bond_type: (str) one of the elements in the variable BOND_TYPES</p></li>
</ul>
<ul class="simple">
<li><p>resis: (list) a list of tuples, where each tuple is an edge.</p></li>
</ul>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.get_interacting_atoms">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">get_interacting_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">angstroms</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#get_interacting_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.get_interacting_atoms" title="Permalink to this definition">#</a></dt>
<dd><p>Find the atoms that are within a particular radius of one another.</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.get_ring_atoms">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">get_ring_atoms</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">aa</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#get_ring_atoms"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.get_ring_atoms" title="Permalink to this definition">#</a></dt>
<dd><p>Return ring atoms from a dataframe.</p>
<p>A helper function for add_aromatic_interactions.</p>
<p>Gets the ring atoms from the particular aromatic amino acid.</p>
<ul class="simple">
<li><p>dataframe: the dataframe containing the atom records.</p></li>
<li><p>aa: the amino acid of interest, passed in as 3-letter string.</p></li>
</ul>
<ul class="simple">
<li><dl class="simple">
<dt>dataframe: a filtered dataframe containing just those atoms from the</dt><dd><p>particular amino acid selected. e.g. equivalent to
selecting just the ring atoms from a particular amino
acid.</p>
</dd>
</dl>
</li>
</ul>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.get_ring_centroids">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">get_ring_centroids</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">ring_atom_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#get_ring_centroids"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.get_ring_centroids" title="Permalink to this definition">#</a></dt>
<dd><p>Return aromatic ring centrods.</p>
<p>A helper function for add_aromatic_interactions.</p>
<p>Computes the ring centroids for each a particular amino acid’s ring
atoms.</p>
<p>Ring centroids are computed by taking the mean of the x, y, and z
coordinates.</p>
<ul class="simple">
<li><p>ring_atom_df: a dataframe computed using get_ring_atoms.</p></li>
<li><p>aa: the amino acid under study</p></li>
</ul>
<ul class="simple">
<li><dl class="simple">
<dt>centroid_df: a dataframe containing just the centroid coordinates of</dt><dd><p>the ring atoms of each residue.</p>
</dd>
</dl>
</li>
</ul>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.distance.node_coords">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.distance.</span></span><span class="sig-name descname"><span class="pre">node_coords</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.10)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/distance.html#node_coords"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.distance.node_coords" title="Permalink to this definition">#</a></dt>
<dd><p>Return the <code class="docutils literal notranslate"><span class="pre">x,</span> <span class="pre">y,</span> <span class="pre">z</span></code> coordinates of a node.
This is a helper function. Simplifies the code.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph to extract coordinates from</p></li>
<li><p><strong>n</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – str node ID in graph to extract coordinates from</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Tuple of coordinates <code class="docutils literal notranslate"><span class="pre">(x,</span> <span class="pre">y,</span> <span class="pre">z)</span></code></p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Tuple[<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>]</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.edges.intramolecular">
<span id="intramolecular"></span><h3>Intramolecular<a class="headerlink" href="#module-graphein.protein.edges.intramolecular" title="Permalink to this headline">#</a></h3>
<p>Featurization functions for graph edges.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.add_contacts_edge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">add_contacts_edge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">interaction_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#add_contacts_edge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.add_contacts_edge" title="Permalink to this definition">#</a></dt>
<dd><p>Adds specific interaction types to the protein graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – networkx protein graph</p></li>
<li><p><strong>interaction_type</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – interaction type to be added</p></li>
</ul>
</dd>
<dt class="field-even">Return G</dt>
<dd class="field-even"><p>nx.Graph with specified interaction-based edges added.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.get_contacts_df">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">get_contacts_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><span class="pre">GetContactsConfig</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pd.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#get_contacts_df"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.get_contacts_df" title="Permalink to this definition">#</a></dt>
<dd><p>Reads GetContact File and returns it as a pd.DataFrame</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>GetContactsConfig</em></a>) – GetContactsConfig object</p></li>
<li><p><strong>pdb_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Name of PDB file. Contacts files are name {pdb_name}_contacts.tsv</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>DataFrame of prased GetContacts output</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.hydrogen_bond">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">hydrogen_bond</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#hydrogen_bond"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.hydrogen_bond" title="Permalink to this definition">#</a></dt>
<dd><p>Adds hydrogen bonds to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add hydrogen bonds to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with hydrogen bonds added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.hydrophobic">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">hydrophobic</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#hydrophobic"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.hydrophobic" title="Permalink to this definition">#</a></dt>
<dd><p>Adds hydrophobic interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add hydrophobic interaction edges to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with hydrophobic interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.peptide_bonds">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">peptide_bonds</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#peptide_bonds"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.peptide_bonds" title="Permalink to this definition">#</a></dt>
<dd><p>Adds peptide backbone to residues in each chain</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph protein graph with added peptide bonds</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.pi_cation">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">pi_cation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#pi_cation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.pi_cation" title="Permalink to this definition">#</a></dt>
<dd><p>Adds pi-cation interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add pi-cation interactions to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with pi-pi_cation interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.pi_stacking">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">pi_stacking</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#pi_stacking"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.pi_stacking" title="Permalink to this definition">#</a></dt>
<dd><p>Adds pi-stacking interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add pi-stacking interactions to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with pi-stacking interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.read_contacts_file">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">read_contacts_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><span class="pre">GetContactsConfig</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">contacts_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">pd.DataFrame</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#read_contacts_file"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.read_contacts_file" title="Permalink to this definition">#</a></dt>
<dd><p>Parses GetContacts file to an edgelist (pd.DataFrame)</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>GetContactsConfig</em></a>) – GetContactsConfig object (graphein.protein.config.GetContactsConfig)</p></li>
<li><p><strong>contacts_file</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – file name of contacts file</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Pandas Dataframe of edge list</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.run_get_contacts">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">run_get_contacts</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><span class="pre">GetContactsConfig</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#run_get_contacts"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.run_get_contacts" title="Permalink to this definition">#</a></dt>
<dd><p>Runs GetContacts on a protein structure. If no file_name is provided, a PDB file is downloaded for the pdb_id</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.GetContactsConfig" title="graphein.protein.config.GetContactsConfig"><em>graphein.protein.config.GetContactsConfig</em></a>) – GetContactsConfig object containing GetContacts parameters</p></li>
<li><p><strong>pdb_id</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – 4-character PDB accession code</p></li>
<li><p><strong>file_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – PDB_name file to use, if annotations to be retrieved from the PDB</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.salt_bridge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">salt_bridge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#salt_bridge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.salt_bridge" title="Permalink to this definition">#</a></dt>
<dd><p>Adds salt bridges to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add salt bridges to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with salt bridges added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.t_stacking">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">t_stacking</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#t_stacking"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.t_stacking" title="Permalink to this definition">#</a></dt>
<dd><p>Adds t-stacking interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add t-stacking interactions to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with t-stacking interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.intramolecular.van_der_waals">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.intramolecular.</span></span><span class="sig-name descname"><span class="pre">van_der_waals</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/intramolecular.html#van_der_waals"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.intramolecular.van_der_waals" title="Permalink to this definition">#</a></dt>
<dd><p>Adds van der Waals interactions to protein structure graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph to add van der Waals interactions to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph with van der Waals interactions added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.edges.atomic">
<span id="atomic"></span><h3>Atomic<a class="headerlink" href="#module-graphein.protein.edges.atomic" title="Permalink to this headline">#</a></h3>
<p>Functions for computing atomic structure of proteins.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.add_atomic_edges">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">add_atomic_edges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tolerance</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.56</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#add_atomic_edges"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.add_atomic_edges" title="Permalink to this definition">#</a></dt>
<dd><p>Computes covalent edges based on atomic distances. Covalent radii are assigned to each atom based on its bond assign_bond_states_to_dataframe
The distance matrix is then thresholded to entries less than this distance plus some tolerance to create an adjacency matrix.
This adjacency matrix is then parsed into an edge list and covalent edges added</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Atomic graph (nodes correspond to atoms) to populate with atomic bonds as edges</p></li>
<li><p><strong>tolerance</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Tolerance for atomic distance. Default is <code class="docutils literal notranslate"><span class="pre">0.56</span></code> Angstroms. Commonly used values are: <code class="docutils literal notranslate"><span class="pre">0.4,</span> <span class="pre">0.45,</span> <span class="pre">0.56</span></code></p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Atomic graph with edges between bonded atoms added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.add_bond_order">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">add_bond_order</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#add_bond_order"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.add_bond_order" title="Permalink to this definition">#</a></dt>
<dd><p>Assign bond orders to the covalent bond edges between atoms on the basis of bond length. Values are taken from:</p>
<blockquote>
<div><p>Automatic Assignment of Chemical Connectivity to Organic Molecules in the Cambridge Structural Database.
Jon C. Baber and Edward E. Hodgkin*</p>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Atomic-level protein graph with covalent edges.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Atomic-level protein graph with covalent edges annotated with putative bond order.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>mx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.add_ring_status">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">add_ring_status</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#add_ring_status"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.add_ring_status" title="Permalink to this definition">#</a></dt>
<dd><p>Identifies rings in the atomic graph. Assigns the edge attribute <code class="docutils literal notranslate"><span class="pre">"RING"</span></code> to edges in the ring. We do not distinguish
between aromatic and non-aromatic rings. Functions by identifying all cycles in the graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Atom-level protein structure graph to add ring edge types to</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Atom-level protein structure graph with added <code class="docutils literal notranslate"><span class="pre">"RING"</span></code> edge attribute</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.assign_bond_states_to_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">assign_bond_states_to_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#assign_bond_states_to_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.assign_bond_states_to_dataframe" title="Permalink to this definition">#</a></dt>
<dd><p>Takes a <code class="docutils literal notranslate"><span class="pre">PandasPDB</span></code> atom dataframe and assigns bond states to each atom based on:</p>
<blockquote>
<div><p>Atomic Structures of all the Twenty Essential Amino Acids and a Tripeptide, with Bond Lengths as Sums of Atomic Covalent Radii
Heyrovska, 2008</p>
</div></blockquote>
<p>First, maps atoms to their standard bond states (<a class="reference internal" href="#graphein.protein.resi_atoms.DEFAULT_BOND_STATE" title="graphein.protein.resi_atoms.DEFAULT_BOND_STATE"><code class="xref py py-const docutils literal notranslate"><span class="pre">DEFAULT_BOND_STATE</span></code></a>).
Second, maps non-standard bonds states (<a class="reference internal" href="#graphein.protein.resi_atoms.RESIDUE_ATOM_BOND_STATE" title="graphein.protein.resi_atoms.RESIDUE_ATOM_BOND_STATE"><code class="xref py py-const docutils literal notranslate"><span class="pre">RESIDUE_ATOM_BOND_STATE</span></code></a>).
Fills NaNs with standard bond states.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Pandas PDB dataframe</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Dataframe with added <code class="docutils literal notranslate"><span class="pre">atom_bond_state</span></code> column</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.assign_covalent_radii_to_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">assign_covalent_radii_to_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#assign_covalent_radii_to_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.assign_covalent_radii_to_dataframe" title="Permalink to this definition">#</a></dt>
<dd><p>Assigns covalent radius (<a class="reference internal" href="#graphein.protein.resi_atoms.COVALENT_RADII" title="graphein.protein.resi_atoms.COVALENT_RADII"><code class="xref py py-const docutils literal notranslate"><span class="pre">COVALENT_RADII</span></code></a>) to each atom based on its bond state. Adds a <code class="docutils literal notranslate"><span class="pre">covalent_radius</span></code> column. Using values from:</p>
<blockquote>
<div><p>Atomic Structures of all the Twenty Essential Amino Acids and a Tripeptide, with Bond Lengths as Sums of Atomic Covalent Radii
Heyrovska, 2008</p>
</div></blockquote>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – Pandas PDB dataframe with a <code class="docutils literal notranslate"><span class="pre">bond_states_column</span></code></p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Pandas PDB dataframe with added <code class="docutils literal notranslate"><span class="pre">covalent_radius</span></code> column</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.edges.atomic.identify_bond_type_from_mapping">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.edges.atomic.</span></span><span class="sig-name descname"><span class="pre">identify_bond_type_from_mapping</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">u</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">v</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">a</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Dict" title="(in Python v3.10)"><span class="pre">Dict</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">query</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/edges/atomic.html#identify_bond_type_from_mapping"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.edges.atomic.identify_bond_type_from_mapping" title="Permalink to this definition">#</a></dt>
<dd><p>Compares the bond length between two atoms in the graph, and the relevant experimental value by performing a lookup
against the watershed values in:</p>
<blockquote>
<div><p>Automatic Assignment of Chemical Connectivity to Organic Molecules in the Cambridge
Structural Database. Jon C. Baber and Edward E. Hodgkin*</p>
</div></blockquote>
<p>Bond orders are assigned in the order <code class="docutils literal notranslate"><span class="pre">triple</span></code> &lt; <code class="docutils literal notranslate"><span class="pre">double</span></code> &lt; <code class="docutils literal notranslate"><span class="pre">single</span></code> (e.g. if a bond is shorter than the triple bond
watershed (<code class="docutils literal notranslate"><span class="pre">w_dt</span></code>) then it is assigned as a triple bond. Similarly, if a bond is longer than this but shorter than the
double bond watershed (<code class="docutils literal notranslate"><span class="pre">w_sd</span></code>), it is assigned double bond status.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> of atom-protein structure with atomic edges added</p></li>
<li><p><strong>u</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – node 1 in edge</p></li>
<li><p><strong>v</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – node 2 in edge</p></li>
<li><p><strong>a</strong> (<em>Dict</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>Any</em><em>]</em>) – edge data</p></li>
<li><p><strong>query</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – <code class="docutils literal notranslate"><span class="pre">"ELEMENTX-ELEMENTY"</span></code> to perform lookup with (E.g. <code class="docutils literal notranslate"><span class="pre">"C-O"</span></code>,``”N-N”<a href="#id1"><span class="problematic" id="id2">``</span></a>)</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Graph with atomic edge bond order assigned</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
</div>
<div class="section" id="features">
<h2>Features<a class="headerlink" href="#features" title="Permalink to this headline">#</a></h2>
<div class="section" id="module-graphein.protein.features.nodes.aaindex">
<span id="node"></span><h3>Node<a class="headerlink" href="#module-graphein.protein.features.nodes.aaindex" title="Permalink to this headline">#</a></h3>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.aaindex.aaindex1">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.aaindex.</span></span><span class="sig-name descname"><span class="pre">aaindex1</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">accession</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/aaindex.html#aaindex1"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.aaindex.aaindex1" title="Permalink to this definition">#</a></dt>
<dd><p>Adds AAIndex1 datavalues for a given accession as node features.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graphein to featurise</p></li>
<li><p><strong>accession</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – AAIndex1 accession code for values to use</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Structure graph with AAindex1 node features added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.aaindex.fetch_AAIndex">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.aaindex.</span></span><span class="sig-name descname"><span class="pre">fetch_AAIndex</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">accession</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.10)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Dict" title="(in Python v3.10)"><span class="pre">Dict</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/aaindex.html#fetch_AAIndex"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.aaindex.fetch_AAIndex" title="Permalink to this definition">#</a></dt>
<dd><p>Fetches AAindex1 dictionary from an accession code. The dictionary maps one-letter AA codes to float values</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>accession</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Aaindex1 accession code</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>tuple of record titel(str) and dictionary of AA:value mappings</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Tuple[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>, Dict[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>]]</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.nodes.amino_acid"></span><p>Featurization functions for amino acids.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.amino_acid_one_hot">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">amino_acid_one_hot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Dict" title="(in Python v3.10)"><span class="pre">Dict</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_array</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">allowable_set</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v1.4.1)"><span class="pre">pandas.core.series.Series</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.22)"><span class="pre">numpy.ndarray</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#amino_acid_one_hot"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.amino_acid_one_hot" title="Permalink to this definition">#</a></dt>
<dd><p>Adds a one-hot encoding of amino acid types as a node attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – node name, this is unused and only included for compatibility with the other functions</p></li>
<li><p><strong>d</strong> (<em>Dict</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>Any</em><em>]</em>) – Node data</p></li>
<li><p><strong>return_array</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – If True, returns a numpy array of one-hot encoding, otherwise returns a pd.Series. Default is True.</p></li>
<li><p><strong>allowable_set</strong> – Specifies vocabulary of amino acids. Default is None (which uses <cite>graphein.protein.resi_atoms.STANDARD_AMINO_ACIDS</cite>).</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>One-hot encoding of amino acid types</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[pd.Series, np.ndarray]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.expasy_protein_scale">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">expasy_protein_scale</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">selection</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">add_separate</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_array</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v1.4.1)"><span class="pre">pandas.core.series.Series</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.22)"><span class="pre">numpy.ndarray</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#expasy_protein_scale"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.expasy_protein_scale" title="Permalink to this definition">#</a></dt>
<dd><p>Return amino acid features that come from the EXPASY protein scale.</p>
<p>Source: <a class="reference external" href="https://web.expasy.org/protscale/">https://web.expasy.org/protscale/</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n</strong> – Node in a NetworkX graph</p></li>
<li><p><strong>d</strong> – NetworkX node attributes.</p></li>
<li><p><strong>selection</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – List of columns to select. Viewable in graphein.protein.features.nodes.meiler_embeddings</p></li>
<li><p><strong>add_separate</strong> – Whether or not to add the expasy features as indvidual entries or as a series.</p></li>
<li><p><strong>return_array</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Bool indicating whether or not to return a np.ndarray of the features. Default is pd.Series</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.Series of amino acid features</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.Series</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.load_expasy_scales">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">load_expasy_scales</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#load_expasy_scales"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.load_expasy_scales" title="Permalink to this definition">#</a></dt>
<dd><p>Load pre-downloaded EXPASY scales.</p>
<p>This helps with node featuarization.</p>
<p>The function is LRU-cached in memory for fast access
on each function call.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>pd.DataFrame containing expasy scales</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.load_meiler_embeddings">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">load_meiler_embeddings</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#load_meiler_embeddings"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.load_meiler_embeddings" title="Permalink to this definition">#</a></dt>
<dd><p>Load pre-downloaded Meiler embeddings.</p>
<p>This helps with node featurization.</p>
<p>The function is LRU-cached in memory for fast access
on each function call.</p>
<dl class="field-list simple">
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>pd.DataFrame containing Meiler Embeddings from Meiler et al. 2001</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.amino_acid.meiler_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.amino_acid.</span></span><span class="sig-name descname"><span class="pre">meiler_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">n</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">d</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_array</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v1.4.1)"><span class="pre">pandas.core.series.Series</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">numpy.array</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/amino_acid.html#meiler_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.amino_acid.meiler_embedding" title="Permalink to this definition">#</a></dt>
<dd><p>Return amino acid features from reduced dimensional embeddings of amino acid physicochemical properties.</p>
<p>Source: <a class="reference external" href="https://link.springer.com/article/10.1007/s008940100038">https://link.springer.com/article/10.1007/s008940100038</a>
doi: <a class="reference external" href="https://doi.org/10.1007/s008940100038">https://doi.org/10.1007/s008940100038</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>n</strong> – Node in a NetworkX graph</p></li>
<li><p><strong>d</strong> – NetworkX node attributes.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.Series of amino acid features</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.Series</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.nodes.dssp"></span><p>Featurization functions for graph nodes using DSSP predicted features.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.add_dssp_df">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">add_dssp_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">nx.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dssp_config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">DSSPConfig</span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">nx.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#add_dssp_df"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.add_dssp_df" title="Permalink to this definition">#</a></dt>
<dd><p>Construct DSSP dataframe and add as graph level variable to protein graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> – Input protein graph</p></li>
<li><p><strong>G</strong> – nx.Graph</p></li>
<li><p><strong>dssp_config</strong> (<em>DSSPConfig</em><em>, </em><em>optional</em>) – DSSPConfig object. Specifies which executable to run. Located in graphein.protein.config</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with DSSP dataframe added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.add_dssp_feature">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">add_dssp_feature</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#add_dssp_feature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.add_dssp_feature" title="Permalink to this definition">#</a></dt>
<dd><p>Adds add_dssp_feature specified amino acid feature as calculated
by DSSP to every node in a protein graph
:param G: Protein structure graph to add dssp feature to
:param feature: string specifying name of DSSP feature to add:
“chain”,
“resnum”,
“icode”,
“aa”,
“ss”,
“asa”,
“phi”,
“psi”,
“dssp_index”,
“NH_O_1_relidx”,
“NH_O_1_energy”,
“O_NH_1_relidx”,
“O_NH_1_energy”,
“NH_O_2_relidx”,
“NH_O_2_energy”,
“O_NH_2_relidx”,
“O_NH_2_energy”,</p>
<p>These names parse_dssp_df accessible in the DSSP_COLS list
:param G: Protein Graph to add features to
:type G: nx.Graph
:return: Protein structure graph with DSSP feature added to nodes
:rtype: nx.Graph</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.asa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">asa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#asa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.asa" title="Permalink to this definition">#</a></dt>
<dd><p>Adds ASA of each residue in protein graph as calculated by DSSP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with asa values added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.parse_dssp_df">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">parse_dssp_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dssp</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Dict" title="(in Python v3.10)"><span class="pre">Dict</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#parse_dssp_df"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.parse_dssp_df" title="Permalink to this definition">#</a></dt>
<dd><p>Parse DSSP output to DataFrame</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>dssp</strong> (<em>Dict</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>Any</em><em>]</em>) – Dictionary containing DSSP output</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.Dataframe containing parsed DSSP output</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.phi">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">phi</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#phi"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.phi" title="Permalink to this definition">#</a></dt>
<dd><p>Adds phi-angles of each residue in protein graph as calculated by DSSP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with phi-angles values added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.process_dssp_df">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">process_dssp_df</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#process_dssp_df"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.process_dssp_df" title="Permalink to this definition">#</a></dt>
<dd><p>Processes a DSSP DataFrame to make indexes align with node IDs</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>df</strong> (<em>pd.DataFrame</em>) – pd.DataFrame containing the parsed output from DSSP.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>pd.DataFrame with node IDs</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.psi">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">psi</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#psi"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.psi" title="Permalink to this definition">#</a></dt>
<dd><p>Adds psi-angles of each residue in protein graph as calculated by DSSP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with psi-angles values added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.rsa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">rsa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#rsa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.rsa" title="Permalink to this definition">#</a></dt>
<dd><p>Adds RSA (relative solvent accessibility) of each residue in protein graph
as calculated by DSSP.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with rsa values added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.dssp.secondary_structure">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.dssp.</span></span><span class="sig-name descname"><span class="pre">secondary_structure</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/dssp.html#secondary_structure"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.dssp.secondary_structure" title="Permalink to this definition">#</a></dt>
<dd><p>Adds secondary structure of each residue in protein graph
as calculated by DSSP in the form of a string</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – Input protein graph</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein graph with secondary structure added</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.nodes.geometry"></span><p>Provides geometry-based featurisation functions.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.geometry.add_beta_carbon_vector">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.geometry.</span></span><span class="sig-name descname"><span class="pre">add_beta_carbon_vector</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/geometry.html#add_beta_carbon_vector"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.geometry.add_beta_carbon_vector" title="Permalink to this definition">#</a></dt>
<dd><p>Adds vector from node (typically alpha carbon) to position of beta carbon.</p>
<p>Glycine does not have a beta carbon, so we set it to <code class="docutils literal notranslate"><span class="pre">np.array([0,</span> <span class="pre">0,</span> <span class="pre">0])</span></code>.
We extract the position of the beta carbon from the unprocessed atomic PDB dataframe.
For this we use the <code class="docutils literal notranslate"><span class="pre">raw_pdb_df</span></code> dataframe.
If scale, we scale the vector to the unit vector. If reverse is True,
we reverse the vector (<code class="docutils literal notranslate"><span class="pre">C</span> <span class="pre">beta</span> <span class="pre">-</span> <span class="pre">node</span></code>). If reverse is false (default) we compute (<code class="docutils literal notranslate"><span class="pre">node</span> <span class="pre">-</span> <span class="pre">C</span> <span class="pre">beta</span></code>).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – Graph to add vector to.</p></li>
<li><p><strong>scale</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Scale vector to unit vector. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>reverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Reverse vector. Defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.geometry.add_sequence_neighbour_vector">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.geometry.</span></span><span class="sig-name descname"><span class="pre">add_sequence_neighbour_vector</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n_to_c</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/geometry.html#add_sequence_neighbour_vector"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.geometry.add_sequence_neighbour_vector" title="Permalink to this definition">#</a></dt>
<dd><p>Computes vector from node to adjacent node in sequence.
Typically used with <code class="docutils literal notranslate"><span class="pre">CA</span></code> (alpha carbon) graphs.</p>
<p>If <code class="docutils literal notranslate"><span class="pre">n_to_c</span></code> is <code class="docutils literal notranslate"><span class="pre">True</span></code> (default), we compute the vectors from the N terminus to the C terminus (canonical direction).
If <code class="docutils literal notranslate"><span class="pre">reverse</span></code> is <code class="docutils literal notranslate"><span class="pre">False</span></code> (default), we compute <code class="docutils literal notranslate"><span class="pre">Node_i</span> <span class="pre">-</span> <span class="pre">Node_{i+1}</span></code>.
If <code class="docutils literal notranslate"><span class="pre">reverse</span> <span class="pre">is</span> <span class="pre">``True</span></code>, we compute <code class="docutils literal notranslate"><span class="pre">Node_{i+1}</span> <span class="pre">-</span> <span class="pre">Node_i</span></code>.
:param g: Graph to add vector to.
:type g: nx.Graph
:param scale: Scale vector to unit vector. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.
:type scale: bool
:param reverse: Reverse vector. Defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.
:type reverse: bool
:param n_to_c: Compute vector from N to C or C to N. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.
:type n_to_c: bool</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.nodes.geometry.add_sidechain_vector">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.nodes.geometry.</span></span><span class="sig-name descname"><span class="pre">add_sidechain_vector</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">reverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/features/nodes/geometry.html#add_sidechain_vector"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.nodes.geometry.add_sidechain_vector" title="Permalink to this definition">#</a></dt>
<dd><p>Adds vector from node to average position of sidechain atoms.</p>
<p>We compute the mean of the sidechain atoms for each node. For this we use the <code class="docutils literal notranslate"><span class="pre">rgroup_df</span></code> dataframe.
If the graph does not contain the <code class="docutils literal notranslate"><span class="pre">rgroup_df</span></code> dataframe, we compute it from the <code class="docutils literal notranslate"><span class="pre">raw_pdb_df</span></code>.
If scale, we scale the vector to the unit vector. If reverse is True,
we reverse the vector (<code class="docutils literal notranslate"><span class="pre">sidechain</span> <span class="pre">-</span> <span class="pre">node</span></code>). If reverse is false (default) we compute (<code class="docutils literal notranslate"><span class="pre">node</span> <span class="pre">-</span> <span class="pre">sidechain</span></code>).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – Graph to add vector to.</p></li>
<li><p><strong>scale</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Scale vector to unit vector. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>reverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Reverse vector. Defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.features.sequence.embeddings">
<span id="sequence"></span><h3>Sequence<a class="headerlink" href="#module-graphein.protein.features.sequence.embeddings" title="Permalink to this headline">#</a></h3>
<p>Functions to add embeddings from pre-trained language models protein structure graphs.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.embeddings.biovec_sequence_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">biovec_sequence_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#biovec_sequence_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.embeddings.biovec_sequence_embedding" title="Permalink to this definition">#</a></dt>
<dd><p>Adds BioVec sequence embedding feature to the graph. Computed over chains.</p>
<dl class="simple">
<dt><strong>Source</strong></dt><dd><p>ProtVec: A Continuous Distributed Representation of Biological Sequences</p>
</dd>
</dl>
<p>Paper: <a class="reference external" href="http://arxiv.org/pdf/1503.05140v1.pdf">http://arxiv.org/pdf/1503.05140v1.pdf</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph protein structure graph with biovec embedding added. e.g. <code class="docutils literal notranslate"><span class="pre">G.graph["biovec_embedding_A"]</span></code> for chain <code class="docutils literal notranslate"><span class="pre">A</span></code>.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.embeddings.compute_esm_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">compute_esm_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">representation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'esm1b_t33_650M_UR50S'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_layer</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">33</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">np.ndarray</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#compute_esm_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.embeddings.compute_esm_embedding" title="Permalink to this definition">#</a></dt>
<dd><p>Computes sequence embedding using Pre-trained ESM model from FAIR</p>
<blockquote>
<div><p>Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences (2019)
Rives, Alexander and Meier, Joshua and Sercu, Tom and Goyal, Siddharth and Lin, Zeming and Liu, Jason and Guo,
Demi and Ott, Myle and Zitnick, C. Lawrence and Ma, Jerry and Fergus, Rob</p>
<p>Transformer protein language models are unsupervised structure learners 2020
Rao, Roshan M and Meier, Joshua and Sercu, Tom and Ovchinnikov, Sergey and Rives, Alexander</p>
</div></blockquote>
<p>Pre-trained models:</p>
<p>Full Name layers params Dataset Embedding Dim Model URL
========= ====== ====== ======= ============= =========
ESM-1b esm1b_t33_650M_UR50S 33 650M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt</a>
ESM1-main esm1_t34_670M_UR50S 34 670M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt</a>
esm1_t34_670M_UR50D 34 670M UR50/D 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt</a>
esm1_t34_670M_UR100 34 670M UR100 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt</a>
esm1_t12_85M_UR50S 12 85M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt</a>
esm1_t6_43M_UR50S 6 43M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sequence</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Protein sequence to embed (str)</p></li>
<li><p><strong>representation</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Type of embedding to extract. <code class="docutils literal notranslate"><span class="pre">"residue"</span></code> or <code class="docutils literal notranslate"><span class="pre">"sequence"</span></code>. Sequence-level embeddings are averaged residue embeddings</p></li>
<li><p><strong>model_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Name of pre-trained model to use</p></li>
<li><p><strong>output_layer</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – integer indicating which layer the output should be taken from</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>embedding (<code class="docutils literal notranslate"><span class="pre">np.ndarray</span></code>)</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.embeddings.esm_residue_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">esm_residue_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'esm1b_t33_650M_UR50S'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_layer</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">33</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#esm_residue_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.embeddings.esm_residue_embedding" title="Permalink to this definition">#</a></dt>
<dd><p>Computes ESM residue embeddings from a protein sequence and adds the to the graph.</p>
<blockquote>
<div><p>Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences (2019)
Rives, Alexander and Meier, Joshua and Sercu, Tom and Goyal, Siddharth and Lin, Zeming and Liu, Jason and Guo,
Demi and Ott, Myle and Zitnick, C. Lawrence and Ma, Jerry and Fergus, Rob</p>
<p>Transformer protein language models are unsupervised structure learners 2020
Rao, Roshan M and Meier, Joshua and Sercu, Tom and Ovchinnikov, Sergey and Rives, Alexander</p>
</div></blockquote>
<p><strong>Pre-trained models</strong></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> to add esm embedding to.</p></li>
<li><p><strong>model_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Name of pre-trained model to use.</p></li>
<li><p><strong>output_layer</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – index of output layer in pre-trained model.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> with esm embedding feature added to nodes.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.embeddings.esm_sequence_embedding">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">esm_sequence_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#esm_sequence_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.embeddings.esm_sequence_embedding" title="Permalink to this definition">#</a></dt>
<dd><p>Computes ESM sequence embedding feature over chains in a graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph protein structure graph with esm embedding features added eg. <code class="docutils literal notranslate"><span class="pre">G.graph["esm_embedding_A"]</span></code> for chain A.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.sequence.propy"></span><dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.aa_dipeptide_composition">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">aa_dipeptide_composition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#aa_dipeptide_composition"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.aa_dipeptide_composition" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the composition of AADs, dipeptide and 3-mers for a given protein sequence. Contains all composition</dt><dd><p>values of AADs, dipeptide and 3-mers (8420).</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with aa_dipeptide_composition feature added.
G.graph[“aa_dipeptide_composition_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.aa_spectrum">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">aa_spectrum</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#aa_spectrum"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.aa_spectrum" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate the spectrum descriptors of 3-mers for a given protein. Contains the composition values of 8000 3-mers</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with aa_spectrum feature added. G.graph[“aa_spectrum_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.all_composition_descriptors">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">all_composition_descriptors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#all_composition_descriptors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.all_composition_descriptors" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate all composition descriptors based on seven different properties of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_descriptors feature added.
G.graph[“composition_descriptors_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.all_ctd_descriptors">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">all_ctd_descriptors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#all_ctd_descriptors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.all_ctd_descriptors" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate all CTD descriptors based seven different properties of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with ctd_descriptors feature added.
G.graph[“ctd_descriptors_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.all_distribution_descriptors">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">all_distribution_descriptors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#all_distribution_descriptors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.all_distribution_descriptors" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate all distribution descriptors based on seven different properties of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_descriptors feature added.
G.graph[“distribution_descriptors_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.all_transition_descriptors">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">all_transition_descriptors</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#all_transition_descriptors"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.all_transition_descriptors" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate all transition descriptors based on seven different properties of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_descriptors feature added.
G.graph[“transition_descriptors_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.amino_acid_composition">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">amino_acid_composition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#amino_acid_composition"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.amino_acid_composition" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate the composition of Amino acids for a given protein sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>Optional</em><em>[</em><em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with amino_acid_composition feature added. G.graph[“amino_acid_composition_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_all">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_all</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_all"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_all" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Compute Geary autocorrelation descriptors based on 8 properties of AADs. Result contains 30*8=240 Geary</dt><dd><p>autocorrelation descriptors based on the given properties(i.e., _AAPropert).</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_all feature added.
G.graph[“autocorrelation_geary_all_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_av_flexibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_av_flexibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_av_flexibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_av_flexibility" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on AvFlexibility. contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on AvFlexibility.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_av_flexibility feature added.
G.graph[“autocorrelation_geary_av_flexibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_free_energy">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_free_energy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_free_energy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_free_energy" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on FreeEnergy. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on FreeEnergy.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_free_energy feature added.
G.graph[“autocorrelation_geary_av_free_energy_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_hydrophobicity" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on hydrophobicity. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on hydrophobicity.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_hydrophobicity feature added.
G.graph[“autocorrelation_geary_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_mutability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_mutability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_mutability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_mutability" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on Mutability. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on mutability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_mutability feature added.
G.graph[“autocorrelation_geary_mutability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_polarizability" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on polarizability. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on polarizability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_polarizability feature added.
G.graph[“autocorrelation_geary_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_residue_asa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_residue_asa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_residue_asa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_residue_asa" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on ResidueASA. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on ResidueASA.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_residue_asa feature added.
G.graph[“autocorrelation_geary_residue_asa_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_residue_vol">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_residue_vol</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_residue_vol"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_residue_vol" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on ResidueVol. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on ResidueVol.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_residue_vol feature added.
G.graph[“autocorrelation_geary_residue_vol_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_geary_steric">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_geary_steric</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_geary_steric"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_geary_steric" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Geary Autocorrelation descriptors based on Steric. result contains 30 Geary Autocorrelation</dt><dd><p>descriptors based on Steric</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_geary_steric feature added.
G.graph[“autocorrelation_geary_steric_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_all">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_all</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_all"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_all" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Compute Moran autocorrelation descriptors based on 8 properties of AADs. Result contains 30*8=240 Moran</dt><dd><p>autocorrelation descriptors based on the given properties(i.e., _AAPropert).</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_all feature added.
G.graph[“autocorrelation_moran_all_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_av_flexibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_av_flexibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_av_flexibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_av_flexibility" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on AvFlexibility. contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on AvFlexibility.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_av_flexibility feature added.
G.graph[“autocorrelation_moran_av_flexibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_free_energy">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_free_energy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_free_energy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_free_energy" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on FreeEnergy. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on FreeEnergy.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_free_energy feature added.
G.graph[“autocorrelation_moran_av_free_energy_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_hydrophobicity" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on hydrophobicity. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on hydrophobicity.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_hydrophobicity feature added.
G.graph[“autocorrelation_moran_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_mutability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_mutability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_mutability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_mutability" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on Mutability. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on mutability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_mutability feature added.
G.graph[“autocorrelation_moran_mutability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_polarizability" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on polarizability. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on polarizability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_polarizability feature added.
G.graph[“autocorrelation_moran_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_residue_asa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_residue_asa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_residue_asa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_residue_asa" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on ResidueASA. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on ResidueASA.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_residue_asa feature added.
G.graph[“autocorrelation_moran_residue_asa_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_residue_vol">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_residue_vol</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_residue_vol"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_residue_vol" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on ResidueVol. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on ResidueVol.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_residue_vol feature added.
G.graph[“autocorrelation_moran_residue_vol_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_moran_steric">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_moran_steric</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_moran_steric"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_moran_steric" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the Moran Autocorrelation descriptors based on Steric. result contains 30 Moran Autocorrelation</dt><dd><p>descriptors based on Steric</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_moran_steric feature added.
G.graph[“autocorrelation_moran_steric_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_all">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_all</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_all"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_all" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Compute NormalizedMoreauBroto autocorrelation descriptors based on 8 properties of AADs. Result contains 30*8=240</dt><dd><p>NormalizedMoreauBroto autocorrelation descriptors based on the given properties(i.e., _AAPropert).</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_all feature added.
G.graph[“autocorrelation_normalized_moreau_broto_all_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_av_flexibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_av_flexibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_av_flexibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_av_flexibility" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on AvFlexibility. contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on AvFlexibility.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_av_flexibility feature added.
G.graph[“autocorrelation_normalized_moreau_broto_av_flexibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_free_energy">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_free_energy</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_free_energy"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_free_energy" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on FreeEnergy. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on FreeEnergy.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_free_energy feature added.
G.graph[“autocorrelation_normalized_moreau_broto_av_free_energy_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_hydrophobicity" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on hydrophobicity. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on hydrophobicity.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_hydrophobicity feature added.
G.graph[“autocorrelation_normalized_moreau_broto_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_mutability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_mutability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_mutability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_mutability" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on Mutability. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on mutability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_mutability feature added.
G.graph[“autocorrelation_normalized_moreau_broto_mutability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_polarizability" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on polarizability. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on polarizability.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_polarizability feature added.
G.graph[“autocorrelation_normalized_moreau_broto_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_residue_asa">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_residue_asa</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_residue_asa"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_residue_asa" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on ResidueASA. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on ResidueASA.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_residue_asa feature added.
G.graph[“autocorrelation_normalized_moreau_broto_residue_asa_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_residue_vol">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_residue_vol</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_residue_vol"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_residue_vol" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on ResidueVol. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on ResidueVol.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_residue_vol feature added.
G.graph[“autocorrelation_normalized_moreau_broto_residue_vol_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_steric">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_normalized_moreau_broto_steric</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_normalized_moreau_broto_steric"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_normalized_moreau_broto_steric" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Calculate the NormalizedMoreauBroto Autocorrelation descriptors based on Steric. result contains 30</dt><dd><p>NormalizedMoreauBroto Autocorrelation descriptors based on Steric</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_normalized_moreau_broto_steric feature added.
G.graph[“autocorrelation_normalized_moreau_broto_steric_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.autocorrelation_total">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">autocorrelation_total</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#autocorrelation_total"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.autocorrelation_total" title="Permalink to this definition">#</a></dt>
<dd><dl class="simple">
<dt>Compute all autocorrelation descriptors based on 8 properties of AADs. result contains 30*8*3=720 normalized Moreau</dt><dd><p>Broto, Moran, and Geary</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with autocorrelation_total feature added.
G.graph[“autocorrelation_total_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_charge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_charge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_charge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_charge" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate composition descriptors based on Charge of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_charge feature added.
G.graph[“composition_charge_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_descriptor">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_descriptor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAProperty</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Dict" title="(in Python v3.10)"><span class="pre">Dict</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAPName</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_descriptor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_descriptor" title="Permalink to this definition">#</a></dt>
<dd><p>Compute composition descriptors.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>AAProperty</strong> (<em>Dict</em><em>[</em><em>Any</em><em>, </em><em>Any</em><em>]</em>) – contains classification of amino acids such as _Polarizability.</p></li>
<li><p><strong>AAPName</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – used for indicating a AAP name.</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_{AAPName} feature added.
G.graph[“composition_{AAPName}_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_hydrophobicity" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate composition descriptors based on Hydrophobicity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_hydrophobicity feature added.
G.graph[“composition_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_normalized_vdwv">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_normalized_vdwv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_normalized_vdwv"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_normalized_vdwv" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate composition descriptors based on NormalizedVDWV of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_normalized_vdwv feature added.
G.graph[“composition_normalized_vdwv_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_polarity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_polarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_polarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_polarity" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate composition descriptors based on Polarity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_polarity feature added.
G.graph[“composition_polarity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_polarizability" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate composition descriptors based on Polarizability of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_polarizability feature added.
G.graph[“composition_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_secondary_str">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_secondary_str</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_secondary_str"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_secondary_str" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate composition descriptors based on SecondaryStr of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_secondary_str feature added.
G.graph[“composition_secondary_str_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.composition_solvent_accessibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">composition_solvent_accessibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#composition_solvent_accessibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.composition_solvent_accessibility" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate composition descriptors based on SolventAccessibility of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with composition_solvent_accessibility feature added.
G.graph[“composition_solvent_accessibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.compute_propy_feature">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">compute_propy_feature</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">func</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Callable" title="(in Python v3.10)"><span class="pre">Callable</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#compute_propy_feature"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.compute_propy_feature" title="Permalink to this definition">#</a></dt>
<dd><p>Computes Propy Descriptors over chains in a Protein Graph</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph</p></li>
<li><p><strong>func</strong> (<em>Callable</em>) – ProPy wrapper function to compute</p></li>
<li><p><strong>feature_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Name of feature to index it in the nx.Graph object</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – Type of aggregation to use when aggregating a feature over multiple chains. One of:
[“mean”, “man”, “sum”]. Defaults to None.</p></li>
</ul>
</dd>
<dt class="field-even">Return G</dt>
<dd class="field-even"><p>Returns protein Graph with features added. Features are accessible with G.graph[{feature_name}_{chain | aggegation_type}]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.dipeptide_composition">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">dipeptide_composition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#dipeptide_composition"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.dipeptide_composition" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate the composition of dipeptidefor a given protein sequence. Contains composition of 400 dipeptides</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with dipeptide_composition feature added. G.graph[“dipeptide_composition_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_charge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_charge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_charge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_charge" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate distribution descriptors based on Charge of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_charge feature added.
G.graph[“distribution_charge_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_descriptor">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_descriptor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAProperty</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Dict" title="(in Python v3.10)"><span class="pre">Dict</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAPName</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_descriptor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_descriptor" title="Permalink to this definition">#</a></dt>
<dd><p>Compute distribution descriptors.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>AAProperty</strong> (<em>Dict</em><em>[</em><em>Any</em><em>, </em><em>Any</em><em>]</em>) – contains classifciation of amino acids such as _Polarizability.</p></li>
<li><p><strong>AAPName</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – used for indicating a AAP name.</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_{AAPName} feature added.
G.graph[“distribution_{AAPName}_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_hydrophobicity" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate distribution descriptors based on Hydrophobicity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_hydrophobicity feature added.
G.graph[“distribution_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_normalized_vdwv">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_normalized_vdwv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_normalized_vdwv"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_normalized_vdwv" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate distribution descriptors based on NormalizedVDWV of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_normalized_vdwv feature added.
G.graph[“distribution_normalized_vdwv_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_polarity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_polarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_polarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_polarity" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate distribution descriptors based on Polarity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_polarity feature added.
G.graph[“distribution_polarity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_polarizability" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate distribution descriptors based on Polarizability of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_polarizability feature added.
G.graph[“distribution_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_secondary_str">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_secondary_str</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_secondary_str"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_secondary_str" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate distribution descriptors based on SecondaryStr of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_secondary_str feature added.
G.graph[“distribution_secondary_str_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.distribution_solvent_accessibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">distribution_solvent_accessibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#distribution_solvent_accessibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.distribution_solvent_accessibility" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate distribution descriptors based on SolventAccessibility of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with distribution_solvent_accessibility feature added.
G.graph[“distribution_solvent_accessibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.quasi_sequence_order">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">quasi_sequence_order</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxlag</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.1</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#quasi_sequence_order"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.quasi_sequence_order" title="Permalink to this definition">#</a></dt>
<dd><p>Compute quasi-sequence-order descriptors for a given protein.</p>
<dl class="simple">
<dt>Kuo-Chen Chou. Prediction of Protein Subcellar Locations by Incorporating Quasi-Sequence-Order Effect.</dt><dd><p>Biochemical and Biophysical Research Communications 2000, 278, 477-483.</p>
</dd>
</dl>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>maxlag</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – (int, optional (default: 30)) – the maximum lag and the length of the protein should be larger than
maxlag</p></li>
<li><p><strong>weight</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – (float, optional (default: 0.1)) – a weight factor. Please see reference 1 for its choice.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.sequence_order_coupling_number_total">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">sequence_order_coupling_number_total</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">maxlag</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">30</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#sequence_order_coupling_number_total"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.sequence_order_coupling_number_total" title="Permalink to this definition">#</a></dt>
<dd><p>Compute the sequence order coupling numbers from 1 to maxlag for a given protein sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph</p></li>
<li><p><strong>(</strong><strong>default</strong> (<em>maxlag</em><em> (</em><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>, </em><em>optional</em>) – 30)) – the maximum lag and the length of the protein should be larger</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p></p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_charge">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_charge</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_charge"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_charge" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate transition descriptors based on Charge of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_charge feature added.
G.graph[“transition_charge_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_descriptor">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_descriptor</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAProperty</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Dict" title="(in Python v3.10)"><span class="pre">Dict</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">AAPName</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_descriptor"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_descriptor" title="Permalink to this definition">#</a></dt>
<dd><p>Compute transition descriptors.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>AAProperty</strong> (<em>Dict</em><em>[</em><em>Any</em><em>, </em><em>Any</em><em>]</em>) – contains classifciation of amino acids such as _Polarizability.</p></li>
<li><p><strong>AAPName</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – used for indicating a AAP name.</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_{AAPName} feature added.
G.graph[“transition_{AAPName}_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_hydrophobicity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_hydrophobicity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_hydrophobicity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_hydrophobicity" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate transition descriptors based on Hydrophobicity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_hydrophobicity feature added.
G.graph[“transition_hydrophobicity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_normalized_vdwv">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_normalized_vdwv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_normalized_vdwv"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_normalized_vdwv" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate transition descriptors based on NormalizedVDWV of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_normalized_vdwv feature added.
G.graph[“transition_normalized_vdwv_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_polarity">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_polarity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_polarity"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_polarity" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate transition descriptors based on Polarity of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein Graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_polarity feature added.
G.graph[“transition_polarity_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_polarizability">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_polarizability</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_polarizability"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_polarizability" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate transition descriptors based on Polarizability of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_polarizability feature added.
G.graph[“transition_polarizability_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_secondary_str">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_secondary_str</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_secondary_str"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_secondary_str" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate transition descriptors based on SecondaryStr of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_secondary_str feature added.
G.graph[“transition_secondary_str_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.sequence.propy.transition_solvent_accessibility">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.propy.</span></span><span class="sig-name descname"><span class="pre">transition_solvent_accessibility</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/propy.html#transition_solvent_accessibility"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.sequence.propy.transition_solvent_accessibility" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate transition descriptors based on SolventAccessibility of AADs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Protein graph to featurise</p></li>
<li><p><strong>aggregation_type</strong> (<em>List</em><em>[</em><em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>]</em>) – Aggregation types to use over chains</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Protein Graph with transition_solvent_accessibility feature added.
G.graph[“transition_solvent_accessibility_{chain | aggregation_type}”]</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<span class="target" id="module-graphein.protein.features.sequence.sequence"></span><p>Functions for graph-level featurization of the sequence of a protein. This submodule is focussed on physicochemical
proporties of the sequence.</p>
</div>
<div class="section" id="module-0">
<span id="sequence-utils"></span><h3>Sequence Utils<a class="headerlink" href="#module-0" title="Permalink to this headline">#</a></h3>
<p>Functions to add embeddings from pre-trained language models protein structure graphs.</p>
<dl class="py function">
<dt class="sig sig-object py" id="id0">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">biovec_sequence_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#biovec_sequence_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#id0" title="Permalink to this definition">#</a></dt>
<dd><p>Adds BioVec sequence embedding feature to the graph. Computed over chains.</p>
<dl class="simple">
<dt><strong>Source</strong></dt><dd><p>ProtVec: A Continuous Distributed Representation of Biological Sequences</p>
</dd>
</dl>
<p>Paper: <a class="reference external" href="http://arxiv.org/pdf/1503.05140v1.pdf">http://arxiv.org/pdf/1503.05140v1.pdf</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph protein structure graph with biovec embedding added. e.g. <code class="docutils literal notranslate"><span class="pre">G.graph["biovec_embedding_A"]</span></code> for chain <code class="docutils literal notranslate"><span class="pre">A</span></code>.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="id3">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">compute_esm_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sequence</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">representation</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'esm1b_t33_650M_UR50S'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_layer</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">33</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">np.ndarray</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#compute_esm_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#id3" title="Permalink to this definition">#</a></dt>
<dd><p>Computes sequence embedding using Pre-trained ESM model from FAIR</p>
<blockquote>
<div><p>Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences (2019)
Rives, Alexander and Meier, Joshua and Sercu, Tom and Goyal, Siddharth and Lin, Zeming and Liu, Jason and Guo,
Demi and Ott, Myle and Zitnick, C. Lawrence and Ma, Jerry and Fergus, Rob</p>
<p>Transformer protein language models are unsupervised structure learners 2020
Rao, Roshan M and Meier, Joshua and Sercu, Tom and Ovchinnikov, Sergey and Rives, Alexander</p>
</div></blockquote>
<p>Pre-trained models:</p>
<p>Full Name layers params Dataset Embedding Dim Model URL
========= ====== ====== ======= ============= =========
ESM-1b esm1b_t33_650M_UR50S 33 650M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1b_t33_650M_UR50S.pt</a>
ESM1-main esm1_t34_670M_UR50S 34 670M UR50/S 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50S.pt</a>
esm1_t34_670M_UR50D 34 670M UR50/D 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR50D.pt</a>
esm1_t34_670M_UR100 34 670M UR100 1280 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t34_670M_UR100.pt</a>
esm1_t12_85M_UR50S 12 85M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t12_85M_UR50S.pt</a>
esm1_t6_43M_UR50S 6 43M UR50/S 768 <a class="reference external" href="https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt">https://dl.fbaipublicfiles.com/fair-esm/models/esm1_t6_43M_UR50S.pt</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sequence</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Protein sequence to embed (str)</p></li>
<li><p><strong>representation</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Type of embedding to extract. <code class="docutils literal notranslate"><span class="pre">"residue"</span></code> or <code class="docutils literal notranslate"><span class="pre">"sequence"</span></code>. Sequence-level embeddings are averaged residue embeddings</p></li>
<li><p><strong>model_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Name of pre-trained model to use</p></li>
<li><p><strong>output_layer</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – integer indicating which layer the output should be taken from</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>embedding (<code class="docutils literal notranslate"><span class="pre">np.ndarray</span></code>)</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="id4">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">esm_residue_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'esm1b_t33_650M_UR50S'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_layer</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">33</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#esm_residue_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#id4" title="Permalink to this definition">#</a></dt>
<dd><p>Computes ESM residue embeddings from a protein sequence and adds the to the graph.</p>
<blockquote>
<div><p>Biological Structure and Function Emerge from Scaling Unsupervised Learning to 250 Million Protein Sequences (2019)
Rives, Alexander and Meier, Joshua and Sercu, Tom and Goyal, Siddharth and Lin, Zeming and Liu, Jason and Guo,
Demi and Ott, Myle and Zitnick, C. Lawrence and Ma, Jerry and Fergus, Rob</p>
<p>Transformer protein language models are unsupervised structure learners 2020
Rao, Roshan M and Meier, Joshua and Sercu, Tom and Ovchinnikov, Sergey and Rives, Alexander</p>
</div></blockquote>
<p><strong>Pre-trained models</strong></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> to add esm embedding to.</p></li>
<li><p><strong>model_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Name of pre-trained model to use.</p></li>
<li><p><strong>output_layer</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – index of output layer in pre-trained model.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> with esm embedding feature added to nodes.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="id5">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.sequence.embeddings.</span></span><span class="sig-name descname"><span class="pre">esm_sequence_embedding</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/sequence/embeddings.html#esm_sequence_embedding"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#id5" title="Permalink to this definition">#</a></dt>
<dd><p>Computes ESM sequence embedding feature over chains in a graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph protein structure graph with esm embedding features added eg. <code class="docutils literal notranslate"><span class="pre">G.graph["esm_embedding_A"]</span></code> for chain A.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.features.utils">
<span id="utils"></span><h3>Utils<a class="headerlink" href="#module-graphein.protein.features.utils" title="Permalink to this headline">#</a></h3>
<p>Utility functions to work with graph-level features.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.utils.aggregate_graph_feature_over_chains">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.utils.</span></span><span class="sig-name descname"><span class="pre">aggregate_graph_feature_over_chains</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">aggregation_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/utils.html#aggregate_graph_feature_over_chains"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.utils.aggregate_graph_feature_over_chains" title="Permalink to this definition">#</a></dt>
<dd><p>Performs aggregation of a feature over the chains. E.g. sums/averages/min/max molecular weights for each chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph of protein containing chain-specific features.</p></li>
<li><p><strong>feature_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Name of features to aggregate.</p></li>
<li><p><strong>aggregation_type</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Type of aggregation to perform (<code class="docutils literal notranslate"><span class="pre">"min"`,</span> <span class="pre">``"max"</span></code>, <code class="docutils literal notranslate"><span class="pre">"sum"</span></code>, <code class="docutils literal notranslate"><span class="pre">"mean"</span></code>).</p></li>
</ul>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><p><a class="reference external" href="https://docs.python.org/3/library/exceptions.html#NameError" title="(in Python v3.10)"><strong>NameError</strong></a> – If <code class="docutils literal notranslate"><span class="pre">aggregation_type</span></code> is not one of <code class="docutils literal notranslate"><span class="pre">"min"`,</span> <span class="pre">``"max"</span></code>, <code class="docutils literal notranslate"><span class="pre">"sum"</span></code>, <code class="docutils literal notranslate"><span class="pre">"mean"</span></code>.</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>nx.Graph of protein with a new aggregated feature <code class="docutils literal notranslate"><span class="pre">G.graph[f"{feature_name}_{aggregation_type}"]</span></code>.</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.features.utils.convert_graph_dict_feat_to_series">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.features.utils.</span></span><span class="sig-name descname"><span class="pre">convert_graph_dict_feat_to_series</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">networkx.classes.graph.Graph</span></span></span><a class="reference internal" href="../_modules/graphein/protein/features/utils.html#convert_graph_dict_feat_to_series"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.features.utils.convert_graph_dict_feat_to_series" title="Permalink to this definition">#</a></dt>
<dd><p>Takes in a graph and a graph-level <code class="docutils literal notranslate"><span class="pre">feature_name</span></code>. Converts this feature to a <code class="docutils literal notranslate"><span class="pre">pd.Series</span></code>.
This is useful as some features are output as dictionaries and we wish to standardise this.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph containing <code class="docutils literal notranslate"><span class="pre">G.graph[f"{feature_name}"]</span></code> (<code class="docutils literal notranslate"><span class="pre">Dict[Any,</span> <span class="pre">Any]</span></code>).</p></li>
<li><p><strong>feature_name</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Name of feature to convert to dictionary.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>nx.Graph containing <code class="docutils literal notranslate"><span class="pre">G.graph[f"{feature_name}"]:</span> <span class="pre">pd.Series</span></code>.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>nx.Graph</p>
</dd>
</dl>
</dd></dl>
</div>
</div>
<div class="section" id="module-graphein.protein.subgraphs">
<span id="subgraphs"></span><h2>Subgraphs<a class="headerlink" href="#module-graphein.protein.subgraphs" title="Permalink to this headline">#</a></h2>
<p>Provides functions for extracting subgraphs from protein graphs.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_k_hop_subgraph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_k_hop_subgraph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">central_node</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">k</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">k_only</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_k_hop_subgraph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_k_hop_subgraph" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a k-hop subgraph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>central_node</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – The central node to extract the subgraph from.</p></li>
<li><p><strong>k</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – The number of hops to extract.</p></li>
<li><p><strong>k_only</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to only extract the exact k-hop subgraph (e.g. include 2-hop neighbours in 5-hop graph). Defaults to False.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to filter the pdb_df of the graph, defaults to True</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to inverse the selection, defaults to False</p></li>
<li><p><strong>return_node_list</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to return the node list. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence_positions</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chains</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">residue_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bond_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centre_point</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.22)"><span class="pre">numpy.ndarray</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.10)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ss_elements</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rsa_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">k_hop_central_node</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">k_hops</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">k_only</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph from a graph based on a list of nodes, sequence positions, chains, residue types, atom types, centre point and radius.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>node_list</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – List of nodes to extract specified by their node_id. Defaults to None.</p></li>
<li><p><strong>sequence_positions</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>]</em><em>, </em><em>optional</em>) – The sequence positions to extract. Defaults to None.</p></li>
<li><p><strong>chains</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – The chain(s) to extract. Defaults to None.</p></li>
<li><p><strong>residue_types</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – List of allowable residue types (3 letter residue names). Defaults to None.</p></li>
<li><p><strong>atom_types</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – List of allowable atom types. Defaults to None.</p></li>
<li><p><strong>centre_point</strong> (<em>Union</em><em>[</em><em>np.ndarray</em><em>, </em><em>Tuple</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>, </em><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>, </em><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>]</em><em>]</em><em>, </em><em>optional</em>) – The centre point to extract the subgraph from. Defaults to None.</p></li>
<li><p><strong>radius</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>, </em><em>optional</em>) – The radius to extract the subgraph from. Defaults to None.</p></li>
<li><p><strong>ss_elements</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – List of secondary structure elements to extract. <cite>[“H”, “B”, “E”, “G”, “I”, “T”, “S”, “-“]</cite> corresponding to Alpha helix
Beta bridge, Strand, Helix-3, Helix-5, Turn, Bend, None. Defaults to None.</p></li>
<li><p><strong>rsa_threshold</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>, </em><em>optional</em>) – The threshold to use for the RSA. Defaults to None.</p></li>
<li><p><strong>central_node</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – The central node to extract the subgraph from. Defaults to None.</p></li>
<li><p><strong>k</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – The number of hops to extract.</p></li>
<li><p><strong>k_only</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to only extract the exact k-hop subgraph (e.g. include 2-hop neighbours in 5-hop graph). Defaults to False.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to filter the pdb_df dataframe of the graph. Defaults to True. Defaults to None.</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to inverse the selection. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph_by_bond_type">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph_by_bond_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bond_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph_by_bond_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph_by_bond_type" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph from a graph based on a list of allowable bond types.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>bond_types</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – List of allowable bond types.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to filter the pdb_df of the graph, defaults to True</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to inverse the selection, defaults to False</p></li>
<li><p><strong>return_node_list</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to return the node list, defaults to False</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph_by_sequence_position">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph_by_sequence_position</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sequence_positions</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph_by_sequence_position"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph_by_sequence_position" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph from a graph based on a chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>chain</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>]</em>) – The sequence positions to extract.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to filter the pdb_df dataframe of the graph. Defaults to True.</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to inverse the selection. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph_from_atom_types">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph_from_atom_types</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atom_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph_from_atom_types"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph_from_atom_types" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph from a graph based on a list of atom types.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>atom_types</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – The list of atom types to extract.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to filter the pdb_df dataframe of the graph. Defaults to True.</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to inverse the selection. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph_from_chains">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph_from_chains</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chains</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph_from_chains"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph_from_chains" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph from a graph based on a chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>chain</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – The chain(s) to extract.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to filter the pdb_df dataframe of the graph. Defaults to True.</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to inverse the selection. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph_from_node_list">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph_from_node_list</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph_from_node_list"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph_from_node_list" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph from a graph based on a list of nodes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>node_list</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – The list of nodes to extract.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to filter the pdb_df dataframe of the graph. Defaults to True.</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to inverse the selection. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph_from_point">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph_from_point</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">centre_point</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.22)"><span class="pre">numpy.ndarray</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.10)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">radius</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph_from_point"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph_from_point" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph from a graph based on a centre point and radius.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>centre_point</strong> (<em>Tuple</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>, </em><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>, </em><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>]</em>) – The centre point of the subgraph.</p></li>
<li><p><strong>radius</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – The radius of the subgraph.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to filter the pdb_df dataframe of the graph. Defaults to True.</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to inverse the selection. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph_from_residue_types">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph_from_residue_types</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">residue_types</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph_from_residue_types"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph_from_residue_types" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph from a graph based on a list of allowable residue types.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>residue_types</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – List of allowable residue types (3 letter residue names).</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to filer the pdb_df of the graph, defaults to True</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to inverse the selection. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_subgraph_from_secondary_structure">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_subgraph_from_secondary_structure</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ss_elements</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_subgraph_from_secondary_structure"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_subgraph_from_secondary_structure" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts subgraphs for nodes that have a secondary structure element in the list.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>ss_elements</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – List of secondary structure elements to extract.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to inverse the selection. Defaults to False.</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to filter the pdb_df of the graph, defaults to True</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>return_node_list</strong> – Whether to return the node list. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><p><a class="reference internal" href="#graphein.protein.utils.ProteinGraphConfigurationError" title="graphein.protein.utils.ProteinGraphConfigurationError"><strong>ProteinGraphConfigurationError</strong></a> – If the graph does not contain ss features on the nodes (<cite>d[‘ss’] not in d.keys() for _, d in g.nodes(data=True)</cite>).</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.subgraphs.extract_surface_subgraph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.subgraphs.</span></span><span class="sig-name descname"><span class="pre">extract_surface_subgraph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rsa_threshold</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">inverse</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">filter_dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">recompute_distmat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">update_coords</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">return_node_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/subgraphs.html#extract_surface_subgraph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.subgraphs.extract_surface_subgraph" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a subgraph based on thresholding the Relative Solvent Accessibility (RSA). This can be used for extracting a surface graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – The graph to extract the subgraph from.</p></li>
<li><p><strong>rsa_threshold</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – The threshold to use for the RSA. Defaults to 0.2 (20%)</p></li>
<li><p><strong>filter_dataframe</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to filter the pdb_df of the graph, defaults to True</p></li>
<li><p><strong>update_coords</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to update the coordinates of the graph. Defaults to True.</p></li>
<li><p><strong>recompute_distmat</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to recompute the distance matrix of the graph. Defaults to False.</p></li>
<li><p><strong>inverse</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether to inverse the selection, defaults to False</p></li>
<li><p><strong>return_node_list</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to return the node list. Defaults to False.</p></li>
</ul>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><p><a class="reference internal" href="#graphein.protein.utils.ProteinGraphConfigurationError" title="graphein.protein.utils.ProteinGraphConfigurationError"><strong>ProteinGraphConfigurationError</strong></a> – If the graph does not contain RSA features on the nodes (<cite>d[‘rsa’] not in d.keys() for _, d in g.nodes(data=True)</cite>).</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>The subgraph or node list if return_node_list is True.</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>Union[nx.Graph, List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]]</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.analysis">
<span id="analysis"></span><h2>Analysis<a class="headerlink" href="#module-graphein.protein.analysis" title="Permalink to this headline">#</a></h2>
<p>Contains utilities for computing analytics on and plotting summaries of Protein Structure Graphs.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.analysis.graph_summary">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.analysis.</span></span><span class="sig-name descname"><span class="pre">graph_summary</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">summary_statistics</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">['degree',</span> <span class="pre">'betweenness_centrality',</span> <span class="pre">'closeness_centrality',</span> <span class="pre">'eigenvector_centrality',</span> <span class="pre">'communicability_betweenness_centrality']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">custom_data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Union" title="(in Python v3.10)"><span class="pre">Union</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.Series.html#pandas.Series" title="(in pandas v1.4.1)"><span class="pre">pandas.core.series.Series</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/analysis.html#graph_summary"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.analysis.graph_summary" title="Permalink to this definition">#</a></dt>
<dd><p>Returns a summary of the graph in a dataframe.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – NetworkX graph to get summary of.</p></li>
<li><p><strong>plot</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether or not to plot the summary as a heatmap, defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Dataframe of summary or plot.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.analysis.plot_degree_by_residue_type">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.analysis.</span></span><span class="sig-name descname"><span class="pre">plot_degree_by_residue_type</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">nx.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">normalise_by_residue_occurrence</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objects.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/analysis.html#plot_degree_by_residue_type"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.analysis.plot_degree_by_residue_type" title="Permalink to this definition">#</a></dt>
<dd><p>Plots the distribution of node degrees in the graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – networkx graph to plot the distribution of node degrees by residue type of.</p></li>
<li><p><strong>normalise_by_residue_occurrence</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to normalise the degree by the number of residues of the same type.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Plotly figure.</p>
</dd>
<dt class="field-odd">Rtpe</dt>
<dd class="field-odd"><p>plotly.graph_objects.Figure</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.analysis.plot_degree_distribution">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.analysis.</span></span><span class="sig-name descname"><span class="pre">plot_degree_distribution</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">nx.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objects.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/analysis.html#plot_degree_distribution"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.analysis.plot_degree_distribution" title="Permalink to this definition">#</a></dt>
<dd><p>Plots the distribution of node degrees in the graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – networkx graph to plot the distribution of node degrees in.</p></li>
<li><p><strong>title</strong> (<em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – Title of plot. defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Plotly figure.</p>
</dd>
<dt class="field-odd">Rtpe</dt>
<dd class="field-odd"><p>plotly.graph_objects.Figure</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.analysis.plot_edge_type_distribution">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.analysis.</span></span><span class="sig-name descname"><span class="pre">plot_edge_type_distribution</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">nx.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'bar'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objects.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/analysis.html#plot_edge_type_distribution"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.analysis.plot_edge_type_distribution" title="Permalink to this definition">#</a></dt>
<dd><p>Plots the distribution of edge types in the graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – NetworkX graph to plot the distribution of edge types in.</p></li>
<li><p><strong>plot_type</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – Type of plot to produce, defaults to <code class="docutils literal notranslate"><span class="pre">"bar"</span></code>. One of <code class="docutils literal notranslate"><span class="pre">"bar"</span></code>, <code class="docutils literal notranslate"><span class="pre">"pie"</span></code>.</p></li>
<li><p><strong>title</strong> (<em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – Title of plot. defaults to None</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Plotly figure.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>plotly.graph_objects.Figure</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.analysis.plot_graph_metric_property_correlation">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.analysis.</span></span><span class="sig-name descname"><span class="pre">plot_graph_metric_property_correlation</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">nx.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">summary_statistics</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">['degree',</span> <span class="pre">'betweenness_centrality',</span> <span class="pre">'closeness_centrality',</span> <span class="pre">'eigenvector_centrality',</span> <span class="pre">'communicability_betweenness_centrality']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">properties</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">['asa']</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_by</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'residue_type'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">opacity</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diagonal_visible</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">font_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objects.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/analysis.html#plot_graph_metric_property_correlation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.analysis.plot_graph_metric_property_correlation" title="Permalink to this definition">#</a></dt>
<dd><p>Plots the correlation between graph metrics and properties.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – Protein graph to plot the correlation of.</p></li>
<li><p><strong>summary_statistics</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – List of graph metrics to employ in plot, defaults to
<code class="docutils literal notranslate"><span class="pre">["degree",</span> <span class="pre">"betweenness_centrality",</span> <span class="pre">"closeness_centrality",</span> <span class="pre">"eigenvector_centrality",</span> <span class="pre">"communicability_betweenness_centrality"]</span></code>.</p></li>
<li><p><strong>properties</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – List of node properties to use in plot, defaults to <code class="docutils literal notranslate"><span class="pre">["asa"]</span></code>.</p></li>
<li><p><strong>colour_by</strong> (<em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – Controls colouring of points in plot. Options: <code class="docutils literal notranslate"><span class="pre">"residue_type"</span></code>, <code class="docutils literal notranslate"><span class="pre">"position"</span></code>, <code class="docutils literal notranslate"><span class="pre">"chain"</span></code>, defaults to <code class="docutils literal notranslate"><span class="pre">"residue_type"</span></code>.</p></li>
<li><p><strong>opacity</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>, </em><em>optional</em>) – Opacity of plot points, defaults to <code class="docutils literal notranslate"><span class="pre">0.2</span></code>.</p></li>
<li><p><strong>diagonal_visible</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether or not to show the diagonal plots, defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>title</strong> (<em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – Title of plot, defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>height</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>, </em><em>optional</em>) – Height of plot, defaults to <code class="docutils literal notranslate"><span class="pre">1000</span></code>.</p></li>
<li><p><strong>width</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>, </em><em>optional</em>) – Width of plot, defaults to <code class="docutils literal notranslate"><span class="pre">1000</span></code>.</p></li>
<li><p><strong>font_size</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>, </em><em>optional</em>) – Font size for plot text, defaults to <code class="docutils literal notranslate"><span class="pre">10</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Scatter plot matrix of graph metrics and protein properties.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>plotly.graph_objects.Figure</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.analysis.plot_residue_composition">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.analysis.</span></span><span class="sig-name descname"><span class="pre">plot_residue_composition</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">nx.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sort_by</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'bar'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objects.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/analysis.html#plot_residue_composition"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.analysis.plot_residue_composition" title="Permalink to this definition">#</a></dt>
<dd><p>Plots the residue composition of the graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – Protein graph to plot the residue composition of.</p></li>
<li><p><strong>sort_by</strong> (<em>Optional</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em><em>, </em><em>optional</em>) – How to sort the values (<code class="docutils literal notranslate"><span class="pre">"alphabetical"</span></code>, <code class="docutils literal notranslate"><span class="pre">"count"</span></code>), defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code> (no sorting).</p></li>
<li><p><strong>plot_type</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – How to plot the composition (<code class="docutils literal notranslate"><span class="pre">"bar"</span></code>, <code class="docutils literal notranslate"><span class="pre">"pie"</span></code>), defaults to <code class="docutils literal notranslate"><span class="pre">"bar"</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><p><a class="reference external" href="https://docs.python.org/3/library/exceptions.html#ValueError" title="(in Python v3.10)"><strong>ValueError</strong></a> – Raises ValueError if <code class="docutils literal notranslate"><span class="pre">sort_by</span></code> is not one of <code class="docutils literal notranslate"><span class="pre">"alphabetical"</span></code>, <code class="docutils literal notranslate"><span class="pre">"count"</span></code>.</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>Plotly figure.</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p>plotly.graph_objects.Figure</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.meshes">
<span id="meshes"></span><h2>Meshes<a class="headerlink" href="#module-graphein.protein.meshes" title="Permalink to this headline">#</a></h2>
<p>Functions to create protein meshes via pymol.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.check_for_pymol_installation">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">check_for_pymol_installation</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#check_for_pymol_installation"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.check_for_pymol_installation" title="Permalink to this definition">#</a></dt>
<dd><p>Checks for presence of a pymol installation</p>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.configure_pymol_session">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">configure_pymol_session</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><span class="pre">graphein.protein.config.ProteinMeshConfig</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#configure_pymol_session"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.configure_pymol_session" title="Permalink to this definition">#</a></dt>
<dd><p>Configures a PyMol session based on <code class="docutils literal notranslate"><span class="pre">config.parse_pymol_commands</span></code>. Uses default parameters <code class="docutils literal notranslate"><span class="pre">"-cKq"</span></code>.</p>
<p>See: <a class="reference external" href="https://pymolwiki.org/index.php/Command_Line_Options">https://pymolwiki.org/index.php/Command_Line_Options</a></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><em>graphein.protein.config.ProteinMeshConfig</em></a>) – <a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMeshConfig</span></code></a> to use. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code> which uses default config.</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.convert_verts_and_face_to_mesh">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">convert_verts_and_face_to_mesh</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">verts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">torch.FloatTensor</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">faces</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">NamedTuple</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Meshes</span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#convert_verts_and_face_to_mesh"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.convert_verts_and_face_to_mesh" title="Permalink to this definition">#</a></dt>
<dd><p>Converts vertices and faces into a <code class="docutils literal notranslate"><span class="pre">pytorch3d.structures</span></code> Meshes object.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>verts</strong> (<em>torch.FloatTensor</em>) – Vertices.</p></li>
<li><p><strong>faces</strong> (<em>NamedTuple</em>) – Faces.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Meshes object.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pytorch3d.structures.Meshes</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.create_mesh">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">create_mesh</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><span class="pre">ProteinMeshConfig</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">torch.FloatTensor</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">NamedTuple</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">NamedTuple</span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#create_mesh"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.create_mesh" title="Permalink to this definition">#</a></dt>
<dd><p>Creates a <code class="docutils literal notranslate"><span class="pre">PyTorch3D</span></code> mesh from a <code class="docutils literal notranslate"><span class="pre">pdb_file</span></code> or <code class="docutils literal notranslate"><span class="pre">pdb_code</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb_file</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – path to <code class="docutils literal notranslate"><span class="pre">pdb_file</span></code>. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>pdb_code</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – 4-letter PDB accession code. Defaults to None.</p></li>
<li><p><strong>out_dir</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – output directory to store <code class="docutils literal notranslate"><span class="pre">.obj</span></code> file. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><em>graphein.protein.config.ProteinMeshConfig</em></a>) – <a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMeshConfig</span></code></a> config to use. Defaults to default config in <code class="docutils literal notranslate"><span class="pre">graphein.protein.config</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p><code class="docutils literal notranslate"><span class="pre">verts</span></code>, <code class="docutils literal notranslate"><span class="pre">faces</span></code>, <code class="docutils literal notranslate"><span class="pre">aux</span></code>.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Tuple[torch.Tensor, torch.Tensor, torch.Tensor]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.get_obj_file">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">get_obj_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><span class="pre">graphein.protein.config.ProteinMeshConfig</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#get_obj_file"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.get_obj_file" title="Permalink to this definition">#</a></dt>
<dd><p>Runs PyMol to compute surface/mesh for a given protein.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>pdb_file</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – path to <code class="docutils literal notranslate"><span class="pre">pdb_file</span></code> to use. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>pdb_code</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – 4-letter pdb accession code. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>out_dir</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – path to output. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><em>graphein.protein.config.ProteinMeshConfig</em></a>) – <a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><code class="xref py py-class docutils literal notranslate"><span class="pre">ProteinMeshConfig</span></code></a> containing pymol commands to run. Default is <code class="docutils literal notranslate"><span class="pre">None</span></code> (<code class="docutils literal notranslate"><span class="pre">"show</span> <span class="pre">surface"</span></code>).</p></li>
</ul>
</dd>
<dt class="field-even">Raises</dt>
<dd class="field-even"><p>ValueError if both or neither <code class="docutils literal notranslate"><span class="pre">pdb_file</span></code> or <code class="docutils literal notranslate"><span class="pre">pdb_code</span></code> are provided.</p>
</dd>
<dt class="field-odd">Returns</dt>
<dd class="field-odd"><p>returns path to <code class="docutils literal notranslate"><span class="pre">.obj</span></code> file (str)</p>
</dd>
<dt class="field-even">Return type</dt>
<dd class="field-even"><p><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a></p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.normalize_and_center_mesh_vertices">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">normalize_and_center_mesh_vertices</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">verts</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">torch.FloatTensor</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">torch.FloatTensor</span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#normalize_and_center_mesh_vertices"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.normalize_and_center_mesh_vertices" title="Permalink to this definition">#</a></dt>
<dd><p>We scale normalize and center the target mesh to fit in a sphere of radius 1 centered at <code class="docutils literal notranslate"><span class="pre">(0,0,0)</span></code>.</p>
<p><code class="docutils literal notranslate"><span class="pre">(scale,</span> <span class="pre">center)</span></code> will be used to bring the predicted mesh to its original center and scale
Note that normalizing the target mesh, speeds up the optimization but is not necessary!</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>verts</strong> (<em>torch.FloatTensor</em>) – Mesh vertices.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Normalized and centered vertices.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>torch.FloatTensor</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.parse_pymol_commands">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">parse_pymol_commands</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><span class="pre">graphein.protein.config.ProteinMeshConfig</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#parse_pymol_commands"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.parse_pymol_commands" title="Permalink to this definition">#</a></dt>
<dd><p>Parses pymol commands from config. At the moment users can only supply a list of string commands.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>config</strong> (<a class="reference internal" href="#graphein.protein.config.ProteinMeshConfig" title="graphein.protein.config.ProteinMeshConfig"><em>ProteinMeshConfig</em></a>) – ProteinMeshConfig containing pymol commands to run in <code class="docutils literal notranslate"><span class="pre">config.pymol_commands</span></code>.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>list of pymol commands to run</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>List[<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a>]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.meshes.run_pymol_commands">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.meshes.</span></span><span class="sig-name descname"><span class="pre">run_pymol_commands</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">commands</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/constants.html#None" title="(in Python v3.10)"><span class="pre">None</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/meshes.html#run_pymol_commands"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.meshes.run_pymol_commands" title="Permalink to this definition">#</a></dt>
<dd><p>Runs Pymol Commands.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>commands</strong> (<em>List</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>]</em>) – List of commands to pass to PyMol.</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.visualisation">
<span id="visualisation"></span><h2>Visualisation<a class="headerlink" href="#module-graphein.protein.visualisation" title="Permalink to this headline">#</a></h2>
<p>Functions for plotting protein graphs and meshes.</p>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.add_vector_to_plot">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">add_vector_to_plot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">fig</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">vector</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'sidechain_vector'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'red'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">10</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objs._figure.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#add_vector_to_plot"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.add_vector_to_plot" title="Permalink to this definition">#</a></dt>
<dd><p>Adds representations of vector features to the protein graph.</p>
<p>Requires all nodes have a vector feature (1 x 3 array).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – Protein graph containing vector features</p></li>
<li><p><strong>fig</strong> (<em>go.Figure</em>) – 3D plotly figure to add vectors to.</p></li>
<li><p><strong>vector</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – Name of node vector feature to add, defaults to “sidechain_vector”</p></li>
<li><p><strong>scale</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a><em>, </em><em>optional</em>) – How much to scale the vectors by, defaults to 5</p></li>
<li><p><strong>colour</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – Colours for vectors, defaults to “red”</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>3D Plotly plot with vectors added.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>go.Figure</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.asteroid_plot">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">asteroid_plot</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g:</span> <span class="pre">nx.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_id:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">k:</span> <span class="pre">int</span> <span class="pre">=</span> <span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_nodes_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'shell'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_edges_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'kind'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_colour_map:</span> <span class="pre">plt.cm.Colormap</span> <span class="pre">=</span> <span class="pre">&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_labels:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title:</span> <span class="pre">Optional[str]</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width:</span> <span class="pre">int</span> <span class="pre">=</span> <span class="pre">600</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height:</span> <span class="pre">int</span> <span class="pre">=</span> <span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_plotly:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_edges:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size_multiplier:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">10</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Union</span><span class="p"><span class="pre">[</span></span><span class="pre">plotly.graph_objects.Figure</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.Figure" title="(in Matplotlib v3.5.1)"><span class="pre">matplotlib.figure.Figure</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#asteroid_plot"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.asteroid_plot" title="Permalink to this definition">#</a></dt>
<dd><p>Plots a k-hop subgraph around a node as concentric shells.</p>
<p>Radius of each point is proportional to the degree of the node (modified by node_size_multiplier).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – NetworkX graph to plot.</p></li>
<li><p><strong>node_id</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Node to centre the plot around.</p></li>
<li><p><strong>k</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – Number of hops to plot. Defaults to <code class="docutils literal notranslate"><span class="pre">2</span></code>.</p></li>
<li><p><strong>colour_nodes_by</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Colour the nodes by this attribute. Currently only <code class="docutils literal notranslate"><span class="pre">"shell"</span></code> is supported.</p></li>
<li><p><strong>colour_edges_by</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Colour the edges by this attribute. Currently only <code class="docutils literal notranslate"><span class="pre">"kind"</span></code> is supported.</p></li>
<li><p><strong>edge_colour_map</strong> (<em>plt.cm.Colormap</em>) – Colour map for edges. Defaults to <code class="docutils literal notranslate"><span class="pre">plt.cm.plasma</span></code>.</p></li>
<li><p><strong>title</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Title of the plot. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>width</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – Width of the plot. Defaults to <code class="docutils literal notranslate"><span class="pre">600</span></code>.</p></li>
<li><p><strong>height</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – Height of the plot. Defaults to <code class="docutils literal notranslate"><span class="pre">500</span></code>.</p></li>
<li><p><strong>use_plotly</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Use plotly to render the graph. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>show_edges</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether or not to show edges in the plot. Defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
<li><p><strong>node_size_multiplier</strong> (<em>float.</em>) – Multiplier for the size of the nodes. Defaults to <code class="docutils literal notranslate"><span class="pre">10</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Plotly figure or matplotlib figure.</p>
</dd>
<dt class="field-odd">Rtpye</dt>
<dd class="field-odd"><p>Union[plotly.graph_objects.Figure, matplotlib.figure.Figure]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.colour_edges">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">colour_edges</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_map</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.ListedColormap.html#matplotlib.colors.ListedColormap" title="(in Matplotlib v3.5.1)"><span class="pre">matplotlib.colors.ListedColormap</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_by</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'kind'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.10)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#colour_edges"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.colour_edges" title="Permalink to this definition">#</a></dt>
<dd><p>Computes edge colours based on the kind of bond/interaction.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph protein structure graph to compute edge colours from.</p></li>
<li><p><strong>colour_map</strong> (<a class="reference external" href="https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.ListedColormap.html#matplotlib.colors.ListedColormap" title="(in Matplotlib v3.5.1)"><em>matplotlib.colors.ListedColormap</em></a>) – Colourmap to use.</p></li>
<li><p><strong>colour_by</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Edge attribute to colour by. Currently only <code class="docutils literal notranslate"><span class="pre">"kind"</span></code> is supported.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>List of edge colours.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>List[Tuple[<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.colour_nodes">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">colour_nodes</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G:</span> <span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_by:</span> <span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_map:</span> <span class="pre">matplotlib.colors.ListedColormap</span> <span class="pre">=</span> <span class="pre">&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Tuple" title="(in Python v3.10)"><span class="pre">Tuple</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#colour_nodes"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.colour_nodes" title="Permalink to this definition">#</a></dt>
<dd><p>Computes node colours based on <code class="docutils literal notranslate"><span class="pre">"degree"</span></code>, <code class="docutils literal notranslate"><span class="pre">"seq_position"</span></code> or node attributes.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – Graph to compute node colours for</p></li>
<li><p><strong>colour_map</strong> (<a class="reference external" href="https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.ListedColormap.html#matplotlib.colors.ListedColormap" title="(in Matplotlib v3.5.1)"><em>matplotlib.colors.ListedColormap</em></a>) – Colourmap to use.</p></li>
<li><p><strong>colour_by</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Manner in which to colour nodes. If not <code class="docutils literal notranslate"><span class="pre">"degree"</span></code> or <code class="docutils literal notranslate"><span class="pre">"seq_position"</span></code>, this must correspond to a node feature.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>List of node colours</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>List[Tuple[<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>, <a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)">float</a>]]</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.plot_distance_landscape">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">plot_distance_landscape</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_mat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.22)"><span class="pre">numpy.ndarray</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">add_contour</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">width</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">height</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">500</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">autosize</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objs._figure.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#plot_distance_landscape"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.plot_distance_landscape" title="Permalink to this definition">#</a></dt>
<dd><p>Plots a distance landscape of the graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – Graph to plot (must contain a distance matrix in <code class="docutils literal notranslate"><span class="pre">g.graph["dist_mat"]</span></code>).</p></li>
<li><p><strong>add_contour</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether or not to show the contour, defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>width</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>, </em><em>optional</em>) – Plot width, defaults to <code class="docutils literal notranslate"><span class="pre">500</span></code>.</p></li>
<li><p><strong>height</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>, </em><em>optional</em>) – Plot height, defaults to <code class="docutils literal notranslate"><span class="pre">500</span></code>.</p></li>
<li><p><strong>autosize</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a><em>, </em><em>optional</em>) – Whether or not to autosize the plot, defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Plotly figure of distance landscape.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>go.Figure</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.plot_distance_matrix">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">plot_distance_matrix</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><span class="pre">networkx.classes.graph.Graph</span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">dist_mat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html#numpy.ndarray" title="(in NumPy v1.22)"><span class="pre">numpy.ndarray</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">use_plotly</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Optional" title="(in Python v3.10)"><span class="pre">Optional</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">show_residue_labels</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objs._figure.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#plot_distance_matrix"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.plot_distance_matrix" title="Permalink to this definition">#</a></dt>
<dd><p>Plots a distance matrix of the graph.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em><em>, </em><em>optional</em>) – NetworkX graph containing a distance matrix as a graph attribute (<code class="docutils literal notranslate"><span class="pre">g.graph['dist_mat']</span></code>).</p></li>
<li><p><strong>dist_mat</strong> (<em>np.ndarray</em><em>, </em><em>optional</em>) – Distance matrix to plot. If not provided, the distance matrix is taken from the graph. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>use_plotly</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to use <code class="docutils literal notranslate"><span class="pre">plotly</span></code> or <code class="docutils literal notranslate"><span class="pre">seaborn</span></code> for plotting. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>title</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – Title of the plot.Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
</ul>
</dd>
<dt class="field-even">Show_residue_labels</dt>
<dd class="field-even"><p>Whether to show residue labels on the plot. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p>
</dd>
<dt class="field-odd">Raises</dt>
<dd class="field-odd"><p>ValueError if neither a graph <code class="docutils literal notranslate"><span class="pre">g</span></code> or a <code class="docutils literal notranslate"><span class="pre">dist_mat</span></code> are provided.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Plotly figure.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>px.Figure</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.plot_pointcloud">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">plot_pointcloud</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">mesh</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Meshes</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">title</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">Axes3D</span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#plot_pointcloud"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.plot_pointcloud" title="Permalink to this definition">#</a></dt>
<dd><p>Plots pytorch3d Meshes object as pointcloud.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>mesh</strong> (<em>pytorch3d.structures.meshes.Meshes</em>) – Meshes object to plot.</p></li>
<li><p><strong>title</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Title of plot.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>returns Axes3D containing plot</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Axes3D</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.plot_protein_structure_graph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">plot_protein_structure_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G:</span> <span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">angle:</span> <span class="pre">int</span> <span class="pre">=</span> <span class="pre">30</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_title:</span> <span class="pre">typing.Optional[str]</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize:</span> <span class="pre">typing.Tuple[int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">int]</span> <span class="pre">=</span> <span class="pre">(10</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">7)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_alpha:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">0.7</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size_min:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">20.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size_multiplier:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">20.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">label_node_ids:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_colour_map=&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_color_map=&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_nodes_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'degree'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_edges_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'kind'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_alpha:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_style:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'ggplot'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_path:</span> <span class="pre">typing.Optional[str]</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_format:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'.png'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://matplotlib.org/stable/api/_as_gen/mpl_toolkits.mplot3d.axes3d.Axes3D.html#mpl_toolkits.mplot3d.axes3d.Axes3D" title="(in Matplotlib v3.5.1)"><span class="pre">mpl_toolkits.mplot3d.axes3d.Axes3D</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#plot_protein_structure_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.plot_protein_structure_graph" title="Permalink to this definition">#</a></dt>
<dd><p>Plots protein structure graph in <code class="docutils literal notranslate"><span class="pre">Axes3D</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph Protein Structure graph to plot.</p></li>
<li><p><strong>angle</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a>) – View angle. Defaults to <code class="docutils literal notranslate"><span class="pre">30</span></code>.</p></li>
<li><p><strong>plot_title</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – Title of plot. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>figsize</strong> (<em>Tuple</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>, </em><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>]</em>) – Size of figure, defaults to <code class="docutils literal notranslate"><span class="pre">(10,</span> <span class="pre">7)</span></code>.</p></li>
<li><p><strong>node_alpha</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Controls node transparency, defaults to <code class="docutils literal notranslate"><span class="pre">0.7</span></code>.</p></li>
<li><p><strong>node_size_min</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Specifies node minimum size, defaults to <code class="docutils literal notranslate"><span class="pre">20</span></code>.</p></li>
<li><p><strong>node_size_multiplier</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Scales node size by a constant. Node sizes reflect degree. Defaults to <code class="docutils literal notranslate"><span class="pre">20</span></code>.</p></li>
<li><p><strong>label_node_ids</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – bool indicating whether or not to plot <code class="docutils literal notranslate"><span class="pre">node_id</span></code> labels. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>node_colour_map</strong> (<em>plt.cm</em>) – colour map to use for nodes. Defaults to <code class="docutils literal notranslate"><span class="pre">plt.cm.plasma</span></code>.</p></li>
<li><p><strong>edge_color_map</strong> (<em>plt.cm</em>) – colour map to use for edges. Defaults to <code class="docutils literal notranslate"><span class="pre">plt.cm.plasma</span></code>.</p></li>
<li><p><strong>colour_nodes_by</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Specifies how to colour nodes. <code class="docutils literal notranslate"><span class="pre">"degree"</span></code>, <code class="docutils literal notranslate"><span class="pre">"seq_position"</span></code> or a node feature.</p></li>
<li><p><strong>colour_edges_by</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Specifies how to colour edges. Currently only <code class="docutils literal notranslate"><span class="pre">"kind"</span></code> is supported.</p></li>
<li><p><strong>edge_alpha</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Controls edge transparency. Defaults to <code class="docutils literal notranslate"><span class="pre">0.5</span></code>.</p></li>
<li><p><strong>plot_style</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – matplotlib style sheet to use. Defaults to <code class="docutils literal notranslate"><span class="pre">"ggplot"</span></code>.</p></li>
<li><p><strong>out_path</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – If not none, writes plot to this location. Defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code> (does not save).</p></li>
<li><p><strong>out_format</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Fileformat to use for plot</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>matplotlib Axes3D object.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>Axes3D</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.visualisation.plotly_protein_structure_graph">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.visualisation.</span></span><span class="sig-name descname"><span class="pre">plotly_protein_structure_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">G:</span> <span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">plot_title:</span> <span class="pre">typing.Optional[str]</span> <span class="pre">=</span> <span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">figsize:</span> <span class="pre">typing.Tuple[int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">int]</span> <span class="pre">=</span> <span class="pre">(620</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">650)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_alpha:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">0.7</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size_min:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">20.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_size_multiplier:</span> <span class="pre">float</span> <span class="pre">=</span> <span class="pre">20.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">label_node_ids:</span> <span class="pre">bool</span> <span class="pre">=</span> <span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">node_colour_map=&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">edge_color_map=&lt;matplotlib.colors.ListedColormap</span> <span class="pre">object&gt;</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_nodes_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'degree'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">colour_edges_by:</span> <span class="pre">str</span> <span class="pre">=</span> <span class="pre">'kind'</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><span class="pre">plotly.graph_objs._figure.Figure</span></span></span><a class="reference internal" href="../_modules/graphein/protein/visualisation.html#plotly_protein_structure_graph"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.visualisation.plotly_protein_structure_graph" title="Permalink to this definition">#</a></dt>
<dd><p>Plots protein structure graph using plotly.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>G</strong> (<em>nx.Graph</em>) – nx.Graph Protein Structure graph to plot</p></li>
<li><p><strong>plot_title</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a><em>, </em><em>optional</em>) – Title of plot, defaults to <code class="docutils literal notranslate"><span class="pre">None</span></code>.</p></li>
<li><p><strong>figsize</strong> (<em>Tuple</em><em>[</em><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>, </em><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><em>int</em></a><em>]</em>) – Size of figure, defaults to <code class="docutils literal notranslate"><span class="pre">(620,</span> <span class="pre">650)</span></code>.</p></li>
<li><p><strong>node_alpha</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Controls node transparency, defaults to <code class="docutils literal notranslate"><span class="pre">0.7</span></code>.</p></li>
<li><p><strong>node_size_min</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Specifies node minimum size. Defaults to <code class="docutils literal notranslate"><span class="pre">20.0</span></code>.</p></li>
<li><p><strong>node_size_multiplier</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><em>float</em></a>) – Scales node size by a constant. Node sizes reflect degree. Defaults to <code class="docutils literal notranslate"><span class="pre">20.0</span></code>.</p></li>
<li><p><strong>label_node_ids</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – bool indicating whether or not to plot <code class="docutils literal notranslate"><span class="pre">node_id</span></code> labels. Defaults to <code class="docutils literal notranslate"><span class="pre">True</span></code>.</p></li>
<li><p><strong>node_colour_map</strong> (<em>plt.cm</em>) – colour map to use for nodes. Defaults to <code class="docutils literal notranslate"><span class="pre">plt.cm.plasma</span></code>.</p></li>
<li><p><strong>edge_color_map</strong> (<em>plt.cm</em>) – colour map to use for edges. Defaults to <code class="docutils literal notranslate"><span class="pre">plt.cm.plasma</span></code>.</p></li>
<li><p><strong>colour_nodes_by</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Specifies how to colour nodes. <code class="docutils literal notranslate"><span class="pre">"degree"</span></code>, <code class="docutils literal notranslate"><span class="pre">"seq_position"</span></code> or a node feature.</p></li>
<li><p><strong>colour_edges_by</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Specifies how to colour edges. Currently only <code class="docutils literal notranslate"><span class="pre">"kind"</span></code> is supported.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Plotly Graph Objects plot</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>go.Figure</p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="id6">
<h2>Utils<a class="headerlink" href="#id6" title="Permalink to this headline">#</a></h2>
<span class="target" id="module-graphein.protein.utils"></span><p>Provides utility functions for use across Graphein.</p>
<dl class="py exception">
<dt class="sig sig-object py" id="graphein.protein.utils.ProteinGraphConfigurationError">
<em class="property"><span class="pre">exception</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">ProteinGraphConfigurationError</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">message</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/utils.html#ProteinGraphConfigurationError"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.ProteinGraphConfigurationError" title="Permalink to this definition">#</a></dt>
<dd><p>Exception when an invalid Graph configuration if provided to a downstream function or method.</p>
<dl class="py method">
<dt class="sig sig-object py" id="graphein.protein.utils.ProteinGraphConfigurationError.__init__">
<span class="sig-name descname"><span class="pre">__init__</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">message</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/utils.html#ProteinGraphConfigurationError.__init__"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.ProteinGraphConfigurationError.__init__" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.compute_rgroup_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">compute_rgroup_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#compute_rgroup_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.compute_rgroup_dataframe" title="Permalink to this definition">#</a></dt>
<dd><p>Return the atoms that are in R-groups and not the backbone chain.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pdb_df</strong> (<em>pd.DataFrame</em>) – DataFrame to compute R group dataframe from.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Dataframe containing R-groups only (backbone atoms removed).</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.download_pdb">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">download_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">config</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">pdb_code</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/pathlib.html#pathlib.Path" title="(in Python v3.10)"><span class="pre">pathlib.Path</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#download_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.download_pdb" title="Permalink to this definition">#</a></dt>
<dd><p>Download PDB structure from PDB.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pdb_code</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – 4 character PDB accession code.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>returns filepath to downloaded structure.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a></p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.filter_dataframe">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">filter_dataframe</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">dataframe</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">by_column</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">list_of_values</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.List" title="(in Python v3.10)"><span class="pre">List</span></a><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/typing.html#typing.Any" title="(in Python v3.10)"><span class="pre">Any</span></a><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">boolean</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#filter_dataframe"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.filter_dataframe" title="Permalink to this definition">#</a></dt>
<dd><p>Filter function for dataframe.</p>
<p>Filters the dataframe such that the <code class="docutils literal notranslate"><span class="pre">by_column</span></code> values have to be
in the <code class="docutils literal notranslate"><span class="pre">list_of_values</span></code> list if <code class="docutils literal notranslate"><span class="pre">boolean</span> <span class="pre">==</span> <span class="pre">True</span></code>, or not in the list
if <code class="docutils literal notranslate"><span class="pre">boolean</span> <span class="pre">==</span> <span class="pre">False</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>dataframe</strong> (<em>pd.DataFrame</em>) – pd.DataFrame to filter.</p></li>
<li><p><strong>by_column</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – str denoting column of dataframe to filter.</p></li>
<li><p><strong>list_of_values</strong> (<em>List</em><em>[</em><em>Any</em><em>]</em>) – List of values to filter with.</p></li>
<li><p><strong>boolean</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – indicates whether to keep or exclude matching <code class="docutils literal notranslate"><span class="pre">list_of_values</span></code>. <code class="docutils literal notranslate"><span class="pre">True</span></code> -&gt; in list, <code class="docutils literal notranslate"><span class="pre">False</span></code> -&gt; not in list.</p></li>
</ul>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>Filtered dataframe.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.get_protein_name_from_filename">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">get_protein_name_from_filename</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">pdb_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#get_protein_name_from_filename"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.get_protein_name_from_filename" title="Permalink to this definition">#</a></dt>
<dd><p>Extracts a filename from a <code class="docutils literal notranslate"><span class="pre">pdb_path</span></code></p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>pdb_path</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Path to extract filename from.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>file name.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a></p>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.save_graph_to_pdb">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">save_graph_to_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">gz</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/utils.html#save_graph_to_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.save_graph_to_pdb" title="Permalink to this definition">#</a></dt>
<dd><p>Saves processed <code class="docutils literal notranslate"><span class="pre">pdb_df</span></code> (<code class="docutils literal notranslate"><span class="pre">g.graph["pdb_df"]</span></code>) dataframe to a PDB file.</p>
<p>N.B. PDBs do not contain connectivity information.
This only captures the nodes in the graph.
Connectivity is filled in according to standard rules by visualisation programs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – Protein graph to save dataframe from.</p></li>
<li><p><strong>path</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Path to save PDB file to.</p></li>
<li><p><strong>gz</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to gzip the file. Defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.save_pdb_df_to_pdb">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">save_pdb_df_to_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">df</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html#pandas.DataFrame" title="(in pandas v1.4.1)"><span class="pre">pandas.core.frame.DataFrame</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">gz</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/utils.html#save_pdb_df_to_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.save_pdb_df_to_pdb" title="Permalink to this definition">#</a></dt>
<dd><p>Saves pdb dataframe to a PDB file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>pd.DataFrame</em>) – Dataframe to save as PDB</p></li>
<li><p><strong>path</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Path to save PDB file to.</p></li>
<li><p><strong>gz</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to gzip the file. Defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.save_rgroup_df_to_pdb">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">save_rgroup_df_to_pdb</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">networkx.classes.graph.Graph</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">gz</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><span class="pre">bool</span></a></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../_modules/graphein/protein/utils.html#save_rgroup_df_to_pdb"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.save_rgroup_df_to_pdb" title="Permalink to this definition">#</a></dt>
<dd><p>Saves R-group (<code class="docutils literal notranslate"><span class="pre">g.graph["rgroup_df"]</span></code>) dataframe to a PDB file.</p>
<p>N.B. PDBs do not contain connectivity information.
This only captures the atoms in the r groups.
Connectivity is filled in according to standard rules by visualisation programs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>g</strong> (<em>nx.Graph</em>) – Protein graph to save R group dataframe from.</p></li>
<li><p><strong>path</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Path to save PDB file to.</p></li>
<li><p><strong>gz</strong> (<a class="reference external" href="https://docs.python.org/3/library/functions.html#bool" title="(in Python v3.10)"><em>bool</em></a>) – Whether to gzip the file. Defaults to <code class="docutils literal notranslate"><span class="pre">False</span></code>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>
<dl class="py function">
<dt class="sig sig-object py" id="graphein.protein.utils.three_to_one_with_mods">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.utils.</span></span><span class="sig-name descname"><span class="pre">three_to_one_with_mods</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">res</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">→</span> <span class="sig-return-typehint"><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a></span></span><a class="reference internal" href="../_modules/graphein/protein/utils.html#three_to_one_with_mods"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#graphein.protein.utils.three_to_one_with_mods" title="Permalink to this definition">#</a></dt>
<dd><p>Converts three letter AA codes into 1 letter. Allows for modified residues.</p>
<p>See: <a class="reference internal" href="#graphein.protein.resi_atoms.RESI_THREE_TO_1" title="graphein.protein.resi_atoms.RESI_THREE_TO_1"><code class="xref py py-const docutils literal notranslate"><span class="pre">RESI_THREE_TO_1</span></code></a>.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>res</strong> (<a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><em>str</em></a>) – Three letter residue code string.</p>
</dd>
<dt class="field-even">Returns</dt>
<dd class="field-even"><p>1-letter residue code.</p>
</dd>
<dt class="field-odd">Return type</dt>
<dd class="field-odd"><p><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)">str</a></p>
</dd>
</dl>
</dd></dl>
</div>
<div class="section" id="module-graphein.protein.resi_atoms">
<span id="constants"></span><h2>Constants<a class="headerlink" href="#module-graphein.protein.resi_atoms" title="Permalink to this headline">#</a></h2>
<p>Author: Eric J. Ma, Arian Jamasb
Purpose: This is a set of utility variables and functions that can be used
across the Graphein project.</p>
<p>These include various collections of standard &amp; non-standard/modified amino acids and their names, identifiers and properties.</p>
<p>We also include mappings of covalent radii and bond lengths for the amino acids used in assembling atomic protein graphs.</p>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.AA_RING_ATOMS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">AA_RING_ATOMS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'HIS':</span> <span class="pre">['CG',</span> <span class="pre">'CD',</span> <span class="pre">'CE',</span> <span class="pre">'ND',</span> <span class="pre">'NE'],</span> <span class="pre">'PHE':</span> <span class="pre">['CG',</span> <span class="pre">'CD',</span> <span class="pre">'CE',</span> <span class="pre">'CZ'],</span> <span class="pre">'TRP':</span> <span class="pre">['CD',</span> <span class="pre">'CE',</span> <span class="pre">'CH',</span> <span class="pre">'CZ'],</span> <span class="pre">'TYR':</span> <span class="pre">['CG',</span> <span class="pre">'CD',</span> <span class="pre">'CE',</span> <span class="pre">'CZ']}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.AA_RING_ATOMS" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary mapping amino acid 3-letter codes to lists of atoms that are part of rings.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.AMINO_ACIDS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">AMINO_ACIDS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['A',</span> <span class="pre">'B',</span> <span class="pre">'C',</span> <span class="pre">'D',</span> <span class="pre">'E',</span> <span class="pre">'F',</span> <span class="pre">'G',</span> <span class="pre">'H',</span> <span class="pre">'I',</span> <span class="pre">'J',</span> <span class="pre">'K',</span> <span class="pre">'L',</span> <span class="pre">'M',</span> <span class="pre">'N',</span> <span class="pre">'O',</span> <span class="pre">'P',</span> <span class="pre">'Q',</span> <span class="pre">'R',</span> <span class="pre">'S',</span> <span class="pre">'T',</span> <span class="pre">'U',</span> <span class="pre">'V',</span> <span class="pre">'W',</span> <span class="pre">'X',</span> <span class="pre">'Y',</span> <span class="pre">'Z']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.AMINO_ACIDS" title="Permalink to this definition">#</a></dt>
<dd><p>Vocabulary of amino acids with one-letter codes. Includes <cite>fuzzy</cite> standard amino acids:
<code class="docutils literal notranslate"><span class="pre">"B"</span></code> denotes <code class="docutils literal notranslate"><span class="pre">"ASX"</span></code> which corresponds to <code class="docutils literal notranslate"><span class="pre">"ASP"</span></code> (<code class="docutils literal notranslate"><span class="pre">"D"</span></code>) <strong>or</strong> <code class="docutils literal notranslate"><span class="pre">"ASN"</span></code> (<code class="docutils literal notranslate"><span class="pre">"N"</span></code>) 
and <code class="docutils literal notranslate"><span class="pre">"Z"</span></code> denotes <code class="docutils literal notranslate"><span class="pre">"GLX"</span></code> which corresponds to``”GLU”`` (<code class="docutils literal notranslate"><span class="pre">"E"</span></code>) <strong>or</strong> <code class="docutils literal notranslate"><span class="pre">"GLN"</span></code> (<code class="docutils literal notranslate"><span class="pre">"Q"</span></code>).</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.AROMATIC_RESIS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">AROMATIC_RESIS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['PHE',</span> <span class="pre">'TRP',</span> <span class="pre">'HIS',</span> <span class="pre">'TYR']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.AROMATIC_RESIS" title="Permalink to this definition">#</a></dt>
<dd><p>List of aromatic residues.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.BACKBONE_ATOMS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">BACKBONE_ATOMS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['N',</span> <span class="pre">'CA',</span> <span class="pre">'C',</span> <span class="pre">'O']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.BACKBONE_ATOMS" title="Permalink to this definition">#</a></dt>
<dd><p>Atoms present in Amino Acid Backbones.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.BASE_AMINO_ACIDS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">BASE_AMINO_ACIDS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['A',</span> <span class="pre">'C',</span> <span class="pre">'D',</span> <span class="pre">'E',</span> <span class="pre">'F',</span> <span class="pre">'G',</span> <span class="pre">'H',</span> <span class="pre">'I',</span> <span class="pre">'K',</span> <span class="pre">'L',</span> <span class="pre">'M',</span> <span class="pre">'N',</span> <span class="pre">'P',</span> <span class="pre">'Q',</span> <span class="pre">'R',</span> <span class="pre">'S',</span> <span class="pre">'T',</span> <span class="pre">'V',</span> <span class="pre">'W',</span> <span class="pre">'Y']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.BASE_AMINO_ACIDS" title="Permalink to this definition">#</a></dt>
<dd><p>Vocabulary of 20 standard amino acids.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.BOND_LENGTHS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">BOND_LENGTHS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'As-N':</span> <span class="pre">{'i_d':</span> <span class="pre">1.835,</span> <span class="pre">'i_s':</span> <span class="pre">1.86,</span> <span class="pre">'w_sd':</span> <span class="pre">1.845},</span> <span class="pre">'As-O':</span> <span class="pre">{'i_d':</span> <span class="pre">1.66,</span> <span class="pre">'i_s':</span> <span class="pre">1.71,</span> <span class="pre">'w_sd':</span> <span class="pre">1.68},</span> <span class="pre">'As-S':</span> <span class="pre">{'i_d':</span> <span class="pre">2.08,</span> <span class="pre">'i_s':</span> <span class="pre">2.28,</span> <span class="pre">'w_sd':</span> <span class="pre">2.15},</span> <span class="pre">'C-C':</span> <span class="pre">{'i_d':</span> <span class="pre">1.31,</span> <span class="pre">'i_s':</span> <span class="pre">1.49,</span> <span class="pre">'i_t':</span> <span class="pre">1.18,</span> <span class="pre">'w_dt':</span> <span class="pre">1.21,</span> <span class="pre">'w_sd':</span> <span class="pre">1.38},</span> <span class="pre">'C-N':</span> <span class="pre">{'i_d':</span> <span class="pre">1.32,</span> <span class="pre">'i_s':</span> <span class="pre">1.42,</span> <span class="pre">'i_t':</span> <span class="pre">1.14,</span> <span class="pre">'w_dt':</span> <span class="pre">1.2,</span> <span class="pre">'w_sd':</span> <span class="pre">1.34},</span> <span class="pre">'C-O':</span> <span class="pre">{'i_d':</span> <span class="pre">1.22,</span> <span class="pre">'i_s':</span> <span class="pre">1.41,</span> <span class="pre">'w_sd':</span> <span class="pre">1.28},</span> <span class="pre">'C-S':</span> <span class="pre">{'i_d':</span> <span class="pre">1.68,</span> <span class="pre">'i_s':</span> <span class="pre">1.78,</span> <span class="pre">'w_sd':</span> <span class="pre">1.7},</span> <span class="pre">'C-Te':</span> <span class="pre">{'i_d':</span> <span class="pre">1.8,</span> <span class="pre">'i_s':</span> <span class="pre">2.2,</span> <span class="pre">'w_sd':</span> <span class="pre">2.1},</span> <span class="pre">'N-N':</span> <span class="pre">{'i_d':</span> <span class="pre">1.22,</span> <span class="pre">'i_s':</span> <span class="pre">1.4,</span> <span class="pre">'w_sd':</span> <span class="pre">1.32},</span> <span class="pre">'N-O':</span> <span class="pre">{'i_d':</span> <span class="pre">1.22,</span> <span class="pre">'i_s':</span> <span class="pre">1.39,</span> <span class="pre">'w_sd':</span> <span class="pre">1.25},</span> <span class="pre">'N-P':</span> <span class="pre">{'i_d':</span> <span class="pre">1.59,</span> <span class="pre">'i_s':</span> <span class="pre">1.69,</span> <span class="pre">'w_sd':</span> <span class="pre">1.62},</span> <span class="pre">'N-S':</span> <span class="pre">{'i_d':</span> <span class="pre">1.54,</span> <span class="pre">'i_s':</span> <span class="pre">1.66,</span> <span class="pre">'w_sd':</span> <span class="pre">1.58},</span> <span class="pre">'N-Se':</span> <span class="pre">{'i_d':</span> <span class="pre">1.79,</span> <span class="pre">'i_s':</span> <span class="pre">1.83,</span> <span class="pre">'w_sd':</span> <span class="pre">1.8},</span> <span class="pre">'O-P':</span> <span class="pre">{'i_d':</span> <span class="pre">1.48,</span> <span class="pre">'i_s':</span> <span class="pre">1.6,</span> <span class="pre">'w_sd':</span> <span class="pre">1.52},</span> <span class="pre">'O-S':</span> <span class="pre">{'i_d':</span> <span class="pre">1.45,</span> <span class="pre">'i_s':</span> <span class="pre">1.58,</span> <span class="pre">'w_sd':</span> <span class="pre">1.54},</span> <span class="pre">'P-P':</span> <span class="pre">{'i_d':</span> <span class="pre">2.04,</span> <span class="pre">'i_s':</span> <span class="pre">2.23,</span> <span class="pre">'w_sd':</span> <span class="pre">2.06}}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.BOND_LENGTHS" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary containing idealised single, double and triple bond lengths (<code class="docutils literal notranslate"><span class="pre">i_s</span></code>, <code class="docutils literal notranslate"><span class="pre">i_d</span></code>, <code class="docutils literal notranslate"><span class="pre">i_t</span></code>) and watersheds (<code class="docutils literal notranslate"><span class="pre">w_sd</span></code>, <code class="docutils literal notranslate"><span class="pre">w_dt</span></code>),
below which a bond is probably double/triple (e.g. <code class="docutils literal notranslate"><span class="pre">triple</span></code> &lt; <code class="docutils literal notranslate"><span class="pre">double</span></code> &lt; <code class="docutils literal notranslate"><span class="pre">single</span></code>). All lengths are in Angstroms.</p>
<p>Taken from:</p>
<blockquote>
<div><p>Automatic Assignment of Chemical Connectivity to Organic Molecules in the Cambridge Structural Database
Jon C. Baber and Edward E. Hodgkin*
J. Chem. Inf. Comput. Sci. 1992, 32. 401-406</p>
</div></blockquote>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.BOND_ORDERS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">BOND_ORDERS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'As-N':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'As-O':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'As-S':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'C-C':</span> <span class="pre">[1,</span> <span class="pre">2,</span> <span class="pre">3],</span> <span class="pre">'C-N':</span> <span class="pre">[1,</span> <span class="pre">2,</span> <span class="pre">3],</span> <span class="pre">'C-O':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'C-S':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'C-Te':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'N-N':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'N-O':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'N-P':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'N-S':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'N-Se':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'O-P':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'O-S':</span> <span class="pre">[1,</span> <span class="pre">2],</span> <span class="pre">'P-P':</span> <span class="pre">[1,</span> <span class="pre">2]}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.BOND_ORDERS" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of allowable bond orders for each covalent bond type.</p>
<p>Taken from:</p>
<blockquote>
<div><p>Automatic Assignment of Chemical Connectivity to Organic Molecules in the Cambridge Structural Database
Jon C. Baber and Edward E. Hodgkin*
J. Chem. Inf. Comput. Sci. 1992, 32. 401-406</p>
</div></blockquote>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.BOND_TYPES">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">BOND_TYPES</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['hydrophobic',</span> <span class="pre">'disulfide',</span> <span class="pre">'hbond',</span> <span class="pre">'ionic',</span> <span class="pre">'aromatic',</span> <span class="pre">'aromatic_sulphur',</span> <span class="pre">'cation_pi',</span> <span class="pre">'backbone',</span> <span class="pre">'delaunay']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.BOND_TYPES" title="Permalink to this definition">#</a></dt>
<dd><p>List of supported bond types.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.CARBOHYDRATE_CODES">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">CARBOHYDRATE_CODES</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['BGC',</span> <span class="pre">'GLC',</span> <span class="pre">'MAN',</span> <span class="pre">'BMA',</span> <span class="pre">'FUC',</span> <span class="pre">'GAL',</span> <span class="pre">'GLA',</span> <span class="pre">'NAG',</span> <span class="pre">'NGA',</span> <span class="pre">'SIA',</span> <span class="pre">'XYS']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.CARBOHYDRATE_CODES" title="Permalink to this definition">#</a></dt>
<dd><p>Three letter codes of commonly found carbohydrates in protein structures.</p>
<p>See <a class="reference external" href="http://ligand-expo.rcsb.org/">http://ligand-expo.rcsb.org/</a> and <a class="reference external" href="https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html">https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.CARBOHYDRATE_CODE_NAME_MAPPING">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">CARBOHYDRATE_CODE_NAME_MAPPING</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'BGC':</span> <span class="pre">'D-GLUCOSE',</span> <span class="pre">'BMA':</span> <span class="pre">'D-MANNOSE',</span> <span class="pre">'FUC':</span> <span class="pre">'FUCOSE',</span> <span class="pre">'GAL':</span> <span class="pre">'D-GALACTOSE',</span> <span class="pre">'GLA':</span> <span class="pre">'D-GALACTOSE',</span> <span class="pre">'GLC':</span> <span class="pre">'D-GLUCOSE',</span> <span class="pre">'MAN':</span> <span class="pre">'D-MANNOSE',</span> <span class="pre">'NAG':</span> <span class="pre">'N-ACETYL-D-GLUCOSAMINE',</span> <span class="pre">'NGA':</span> <span class="pre">'N-ACETYL-D-GALACTOSAMINE',</span> <span class="pre">'SIA':</span> <span class="pre">'O-SIALIC_ACID',</span> <span class="pre">'XYS':</span> <span class="pre">'D-XYLOPYRANOSE'}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.CARBOHYDRATE_CODE_NAME_MAPPING" title="Permalink to this definition">#</a></dt>
<dd><p>Mapping of 3-letter PDB ligand accession codes for common carbohydrates to their full names.</p>
<p>See <a class="reference external" href="http://ligand-expo.rcsb.org/">http://ligand-expo.rcsb.org/</a> and <a class="reference external" href="https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html">https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.CARBOHYDRATE_NAMES">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">CARBOHYDRATE_NAMES</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['D-GLUCOSE',</span> <span class="pre">'D-MANNOSE',</span> <span class="pre">'FUCOSE',</span> <span class="pre">'D-GALACTOSE',</span> <span class="pre">'N-ACETYL-D-GLUCOSAMINE',</span> <span class="pre">'N-ACETYL-D-GALACTOSAMINE',</span> <span class="pre">'O-SIALIC_ACID',</span> <span class="pre">'D-XYLOPYRANOSE']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.CARBOHYDRATE_NAMES" title="Permalink to this definition">#</a></dt>
<dd><p>Names of commonly found carbohydrates in protein structures.</p>
<p>See <a class="reference external" href="http://ligand-expo.rcsb.org/">http://ligand-expo.rcsb.org/</a> and <a class="reference external" href="https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html">https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.CATION_PI_RESIS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">CATION_PI_RESIS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['LYS',</span> <span class="pre">'ARG',</span> <span class="pre">'PHE',</span> <span class="pre">'TYR',</span> <span class="pre">'TRP']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.CATION_PI_RESIS" title="Permalink to this definition">#</a></dt>
<dd><p>List of residues involved in cation-pi interactions.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.CATION_RESIS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">CATION_RESIS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['LYS',</span> <span class="pre">'ARG']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.CATION_RESIS" title="Permalink to this definition">#</a></dt>
<dd><p>List of cationic residues.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.COFACTOR_CODES">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">COFACTOR_CODES</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['ADP',</span> <span class="pre">'AMP',</span> <span class="pre">'ATP',</span> <span class="pre">'CMP',</span> <span class="pre">'COA',</span> <span class="pre">'FAD',</span> <span class="pre">'FMN',</span> <span class="pre">'NAP',</span> <span class="pre">'NDP']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.COFACTOR_CODES" title="Permalink to this definition">#</a></dt>
<dd><p>Three letter codes of cofactors commonly found in PDB structures.</p>
<p>See: <a class="reference external" href="http://ligand-expo.rcsb.org/">http://ligand-expo.rcsb.org/</a> and <a class="reference external" href="https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html">https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.COFACTOR_CODE_NAME_MAPPING">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">COFACTOR_CODE_NAME_MAPPING</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'ADP':</span> <span class="pre">'ADP',</span> <span class="pre">'AMP':</span> <span class="pre">'AMP',</span> <span class="pre">'ATP':</span> <span class="pre">'ATP',</span> <span class="pre">'CMP':</span> <span class="pre">'cAMP',</span> <span class="pre">'COA':</span> <span class="pre">'COENZYME_A',</span> <span class="pre">'FAD':</span> <span class="pre">'FAD',</span> <span class="pre">'FMN':</span> <span class="pre">'FLAVIN_MONONUCLEOTIDE',</span> <span class="pre">'NAP':</span> <span class="pre">'NADP',</span> <span class="pre">'NDP':</span> <span class="pre">'NADPH'}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.COFACTOR_CODE_NAME_MAPPING" title="Permalink to this definition">#</a></dt>
<dd><p>Mapping between 3-letter PDB ligand codes and cofactor names.</p>
<p>See <a class="reference external" href="http://ligand-expo.rcsb.org/">http://ligand-expo.rcsb.org/</a> and <a class="reference external" href="https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html">https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.COFACTOR_NAMES">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">COFACTOR_NAMES</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['ADP',</span> <span class="pre">'AMP',</span> <span class="pre">'ATP',</span> <span class="pre">'cAMP',</span> <span class="pre">'COENZYME_A',</span> <span class="pre">'FAD',</span> <span class="pre">'FLAVIN_MONONUCLEOTIDE',</span> <span class="pre">'NADP',</span> <span class="pre">'NADPH']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.COFACTOR_NAMES" title="Permalink to this definition">#</a></dt>
<dd><p>Names of cofactors commonly found in PDB structures.</p>
<p>See: <a class="reference external" href="http://ligand-expo.rcsb.org/">http://ligand-expo.rcsb.org/</a> and <a class="reference external" href="https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html">https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.COVALENT_RADII">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">COVALENT_RADII</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'Cdb':</span> <span class="pre">0.67,</span> <span class="pre">'Cres':</span> <span class="pre">0.72,</span> <span class="pre">'Csb':</span> <span class="pre">0.77,</span> <span class="pre">'Hsb':</span> <span class="pre">0.37,</span> <span class="pre">'Ndb':</span> <span class="pre">0.62,</span> <span class="pre">'Nres':</span> <span class="pre">0.66,</span> <span class="pre">'Nsb':</span> <span class="pre">0.7,</span> <span class="pre">'Odb':</span> <span class="pre">0.6,</span> <span class="pre">'Ores':</span> <span class="pre">0.635,</span> <span class="pre">'Osb':</span> <span class="pre">0.67,</span> <span class="pre">'Ssb':</span> <span class="pre">1.04}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.COVALENT_RADII" title="Permalink to this definition">#</a></dt>
<dd><p>Covalent radii for OpenSCAD output.
Adding <code class="docutils literal notranslate"><span class="pre">Ores</span></code> between <code class="docutils literal notranslate"><span class="pre">Osb</span></code> and <code class="docutils literal notranslate"><span class="pre">Odb</span></code> for <code class="docutils literal notranslate"><span class="pre">Asp</span></code> and <code class="docutils literal notranslate"><span class="pre">Glu</span></code>, <code class="docutils literal notranslate"><span class="pre">Nres</span></code> between <code class="docutils literal notranslate"><span class="pre">Nsb</span></code> and <code class="docutils literal notranslate"><span class="pre">Ndb</span></code> for <code class="docutils literal notranslate"><span class="pre">Arg</span></code>, as PDB does not specify</p>
<p>Covalent radii from:</p>
<blockquote>
<div><p>Heyrovska, Raji : ‘Atomic Structures of all the Twenty Essential Amino Acids and a Tripeptide, with Bond Lengths as Sums of Atomic Covalent Radii’</p>
</div></blockquote>
<p>Paper: <a class="reference external" href="https://arxiv.org/pdf/0804.2488.pdf">https://arxiv.org/pdf/0804.2488.pdf</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.DEFAULT_BOND_STATE">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">DEFAULT_BOND_STATE</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'1HD2':</span> <span class="pre">'Hsb',</span> <span class="pre">'1HH1':</span> <span class="pre">'Hsb',</span> <span class="pre">'1HH2':</span> <span class="pre">'Hsb',</span> <span class="pre">'2HD2':</span> <span class="pre">'Hsb',</span> <span class="pre">'2HH1':</span> <span class="pre">'Hsb',</span> <span class="pre">'2HH2':</span> <span class="pre">'Hsb',</span> <span class="pre">'C':</span> <span class="pre">'Cdb',</span> <span class="pre">'CA':</span> <span class="pre">'Csb',</span> <span class="pre">'CB':</span> <span class="pre">'Csb',</span> <span class="pre">'H':</span> <span class="pre">'Hsb',</span> <span class="pre">'HE':</span> <span class="pre">'Hsb',</span> <span class="pre">'HG':</span> <span class="pre">'Hsb',</span> <span class="pre">'HG1':</span> <span class="pre">'Hsb',</span> <span class="pre">'HH':</span> <span class="pre">'Hsb',</span> <span class="pre">'HZ1':</span> <span class="pre">'Hsb',</span> <span class="pre">'HZ2':</span> <span class="pre">'Hsb',</span> <span class="pre">'HZ3':</span> <span class="pre">'Hsb',</span> <span class="pre">'N':</span> <span class="pre">'Nsb',</span> <span class="pre">'O':</span> <span class="pre">'Odb',</span> <span class="pre">'OXT':</span> <span class="pre">'Osb'}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.DEFAULT_BOND_STATE" title="Permalink to this definition">#</a></dt>
<dd><p>Assignment of atom classes to atomic radii.</p>
<p>Covalent radii from:</p>
<blockquote>
<div><p>Heyrovska, Raji : ‘Atomic Structures of all the Twenty Essential Amino Acids and a Tripeptide, with Bond Lengths as Sums of Atomic Covalent Radii’</p>
</div></blockquote>
<p>Paper: <a class="reference external" href="https://arxiv.org/pdf/0804.2488.pdf">https://arxiv.org/pdf/0804.2488.pdf</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.DISULFIDE_ATOMS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">DISULFIDE_ATOMS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['SG']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.DISULFIDE_ATOMS" title="Permalink to this definition">#</a></dt>
<dd><p>List of atoms capable of forming disulphide bonds.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.DISULFIDE_RESIS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">DISULFIDE_RESIS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['CYS']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.DISULFIDE_RESIS" title="Permalink to this definition">#</a></dt>
<dd><p>Residues capable of forming disulfide bonds.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.GRANTHAM_CHEMICAL_DISTANCE_MATRIX">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">GRANTHAM_CHEMICAL_DISTANCE_MATRIX</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'AA':</span> <span class="pre">0.0,</span> <span class="pre">'AC':</span> <span class="pre">0.112,</span> <span class="pre">'AD':</span> <span class="pre">0.819,</span> <span class="pre">'AE':</span> <span class="pre">0.827,</span> <span class="pre">'AF':</span> <span class="pre">0.54,</span> <span class="pre">'AG':</span> <span class="pre">0.208,</span> <span class="pre">'AH':</span> <span class="pre">0.696,</span> <span class="pre">'AI':</span> <span class="pre">0.407,</span> <span class="pre">'AK':</span> <span class="pre">0.891,</span> <span class="pre">'AL':</span> <span class="pre">0.406,</span> <span class="pre">'AM':</span> <span class="pre">0.379,</span> <span class="pre">'AN':</span> <span class="pre">0.318,</span> <span class="pre">'AP':</span> <span class="pre">0.191,</span> <span class="pre">'AQ':</span> <span class="pre">0.372,</span> <span class="pre">'AR':</span> <span class="pre">1.0,</span> <span class="pre">'AS':</span> <span class="pre">0.094,</span> <span class="pre">'AT':</span> <span class="pre">0.22,</span> <span class="pre">'AV':</span> <span class="pre">0.273,</span> <span class="pre">'AW':</span> <span class="pre">0.739,</span> <span class="pre">'AY':</span> <span class="pre">0.552,</span> <span class="pre">'CA':</span> <span class="pre">0.114,</span> <span class="pre">'CC':</span> <span class="pre">0.0,</span> <span class="pre">'CD':</span> <span class="pre">0.847,</span> <span class="pre">'CE':</span> <span class="pre">0.838,</span> <span class="pre">'CF':</span> <span class="pre">0.437,</span> <span class="pre">'CG':</span> <span class="pre">0.32,</span> <span class="pre">'CH':</span> <span class="pre">0.66,</span> <span class="pre">'CI':</span> <span class="pre">0.304,</span> <span class="pre">'CK':</span> <span class="pre">0.887,</span> <span class="pre">'CL':</span> <span class="pre">0.301,</span> <span class="pre">'CM':</span> <span class="pre">0.277,</span> <span class="pre">'CN':</span> <span class="pre">0.324,</span> <span class="pre">'CP':</span> <span class="pre">0.157,</span> <span class="pre">'CQ':</span> <span class="pre">0.341,</span> <span class="pre">'CR':</span> <span class="pre">1.0,</span> <span class="pre">'CS':</span> <span class="pre">0.176,</span> <span class="pre">'CT':</span> <span class="pre">0.233,</span> <span class="pre">'CV':</span> <span class="pre">0.167,</span> <span class="pre">'CW':</span> <span class="pre">0.639,</span> <span class="pre">'CY':</span> <span class="pre">0.457,</span> <span class="pre">'DA':</span> <span class="pre">0.729,</span> <span class="pre">'DC':</span> <span class="pre">0.742,</span> <span class="pre">'DD':</span> <span class="pre">0.0,</span> <span class="pre">'DE':</span> <span class="pre">0.124,</span> <span class="pre">'DF':</span> <span class="pre">0.924,</span> <span class="pre">'DG':</span> <span class="pre">0.697,</span> <span class="pre">'DH':</span> <span class="pre">0.435,</span> <span class="pre">'DI':</span> <span class="pre">0.847,</span> <span class="pre">'DK':</span> <span class="pre">0.249,</span> <span class="pre">'DL':</span> <span class="pre">0.841,</span> <span class="pre">'DM':</span> <span class="pre">0.819,</span> <span class="pre">'DN':</span> <span class="pre">0.56,</span> <span class="pre">'DP':</span> <span class="pre">0.657,</span> <span class="pre">'DQ':</span> <span class="pre">0.584,</span> <span class="pre">'DR':</span> <span class="pre">0.295,</span> <span class="pre">'DS':</span> <span class="pre">0.667,</span> <span class="pre">'DT':</span> <span class="pre">0.649,</span> <span class="pre">'DV':</span> <span class="pre">0.797,</span> <span class="pre">'DW':</span> <span class="pre">1.0,</span> <span class="pre">'DY':</span> <span class="pre">0.836,</span> <span class="pre">'EA':</span> <span class="pre">0.79,</span> <span class="pre">'EC':</span> <span class="pre">0.788,</span> <span class="pre">'ED':</span> <span class="pre">0.133,</span> <span class="pre">'EE':</span> <span class="pre">0.0,</span> <span class="pre">'EF':</span> <span class="pre">0.932,</span> <span class="pre">'EG':</span> <span class="pre">0.779,</span> <span class="pre">'EH':</span> <span class="pre">0.406,</span> <span class="pre">'EI':</span> <span class="pre">0.86,</span> <span class="pre">'EK':</span> <span class="pre">0.143,</span> <span class="pre">'EL':</span> <span class="pre">0.854,</span> <span class="pre">'EM':</span> <span class="pre">0.83,</span> <span class="pre">'EN':</span> <span class="pre">0.599,</span> <span class="pre">'EP':</span> <span class="pre">0.688,</span> <span class="pre">'EQ':</span> <span class="pre">0.598,</span> <span class="pre">'ER':</span> <span class="pre">0.234,</span> <span class="pre">'ES':</span> <span class="pre">0.726,</span> <span class="pre">'ET':</span> <span class="pre">0.682,</span> <span class="pre">'EV':</span> <span class="pre">0.824,</span> <span class="pre">'EW':</span> <span class="pre">1.0,</span> <span class="pre">'EY':</span> <span class="pre">0.837,</span> <span class="pre">'FA':</span> <span class="pre">0.508,</span> <span class="pre">'FC':</span> <span class="pre">0.405,</span> <span class="pre">'FD':</span> <span class="pre">0.977,</span> <span class="pre">'FE':</span> <span class="pre">0.918,</span> <span class="pre">'FF':</span> <span class="pre">0.0,</span> <span class="pre">'FG':</span> <span class="pre">0.69,</span> <span class="pre">'FH':</span> <span class="pre">0.663,</span> <span class="pre">'FI':</span> <span class="pre">0.128,</span> <span class="pre">'FK':</span> <span class="pre">0.903,</span> <span class="pre">'FL':</span> <span class="pre">0.131,</span> <span class="pre">'FM':</span> <span class="pre">0.169,</span> <span class="pre">'FN':</span> <span class="pre">0.541,</span> <span class="pre">'FP':</span> <span class="pre">0.42,</span> <span class="pre">'FQ':</span> <span class="pre">0.459,</span> <span class="pre">'FR':</span> <span class="pre">1.0,</span> <span class="pre">'FS':</span> <span class="pre">0.548,</span> <span class="pre">'FT':</span> <span class="pre">0.499,</span> <span class="pre">'FV':</span> <span class="pre">0.252,</span> <span class="pre">'FW':</span> <span class="pre">0.207,</span> <span class="pre">'FY':</span> <span class="pre">0.179,</span> <span class="pre">'GA':</span> <span class="pre">0.206,</span> <span class="pre">'GC':</span> <span class="pre">0.312,</span> <span class="pre">'GD':</span> <span class="pre">0.776,</span> <span class="pre">'GE':</span> <span class="pre">0.807,</span> <span class="pre">'GF':</span> <span class="pre">0.727,</span> <span class="pre">'GG':</span> <span class="pre">0.0,</span> <span class="pre">'GH':</span> <span class="pre">0.769,</span> <span class="pre">'GI':</span> <span class="pre">0.592,</span> <span class="pre">'GK':</span> <span class="pre">0.894,</span> <span class="pre">'GL':</span> <span class="pre">0.591,</span> <span class="pre">'GM':</span> <span class="pre">0.557,</span> <span class="pre">'GN':</span> <span class="pre">0.381,</span> <span class="pre">'GP':</span> <span class="pre">0.323,</span> <span class="pre">'GQ':</span> <span class="pre">0.467,</span> <span class="pre">'GR':</span> <span class="pre">1.0,</span> <span class="pre">'GS':</span> <span class="pre">0.158,</span> <span class="pre">'GT':</span> <span class="pre">0.272,</span> <span class="pre">'GV':</span> <span class="pre">0.464,</span> <span class="pre">'GW':</span> <span class="pre">0.923,</span> <span class="pre">'GY':</span> <span class="pre">0.728,</span> <span class="pre">'HA':</span> <span class="pre">0.896,</span> <span class="pre">'HC':</span> <span class="pre">0.836,</span> <span class="pre">'HD':</span> <span class="pre">0.629,</span> <span class="pre">'HE':</span> <span class="pre">0.547,</span> <span class="pre">'HF':</span> <span class="pre">0.907,</span> <span class="pre">'HG':</span> <span class="pre">1.0,</span> <span class="pre">'HH':</span> <span class="pre">0.0,</span> <span class="pre">'HI':</span> <span class="pre">0.848,</span> <span class="pre">'HK':</span> <span class="pre">0.566,</span> <span class="pre">'HL':</span> <span class="pre">0.842,</span> <span class="pre">'HM':</span> <span class="pre">0.825,</span> <span class="pre">'HN':</span> <span class="pre">0.754,</span> <span class="pre">'HP':</span> <span class="pre">0.777,</span> <span class="pre">'HQ':</span> <span class="pre">0.716,</span> <span class="pre">'HR':</span> <span class="pre">0.697,</span> <span class="pre">'HS':</span> <span class="pre">0.865,</span> <span class="pre">'HT':</span> <span class="pre">0.834,</span> <span class="pre">'HV':</span> <span class="pre">0.831,</span> <span class="pre">'HW':</span> <span class="pre">0.981,</span> <span class="pre">'HY':</span> <span class="pre">0.821,</span> <span class="pre">'IA':</span> <span class="pre">0.403,</span> <span class="pre">'IC':</span> <span class="pre">0.296,</span> <span class="pre">'ID':</span> <span class="pre">0.942,</span> <span class="pre">'IE':</span> <span class="pre">0.891,</span> <span class="pre">'IF':</span> <span class="pre">0.134,</span> <span class="pre">'IG':</span> <span class="pre">0.592,</span> <span class="pre">'IH':</span> <span class="pre">0.652,</span> <span class="pre">'II':</span> <span class="pre">0.0,</span> <span class="pre">'IK':</span> <span class="pre">0.892,</span> <span class="pre">'IL':</span> <span class="pre">0.013,</span> <span class="pre">'IM':</span> <span class="pre">0.057,</span> <span class="pre">'IN':</span> <span class="pre">0.457,</span> <span class="pre">'IP':</span> <span class="pre">0.311,</span> <span class="pre">'IQ':</span> <span class="pre">0.383,</span> <span class="pre">'IR':</span> <span class="pre">1.0,</span> <span class="pre">'IS':</span> <span class="pre">0.443,</span> <span class="pre">'IT':</span> <span class="pre">0.396,</span> <span class="pre">'IV':</span> <span class="pre">0.133,</span> <span class="pre">'IW':</span> <span class="pre">0.339,</span> <span class="pre">'IY':</span> <span class="pre">0.213,</span> <span class="pre">'KA':</span> <span class="pre">0.889,</span> <span class="pre">'KC':</span> <span class="pre">0.871,</span> <span class="pre">'KD':</span> <span class="pre">0.279,</span> <span class="pre">'KE':</span> <span class="pre">0.149,</span> <span class="pre">'KF':</span> <span class="pre">0.957,</span> <span class="pre">'KG':</span> <span class="pre">0.9,</span> <span class="pre">'KH':</span> <span class="pre">0.438,</span> <span class="pre">'KI':</span> <span class="pre">0.899,</span> <span class="pre">'KK':</span> <span class="pre">0.0,</span> <span class="pre">'KL':</span> <span class="pre">0.892,</span> <span class="pre">'KM':</span> <span class="pre">0.871,</span> <span class="pre">'KN':</span> <span class="pre">0.667,</span> <span class="pre">'KP':</span> <span class="pre">0.757,</span> <span class="pre">'KQ':</span> <span class="pre">0.639,</span> <span class="pre">'KR':</span> <span class="pre">0.154,</span> <span class="pre">'KS':</span> <span class="pre">0.825,</span> <span class="pre">'KT':</span> <span class="pre">0.759,</span> <span class="pre">'KV':</span> <span class="pre">0.882,</span> <span class="pre">'KW':</span> <span class="pre">1.0,</span> <span class="pre">'KY':</span> <span class="pre">0.848,</span> <span class="pre">'LA':</span> <span class="pre">0.405,</span> <span class="pre">'LC':</span> <span class="pre">0.296,</span> <span class="pre">'LD':</span> <span class="pre">0.944,</span> <span class="pre">'LE':</span> <span class="pre">0.892,</span> <span class="pre">'LF':</span> <span class="pre">0.139,</span> <span class="pre">'LG':</span> <span class="pre">0.596,</span> <span class="pre">'LH':</span> <span class="pre">0.653,</span> <span class="pre">'LI':</span> <span class="pre">0.013,</span> <span class="pre">'LK':</span> <span class="pre">0.893,</span> <span class="pre">'LL':</span> <span class="pre">0.0,</span> <span class="pre">'LM':</span> <span class="pre">0.062,</span> <span class="pre">'LN':</span> <span class="pre">0.452,</span> <span class="pre">'LP':</span> <span class="pre">0.309,</span> <span class="pre">'LQ':</span> <span class="pre">0.376,</span> <span class="pre">'LR':</span> <span class="pre">1.0,</span> <span class="pre">'LS':</span> <span class="pre">0.443,</span> <span class="pre">'LT':</span> <span class="pre">0.397,</span> <span class="pre">'LV':</span> <span class="pre">0.133,</span> <span class="pre">'LW':</span> <span class="pre">0.341,</span> <span class="pre">'LY':</span> <span class="pre">0.205,</span> <span class="pre">'MA':</span> <span class="pre">0.383,</span> <span class="pre">'MC':</span> <span class="pre">0.276,</span> <span class="pre">'MD':</span> <span class="pre">0.932,</span> <span class="pre">'ME':</span> <span class="pre">0.879,</span> <span class="pre">'MF':</span> <span class="pre">0.182,</span> <span class="pre">'MG':</span> <span class="pre">0.569,</span> <span class="pre">'MH':</span> <span class="pre">0.648,</span> <span class="pre">'MI':</span> <span class="pre">0.058,</span> <span class="pre">'MK':</span> <span class="pre">0.884,</span> <span class="pre">'ML':</span> <span class="pre">0.062,</span> <span class="pre">'MM':</span> <span class="pre">0.0,</span> <span class="pre">'MN':</span> <span class="pre">0.447,</span> <span class="pre">'MP':</span> <span class="pre">0.285,</span> <span class="pre">'MQ':</span> <span class="pre">0.372,</span> <span class="pre">'MR':</span> <span class="pre">1.0,</span> <span class="pre">'MS':</span> <span class="pre">0.417,</span> <span class="pre">'MT':</span> <span class="pre">0.358,</span> <span class="pre">'MV':</span> <span class="pre">0.12,</span> <span class="pre">'MW':</span> <span class="pre">0.391,</span> <span class="pre">'MY':</span> <span class="pre">0.255,</span> <span class="pre">'NA':</span> <span class="pre">0.424,</span> <span class="pre">'NC':</span> <span class="pre">0.425,</span> <span class="pre">'ND':</span> <span class="pre">0.838,</span> <span class="pre">'NE':</span> <span class="pre">0.835,</span> <span class="pre">'NF':</span> <span class="pre">0.766,</span> <span class="pre">'NG':</span> <span class="pre">0.512,</span> <span class="pre">'NH':</span> <span class="pre">0.78,</span> <span class="pre">'NI':</span> <span class="pre">0.615,</span> <span class="pre">'NK':</span> <span class="pre">0.891,</span> <span class="pre">'NL':</span> <span class="pre">0.603,</span> <span class="pre">'NM':</span> <span class="pre">0.588,</span> <span class="pre">'NN':</span> <span class="pre">0.0,</span> <span class="pre">'NP':</span> <span class="pre">0.266,</span> <span class="pre">'NQ':</span> <span class="pre">0.175,</span> <span class="pre">'NR':</span> <span class="pre">1.0,</span> <span class="pre">'NS':</span> <span class="pre">0.361,</span> <span class="pre">'NT':</span> <span class="pre">0.368,</span> <span class="pre">'NV':</span> <span class="pre">0.503,</span> <span class="pre">'NW':</span> <span class="pre">0.945,</span> <span class="pre">'NY':</span> <span class="pre">0.641,</span> <span class="pre">'PA':</span> <span class="pre">0.22,</span> <span class="pre">'PC':</span> <span class="pre">0.179,</span> <span class="pre">'PD':</span> <span class="pre">0.852,</span> <span class="pre">'PE':</span> <span class="pre">0.831,</span> <span class="pre">'PF':</span> <span class="pre">0.515,</span> <span class="pre">'PG':</span> <span class="pre">0.376,</span> <span class="pre">'PH':</span> <span class="pre">0.696,</span> <span class="pre">'PI':</span> <span class="pre">0.363,</span> <span class="pre">'PK':</span> <span class="pre">0.875,</span> <span class="pre">'PL':</span> <span class="pre">0.357,</span> <span class="pre">'PM':</span> <span class="pre">0.326,</span> <span class="pre">'PN':</span> <span class="pre">0.231,</span> <span class="pre">'PP':</span> <span class="pre">0.0,</span> <span class="pre">'PQ':</span> <span class="pre">0.228,</span> <span class="pre">'PR':</span> <span class="pre">1.0,</span> <span class="pre">'PS':</span> <span class="pre">0.196,</span> <span class="pre">'PT':</span> <span class="pre">0.161,</span> <span class="pre">'PV':</span> <span class="pre">0.244,</span> <span class="pre">'PW':</span> <span class="pre">0.72,</span> <span class="pre">'PY':</span> <span class="pre">0.481,</span> <span class="pre">'QA':</span> <span class="pre">0.512,</span> <span class="pre">'QC':</span> <span class="pre">0.462,</span> <span class="pre">'QD':</span> <span class="pre">0.903,</span> <span class="pre">'QE':</span> <span class="pre">0.861,</span> <span class="pre">'QF':</span> <span class="pre">0.671,</span> <span class="pre">'QG':</span> <span class="pre">0.648,</span> <span class="pre">'QH':</span> <span class="pre">0.765,</span> <span class="pre">'QI':</span> <span class="pre">0.532,</span> <span class="pre">'QK':</span> <span class="pre">0.881,</span> <span class="pre">'QL':</span> <span class="pre">0.518,</span> <span class="pre">'QM':</span> <span class="pre">0.505,</span> <span class="pre">'QN':</span> <span class="pre">0.181,</span> <span class="pre">'QP':</span> <span class="pre">0.272,</span> <span class="pre">'QQ':</span> <span class="pre">0.0,</span> <span class="pre">'QR':</span> <span class="pre">1.0,</span> <span class="pre">'QS':</span> <span class="pre">0.461,</span> <span class="pre">'QT':</span> <span class="pre">0.389,</span> <span class="pre">'QV':</span> <span class="pre">0.464,</span> <span class="pre">'QW':</span> <span class="pre">0.831,</span> <span class="pre">'QY':</span> <span class="pre">0.522,</span> <span class="pre">'RA':</span> <span class="pre">0.919,</span> <span class="pre">'RC':</span> <span class="pre">0.905,</span> <span class="pre">'RD':</span> <span class="pre">0.305,</span> <span class="pre">'RE':</span> <span class="pre">0.225,</span> <span class="pre">'RF':</span> <span class="pre">0.977,</span> <span class="pre">'RG':</span> <span class="pre">0.928,</span> <span class="pre">'RH':</span> <span class="pre">0.498,</span> <span class="pre">'RI':</span> <span class="pre">0.929,</span> <span class="pre">'RK':</span> <span class="pre">0.141,</span> <span class="pre">'RL':</span> <span class="pre">0.92,</span> <span class="pre">'RM':</span> <span class="pre">0.908,</span> <span class="pre">'RN':</span> <span class="pre">0.69,</span> <span class="pre">'RP':</span> <span class="pre">0.796,</span> <span class="pre">'RQ':</span> <span class="pre">0.668,</span> <span class="pre">'RR':</span> <span class="pre">0.0,</span> <span class="pre">'RS':</span> <span class="pre">0.86,</span> <span class="pre">'RT':</span> <span class="pre">0.808,</span> <span class="pre">'RV':</span> <span class="pre">0.914,</span> <span class="pre">'RW':</span> <span class="pre">1.0,</span> <span class="pre">'RY':</span> <span class="pre">0.859,</span> <span class="pre">'SA':</span> <span class="pre">0.1,</span> <span class="pre">'SC':</span> <span class="pre">0.185,</span> <span class="pre">'SD':</span> <span class="pre">0.801,</span> <span class="pre">'SE':</span> <span class="pre">0.812,</span> <span class="pre">'SF':</span> <span class="pre">0.622,</span> <span class="pre">'SG':</span> <span class="pre">0.17,</span> <span class="pre">'SH':</span> <span class="pre">0.718,</span> <span class="pre">'SI':</span> <span class="pre">0.478,</span> <span class="pre">'SK':</span> <span class="pre">0.883,</span> <span class="pre">'SL':</span> <span class="pre">0.474,</span> <span class="pre">'SM':</span> <span class="pre">0.44,</span> <span class="pre">'SN':</span> <span class="pre">0.289,</span> <span class="pre">'SP':</span> <span class="pre">0.181,</span> <span class="pre">'SQ':</span> <span class="pre">0.358,</span> <span class="pre">'SR':</span> <span class="pre">1.0,</span> <span class="pre">'SS':</span> <span class="pre">0.0,</span> <span class="pre">'ST':</span> <span class="pre">0.174,</span> <span class="pre">'SV':</span> <span class="pre">0.342,</span> <span class="pre">'SW':</span> <span class="pre">0.827,</span> <span class="pre">'SY':</span> <span class="pre">0.615,</span> <span class="pre">'TA':</span> <span class="pre">0.251,</span> <span class="pre">'TC':</span> <span class="pre">0.261,</span> <span class="pre">'TD':</span> <span class="pre">0.83,</span> <span class="pre">'TE':</span> <span class="pre">0.812,</span> <span class="pre">'TF':</span> <span class="pre">0.604,</span> <span class="pre">'TG':</span> <span class="pre">0.312,</span> <span class="pre">'TH':</span> <span class="pre">0.737,</span> <span class="pre">'TI':</span> <span class="pre">0.455,</span> <span class="pre">'TK':</span> <span class="pre">0.866,</span> <span class="pre">'TL':</span> <span class="pre">0.453,</span> <span class="pre">'TM':</span> <span class="pre">0.403,</span> <span class="pre">'TN':</span> <span class="pre">0.315,</span> <span class="pre">'TP':</span> <span class="pre">0.159,</span> <span class="pre">'TQ':</span> <span class="pre">0.322,</span> <span class="pre">'TR':</span> <span class="pre">1.0,</span> <span class="pre">'TS':</span> <span class="pre">0.185,</span> <span class="pre">'TT':</span> <span class="pre">0.0,</span> <span class="pre">'TV':</span> <span class="pre">0.345,</span> <span class="pre">'TW':</span> <span class="pre">0.816,</span> <span class="pre">'TY':</span> <span class="pre">0.596,</span> <span class="pre">'VA':</span> <span class="pre">0.275,</span> <span class="pre">'VC':</span> <span class="pre">0.165,</span> <span class="pre">'VD':</span> <span class="pre">0.9,</span> <span class="pre">'VE':</span> <span class="pre">0.867,</span> <span class="pre">'VF':</span> <span class="pre">0.269,</span> <span class="pre">'VG':</span> <span class="pre">0.471,</span> <span class="pre">'VH':</span> <span class="pre">0.649,</span> <span class="pre">'VI':</span> <span class="pre">0.135,</span> <span class="pre">'VK':</span> <span class="pre">0.889,</span> <span class="pre">'VL':</span> <span class="pre">0.134,</span> <span class="pre">'VM':</span> <span class="pre">0.12,</span> <span class="pre">'VN':</span> <span class="pre">0.38,</span> <span class="pre">'VP':</span> <span class="pre">0.212,</span> <span class="pre">'VQ':</span> <span class="pre">0.339,</span> <span class="pre">'VR':</span> <span class="pre">1.0,</span> <span class="pre">'VS':</span> <span class="pre">0.322,</span> <span class="pre">'VT':</span> <span class="pre">0.305,</span> <span class="pre">'VV':</span> <span class="pre">0.0,</span> <span class="pre">'VW':</span> <span class="pre">0.472,</span> <span class="pre">'VY':</span> <span class="pre">0.31,</span> <span class="pre">'WA':</span> <span class="pre">0.658,</span> <span class="pre">'WC':</span> <span class="pre">0.56,</span> <span class="pre">'WD':</span> <span class="pre">1.0,</span> <span class="pre">'WE':</span> <span class="pre">0.931,</span> <span class="pre">'WF':</span> <span class="pre">0.196,</span> <span class="pre">'WG':</span> <span class="pre">0.829,</span> <span class="pre">'WH':</span> <span class="pre">0.678,</span> <span class="pre">'WI':</span> <span class="pre">0.305,</span> <span class="pre">'WK':</span> <span class="pre">0.892,</span> <span class="pre">'WL':</span> <span class="pre">0.304,</span> <span class="pre">'WM':</span> <span class="pre">0.344,</span> <span class="pre">'WN':</span> <span class="pre">0.631,</span> <span class="pre">'WP':</span> <span class="pre">0.555,</span> <span class="pre">'WQ':</span> <span class="pre">0.538,</span> <span class="pre">'WR':</span> <span class="pre">0.968,</span> <span class="pre">'WS':</span> <span class="pre">0.689,</span> <span class="pre">'WT':</span> <span class="pre">0.638,</span> <span class="pre">'WV':</span> <span class="pre">0.418,</span> <span class="pre">'WW':</span> <span class="pre">0.0,</span> <span class="pre">'WY':</span> <span class="pre">0.204,</span> <span class="pre">'YA':</span> <span class="pre">0.587,</span> <span class="pre">'YC':</span> <span class="pre">0.478,</span> <span class="pre">'YD':</span> <span class="pre">1.0,</span> <span class="pre">'YE':</span> <span class="pre">0.932,</span> <span class="pre">'YF':</span> <span class="pre">0.202,</span> <span class="pre">'YG':</span> <span class="pre">0.782,</span> <span class="pre">'YH':</span> <span class="pre">0.678,</span> <span class="pre">'YI':</span> <span class="pre">0.23,</span> <span class="pre">'YK':</span> <span class="pre">0.904,</span> <span class="pre">'YL':</span> <span class="pre">0.219,</span> <span class="pre">'YM':</span> <span class="pre">0.268,</span> <span class="pre">'YN':</span> <span class="pre">0.512,</span> <span class="pre">'YP':</span> <span class="pre">0.444,</span> <span class="pre">'YQ':</span> <span class="pre">0.404,</span> <span class="pre">'YR':</span> <span class="pre">0.995,</span> <span class="pre">'YS':</span> <span class="pre">0.612,</span> <span class="pre">'YT':</span> <span class="pre">0.557,</span> <span class="pre">'YV':</span> <span class="pre">0.328,</span> <span class="pre">'YW':</span> <span class="pre">0.244,</span> <span class="pre">'YY':</span> <span class="pre">0.0}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.GRANTHAM_CHEMICAL_DISTANCE_MATRIX" title="Permalink to this definition">#</a></dt>
<dd><p>Grantham Chemical Distance Matrix. Taken from ProPy3 <a class="reference external" href="https://github.com/MartinThoma/propy3">https://github.com/MartinThoma/propy3</a></p>
<blockquote>
<div><p>Amino Acid Difference Formula to Help Explain Protein Evolution
R. Grantham
Science
Vol 185, Issue 4154
06 September 1974</p>
</div></blockquote>
<p>Paper: <a class="reference external" href="https://science.sciencemag.org/content/185/4154/862/tab-pdf">https://science.sciencemag.org/content/185/4154/862/tab-pdf</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.HYDROPHOBIC_RESIS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">HYDROPHOBIC_RESIS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['ALA',</span> <span class="pre">'VAL',</span> <span class="pre">'LEU',</span> <span class="pre">'ILE',</span> <span class="pre">'MET',</span> <span class="pre">'PHE',</span> <span class="pre">'TRP',</span> <span class="pre">'PRO',</span> <span class="pre">'TYR']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.HYDROPHOBIC_RESIS" title="Permalink to this definition">#</a></dt>
<dd><p>List of residues that are considered to be hydrophobic.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.IONIC_RESIS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">IONIC_RESIS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['ARG',</span> <span class="pre">'LYS',</span> <span class="pre">'HIS',</span> <span class="pre">'ASP',</span> <span class="pre">'GLU']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.IONIC_RESIS" title="Permalink to this definition">#</a></dt>
<dd><p>Residues capable of forming ionic interactions.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.ISOELECTRIC_POINTS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">ISOELECTRIC_POINTS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'ALA':</span> <span class="pre">6.11,</span> <span class="pre">'ARG':</span> <span class="pre">10.76,</span> <span class="pre">'ASN':</span> <span class="pre">10.76,</span> <span class="pre">'ASP':</span> <span class="pre">2.98,</span> <span class="pre">'ASX':</span> <span class="pre">6.87,</span> <span class="pre">'CYS':</span> <span class="pre">5.02,</span> <span class="pre">'GLN':</span> <span class="pre">5.65,</span> <span class="pre">'GLU':</span> <span class="pre">3.08,</span> <span class="pre">'GLX':</span> <span class="pre">4.35,</span> <span class="pre">'GLY':</span> <span class="pre">6.06,</span> <span class="pre">'HIS':</span> <span class="pre">7.64,</span> <span class="pre">'ILE':</span> <span class="pre">6.04,</span> <span class="pre">'LEU':</span> <span class="pre">6.04,</span> <span class="pre">'LYS':</span> <span class="pre">9.74,</span> <span class="pre">'MET':</span> <span class="pre">5.74,</span> <span class="pre">'PHE':</span> <span class="pre">5.91,</span> <span class="pre">'PRO':</span> <span class="pre">6.3,</span> <span class="pre">'SER':</span> <span class="pre">5.68,</span> <span class="pre">'THR':</span> <span class="pre">5.6,</span> <span class="pre">'TRP':</span> <span class="pre">5.88,</span> <span class="pre">'TYR':</span> <span class="pre">5.63,</span> <span class="pre">'UNK':</span> <span class="pre">7.0,</span> <span class="pre">'VAL':</span> <span class="pre">6.02}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.ISOELECTRIC_POINTS" title="Permalink to this definition">#</a></dt>
<dd><p>Dictionary of isoelectric points for standard amino acids. For <code class="docutils literal notranslate"><span class="pre">"UNK"</span></code> residues, neutral (pH 7.0) is assigned.
For <code class="docutils literal notranslate"><span class="pre">"ASX"</span></code> and <code class="docutils literal notranslate"><span class="pre">"GLX"</span></code> the average of their constituents (<code class="docutils literal notranslate"><span class="pre">"D"</span></code> and <code class="docutils literal notranslate"><span class="pre">"N"</span></code>, and <code class="docutils literal notranslate"><span class="pre">"E"</span></code> and <code class="docutils literal notranslate"><span class="pre">"Q"</span></code>, respectively) is assigned.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.ISOELECTRIC_POINTS_STD">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">ISOELECTRIC_POINTS_STD</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'ALA':</span> <span class="pre">array([[-0.0986554]]),</span> <span class="pre">'ARG':</span> <span class="pre">array([[2.33811019]]),</span> <span class="pre">'ASN':</span> <span class="pre">array([[2.33811019]]),</span> <span class="pre">'ASP':</span> <span class="pre">array([[-1.73888686]]),</span> <span class="pre">'ASX':</span> <span class="pre">array([[0.29961166]]),</span> <span class="pre">'CYS':</span> <span class="pre">array([[-0.66985422]]),</span> <span class="pre">'GLN':</span> <span class="pre">array([[-0.33971178]]),</span> <span class="pre">'GLU':</span> <span class="pre">array([[-1.6864833]]),</span> <span class="pre">'GLX':</span> <span class="pre">array([[-1.02095808]]),</span> <span class="pre">'GLY':</span> <span class="pre">array([[-0.12485718]]),</span> <span class="pre">'HIS':</span> <span class="pre">array([[0.70311909]]),</span> <span class="pre">'ILE':</span> <span class="pre">array([[-0.13533789]]),</span> <span class="pre">'LEU':</span> <span class="pre">array([[-0.13533789]]),</span> <span class="pre">'LYS':</span> <span class="pre">array([[1.80359387]]),</span> <span class="pre">'MET':</span> <span class="pre">array([[-0.29254858]]),</span> <span class="pre">'PHE':</span> <span class="pre">array([[-0.20346252]]),</span> <span class="pre">'PRO':</span> <span class="pre">array([[0.00091137]]),</span> <span class="pre">'SER':</span> <span class="pre">array([[-0.32399071]]),</span> <span class="pre">'THR':</span> <span class="pre">array([[-0.36591356]]),</span> <span class="pre">'TRP':</span> <span class="pre">array([[-0.21918359]]),</span> <span class="pre">'TYR':</span> <span class="pre">array([[-0.35019249]]),</span> <span class="pre">'UNK':</span> <span class="pre">array([[0.36773629]]),</span> <span class="pre">'VAL':</span> <span class="pre">array([[-0.1458186]])}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.ISOELECTRIC_POINTS_STD" title="Permalink to this definition">#</a></dt>
<dd><p>Standardized (sklearn.StandardScaler) isoelectric points for standard amino acids.</p>
<p>See <a class="reference internal" href="#graphein.protein.resi_atoms.ISOELECTRIC_POINTS" title="graphein.protein.resi_atoms.ISOELECTRIC_POINTS"><code class="xref py py-const docutils literal notranslate"><span class="pre">ISOELECTRIC_POINTS</span></code></a> for details.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.MAX_NEIGHBOURS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">MAX_NEIGHBOURS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#int" title="(in Python v3.10)"><span class="pre">int</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'B':</span> <span class="pre">3,</span> <span class="pre">'Br':</span> <span class="pre">1,</span> <span class="pre">'C':</span> <span class="pre">4,</span> <span class="pre">'F':</span> <span class="pre">1,</span> <span class="pre">'H':</span> <span class="pre">1,</span> <span class="pre">'I':</span> <span class="pre">3,</span> <span class="pre">'O':</span> <span class="pre">2}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.MAX_NEIGHBOURS" title="Permalink to this definition">#</a></dt>
<dd><p>Maximum number of neighbours an atom can have.</p>
<p>Taken from: <a class="reference external" href="https://www.daylight.com/meetings/mug01/Sayle/m4xbondage.html">https://www.daylight.com/meetings/mug01/Sayle/m4xbondage.html</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.MOLECULAR_WEIGHTS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">MOLECULAR_WEIGHTS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'ALA':</span> <span class="pre">89.0935,</span> <span class="pre">'ARG':</span> <span class="pre">174.2017,</span> <span class="pre">'ASN':</span> <span class="pre">132.1184,</span> <span class="pre">'ASP':</span> <span class="pre">133.1032,</span> <span class="pre">'ASX':</span> <span class="pre">132.6108,</span> <span class="pre">'CYS':</span> <span class="pre">121.159,</span> <span class="pre">'GLN':</span> <span class="pre">146.1451,</span> <span class="pre">'GLU':</span> <span class="pre">147.1299,</span> <span class="pre">'GLX':</span> <span class="pre">146.6375,</span> <span class="pre">'GLY':</span> <span class="pre">75.0669,</span> <span class="pre">'HIS':</span> <span class="pre">155.1552,</span> <span class="pre">'ILE':</span> <span class="pre">131.1736,</span> <span class="pre">'LEU':</span> <span class="pre">131.1736,</span> <span class="pre">'LYS':</span> <span class="pre">146.1882,</span> <span class="pre">'MET':</span> <span class="pre">149.2124,</span> <span class="pre">'PHE':</span> <span class="pre">165.19,</span> <span class="pre">'PRO':</span> <span class="pre">115.131,</span> <span class="pre">'SER':</span> <span class="pre">105.093,</span> <span class="pre">'THR':</span> <span class="pre">119.1197,</span> <span class="pre">'TRP':</span> <span class="pre">204.2262,</span> <span class="pre">'TYR':</span> <span class="pre">181.1894,</span> <span class="pre">'UNK':</span> <span class="pre">137.1484,</span> <span class="pre">'VAL':</span> <span class="pre">117.1469}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.MOLECULAR_WEIGHTS" title="Permalink to this definition">#</a></dt>
<dd><p>Mapping of 3-letter amino acid names to molecular weights. 
<code class="docutils literal notranslate"><span class="pre">UNK</span></code> is used for unknown residues and takes the mean of known weights.
For <code class="docutils literal notranslate"><span class="pre">"ASX"</span></code> and <code class="docutils literal notranslate"><span class="pre">"GLX"</span></code> the average of their constituents (<code class="docutils literal notranslate"><span class="pre">"D"</span></code> and <code class="docutils literal notranslate"><span class="pre">"N"</span></code>, and <code class="docutils literal notranslate"><span class="pre">"E"</span></code> and <code class="docutils literal notranslate"><span class="pre">"Q"</span></code>, respectively) is assigned.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.MOLECULAR_WEIGHTS_STD">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">MOLECULAR_WEIGHTS_STD</span></span><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'ALA':</span> <span class="pre">array([[-1.70781298]]),</span> <span class="pre">'ARG':</span> <span class="pre">array([[1.31682834]]),</span> <span class="pre">'ASN':</span> <span class="pre">array([[-0.17876066]]),</span> <span class="pre">'ASP':</span> <span class="pre">array([[-0.14376208]]),</span> <span class="pre">'ASX':</span> <span class="pre">array([[-0.16126137]]),</span> <span class="pre">'CYS':</span> <span class="pre">array([[-0.56824433]]),</span> <span class="pre">'GLN':</span> <span class="pre">array([[0.31973109]]),</span> <span class="pre">'GLU':</span> <span class="pre">array([[0.35472968]]),</span> <span class="pre">'GLX':</span> <span class="pre">array([[0.33723039]]),</span> <span class="pre">'GLY':</span> <span class="pre">array([[-2.20630119]]),</span> <span class="pre">'HIS':</span> <span class="pre">array([[0.63993903]]),</span> <span class="pre">'ILE':</span> <span class="pre">array([[-0.2123377]]),</span> <span class="pre">'LEU':</span> <span class="pre">array([[-0.2123377]]),</span> <span class="pre">'LYS':</span> <span class="pre">array([[0.32126282]]),</span> <span class="pre">'MET':</span> <span class="pre">array([[0.42873918]]),</span> <span class="pre">'PHE':</span> <span class="pre">array([[0.99656354]]),</span> <span class="pre">'PRO':</span> <span class="pre">array([[-0.78247208]]),</span> <span class="pre">'SER':</span> <span class="pre">array([[-1.13921032]]),</span> <span class="pre">'THR':</span> <span class="pre">array([[-0.64071856]]),</span> <span class="pre">'TRP':</span> <span class="pre">array([[2.38386234]]),</span> <span class="pre">'TYR':</span> <span class="pre">array([[1.56516265]]),</span> <span class="pre">'UNK':</span> <span class="pre">array([[-6.18065683e-07]]),</span> <span class="pre">'VAL':</span> <span class="pre">array([[-0.71082946]])}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.MOLECULAR_WEIGHTS_STD" title="Permalink to this definition">#</a></dt>
<dd><p>Standardized (sklearn.StandardScaler) molecular weights for standard amino acids.</p>
<p>See <a class="reference internal" href="#graphein.protein.resi_atoms.MOLECULAR_WEIGHTS" title="graphein.protein.resi_atoms.MOLECULAR_WEIGHTS"><code class="xref py py-const docutils literal notranslate"><span class="pre">MOLECULAR_WEIGHTS</span></code></a> for details.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.NEG_AA">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">NEG_AA</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['GLU',</span> <span class="pre">'ASP']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.NEG_AA" title="Permalink to this definition">#</a></dt>
<dd><p>Negatively charged amino acids.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.NON_STANDARD_AMINO_ACIDS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">NON_STANDARD_AMINO_ACIDS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['O',</span> <span class="pre">'U']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.NON_STANDARD_AMINO_ACIDS" title="Permalink to this definition">#</a></dt>
<dd><p>Non-standard amino acids with one-letter codes.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.NON_STANDARD_AMINO_ACID_MAPPING_3_TO_1">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">NON_STANDARD_AMINO_ACID_MAPPING_3_TO_1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'PYL':</span> <span class="pre">'O',</span> <span class="pre">'SEC':</span> <span class="pre">'U'}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.NON_STANDARD_AMINO_ACID_MAPPING_3_TO_1" title="Permalink to this definition">#</a></dt>
<dd><p>Mapping of 3-letter non-standard amino acids codes to their one-letter form.</p>
<p>See: <a class="reference external" href="http://ligand-expo.rcsb.org/">http://ligand-expo.rcsb.org/</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.NON_STANDARD_RESIS_NAME">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">NON_STANDARD_RESIS_NAME</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['3-SULFINOALANINE',</span> <span class="pre">'4-HYDROXYPROLINE',</span> <span class="pre">'4-METHYL-4-[(E)-2-BUTENYL]-4,N-METHYL-THREONINE',</span> <span class="pre">'5-HYDROXYPROLINE',</span> <span class="pre">'ACETYL_GROUP',</span> <span class="pre">'ALPHA-AMINOBUTYRIC_ACID',</span> <span class="pre">'ALPHA-AMINOISOBUTYRIC_ACID',</span> <span class="pre">'AMINO_GROUP',</span> <span class="pre">'CARBOXY_GROUP',</span> <span class="pre">'CYSTEINE-S-DIOXIDE',</span> <span class="pre">'CYSTEINESULFONIC_ACID',</span> <span class="pre">'D-ALANINE',</span> <span class="pre">'D-ARGININE',</span> <span class="pre">'D-ASPARAGINE',</span> <span class="pre">'D-ASPARTATE',</span> <span class="pre">'D-CYSTEINE',</span> <span class="pre">'DECARBOXY(PARAHYDROXYBENZYLIDENE-IMIDAZOLIDINONE)THREONINE',</span> <span class="pre">'D-GLUTAMATE',</span> <span class="pre">'D-GLUTAMINE',</span> <span class="pre">'D-HISTIDINE',</span> <span class="pre">'D-ISOLEUCINE',</span> <span class="pre">'D-ISOVALINE',</span> <span class="pre">'D-LEUCINE',</span> <span class="pre">'D-LYSINE',</span> <span class="pre">'D-PHENYLALANINE',</span> <span class="pre">'D-PROLINE',</span> <span class="pre">'D-SERINE',</span> <span class="pre">'D-THREONINE',</span> <span class="pre">'D-TRYPTOPHANE',</span> <span class="pre">'D-TYROSINE',</span> <span class="pre">'D-VALINE',</span> <span class="pre">'FORMYL_GROUP',</span> <span class="pre">'GAMMA-CARBOXY-GLUTAMIC_ACID',</span> <span class="pre">'ISOVALERIC_ACID',</span> <span class="pre">'LYSINE_NZ-CARBOXYLIC_ACID',</span> <span class="pre">"LYSINE-PYRIDOXAL-5'-PHOSPHATE",</span> <span class="pre">'N-CARBOXYMETHIONINE',</span> <span class="pre">'N-FORMYLMETHIONINE',</span> <span class="pre">'N-METHYLLEUCINE',</span> <span class="pre">'N-METHYLVALINE',</span> <span class="pre">'NORLEUCINE',</span> <span class="pre">'O-PHOSPHOTYROSINE',</span> <span class="pre">'ORNITHINE',</span> <span class="pre">'PHOSPHOSERINE',</span> <span class="pre">'PHOSPHOTHREONINE',</span> <span class="pre">'PYROGLUTAMIC_ACID',</span> <span class="pre">'PYRUVOYL_GROUP',</span> <span class="pre">'SARCOSINE',</span> <span class="pre">'S-HYDROXY-CYSTEINE',</span> <span class="pre">'S-HYDROXYCYSTEINE',</span> <span class="pre">'S-MERCAPTOCYSTEINE',</span> <span class="pre">'S-OXY_CYSTEINE',</span> <span class="pre">'S,S-(2-HYDROXYETHYL)THIOCYSTEINE',</span> <span class="pre">'SULFONATED_TYROSINE',</span> <span class="pre">'TERT-BUTYLOXYCARBONYL_GROUP',</span> <span class="pre">'TOPO-QUINONE',</span> <span class="pre">'TYROSINE-O-SULPHONIC_ACID']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.NON_STANDARD_RESIS_NAME" title="Permalink to this definition">#</a></dt>
<dd><p>Non-standard residue info taken from: <a class="reference external" href="https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html">https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html</a>
<code class="docutils literal notranslate"><span class="pre">PYL</span></code> (pyrolysine) and <code class="docutils literal notranslate"><span class="pre">SEC</span></code> are added</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.NON_STANDARD_RESIS_PARENT">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">NON_STANDARD_RESIS_PARENT</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'5HP':</span> <span class="pre">'GLU',</span> <span class="pre">'ABA':</span> <span class="pre">'ALA',</span> <span class="pre">'ACE':</span> <span class="pre">'-',</span> <span class="pre">'AIB':</span> <span class="pre">'ALA',</span> <span class="pre">'BMT':</span> <span class="pre">'THR',</span> <span class="pre">'BOC':</span> <span class="pre">'-',</span> <span class="pre">'CBX':</span> <span class="pre">'-',</span> <span class="pre">'CEA':</span> <span class="pre">'CYS',</span> <span class="pre">'CGU':</span> <span class="pre">'GLU',</span> <span class="pre">'CME':</span> <span class="pre">'CYS',</span> <span class="pre">'CRO':</span> <span class="pre">'CRO',</span> <span class="pre">'CSD':</span> <span class="pre">'CYS',</span> <span class="pre">'CSO':</span> <span class="pre">'CYS',</span> <span class="pre">'CSS':</span> <span class="pre">'CYS',</span> <span class="pre">'CSW':</span> <span class="pre">'CYS',</span> <span class="pre">'CSX':</span> <span class="pre">'CYS',</span> <span class="pre">'CXM':</span> <span class="pre">'MET',</span> <span class="pre">'DAL':</span> <span class="pre">'ALA',</span> <span class="pre">'DAR':</span> <span class="pre">'ARG',</span> <span class="pre">'DCY':</span> <span class="pre">'CYS',</span> <span class="pre">'DGL':</span> <span class="pre">'GLU',</span> <span class="pre">'DGN':</span> <span class="pre">'GLN',</span> <span class="pre">'DHI':</span> <span class="pre">'HIS',</span> <span class="pre">'DIL':</span> <span class="pre">'ILE',</span> <span class="pre">'DIV':</span> <span class="pre">'VAL',</span> <span class="pre">'DLE':</span> <span class="pre">'LEU',</span> <span class="pre">'DLY':</span> <span class="pre">'LYS',</span> <span class="pre">'DPN':</span> <span class="pre">'PHE',</span> <span class="pre">'DPR':</span> <span class="pre">'PRO',</span> <span class="pre">'DSG':</span> <span class="pre">'ASN',</span> <span class="pre">'DSN':</span> <span class="pre">'SER',</span> <span class="pre">'DSP':</span> <span class="pre">'ASP',</span> <span class="pre">'DTH':</span> <span class="pre">'THR',</span> <span class="pre">'DTR':</span> <span class="pre">'DTR',</span> <span class="pre">'DTY':</span> <span class="pre">'TYR',</span> <span class="pre">'DVA':</span> <span class="pre">'VAL',</span> <span class="pre">'FME':</span> <span class="pre">'MET',</span> <span class="pre">'FOR':</span> <span class="pre">'-',</span> <span class="pre">'HYP':</span> <span class="pre">'PRO',</span> <span class="pre">'IVA':</span> <span class="pre">'-',</span> <span class="pre">'KCX':</span> <span class="pre">'LYS',</span> <span class="pre">'LLP':</span> <span class="pre">'LYS',</span> <span class="pre">'MLE':</span> <span class="pre">'LEU',</span> <span class="pre">'MVA':</span> <span class="pre">'VAL',</span> <span class="pre">'NH2':</span> <span class="pre">'-',</span> <span class="pre">'NLE':</span> <span class="pre">'LEU',</span> <span class="pre">'OCS':</span> <span class="pre">'CYS',</span> <span class="pre">'ORN':</span> <span class="pre">'ALA',</span> <span class="pre">'PCA':</span> <span class="pre">'GLU',</span> <span class="pre">'PTR':</span> <span class="pre">'TYR',</span> <span class="pre">'PVL':</span> <span class="pre">'-',</span> <span class="pre">'PYL':</span> <span class="pre">'LYS',</span> <span class="pre">'SAR':</span> <span class="pre">'GLY',</span> <span class="pre">'SEC':</span> <span class="pre">'CYS',</span> <span class="pre">'SEP':</span> <span class="pre">'SER',</span> <span class="pre">'STY':</span> <span class="pre">'TYR',</span> <span class="pre">'TPO':</span> <span class="pre">'THR',</span> <span class="pre">'TPQ':</span> <span class="pre">'PHE',</span> <span class="pre">'TYS':</span> <span class="pre">'TYR'}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.NON_STANDARD_RESIS_PARENT" title="Permalink to this definition">#</a></dt>
<dd><p>Mapping of 3-letter non-standard/modified residues to their 3-letter parent residue names.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.NON_STANDARD_RESI_NAMES">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">NON_STANDARD_RESI_NAMES</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['CSD',</span> <span class="pre">'HYP',</span> <span class="pre">'BMT',</span> <span class="pre">'5HP',</span> <span class="pre">'ACE',</span> <span class="pre">'ABA',</span> <span class="pre">'AIB',</span> <span class="pre">'NH2',</span> <span class="pre">'CBX',</span> <span class="pre">'CSW',</span> <span class="pre">'OCS',</span> <span class="pre">'DAL',</span> <span class="pre">'DAR',</span> <span class="pre">'DSG',</span> <span class="pre">'DSP',</span> <span class="pre">'DCY',</span> <span class="pre">'CRO',</span> <span class="pre">'DGL',</span> <span class="pre">'DGN',</span> <span class="pre">'DHI',</span> <span class="pre">'DIL',</span> <span class="pre">'DIV',</span> <span class="pre">'DLE',</span> <span class="pre">'DLY',</span> <span class="pre">'DPN',</span> <span class="pre">'DPR',</span> <span class="pre">'DSN',</span> <span class="pre">'DTH',</span> <span class="pre">'DTR',</span> <span class="pre">'DTY',</span> <span class="pre">'DVA',</span> <span class="pre">'FOR',</span> <span class="pre">'CGU',</span> <span class="pre">'IVA',</span> <span class="pre">'KCX',</span> <span class="pre">'LLP',</span> <span class="pre">'CXM',</span> <span class="pre">'FME',</span> <span class="pre">'MLE',</span> <span class="pre">'MVA',</span> <span class="pre">'NLE',</span> <span class="pre">'PTR',</span> <span class="pre">'ORN',</span> <span class="pre">'SEP',</span> <span class="pre">'SEC',</span> <span class="pre">'TPO',</span> <span class="pre">'PCA',</span> <span class="pre">'PVL',</span> <span class="pre">'PYL',</span> <span class="pre">'SAR',</span> <span class="pre">'CEA',</span> <span class="pre">'CSO',</span> <span class="pre">'CSS',</span> <span class="pre">'CSX',</span> <span class="pre">'CME',</span> <span class="pre">'TYS',</span> <span class="pre">'BOC',</span> <span class="pre">'TPQ',</span> <span class="pre">'STY']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.NON_STANDARD_RESI_NAMES" title="Permalink to this definition">#</a></dt>
<dd><p>List of non-standard residue 3-letter names.</p>
<p>Collected from: <a class="reference external" href="https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html">https://www.globalphasing.com/buster/manual/maketnt/manual/lib_val/library_validation.html</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.PI_RESIS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">PI_RESIS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['PHE',</span> <span class="pre">'TYR',</span> <span class="pre">'TRP']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.PI_RESIS" title="Permalink to this definition">#</a></dt>
<dd><p>List of residues involved in pi interactions.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.POS_AA">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">POS_AA</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['HIS',</span> <span class="pre">'LYS',</span> <span class="pre">'ARG']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.POS_AA" title="Permalink to this definition">#</a></dt>
<dd><p>Positively charged amino acids.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.RESIDUE_ATOM_BOND_STATE">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">RESIDUE_ATOM_BOND_STATE</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'ARG':</span> <span class="pre">{'CD':</span> <span class="pre">'Csb',</span> <span class="pre">'CG':</span> <span class="pre">'Csb',</span> <span class="pre">'CZ':</span> <span class="pre">'Cdb',</span> <span class="pre">'NE':</span> <span class="pre">'Nsb',</span> <span class="pre">'NH1':</span> <span class="pre">'Nres',</span> <span class="pre">'NH2':</span> <span class="pre">'Nres'},</span> <span class="pre">'ASN':</span> <span class="pre">{'CG':</span> <span class="pre">'Csb',</span> <span class="pre">'ND2':</span> <span class="pre">'Ndb',</span> <span class="pre">'OD1':</span> <span class="pre">'Odb'},</span> <span class="pre">'ASP':</span> <span class="pre">{'CG':</span> <span class="pre">'Csb',</span> <span class="pre">'OD1':</span> <span class="pre">'Ores',</span> <span class="pre">'OD2':</span> <span class="pre">'Ores'},</span> <span class="pre">'CYS':</span> <span class="pre">{'SG':</span> <span class="pre">'Ssb'},</span> <span class="pre">'GLN':</span> <span class="pre">{'CD':</span> <span class="pre">'Csb',</span> <span class="pre">'CG':</span> <span class="pre">'Csb',</span> <span class="pre">'NE2':</span> <span class="pre">'Ndb',</span> <span class="pre">'OE1':</span> <span class="pre">'Odb'},</span> <span class="pre">'GLU':</span> <span class="pre">{'CD':</span> <span class="pre">'Csb',</span> <span class="pre">'CG':</span> <span class="pre">'Csb',</span> <span class="pre">'OE1':</span> <span class="pre">'Ores',</span> <span class="pre">'OE2':</span> <span class="pre">'Ores'},</span> <span class="pre">'HIS':</span> <span class="pre">{'CD2':</span> <span class="pre">'Cdb',</span> <span class="pre">'CE1':</span> <span class="pre">'Cdb',</span> <span class="pre">'CG':</span> <span class="pre">'Cdb',</span> <span class="pre">'ND1':</span> <span class="pre">'Nsb',</span> <span class="pre">'NE2':</span> <span class="pre">'Ndb'},</span> <span class="pre">'ILE':</span> <span class="pre">{'CD1':</span> <span class="pre">'Csb',</span> <span class="pre">'CG1':</span> <span class="pre">'Csb',</span> <span class="pre">'CG2':</span> <span class="pre">'Csb'},</span> <span class="pre">'LEU':</span> <span class="pre">{'CD1':</span> <span class="pre">'Csb',</span> <span class="pre">'CD2':</span> <span class="pre">'Csb',</span> <span class="pre">'CG':</span> <span class="pre">'Csb'},</span> <span class="pre">'LYS':</span> <span class="pre">{'CD':</span> <span class="pre">'Csb',</span> <span class="pre">'CE':</span> <span class="pre">'Csb',</span> <span class="pre">'CG':</span> <span class="pre">'Csb',</span> <span class="pre">'NZ':</span> <span class="pre">'Nsb'},</span> <span class="pre">'MET':</span> <span class="pre">{'CE':</span> <span class="pre">'Csb',</span> <span class="pre">'CG':</span> <span class="pre">'Csb',</span> <span class="pre">'SD':</span> <span class="pre">'Ssb'},</span> <span class="pre">'PHE':</span> <span class="pre">{'CD1':</span> <span class="pre">'Cres',</span> <span class="pre">'CD2':</span> <span class="pre">'Cres',</span> <span class="pre">'CE1':</span> <span class="pre">'Cdb',</span> <span class="pre">'CE2':</span> <span class="pre">'Cdb',</span> <span class="pre">'CG':</span> <span class="pre">'Cdb',</span> <span class="pre">'CZ':</span> <span class="pre">'Cres'},</span> <span class="pre">'PRO':</span> <span class="pre">{'CD':</span> <span class="pre">'Csb',</span> <span class="pre">'CG':</span> <span class="pre">'Csb'},</span> <span class="pre">'SER':</span> <span class="pre">{'OG':</span> <span class="pre">'Osb'},</span> <span class="pre">'THR':</span> <span class="pre">{'CG2':</span> <span class="pre">'Csb',</span> <span class="pre">'OG1':</span> <span class="pre">'Osb'},</span> <span class="pre">'TRP':</span> <span class="pre">{'CD1':</span> <span class="pre">'Cdb',</span> <span class="pre">'CD2':</span> <span class="pre">'Cres',</span> <span class="pre">'CE2':</span> <span class="pre">'Cdb',</span> <span class="pre">'CE3':</span> <span class="pre">'Cdb',</span> <span class="pre">'CG':</span> <span class="pre">'Cdb',</span> <span class="pre">'CH2':</span> <span class="pre">'Cdb',</span> <span class="pre">'CZ2':</span> <span class="pre">'Cres',</span> <span class="pre">'CZ3':</span> <span class="pre">'Cres',</span> <span class="pre">'NE1':</span> <span class="pre">'Nsb'},</span> <span class="pre">'TYR':</span> <span class="pre">{'CD1':</span> <span class="pre">'Cres',</span> <span class="pre">'CD2':</span> <span class="pre">'Cres',</span> <span class="pre">'CE1':</span> <span class="pre">'Cdb',</span> <span class="pre">'CE2':</span> <span class="pre">'Cdb',</span> <span class="pre">'CG':</span> <span class="pre">'Cdb',</span> <span class="pre">'CZ':</span> <span class="pre">'Cres',</span> <span class="pre">'OH':</span> <span class="pre">'Osb'},</span> <span class="pre">'VAL':</span> <span class="pre">{'CG1':</span> <span class="pre">'Csb',</span> <span class="pre">'CG2':</span> <span class="pre">'Csb'},</span> <span class="pre">'XXX':</span> <span class="pre">{'C':</span> <span class="pre">'Cdb',</span> <span class="pre">'CA':</span> <span class="pre">'Csb',</span> <span class="pre">'CB':</span> <span class="pre">'Csb',</span> <span class="pre">'H':</span> <span class="pre">'Hsb',</span> <span class="pre">'N':</span> <span class="pre">'Nsb',</span> <span class="pre">'O':</span> <span class="pre">'Odb',</span> <span class="pre">'OXT':</span> <span class="pre">'Osb'}}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.RESIDUE_ATOM_BOND_STATE" title="Permalink to this definition">#</a></dt>
<dd><p>Assignment of consituent atom classes with each standard residue to atomic radii.</p>
<p>Covalent radii from:</p>
<blockquote>
<div><p>Heyrovska, Raji : ‘Atomic Structures of all the Twenty Essential Amino Acids and a Tripeptide, with Bond Lengths as Sums of Atomic Covalent Radii’</p>
</div></blockquote>
<p>Paper: <a class="reference external" href="https://arxiv.org/pdf/0804.2488.pdf">https://arxiv.org/pdf/0804.2488.pdf</a></p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.RESI_NAMES">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">RESI_NAMES</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['ALA',</span> <span class="pre">'ASX',</span> <span class="pre">'CYS',</span> <span class="pre">'ASP',</span> <span class="pre">'GLU',</span> <span class="pre">'PHE',</span> <span class="pre">'GLY',</span> <span class="pre">'HIS',</span> <span class="pre">'ILE',</span> <span class="pre">'LYS',</span> <span class="pre">'LEU',</span> <span class="pre">'MET',</span> <span class="pre">'ASN',</span> <span class="pre">'PRO',</span> <span class="pre">'GLN',</span> <span class="pre">'ARG',</span> <span class="pre">'SER',</span> <span class="pre">'THR',</span> <span class="pre">'VAL',</span> <span class="pre">'TRP',</span> <span class="pre">'TYR',</span> <span class="pre">'GLX',</span> <span class="pre">'CSD',</span> <span class="pre">'HYP',</span> <span class="pre">'BMT',</span> <span class="pre">'5HP',</span> <span class="pre">'ACE',</span> <span class="pre">'ABA',</span> <span class="pre">'AIB',</span> <span class="pre">'NH2',</span> <span class="pre">'CBX',</span> <span class="pre">'CSW',</span> <span class="pre">'OCS',</span> <span class="pre">'DAL',</span> <span class="pre">'DAR',</span> <span class="pre">'DSG',</span> <span class="pre">'DSP',</span> <span class="pre">'DCY',</span> <span class="pre">'CRO',</span> <span class="pre">'DGL',</span> <span class="pre">'DGN',</span> <span class="pre">'DHI',</span> <span class="pre">'DIL',</span> <span class="pre">'DIV',</span> <span class="pre">'DLE',</span> <span class="pre">'DLY',</span> <span class="pre">'DPN',</span> <span class="pre">'DPR',</span> <span class="pre">'DSN',</span> <span class="pre">'DTH',</span> <span class="pre">'DTR',</span> <span class="pre">'DTY',</span> <span class="pre">'DVA',</span> <span class="pre">'FOR',</span> <span class="pre">'CGU',</span> <span class="pre">'IVA',</span> <span class="pre">'KCX',</span> <span class="pre">'LLP',</span> <span class="pre">'CXM',</span> <span class="pre">'FME',</span> <span class="pre">'MLE',</span> <span class="pre">'MVA',</span> <span class="pre">'NLE',</span> <span class="pre">'PTR',</span> <span class="pre">'ORN',</span> <span class="pre">'SEP',</span> <span class="pre">'SEC',</span> <span class="pre">'TPO',</span> <span class="pre">'PCA',</span> <span class="pre">'PVL',</span> <span class="pre">'PYL',</span> <span class="pre">'SAR',</span> <span class="pre">'CEA',</span> <span class="pre">'CSO',</span> <span class="pre">'CSS',</span> <span class="pre">'CSX',</span> <span class="pre">'CME',</span> <span class="pre">'TYS',</span> <span class="pre">'BOC',</span> <span class="pre">'TPQ',</span> <span class="pre">'STY',</span> <span class="pre">'UNK']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.RESI_NAMES" title="Permalink to this definition">#</a></dt>
<dd><p>3-letter residue names for all amino acids.
Non-standard/modified amino acids are mapped to their parent amino acid.
Includes <code class="docutils literal notranslate"><span class="pre">"UNK"</span></code> to denote unknown residues.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.RESI_THREE_TO_1">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">RESI_THREE_TO_1</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'3HP':</span> <span class="pre">'X',</span> <span class="pre">'4HP':</span> <span class="pre">'X',</span> <span class="pre">'5HP':</span> <span class="pre">'Q',</span> <span class="pre">'ABA':</span> <span class="pre">'A',</span> <span class="pre">'ACE':</span> <span class="pre">'X',</span> <span class="pre">'AIB':</span> <span class="pre">'A',</span> <span class="pre">'ALA':</span> <span class="pre">'A',</span> <span class="pre">'ARG':</span> <span class="pre">'R',</span> <span class="pre">'ASN':</span> <span class="pre">'N',</span> <span class="pre">'ASP':</span> <span class="pre">'D',</span> <span class="pre">'ASX':</span> <span class="pre">'B',</span> <span class="pre">'BMT':</span> <span class="pre">'T',</span> <span class="pre">'BOC':</span> <span class="pre">'X',</span> <span class="pre">'CBX':</span> <span class="pre">'X',</span> <span class="pre">'CEA':</span> <span class="pre">'C',</span> <span class="pre">'CGU':</span> <span class="pre">'E',</span> <span class="pre">'CME':</span> <span class="pre">'C',</span> <span class="pre">'CRO':</span> <span class="pre">'TYG',</span> <span class="pre">'CSD':</span> <span class="pre">'C',</span> <span class="pre">'CSO':</span> <span class="pre">'C',</span> <span class="pre">'CSS':</span> <span class="pre">'C',</span> <span class="pre">'CSW':</span> <span class="pre">'C',</span> <span class="pre">'CSX':</span> <span class="pre">'C',</span> <span class="pre">'CXM':</span> <span class="pre">'M',</span> <span class="pre">'CYS':</span> <span class="pre">'C',</span> <span class="pre">'DAL':</span> <span class="pre">'A',</span> <span class="pre">'DAR':</span> <span class="pre">'R',</span> <span class="pre">'DCY':</span> <span class="pre">'C',</span> <span class="pre">'DGL':</span> <span class="pre">'E',</span> <span class="pre">'DGN':</span> <span class="pre">'Q',</span> <span class="pre">'DHI':</span> <span class="pre">'H',</span> <span class="pre">'DIL':</span> <span class="pre">'I',</span> <span class="pre">'DIV':</span> <span class="pre">'V',</span> <span class="pre">'DLE':</span> <span class="pre">'L',</span> <span class="pre">'DLY':</span> <span class="pre">'K',</span> <span class="pre">'DPN':</span> <span class="pre">'F',</span> <span class="pre">'DPR':</span> <span class="pre">'P',</span> <span class="pre">'DSG':</span> <span class="pre">'N',</span> <span class="pre">'DSN':</span> <span class="pre">'S',</span> <span class="pre">'DSP':</span> <span class="pre">'D',</span> <span class="pre">'DTH':</span> <span class="pre">'T',</span> <span class="pre">'DTR':</span> <span class="pre">'W',</span> <span class="pre">'DTY':</span> <span class="pre">'Y',</span> <span class="pre">'DVA':</span> <span class="pre">'V',</span> <span class="pre">'FME':</span> <span class="pre">'M',</span> <span class="pre">'FOR':</span> <span class="pre">'X',</span> <span class="pre">'GLN':</span> <span class="pre">'Q',</span> <span class="pre">'GLU':</span> <span class="pre">'E',</span> <span class="pre">'GLX':</span> <span class="pre">'Z',</span> <span class="pre">'GLY':</span> <span class="pre">'G',</span> <span class="pre">'HIS':</span> <span class="pre">'H',</span> <span class="pre">'HYP':</span> <span class="pre">'P',</span> <span class="pre">'ILE':</span> <span class="pre">'I',</span> <span class="pre">'IVA':</span> <span class="pre">'X',</span> <span class="pre">'KCX':</span> <span class="pre">'K',</span> <span class="pre">'LEU':</span> <span class="pre">'L',</span> <span class="pre">'LLP':</span> <span class="pre">'K',</span> <span class="pre">'LYS':</span> <span class="pre">'K',</span> <span class="pre">'MET':</span> <span class="pre">'M',</span> <span class="pre">'MLE':</span> <span class="pre">'L',</span> <span class="pre">'MVA':</span> <span class="pre">'V',</span> <span class="pre">'NH2':</span> <span class="pre">'X',</span> <span class="pre">'NLE':</span> <span class="pre">'L',</span> <span class="pre">'OCS':</span> <span class="pre">'C',</span> <span class="pre">'ORN':</span> <span class="pre">'A',</span> <span class="pre">'PCA':</span> <span class="pre">'Q',</span> <span class="pre">'PHE':</span> <span class="pre">'F',</span> <span class="pre">'PRO':</span> <span class="pre">'P',</span> <span class="pre">'PTR':</span> <span class="pre">'Y',</span> <span class="pre">'PVL':</span> <span class="pre">'X',</span> <span class="pre">'PYL':</span> <span class="pre">'O',</span> <span class="pre">'SAR':</span> <span class="pre">'G',</span> <span class="pre">'SEC':</span> <span class="pre">'U',</span> <span class="pre">'SEP':</span> <span class="pre">'S',</span> <span class="pre">'SER':</span> <span class="pre">'S',</span> <span class="pre">'STY':</span> <span class="pre">'Y',</span> <span class="pre">'THR':</span> <span class="pre">'T',</span> <span class="pre">'TPO':</span> <span class="pre">'T',</span> <span class="pre">'TPQ':</span> <span class="pre">'Y',</span> <span class="pre">'TRP':</span> <span class="pre">'W',</span> <span class="pre">'TYR':</span> <span class="pre">'Y',</span> <span class="pre">'TYS':</span> <span class="pre">'Y',</span> <span class="pre">'UNK':</span> <span class="pre">'X',</span> <span class="pre">'VAL':</span> <span class="pre">'V'}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.RESI_THREE_TO_1" title="Permalink to this definition">#</a></dt>
<dd><p>Mapping of 3-letter residue names to 1-letter residue names.
Non-standard/modified amino acids are mapped to their parent amino acid.
Includes <code class="docutils literal notranslate"><span class="pre">"UNK"</span></code> to denote unknown residues.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.SCHNEIDER_WREDE_DISTMAT">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">SCHNEIDER_WREDE_DISTMAT</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.10)"><span class="pre">float</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">{'AA':</span> <span class="pre">0.0,</span> <span class="pre">'AC':</span> <span class="pre">0.112,</span> <span class="pre">'AD':</span> <span class="pre">0.819,</span> <span class="pre">'AE':</span> <span class="pre">0.827,</span> <span class="pre">'AF':</span> <span class="pre">0.54,</span> <span class="pre">'AG':</span> <span class="pre">0.208,</span> <span class="pre">'AH':</span> <span class="pre">0.696,</span> <span class="pre">'AI':</span> <span class="pre">0.407,</span> <span class="pre">'AK':</span> <span class="pre">0.891,</span> <span class="pre">'AL':</span> <span class="pre">0.406,</span> <span class="pre">'AM':</span> <span class="pre">0.379,</span> <span class="pre">'AN':</span> <span class="pre">0.318,</span> <span class="pre">'AP':</span> <span class="pre">0.191,</span> <span class="pre">'AQ':</span> <span class="pre">0.372,</span> <span class="pre">'AR':</span> <span class="pre">1.0,</span> <span class="pre">'AS':</span> <span class="pre">0.094,</span> <span class="pre">'AT':</span> <span class="pre">0.22,</span> <span class="pre">'AV':</span> <span class="pre">0.273,</span> <span class="pre">'AW':</span> <span class="pre">0.739,</span> <span class="pre">'AY':</span> <span class="pre">0.552,</span> <span class="pre">'CA':</span> <span class="pre">0.114,</span> <span class="pre">'CC':</span> <span class="pre">0.0,</span> <span class="pre">'CD':</span> <span class="pre">0.847,</span> <span class="pre">'CE':</span> <span class="pre">0.838,</span> <span class="pre">'CF':</span> <span class="pre">0.437,</span> <span class="pre">'CG':</span> <span class="pre">0.32,</span> <span class="pre">'CH':</span> <span class="pre">0.66,</span> <span class="pre">'CI':</span> <span class="pre">0.304,</span> <span class="pre">'CK':</span> <span class="pre">0.887,</span> <span class="pre">'CL':</span> <span class="pre">0.301,</span> <span class="pre">'CM':</span> <span class="pre">0.277,</span> <span class="pre">'CN':</span> <span class="pre">0.324,</span> <span class="pre">'CP':</span> <span class="pre">0.157,</span> <span class="pre">'CQ':</span> <span class="pre">0.341,</span> <span class="pre">'CR':</span> <span class="pre">1.0,</span> <span class="pre">'CS':</span> <span class="pre">0.176,</span> <span class="pre">'CT':</span> <span class="pre">0.233,</span> <span class="pre">'CV':</span> <span class="pre">0.167,</span> <span class="pre">'CW':</span> <span class="pre">0.639,</span> <span class="pre">'CY':</span> <span class="pre">0.457,</span> <span class="pre">'DA':</span> <span class="pre">0.729,</span> <span class="pre">'DC':</span> <span class="pre">0.742,</span> <span class="pre">'DD':</span> <span class="pre">0.0,</span> <span class="pre">'DE':</span> <span class="pre">0.124,</span> <span class="pre">'DF':</span> <span class="pre">0.924,</span> <span class="pre">'DG':</span> <span class="pre">0.697,</span> <span class="pre">'DH':</span> <span class="pre">0.435,</span> <span class="pre">'DI':</span> <span class="pre">0.847,</span> <span class="pre">'DK':</span> <span class="pre">0.249,</span> <span class="pre">'DL':</span> <span class="pre">0.841,</span> <span class="pre">'DM':</span> <span class="pre">0.819,</span> <span class="pre">'DN':</span> <span class="pre">0.56,</span> <span class="pre">'DP':</span> <span class="pre">0.657,</span> <span class="pre">'DQ':</span> <span class="pre">0.584,</span> <span class="pre">'DR':</span> <span class="pre">0.295,</span> <span class="pre">'DS':</span> <span class="pre">0.667,</span> <span class="pre">'DT':</span> <span class="pre">0.649,</span> <span class="pre">'DV':</span> <span class="pre">0.797,</span> <span class="pre">'DW':</span> <span class="pre">1.0,</span> <span class="pre">'DY':</span> <span class="pre">0.836,</span> <span class="pre">'EA':</span> <span class="pre">0.79,</span> <span class="pre">'EC':</span> <span class="pre">0.788,</span> <span class="pre">'ED':</span> <span class="pre">0.133,</span> <span class="pre">'EE':</span> <span class="pre">0.0,</span> <span class="pre">'EF':</span> <span class="pre">0.932,</span> <span class="pre">'EG':</span> <span class="pre">0.779,</span> <span class="pre">'EH':</span> <span class="pre">0.406,</span> <span class="pre">'EI':</span> <span class="pre">0.86,</span> <span class="pre">'EK':</span> <span class="pre">0.143,</span> <span class="pre">'EL':</span> <span class="pre">0.854,</span> <span class="pre">'EM':</span> <span class="pre">0.83,</span> <span class="pre">'EN':</span> <span class="pre">0.599,</span> <span class="pre">'EP':</span> <span class="pre">0.688,</span> <span class="pre">'EQ':</span> <span class="pre">0.598,</span> <span class="pre">'ER':</span> <span class="pre">0.234,</span> <span class="pre">'ES':</span> <span class="pre">0.726,</span> <span class="pre">'ET':</span> <span class="pre">0.682,</span> <span class="pre">'EV':</span> <span class="pre">0.824,</span> <span class="pre">'EW':</span> <span class="pre">1.0,</span> <span class="pre">'EY':</span> <span class="pre">0.837,</span> <span class="pre">'FA':</span> <span class="pre">0.508,</span> <span class="pre">'FC':</span> <span class="pre">0.405,</span> <span class="pre">'FD':</span> <span class="pre">0.977,</span> <span class="pre">'FE':</span> <span class="pre">0.918,</span> <span class="pre">'FF':</span> <span class="pre">0.0,</span> <span class="pre">'FG':</span> <span class="pre">0.69,</span> <span class="pre">'FH':</span> <span class="pre">0.663,</span> <span class="pre">'FI':</span> <span class="pre">0.128,</span> <span class="pre">'FK':</span> <span class="pre">0.903,</span> <span class="pre">'FL':</span> <span class="pre">0.131,</span> <span class="pre">'FM':</span> <span class="pre">0.169,</span> <span class="pre">'FN':</span> <span class="pre">0.541,</span> <span class="pre">'FP':</span> <span class="pre">0.42,</span> <span class="pre">'FQ':</span> <span class="pre">0.459,</span> <span class="pre">'FR':</span> <span class="pre">1.0,</span> <span class="pre">'FS':</span> <span class="pre">0.548,</span> <span class="pre">'FT':</span> <span class="pre">0.499,</span> <span class="pre">'FV':</span> <span class="pre">0.252,</span> <span class="pre">'FW':</span> <span class="pre">0.207,</span> <span class="pre">'FY':</span> <span class="pre">0.179,</span> <span class="pre">'GA':</span> <span class="pre">0.206,</span> <span class="pre">'GC':</span> <span class="pre">0.312,</span> <span class="pre">'GD':</span> <span class="pre">0.776,</span> <span class="pre">'GE':</span> <span class="pre">0.807,</span> <span class="pre">'GF':</span> <span class="pre">0.727,</span> <span class="pre">'GG':</span> <span class="pre">0.0,</span> <span class="pre">'GH':</span> <span class="pre">0.769,</span> <span class="pre">'GI':</span> <span class="pre">0.592,</span> <span class="pre">'GK':</span> <span class="pre">0.894,</span> <span class="pre">'GL':</span> <span class="pre">0.591,</span> <span class="pre">'GM':</span> <span class="pre">0.557,</span> <span class="pre">'GN':</span> <span class="pre">0.381,</span> <span class="pre">'GP':</span> <span class="pre">0.323,</span> <span class="pre">'GQ':</span> <span class="pre">0.467,</span> <span class="pre">'GR':</span> <span class="pre">1.0,</span> <span class="pre">'GS':</span> <span class="pre">0.158,</span> <span class="pre">'GT':</span> <span class="pre">0.272,</span> <span class="pre">'GV':</span> <span class="pre">0.464,</span> <span class="pre">'GW':</span> <span class="pre">0.923,</span> <span class="pre">'GY':</span> <span class="pre">0.728,</span> <span class="pre">'HA':</span> <span class="pre">0.896,</span> <span class="pre">'HC':</span> <span class="pre">0.836,</span> <span class="pre">'HD':</span> <span class="pre">0.629,</span> <span class="pre">'HE':</span> <span class="pre">0.547,</span> <span class="pre">'HF':</span> <span class="pre">0.907,</span> <span class="pre">'HG':</span> <span class="pre">1.0,</span> <span class="pre">'HH':</span> <span class="pre">0.0,</span> <span class="pre">'HI':</span> <span class="pre">0.848,</span> <span class="pre">'HK':</span> <span class="pre">0.566,</span> <span class="pre">'HL':</span> <span class="pre">0.842,</span> <span class="pre">'HM':</span> <span class="pre">0.825,</span> <span class="pre">'HN':</span> <span class="pre">0.754,</span> <span class="pre">'HP':</span> <span class="pre">0.777,</span> <span class="pre">'HQ':</span> <span class="pre">0.716,</span> <span class="pre">'HR':</span> <span class="pre">0.697,</span> <span class="pre">'HS':</span> <span class="pre">0.865,</span> <span class="pre">'HT':</span> <span class="pre">0.834,</span> <span class="pre">'HV':</span> <span class="pre">0.831,</span> <span class="pre">'HW':</span> <span class="pre">0.981,</span> <span class="pre">'HY':</span> <span class="pre">0.821,</span> <span class="pre">'IA':</span> <span class="pre">0.403,</span> <span class="pre">'IC':</span> <span class="pre">0.296,</span> <span class="pre">'ID':</span> <span class="pre">0.942,</span> <span class="pre">'IE':</span> <span class="pre">0.891,</span> <span class="pre">'IF':</span> <span class="pre">0.134,</span> <span class="pre">'IG':</span> <span class="pre">0.592,</span> <span class="pre">'IH':</span> <span class="pre">0.652,</span> <span class="pre">'II':</span> <span class="pre">0.0,</span> <span class="pre">'IK':</span> <span class="pre">0.892,</span> <span class="pre">'IL':</span> <span class="pre">0.013,</span> <span class="pre">'IM':</span> <span class="pre">0.057,</span> <span class="pre">'IN':</span> <span class="pre">0.457,</span> <span class="pre">'IP':</span> <span class="pre">0.311,</span> <span class="pre">'IQ':</span> <span class="pre">0.383,</span> <span class="pre">'IR':</span> <span class="pre">1.0,</span> <span class="pre">'IS':</span> <span class="pre">0.443,</span> <span class="pre">'IT':</span> <span class="pre">0.396,</span> <span class="pre">'IV':</span> <span class="pre">0.133,</span> <span class="pre">'IW':</span> <span class="pre">0.339,</span> <span class="pre">'IY':</span> <span class="pre">0.213,</span> <span class="pre">'KA':</span> <span class="pre">0.889,</span> <span class="pre">'KC':</span> <span class="pre">0.871,</span> <span class="pre">'KD':</span> <span class="pre">0.279,</span> <span class="pre">'KE':</span> <span class="pre">0.149,</span> <span class="pre">'KF':</span> <span class="pre">0.957,</span> <span class="pre">'KG':</span> <span class="pre">0.9,</span> <span class="pre">'KH':</span> <span class="pre">0.438,</span> <span class="pre">'KI':</span> <span class="pre">0.899,</span> <span class="pre">'KK':</span> <span class="pre">0.0,</span> <span class="pre">'KL':</span> <span class="pre">0.892,</span> <span class="pre">'KM':</span> <span class="pre">0.871,</span> <span class="pre">'KN':</span> <span class="pre">0.667,</span> <span class="pre">'KP':</span> <span class="pre">0.757,</span> <span class="pre">'KQ':</span> <span class="pre">0.639,</span> <span class="pre">'KR':</span> <span class="pre">0.154,</span> <span class="pre">'KS':</span> <span class="pre">0.825,</span> <span class="pre">'KT':</span> <span class="pre">0.759,</span> <span class="pre">'KV':</span> <span class="pre">0.882,</span> <span class="pre">'KW':</span> <span class="pre">1.0,</span> <span class="pre">'KY':</span> <span class="pre">0.848,</span> <span class="pre">'LA':</span> <span class="pre">0.405,</span> <span class="pre">'LC':</span> <span class="pre">0.296,</span> <span class="pre">'LD':</span> <span class="pre">0.944,</span> <span class="pre">'LE':</span> <span class="pre">0.892,</span> <span class="pre">'LF':</span> <span class="pre">0.139,</span> <span class="pre">'LG':</span> <span class="pre">0.596,</span> <span class="pre">'LH':</span> <span class="pre">0.653,</span> <span class="pre">'LI':</span> <span class="pre">0.013,</span> <span class="pre">'LK':</span> <span class="pre">0.893,</span> <span class="pre">'LL':</span> <span class="pre">0.0,</span> <span class="pre">'LM':</span> <span class="pre">0.062,</span> <span class="pre">'LN':</span> <span class="pre">0.452,</span> <span class="pre">'LP':</span> <span class="pre">0.309,</span> <span class="pre">'LQ':</span> <span class="pre">0.376,</span> <span class="pre">'LR':</span> <span class="pre">1.0,</span> <span class="pre">'LS':</span> <span class="pre">0.443,</span> <span class="pre">'LT':</span> <span class="pre">0.397,</span> <span class="pre">'LV':</span> <span class="pre">0.133,</span> <span class="pre">'LW':</span> <span class="pre">0.341,</span> <span class="pre">'LY':</span> <span class="pre">0.205,</span> <span class="pre">'MA':</span> <span class="pre">0.383,</span> <span class="pre">'MC':</span> <span class="pre">0.276,</span> <span class="pre">'MD':</span> <span class="pre">0.932,</span> <span class="pre">'ME':</span> <span class="pre">0.879,</span> <span class="pre">'MF':</span> <span class="pre">0.182,</span> <span class="pre">'MG':</span> <span class="pre">0.569,</span> <span class="pre">'MH':</span> <span class="pre">0.648,</span> <span class="pre">'MI':</span> <span class="pre">0.058,</span> <span class="pre">'MK':</span> <span class="pre">0.884,</span> <span class="pre">'ML':</span> <span class="pre">0.062,</span> <span class="pre">'MM':</span> <span class="pre">0.0,</span> <span class="pre">'MN':</span> <span class="pre">0.447,</span> <span class="pre">'MP':</span> <span class="pre">0.285,</span> <span class="pre">'MQ':</span> <span class="pre">0.372,</span> <span class="pre">'MR':</span> <span class="pre">1.0,</span> <span class="pre">'MS':</span> <span class="pre">0.417,</span> <span class="pre">'MT':</span> <span class="pre">0.358,</span> <span class="pre">'MV':</span> <span class="pre">0.12,</span> <span class="pre">'MW':</span> <span class="pre">0.391,</span> <span class="pre">'MY':</span> <span class="pre">0.255,</span> <span class="pre">'NA':</span> <span class="pre">0.424,</span> <span class="pre">'NC':</span> <span class="pre">0.425,</span> <span class="pre">'ND':</span> <span class="pre">0.838,</span> <span class="pre">'NE':</span> <span class="pre">0.835,</span> <span class="pre">'NF':</span> <span class="pre">0.766,</span> <span class="pre">'NG':</span> <span class="pre">0.512,</span> <span class="pre">'NH':</span> <span class="pre">0.78,</span> <span class="pre">'NI':</span> <span class="pre">0.615,</span> <span class="pre">'NK':</span> <span class="pre">0.891,</span> <span class="pre">'NL':</span> <span class="pre">0.603,</span> <span class="pre">'NM':</span> <span class="pre">0.588,</span> <span class="pre">'NN':</span> <span class="pre">0.0,</span> <span class="pre">'NP':</span> <span class="pre">0.266,</span> <span class="pre">'NQ':</span> <span class="pre">0.175,</span> <span class="pre">'NR':</span> <span class="pre">1.0,</span> <span class="pre">'NS':</span> <span class="pre">0.361,</span> <span class="pre">'NT':</span> <span class="pre">0.368,</span> <span class="pre">'NV':</span> <span class="pre">0.503,</span> <span class="pre">'NW':</span> <span class="pre">0.945,</span> <span class="pre">'NY':</span> <span class="pre">0.641,</span> <span class="pre">'PA':</span> <span class="pre">0.22,</span> <span class="pre">'PC':</span> <span class="pre">0.179,</span> <span class="pre">'PD':</span> <span class="pre">0.852,</span> <span class="pre">'PE':</span> <span class="pre">0.831,</span> <span class="pre">'PF':</span> <span class="pre">0.515,</span> <span class="pre">'PG':</span> <span class="pre">0.376,</span> <span class="pre">'PH':</span> <span class="pre">0.696,</span> <span class="pre">'PI':</span> <span class="pre">0.363,</span> <span class="pre">'PK':</span> <span class="pre">0.875,</span> <span class="pre">'PL':</span> <span class="pre">0.357,</span> <span class="pre">'PM':</span> <span class="pre">0.326,</span> <span class="pre">'PN':</span> <span class="pre">0.231,</span> <span class="pre">'PP':</span> <span class="pre">0.0,</span> <span class="pre">'PQ':</span> <span class="pre">0.228,</span> <span class="pre">'PR':</span> <span class="pre">1.0,</span> <span class="pre">'PS':</span> <span class="pre">0.196,</span> <span class="pre">'PT':</span> <span class="pre">0.161,</span> <span class="pre">'PV':</span> <span class="pre">0.244,</span> <span class="pre">'PW':</span> <span class="pre">0.72,</span> <span class="pre">'PY':</span> <span class="pre">0.481,</span> <span class="pre">'QA':</span> <span class="pre">0.512,</span> <span class="pre">'QC':</span> <span class="pre">0.462,</span> <span class="pre">'QD':</span> <span class="pre">0.903,</span> <span class="pre">'QE':</span> <span class="pre">0.861,</span> <span class="pre">'QF':</span> <span class="pre">0.671,</span> <span class="pre">'QG':</span> <span class="pre">0.648,</span> <span class="pre">'QH':</span> <span class="pre">0.765,</span> <span class="pre">'QI':</span> <span class="pre">0.532,</span> <span class="pre">'QK':</span> <span class="pre">0.881,</span> <span class="pre">'QL':</span> <span class="pre">0.518,</span> <span class="pre">'QM':</span> <span class="pre">0.505,</span> <span class="pre">'QN':</span> <span class="pre">0.181,</span> <span class="pre">'QP':</span> <span class="pre">0.272,</span> <span class="pre">'QQ':</span> <span class="pre">0.0,</span> <span class="pre">'QR':</span> <span class="pre">1.0,</span> <span class="pre">'QS':</span> <span class="pre">0.461,</span> <span class="pre">'QT':</span> <span class="pre">0.389,</span> <span class="pre">'QV':</span> <span class="pre">0.464,</span> <span class="pre">'QW':</span> <span class="pre">0.831,</span> <span class="pre">'QY':</span> <span class="pre">0.522,</span> <span class="pre">'RA':</span> <span class="pre">0.919,</span> <span class="pre">'RC':</span> <span class="pre">0.905,</span> <span class="pre">'RD':</span> <span class="pre">0.305,</span> <span class="pre">'RE':</span> <span class="pre">0.225,</span> <span class="pre">'RF':</span> <span class="pre">0.977,</span> <span class="pre">'RG':</span> <span class="pre">0.928,</span> <span class="pre">'RH':</span> <span class="pre">0.498,</span> <span class="pre">'RI':</span> <span class="pre">0.929,</span> <span class="pre">'RK':</span> <span class="pre">0.141,</span> <span class="pre">'RL':</span> <span class="pre">0.92,</span> <span class="pre">'RM':</span> <span class="pre">0.908,</span> <span class="pre">'RN':</span> <span class="pre">0.69,</span> <span class="pre">'RP':</span> <span class="pre">0.796,</span> <span class="pre">'RQ':</span> <span class="pre">0.668,</span> <span class="pre">'RR':</span> <span class="pre">0.0,</span> <span class="pre">'RS':</span> <span class="pre">0.86,</span> <span class="pre">'RT':</span> <span class="pre">0.808,</span> <span class="pre">'RV':</span> <span class="pre">0.914,</span> <span class="pre">'RW':</span> <span class="pre">1.0,</span> <span class="pre">'RY':</span> <span class="pre">0.859,</span> <span class="pre">'SA':</span> <span class="pre">0.1,</span> <span class="pre">'SC':</span> <span class="pre">0.185,</span> <span class="pre">'SD':</span> <span class="pre">0.801,</span> <span class="pre">'SE':</span> <span class="pre">0.812,</span> <span class="pre">'SF':</span> <span class="pre">0.622,</span> <span class="pre">'SG':</span> <span class="pre">0.17,</span> <span class="pre">'SH':</span> <span class="pre">0.718,</span> <span class="pre">'SI':</span> <span class="pre">0.478,</span> <span class="pre">'SK':</span> <span class="pre">0.883,</span> <span class="pre">'SL':</span> <span class="pre">0.474,</span> <span class="pre">'SM':</span> <span class="pre">0.44,</span> <span class="pre">'SN':</span> <span class="pre">0.289,</span> <span class="pre">'SP':</span> <span class="pre">0.181,</span> <span class="pre">'SQ':</span> <span class="pre">0.358,</span> <span class="pre">'SR':</span> <span class="pre">1.0,</span> <span class="pre">'SS':</span> <span class="pre">0.0,</span> <span class="pre">'ST':</span> <span class="pre">0.174,</span> <span class="pre">'SV':</span> <span class="pre">0.342,</span> <span class="pre">'SW':</span> <span class="pre">0.827,</span> <span class="pre">'SY':</span> <span class="pre">0.615,</span> <span class="pre">'TA':</span> <span class="pre">0.251,</span> <span class="pre">'TC':</span> <span class="pre">0.261,</span> <span class="pre">'TD':</span> <span class="pre">0.83,</span> <span class="pre">'TE':</span> <span class="pre">0.812,</span> <span class="pre">'TF':</span> <span class="pre">0.604,</span> <span class="pre">'TG':</span> <span class="pre">0.312,</span> <span class="pre">'TH':</span> <span class="pre">0.737,</span> <span class="pre">'TI':</span> <span class="pre">0.455,</span> <span class="pre">'TK':</span> <span class="pre">0.866,</span> <span class="pre">'TL':</span> <span class="pre">0.453,</span> <span class="pre">'TM':</span> <span class="pre">0.403,</span> <span class="pre">'TN':</span> <span class="pre">0.315,</span> <span class="pre">'TP':</span> <span class="pre">0.159,</span> <span class="pre">'TQ':</span> <span class="pre">0.322,</span> <span class="pre">'TR':</span> <span class="pre">1.0,</span> <span class="pre">'TS':</span> <span class="pre">0.185,</span> <span class="pre">'TT':</span> <span class="pre">0.0,</span> <span class="pre">'TV':</span> <span class="pre">0.345,</span> <span class="pre">'TW':</span> <span class="pre">0.816,</span> <span class="pre">'TY':</span> <span class="pre">0.596,</span> <span class="pre">'VA':</span> <span class="pre">0.275,</span> <span class="pre">'VC':</span> <span class="pre">0.165,</span> <span class="pre">'VD':</span> <span class="pre">0.9,</span> <span class="pre">'VE':</span> <span class="pre">0.867,</span> <span class="pre">'VF':</span> <span class="pre">0.269,</span> <span class="pre">'VG':</span> <span class="pre">0.471,</span> <span class="pre">'VH':</span> <span class="pre">0.649,</span> <span class="pre">'VI':</span> <span class="pre">0.135,</span> <span class="pre">'VK':</span> <span class="pre">0.889,</span> <span class="pre">'VL':</span> <span class="pre">0.134,</span> <span class="pre">'VM':</span> <span class="pre">0.12,</span> <span class="pre">'VN':</span> <span class="pre">0.38,</span> <span class="pre">'VP':</span> <span class="pre">0.212,</span> <span class="pre">'VQ':</span> <span class="pre">0.339,</span> <span class="pre">'VR':</span> <span class="pre">1.0,</span> <span class="pre">'VS':</span> <span class="pre">0.322,</span> <span class="pre">'VT':</span> <span class="pre">0.305,</span> <span class="pre">'VV':</span> <span class="pre">0.0,</span> <span class="pre">'VW':</span> <span class="pre">0.472,</span> <span class="pre">'VY':</span> <span class="pre">0.31,</span> <span class="pre">'WA':</span> <span class="pre">0.658,</span> <span class="pre">'WC':</span> <span class="pre">0.56,</span> <span class="pre">'WD':</span> <span class="pre">1.0,</span> <span class="pre">'WE':</span> <span class="pre">0.931,</span> <span class="pre">'WF':</span> <span class="pre">0.196,</span> <span class="pre">'WG':</span> <span class="pre">0.829,</span> <span class="pre">'WH':</span> <span class="pre">0.678,</span> <span class="pre">'WI':</span> <span class="pre">0.305,</span> <span class="pre">'WK':</span> <span class="pre">0.892,</span> <span class="pre">'WL':</span> <span class="pre">0.304,</span> <span class="pre">'WM':</span> <span class="pre">0.344,</span> <span class="pre">'WN':</span> <span class="pre">0.631,</span> <span class="pre">'WP':</span> <span class="pre">0.555,</span> <span class="pre">'WQ':</span> <span class="pre">0.538,</span> <span class="pre">'WR':</span> <span class="pre">0.968,</span> <span class="pre">'WS':</span> <span class="pre">0.689,</span> <span class="pre">'WT':</span> <span class="pre">0.638,</span> <span class="pre">'WV':</span> <span class="pre">0.418,</span> <span class="pre">'WW':</span> <span class="pre">0.0,</span> <span class="pre">'WY':</span> <span class="pre">0.204,</span> <span class="pre">'YA':</span> <span class="pre">0.587,</span> <span class="pre">'YC':</span> <span class="pre">0.478,</span> <span class="pre">'YD':</span> <span class="pre">1.0,</span> <span class="pre">'YE':</span> <span class="pre">0.932,</span> <span class="pre">'YF':</span> <span class="pre">0.202,</span> <span class="pre">'YG':</span> <span class="pre">0.782,</span> <span class="pre">'YH':</span> <span class="pre">0.678,</span> <span class="pre">'YI':</span> <span class="pre">0.23,</span> <span class="pre">'YK':</span> <span class="pre">0.904,</span> <span class="pre">'YL':</span> <span class="pre">0.219,</span> <span class="pre">'YM':</span> <span class="pre">0.268,</span> <span class="pre">'YN':</span> <span class="pre">0.512,</span> <span class="pre">'YP':</span> <span class="pre">0.444,</span> <span class="pre">'YQ':</span> <span class="pre">0.404,</span> <span class="pre">'YR':</span> <span class="pre">0.995,</span> <span class="pre">'YS':</span> <span class="pre">0.612,</span> <span class="pre">'YT':</span> <span class="pre">0.557,</span> <span class="pre">'YV':</span> <span class="pre">0.328,</span> <span class="pre">'YW':</span> <span class="pre">0.244,</span> <span class="pre">'YY':</span> <span class="pre">0.0}</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.SCHNEIDER_WREDE_DISTMAT" title="Permalink to this definition">#</a></dt>
<dd><p>Scheider-Wrede Physicochemical Distance Matrix taken from ProPy3 <a class="reference external" href="https://github.com/MartinThoma/propy3">https://github.com/MartinThoma/propy3</a>.</p>
<p><strong>Paper</strong></p>
<blockquote>
<div><p>The rational design of amino acid sequences by artificial neural networks and simulated molecular evolution: de novo design of an idealized leader peptidase cleavage site
Biophysical Journal
Volume 66, Issue 2, Part 1, February 1994, Pages 335-344
G.Schneider, P.Wrede</p>
</div></blockquote>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.STANDARD_AMINO_ACIDS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">STANDARD_AMINO_ACIDS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['A',</span> <span class="pre">'B',</span> <span class="pre">'C',</span> <span class="pre">'D',</span> <span class="pre">'E',</span> <span class="pre">'F',</span> <span class="pre">'G',</span> <span class="pre">'H',</span> <span class="pre">'I',</span> <span class="pre">'J',</span> <span class="pre">'K',</span> <span class="pre">'L',</span> <span class="pre">'M',</span> <span class="pre">'N',</span> <span class="pre">'P',</span> <span class="pre">'Q',</span> <span class="pre">'R',</span> <span class="pre">'S',</span> <span class="pre">'T',</span> <span class="pre">'V',</span> <span class="pre">'W',</span> <span class="pre">'X',</span> <span class="pre">'Y',</span> <span class="pre">'Z']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.STANDARD_AMINO_ACIDS" title="Permalink to this definition">#</a></dt>
<dd><p>Vocabulary of amino acids with one-letter codes. Includes <cite>fuzzy</cite> standard amino acids:
<code class="docutils literal notranslate"><span class="pre">"B"</span></code> denotes <code class="docutils literal notranslate"><span class="pre">"ASX"</span></code> which corresponds to <code class="docutils literal notranslate"><span class="pre">"ASP"</span></code> (<code class="docutils literal notranslate"><span class="pre">"D"</span></code>) <strong>or</strong> <code class="docutils literal notranslate"><span class="pre">"ASN"</span></code> (<code class="docutils literal notranslate"><span class="pre">"N"</span></code>)
and <code class="docutils literal notranslate"><span class="pre">"Z"</span></code> denotes <code class="docutils literal notranslate"><span class="pre">"GLX"</span></code> which corresponds to <code class="docutils literal notranslate"><span class="pre">"GLU"</span></code> (<code class="docutils literal notranslate"><span class="pre">"E"</span></code>) <strong>or</strong> <code class="docutils literal notranslate"><span class="pre">"GLN"</span></code> (<code class="docutils literal notranslate"><span class="pre">"Q"</span></code>).</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.STANDARD_RESI_NAMES">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">STANDARD_RESI_NAMES</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['ALA',</span> <span class="pre">'ASX',</span> <span class="pre">'CYS',</span> <span class="pre">'ASP',</span> <span class="pre">'GLU',</span> <span class="pre">'PHE',</span> <span class="pre">'GLY',</span> <span class="pre">'HIS',</span> <span class="pre">'ILE',</span> <span class="pre">'LYS',</span> <span class="pre">'LEU',</span> <span class="pre">'MET',</span> <span class="pre">'ASN',</span> <span class="pre">'PRO',</span> <span class="pre">'GLN',</span> <span class="pre">'ARG',</span> <span class="pre">'SER',</span> <span class="pre">'THR',</span> <span class="pre">'VAL',</span> <span class="pre">'TRP',</span> <span class="pre">'TYR',</span> <span class="pre">'GLX',</span> <span class="pre">'UNK']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.STANDARD_RESI_NAMES" title="Permalink to this definition">#</a></dt>
<dd><p>List of standard residue 3-letter names.
Includes <code class="docutils literal notranslate"><span class="pre">"UNK"</span></code> for unknown residues.
<code class="docutils literal notranslate"><span class="pre">"ASX"</span></code> denotes <code class="docutils literal notranslate"><span class="pre">"ASP"</span></code> <strong>or</strong> <code class="docutils literal notranslate"><span class="pre">"ASN"</span></code> and <code class="docutils literal notranslate"><span class="pre">"GLX"</span></code> denotes <code class="docutils literal notranslate"><span class="pre">"GLU"</span></code> <strong>or</strong> <code class="docutils literal notranslate"><span class="pre">"GLN"</span></code>.</p>
</dd></dl>
<dl class="py data">
<dt class="sig sig-object py" id="graphein.protein.resi_atoms.SULPHUR_RESIS">
<span class="sig-prename descclassname"><span class="pre">graphein.protein.resi_atoms.</span></span><span class="sig-name descname"><span class="pre">SULPHUR_RESIS</span></span><em class="property"><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="pre">List</span><span class="p"><span class="pre">[</span></span><a class="reference external" href="https://docs.python.org/3/library/stdtypes.html#str" title="(in Python v3.10)"><span class="pre">str</span></a><span class="p"><span class="pre">]</span></span></em><em class="property"><span class="w"> </span><span class="p"><span class="pre">=</span></span><span class="w"> </span><span class="pre">['MET',</span> <span class="pre">'CYS']</span></em><a class="headerlink" href="#graphein.protein.resi_atoms.SULPHUR_RESIS" title="Permalink to this definition">#</a></dt>
<dd><p>Residues containing sulphur atoms.</p>
</dd></dl>
</div>
</div>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="graphein.rna.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">graphein.rna</div>
              </div>
              <svg><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="../datasets/ppisp.html">
              <svg><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">PPISP - Protein Protein Interaction Structure Prediction</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2022, Arian Jamasb
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            <div class="icons">
              
            </div>
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            Contents
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">graphein.protein</a><ul>
<li><a class="reference internal" href="#module-graphein.protein.config">Config</a></li>
<li><a class="reference internal" href="#module-graphein.protein.graphs">Graphs</a></li>
<li><a class="reference internal" href="#edges">Edges</a><ul>
<li><a class="reference internal" href="#module-graphein.protein.edges.distance">Distance</a></li>
<li><a class="reference internal" href="#module-graphein.protein.edges.intramolecular">Intramolecular</a></li>
<li><a class="reference internal" href="#module-graphein.protein.edges.atomic">Atomic</a></li>
</ul>
</li>
<li><a class="reference internal" href="#features">Features</a><ul>
<li><a class="reference internal" href="#module-graphein.protein.features.nodes.aaindex">Node</a></li>
<li><a class="reference internal" href="#module-graphein.protein.features.sequence.embeddings">Sequence</a></li>
<li><a class="reference internal" href="#module-0">Sequence Utils</a></li>
<li><a class="reference internal" href="#module-graphein.protein.features.utils">Utils</a></li>
</ul>
</li>
<li><a class="reference internal" href="#module-graphein.protein.subgraphs">Subgraphs</a></li>
<li><a class="reference internal" href="#module-graphein.protein.analysis">Analysis</a></li>
<li><a class="reference internal" href="#module-graphein.protein.meshes">Meshes</a></li>
<li><a class="reference internal" href="#module-graphein.protein.visualisation">Visualisation</a></li>
<li><a class="reference internal" href="#id6">Utils</a></li>
<li><a class="reference internal" href="#module-graphein.protein.resi_atoms">Constants</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/scripts/furo.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/tabs.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.1.10/require.min.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>