<!doctype html>
<html class="no-js">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />
<link rel="index" title="Index" href="../genindex.html" /><link rel="search" title="Search" href="../search.html" /><link rel="next" title="Graphein - Atom Graph Tutorial" href="atom_graph_tutorial.html" /><link rel="prev" title="Notebooks" href="../notebooks_index.html" />

    <meta name="generator" content="sphinx-4.1.2, furo 2021.08.17.beta43"/>
        <title>Graphein - Residue Graph Tutorial - Graphein 1.0.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo.css?digest=910ac4ccc9d7c023151c34010d7b500fb9777a51" />
    <link rel="stylesheet" type="text/css" href="../_static/styles/furo-extensions.css?digest=16fb25fabf47304eee183a5e9af80b1ba98259b1" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  body[data-theme="dark"] {
    --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
  }
  @media (prefers-color-scheme: dark) {
    body:not([data-theme="light"]) {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
  }
</style>
    <link media="(prefers-color-scheme: dark)" rel="stylesheet" href="../_static/pygments_dark.css"></head>
  <body>
    
<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round">
      <path stroke="none" d="M0 0h24v24H0z" />
      <line x1="4" y1="6" x2="20" y2="6" />
      <line x1="10" y1="12" x2="20" y2="12" />
      <line x1="6" y1="18" x2="20" y2="18" />
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../index.html"><div class="brand">Graphein 1.0.0 documentation</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand centered" href="../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../_static/graphein.png" alt="Logo"/>
  </div>
  
  <span class="sidebar-brand-text">Graphein 1.0.0 documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="../search.html" role="search">
  <input class="sidebar-search" placeholder=Search name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/introduction.html">Graphein</a></li>
<li class="toctree-l1"><a class="reference internal" href="../license.html">License</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1 current has-children"><a class="reference internal" href="../notebooks_index.html">Notebooks</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul class="current">
<li class="toctree-l2 current current-page"><a class="current reference internal" href="#">Graphein - Residue Graph Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="atom_graph_tutorial.html">Graphein - Atom Graph Tutorial</a></li>
<li class="toctree-l2"><a class="reference internal" href="alphafold_protein_graph_tutorial.html">Constructing Graphs from AlphaFold Protein Structure Database</a></li>
<li class="toctree-l2"><a class="reference internal" href="protein_mesh_tutorial.html">Creating Protein Meshes in Graphein</a></li>
<li class="toctree-l2"><a class="reference internal" href="ppi_tutorial.html">PPI Networks and Structural Interactomics</a></li>
<li class="toctree-l2"><a class="reference internal" href="grn_tutorial.html">Gene Regulatory Network Tutorial</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Datasets</span></p>
<ul>
<li class="toctree-l1 has-children"><a class="reference internal" href="../datasets.html">Datasets</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../datasets/pscdb.html">PSCDB - Protein Structural Change Database</a></li>
<li class="toctree-l2"><a class="reference internal" href="pscdb_processing.html">PSCDB - Processing</a></li>
<li class="toctree-l2"><a class="reference internal" href="pscdb_baselines.html">PSCDB - Baselines</a></li>
<li class="toctree-l2"><a class="reference internal" href="../datasets/ppisp.html">PPISP - Protein Protein Interaction Structure Prediction</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../modules/graphein.protein.html">graphein.protein</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/graphein.rna.html">graphein.rna</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/graphein.ppi.html">graphein.ppi</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/graphein.grn.html">graphein.grn</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/graphein.ml.html">graphein.ml</a></li>
</ul>

</div>
</div>
      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <article role="main">
        <div class="content-icon-container">
          <div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        
<style>
/* CSS for nbsphinx extension */

/* remove conflicting styling from Sphinx themes */
div.nbinput.container div.prompt *,
div.nboutput.container div.prompt *,
div.nbinput.container div.input_area pre,
div.nboutput.container div.output_area pre,
div.nbinput.container div.input_area .highlight,
div.nboutput.container div.output_area .highlight {
    border: none;
    padding: 0;
    margin: 0;
    box-shadow: none;
}

div.nbinput.container > div[class*=highlight],
div.nboutput.container > div[class*=highlight] {
    margin: 0;
}

div.nbinput.container div.prompt *,
div.nboutput.container div.prompt * {
    background: none;
}

div.nboutput.container div.output_area .highlight,
div.nboutput.container div.output_area pre {
    background: unset;
}

div.nboutput.container div.output_area div.highlight {
    color: unset;  /* override Pygments text color */
}

/* avoid gaps between output lines */
div.nboutput.container div[class*=highlight] pre {
    line-height: normal;
}

/* input/output containers */
div.nbinput.container,
div.nboutput.container {
    display: -webkit-flex;
    display: flex;
    align-items: flex-start;
    margin: 0;
    width: 100%;
}
@media (max-width: 540px) {
    div.nbinput.container,
    div.nboutput.container {
        flex-direction: column;
    }
}

/* input container */
div.nbinput.container {
    padding-top: 5px;
}

/* last container */
div.nblast.container {
    padding-bottom: 5px;
}

/* input prompt */
div.nbinput.container div.prompt pre {
    color: #307FC1;
}

/* output prompt */
div.nboutput.container div.prompt pre {
    color: #BF5B3D;
}

/* all prompts */
div.nbinput.container div.prompt,
div.nboutput.container div.prompt {
    width: 4.5ex;
    padding-top: 5px;
    position: relative;
    user-select: none;
}

div.nbinput.container div.prompt > div,
div.nboutput.container div.prompt > div {
    position: absolute;
    right: 0;
    margin-right: 0.3ex;
}

@media (max-width: 540px) {
    div.nbinput.container div.prompt,
    div.nboutput.container div.prompt {
        width: unset;
        text-align: left;
        padding: 0.4em;
    }
    div.nboutput.container div.prompt.empty {
        padding: 0;
    }

    div.nbinput.container div.prompt > div,
    div.nboutput.container div.prompt > div {
        position: unset;
    }
}

/* disable scrollbars on prompts */
div.nbinput.container div.prompt pre,
div.nboutput.container div.prompt pre {
    overflow: hidden;
}

/* input/output area */
div.nbinput.container div.input_area,
div.nboutput.container div.output_area {
    -webkit-flex: 1;
    flex: 1;
    overflow: auto;
}
@media (max-width: 540px) {
    div.nbinput.container div.input_area,
    div.nboutput.container div.output_area {
        width: 100%;
    }
}

/* input area */
div.nbinput.container div.input_area {
    border: 1px solid #e0e0e0;
    border-radius: 2px;
    /*background: #f5f5f5;*/
}

/* override MathJax center alignment in output cells */
div.nboutput.container div[class*=MathJax] {
    text-align: left !important;
}

/* override sphinx.ext.imgmath center alignment in output cells */
div.nboutput.container div.math p {
    text-align: left;
}

/* standard error */
div.nboutput.container div.output_area.stderr {
    background: #fdd;
}

/* ANSI colors */
.ansi-black-fg { color: #3E424D; }
.ansi-black-bg { background-color: #3E424D; }
.ansi-black-intense-fg { color: #282C36; }
.ansi-black-intense-bg { background-color: #282C36; }
.ansi-red-fg { color: #E75C58; }
.ansi-red-bg { background-color: #E75C58; }
.ansi-red-intense-fg { color: #B22B31; }
.ansi-red-intense-bg { background-color: #B22B31; }
.ansi-green-fg { color: #00A250; }
.ansi-green-bg { background-color: #00A250; }
.ansi-green-intense-fg { color: #007427; }
.ansi-green-intense-bg { background-color: #007427; }
.ansi-yellow-fg { color: #DDB62B; }
.ansi-yellow-bg { background-color: #DDB62B; }
.ansi-yellow-intense-fg { color: #B27D12; }
.ansi-yellow-intense-bg { background-color: #B27D12; }
.ansi-blue-fg { color: #208FFB; }
.ansi-blue-bg { background-color: #208FFB; }
.ansi-blue-intense-fg { color: #0065CA; }
.ansi-blue-intense-bg { background-color: #0065CA; }
.ansi-magenta-fg { color: #D160C4; }
.ansi-magenta-bg { background-color: #D160C4; }
.ansi-magenta-intense-fg { color: #A03196; }
.ansi-magenta-intense-bg { background-color: #A03196; }
.ansi-cyan-fg { color: #60C6C8; }
.ansi-cyan-bg { background-color: #60C6C8; }
.ansi-cyan-intense-fg { color: #258F8F; }
.ansi-cyan-intense-bg { background-color: #258F8F; }
.ansi-white-fg { color: #C5C1B4; }
.ansi-white-bg { background-color: #C5C1B4; }
.ansi-white-intense-fg { color: #A1A6B2; }
.ansi-white-intense-bg { background-color: #A1A6B2; }

.ansi-default-inverse-fg { color: #FFFFFF; }
.ansi-default-inverse-bg { background-color: #000000; }

.ansi-bold { font-weight: bold; }
.ansi-underline { text-decoration: underline; }


div.nbinput.container div.input_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight] > pre,
div.nboutput.container div.output_area div[class*=highlight].math,
div.nboutput.container div.output_area.rendered_html,
div.nboutput.container div.output_area > div.output_javascript,
div.nboutput.container div.output_area:not(.rendered_html) > img{
    padding: 5px;
    margin: 0;
}

/* fix copybtn overflow problem in chromium (needed for 'sphinx_copybutton') */
div.nbinput.container div.input_area > div[class^='highlight'],
div.nboutput.container div.output_area > div[class^='highlight']{
    overflow-y: hidden;
}

/* hide copybtn icon on prompts (needed for 'sphinx_copybutton') */
.prompt a.copybtn {
    display: none;
}

/* Some additional styling taken form the Jupyter notebook CSS */
div.rendered_html table {
  border: none;
  border-collapse: collapse;
  border-spacing: 0;
  color: black;
  font-size: 12px;
  table-layout: fixed;
}
div.rendered_html thead {
  border-bottom: 1px solid black;
  vertical-align: bottom;
}
div.rendered_html tr,
div.rendered_html th,
div.rendered_html td {
  text-align: right;
  vertical-align: middle;
  padding: 0.5em 0.5em;
  line-height: normal;
  white-space: normal;
  max-width: none;
  border: none;
}
div.rendered_html th {
  font-weight: bold;
}
div.rendered_html tbody tr:nth-child(odd) {
  background: #f5f5f5;
}
div.rendered_html tbody tr:hover {
  background: rgba(66, 165, 245, 0.2);
}
</style>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="o">%</span><span class="k">matplotlib</span> inline
<span class="kn">import</span> <span class="nn">logging</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">(</span><span class="n">level</span><span class="o">=</span><span class="s2">"INFO"</span><span class="p">)</span>
</pre></div>
</div>
</div>
<section id="Graphein---Residue-Graph-Tutorial">
<h1>Graphein - Residue Graph Tutorial<a class="headerlink" href="#Graphein---Residue-Graph-Tutorial" title="Permalink to this headline">¶</a></h1>
<p>In this notebook, we’ll run through residue-level graph construction in Graphein. We start by discsussing the config, the high-level API and spend the bulk of the tutorial running through the low-level API</p>
<section id="Config">
<h2>Config<a class="headerlink" href="#Config" title="Permalink to this headline">¶</a></h2>
<p>Graphein is designed for processing datasets of protein structures into graphs. We rely on a global config object to store various parameters in the high-level API. We use <code class="docutils literal notranslate"><span class="pre">pydantic</span></code> for the config object and provide sane defaults.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.config</span> <span class="kn">import</span> <span class="n">ProteinGraphConfig</span>

<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">()</span>
<span class="n">config</span><span class="o">.</span><span class="n">dict</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
INFO:summarizer.preprocessing.cleaner:'pattern' package not found; tag filters are not available for English
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
{'granularity': 'CA',
 'keep_hets': False,
 'insertions': False,
 'pdb_dir': PosixPath('../examples/pdbs'),
 'verbose': True,
 'exclude_waters': True,
 'deprotonate': False,
 'protein_df_processing_functions': None,
 'edge_construction_functions': [&lt;function graphein.protein.edges.distance.add_peptide_bonds(G: 'nx.Graph') -&gt; 'nx.Graph'&gt;],
 'node_metadata_functions': [&lt;function graphein.protein.features.nodes.amino_acid.meiler_embedding(n, d) -&gt; pandas.core.series.Series&gt;],
 'edge_metadata_functions': None,
 'graph_metadata_functions': None,
 'get_contacts_config': None,
 'dssp_config': None}
</pre></div></div>
</div>
<p>Let’s run through the config:</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">granularity</span></code>: specifies the granularity of the graph (i.e. what should the nodes be). Possible values are: atom identifiers (e.g. <code class="docutils literal notranslate"><span class="pre">"CA"</span></code> for <span class="math notranslate nohighlight">\(\alpha\)</span> carbon, <code class="docutils literal notranslate"><span class="pre">"CB"</span></code> for <span class="math notranslate nohighlight">\(\beta\)</span> carbon), <code class="docutils literal notranslate"><span class="pre">"centroid"</span></code> to use residue centroids (under the hood, this is the same as “CA”, but we use the average x,y,z coordinates for the atoms in the residue) or “atom” for atom-level construction. This is discussed in another notebook.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">keep_hets</span></code>: this is a boolean specifying whether or not to keep heteroatoms present in the .pdb file. Heteroatoms are typically non-protein atoms (waters, metal ions, ligands) but can sometimes contain non-standard or modified residues.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">insertions</span></code>: boolean specifying whether or not to keep insertions in the PDB file</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">pdb_dir</span></code> optional path to a folder in which to save pdb files. Otherwise, <code class="docutils literal notranslate"><span class="pre">/tmp/</span></code> will be used</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">verbose</span></code>: bool controlling amount of info printed</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">exclude_waters</span></code>: not implemented</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">deprotonate</span></code>: bool indicating whether or not to remove Hydrogen atoms</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">protein_df_processing_functions</span></code>: list of functions with which to process the PDB dataframe. Discussed in the low-level API.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">edge_construction_functions</span></code>: list of functions to compute edges with</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">node_metadata_functions</span></code>: list of functions to annotate nodes with</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">edge_metadata_functions</span></code>: list of functions to annotate edges with</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">graph_meta_functions</span></code>: list of functions to annotate graph with</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">get_contacts_config</span></code>: A separate config object if using GetContacts edge construction functions</p></li>
</ul>
<p>If you wish, you can construct an entirely fresh configuration. Or, if we wish to change only some of these parameters, we can pass a dictionary containing our modifications:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">params_to_change</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"granularity"</span><span class="p">:</span> <span class="s2">"centroids"</span><span class="p">}</span>

<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">params_to_change</span><span class="p">)</span>
<span class="n">config</span><span class="o">.</span><span class="n">dict</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
{'granularity': 'centroids',
 'keep_hets': False,
 'insertions': False,
 'pdb_dir': PosixPath('../examples/pdbs'),
 'verbose': True,
 'exclude_waters': True,
 'deprotonate': False,
 'protein_df_processing_functions': None,
 'edge_construction_functions': [&lt;function graphein.protein.edges.distance.add_peptide_bonds(G: 'nx.Graph') -&gt; 'nx.Graph'&gt;],
 'node_metadata_functions': [&lt;function graphein.protein.features.nodes.amino_acid.meiler_embedding(n, d) -&gt; pandas.core.series.Series&gt;],
 'edge_metadata_functions': None,
 'graph_metadata_functions': None,
 'get_contacts_config': None,
 'dssp_config': None}
</pre></div></div>
</div>
</section>
<section id="High-level-API">
<h2>High-level API<a class="headerlink" href="#High-level-API" title="Permalink to this headline">¶</a></h2>
<p>Graphein features a high-level API which should be applicable for most simple graph constructions. This can be used on either a .pdb file (so you can run whatever pre-processing you wish), or we can provide a PDB accession code and retrieve a structure from the PDB itself. If a path is provided, it takes precedence over the PDB code.</p>
<p>To use it we do as follows:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.graphs</span> <span class="kn">import</span> <span class="n">construct_graph</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="c1">#g = construct_graph(config=config, pdb_path="../graphein/examples/pdbs/3eiy.pdb")</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd8a42d2910&gt;
</pre></div></div>
</div>
<p>Let’s check out the results with the in-built visualisation</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.visualisation</span> <span class="kn">import</span> <span class="n">plot_protein_structure_graph</span>

<span class="n">p</span> <span class="o">=</span> <span class="n">plot_protein_structure_graph</span><span class="p">(</span><span class="n">G</span><span class="o">=</span><span class="n">g</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">colour_edges_by</span><span class="o">=</span><span class="s2">"kind"</span><span class="p">,</span> <span class="n">colour_nodes_by</span><span class="o">=</span><span class="s2">"degree"</span><span class="p">,</span> <span class="n">label_node_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s2">"Peptide backbone graph. Nodes coloured by degree."</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/atj39/github/graphein/graphein/protein/visualisation.py:330: MatplotlibDeprecationWarning: Axes3D(fig) adding itself to the figure is deprecated since 3.4. Pass the keyword argument auto_add_to_figure=False and use fig.add_axes(ax) to suppress this warning. The default value of auto_add_to_figure will change to False in mpl3.5 and True values will no longer work in 3.6.  This is consistent with other Axes classes.
  ax = Axes3D(fig, auto_add_to_figure=True)
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 0.98, 'Peptide backbone graph. Nodes coloured by degree.')
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_residue_graphs_10_2.png" src="../_images/notebooks_residue_graphs_10_2.png"/>
</div>
</div>
<p>Cool! So let’s look at what we’ve got here:</p>
<p>We’ve plotted the graph, positioning nodes according to their x,y,z coordinates and coloured them by their degree. As we can see all the nodes are yellow except the two corresponding to the terminal residues. Why is this? Because we’ve only computed the peptide-bond edges. Now, we probably want more so let’s look at how to do that!</p>
<section id="Edge-Functions">
<h3>Edge Functions<a class="headerlink" href="#Edge-Functions" title="Permalink to this headline">¶</a></h3>
<p>Graphein is implemented in a functional fashion. In this case, this means in order to compute edges, we pass a list of edge construction functions to the construction. We have supplied a number of edge computation functions. These are located in: * <code class="docutils literal notranslate"><span class="pre">graphein.protein.edges.distance</span></code> * <code class="docutils literal notranslate"><span class="pre">graphein.protein.edges.intramolecular</span></code>(these rely on an installation of GetContacts, an optional dependency) and a separate get_contacts_config * <code class="docutils literal notranslate"><span class="pre">graphein.protein.edges.atomic</span></code> (these are used in
atomic-level graphs and we discuss these in another notebook)</p>
<p>However, edge functions are simple functions that take in an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> and return an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> with added edges. This means users can easily define their own to suit their purposes! Let’s take a closer look at some of the in-built functions before defining our own.</p>
<section id="Built-in-Edge-Functions">
<h4>Built-in Edge Functions<a class="headerlink" href="#Built-in-Edge-Functions" title="Permalink to this headline">¶</a></h4>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.edges.distance</span> <span class="kn">import</span> <span class="n">add_hydrogen_bond_interactions</span><span class="p">,</span> <span class="n">add_peptide_bonds</span>

<span class="n">new_edge_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"edge_construction_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">add_peptide_bonds</span><span class="p">,</span> <span class="n">add_hydrogen_bond_interactions</span><span class="p">]}</span>
<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_edge_funcs</span><span class="p">)</span>


<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="n">p</span> <span class="o">=</span> <span class="n">plot_protein_structure_graph</span><span class="p">(</span><span class="n">G</span><span class="o">=</span><span class="n">g</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">colour_edges_by</span><span class="o">=</span><span class="s2">"kind"</span><span class="p">,</span> <span class="n">colour_nodes_by</span><span class="o">=</span><span class="s2">"degree"</span><span class="p">,</span> <span class="n">label_node_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s2">"Protein graph with peptide backbone and H-Bonds. </span><span class="se">\n</span><span class="s2"> Nodes coloured by degree. Edges coloured by type."</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd734038490&gt;
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/atj39/github/graphein/graphein/protein/visualisation.py:330: MatplotlibDeprecationWarning: Axes3D(fig) adding itself to the figure is deprecated since 3.4. Pass the keyword argument auto_add_to_figure=False and use fig.add_axes(ax) to suppress this warning. The default value of auto_add_to_figure will change to False in mpl3.5 and True values will no longer work in 3.6.  This is consistent with other Axes classes.
  ax = Axes3D(fig, auto_add_to_figure=True)
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 0.98, 'Protein graph with peptide backbone and H-Bonds. \n Nodes coloured by degree. Edges coloured by type.')
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_residue_graphs_13_3.png" src="../_images/notebooks_residue_graphs_13_3.png"/>
</div>
</div>
<p>Great! So what if we try <strong>a bunch</strong> of them</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.edges.distance</span> <span class="kn">import</span> <span class="p">(</span><span class="n">add_peptide_bonds</span><span class="p">,</span>
                                             <span class="n">add_hydrogen_bond_interactions</span><span class="p">,</span>
                                             <span class="n">add_disulfide_interactions</span><span class="p">,</span>
                                             <span class="n">add_ionic_interactions</span><span class="p">,</span>
                                             <span class="n">add_aromatic_interactions</span><span class="p">,</span>
                                             <span class="n">add_aromatic_sulphur_interactions</span><span class="p">,</span>
                                             <span class="n">add_cation_pi_interactions</span>
                                            <span class="p">)</span>

<span class="n">new_edge_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"edge_construction_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">add_peptide_bonds</span><span class="p">,</span>
                                                  <span class="n">add_hydrogen_bond_interactions</span><span class="p">,</span>
                                                  <span class="n">add_disulfide_interactions</span><span class="p">,</span>
                                                  <span class="n">add_ionic_interactions</span><span class="p">,</span>
                                                  <span class="n">add_aromatic_sulphur_interactions</span><span class="p">,</span>
                                                  <span class="n">add_cation_pi_interactions</span><span class="p">]</span>
                 <span class="p">}</span>

<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_edge_funcs</span><span class="p">)</span>
<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="n">p</span> <span class="o">=</span> <span class="n">plot_protein_structure_graph</span><span class="p">(</span><span class="n">G</span><span class="o">=</span><span class="n">g</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">colour_edges_by</span><span class="o">=</span><span class="s2">"kind"</span><span class="p">,</span> <span class="n">colour_nodes_by</span><span class="o">=</span><span class="s2">"degree"</span><span class="p">,</span> <span class="n">label_node_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s2">"Protein graph with: peptide backbone, H-Bonds, </span><span class="se">\n</span><span class="s2"> Disulphide, ionic, aromatic, aromatic-sulphur and cation-pi interactions. </span><span class="se">\n</span><span class="s2">  Nodes coloured by degree, edges by type"</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd73224d6a0&gt;
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/atj39/github/graphein/graphein/protein/visualisation.py:330: MatplotlibDeprecationWarning: Axes3D(fig) adding itself to the figure is deprecated since 3.4. Pass the keyword argument auto_add_to_figure=False and use fig.add_axes(ax) to suppress this warning. The default value of auto_add_to_figure will change to False in mpl3.5 and True values will no longer work in 3.6.  This is consistent with other Axes classes.
  ax = Axes3D(fig, auto_add_to_figure=True)
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 0.98, 'Protein graph with: peptide backbone, H-Bonds, \n Disulphide, ionic, aromatic, aromatic-sulphur and cation-pi interactions. \n  Nodes coloured by degree, edges by type')
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_residue_graphs_15_3.png" src="../_images/notebooks_residue_graphs_15_3.png"/>
</div>
</div>
<p>Cool! So we’ve added a bunch on distance-based computations of intramolecular edges to our graph! We have a few more distance based edge functions that we can visualise. Let’s check them out!</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">graphein.protein.edges.distance.add_delaunay_triangulation</span></code>: Adds edges based on the delaunay triangulation</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">graphein.protein.edges.distance.add_k_nn_edges</span></code>: Adds edges based on K Nearest neighbours</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">graphein.protein.edges.distance.add_distance_threshold</span></code>: Adds edges based on distance cutoff in Angstroms</p></li>
</ul>
<p>First up, the Delaunay triangulation:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.edges.distance</span> <span class="kn">import</span> <span class="n">add_delaunay_triangulation</span>

<span class="n">new_edge_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"edge_construction_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">add_delaunay_triangulation</span><span class="p">]}</span>
<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_edge_funcs</span><span class="p">)</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="n">nrp</span> <span class="o">=</span> <span class="n">plot_protein_structure_graph</span><span class="p">(</span><span class="n">G</span><span class="o">=</span><span class="n">g</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">colour_edges_by</span><span class="o">=</span><span class="s2">"kind"</span><span class="p">,</span> <span class="n">colour_nodes_by</span><span class="o">=</span><span class="s2">"degree"</span><span class="p">,</span> <span class="n">label_node_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s2">"Protein graph created by the Delaunay trinagulation of the structure. </span><span class="se">\n</span><span class="s2"> Nodes coloured by degree."</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd73224d4c0&gt;
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/atj39/github/graphein/graphein/protein/visualisation.py:330: MatplotlibDeprecationWarning: Axes3D(fig) adding itself to the figure is deprecated since 3.4. Pass the keyword argument auto_add_to_figure=False and use fig.add_axes(ax) to suppress this warning. The default value of auto_add_to_figure will change to False in mpl3.5 and True values will no longer work in 3.6.  This is consistent with other Axes classes.
  ax = Axes3D(fig, auto_add_to_figure=True)
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 0.98, 'Protein graph created by the Delaunay trinagulation of the structure. \n Nodes coloured by degree.')
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_residue_graphs_17_3.png" src="../_images/notebooks_residue_graphs_17_3.png"/>
</div>
</div>
<p>Next, let’s consider the last two edge construction functions <code class="docutils literal notranslate"><span class="pre">add_k_nn_edges</span></code> and<code class="docutils literal notranslate"><span class="pre">add_distance_threshold</span></code>. These two functions take additional parameters. We can manage this with partial functions. These functions also feature a <code class="docutils literal notranslate"><span class="pre">long_interaction_threshold</span></code> parameter which specifies the minimum number of residues in the sequence between two residues in order to add an edge. This is because we may not be so interested in residues close together in the sequence being close together in the
graph.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">functools</span> <span class="kn">import</span> <span class="n">partial</span>
<span class="kn">from</span> <span class="nn">graphein.protein.edges.distance</span> <span class="kn">import</span> <span class="n">add_distance_threshold</span>

<span class="n">new_edge_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"edge_construction_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">partial</span><span class="p">(</span><span class="n">add_distance_threshold</span><span class="p">,</span> <span class="n">long_interaction_threshold</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">threshold</span><span class="o">=</span><span class="mf">10.</span><span class="p">)]}</span>
<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_edge_funcs</span><span class="p">)</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="n">p</span> <span class="o">=</span> <span class="n">plot_protein_structure_graph</span><span class="p">(</span><span class="n">G</span><span class="o">=</span><span class="n">g</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">colour_edges_by</span><span class="o">=</span><span class="s2">"kind"</span><span class="p">,</span> <span class="n">colour_nodes_by</span><span class="o">=</span><span class="s2">"degree"</span><span class="p">,</span> <span class="n">label_node_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s2">"Protein graph created by thresholding distance between nodes. </span><span class="se">\n</span><span class="s2"> Nodes must be &lt;10A apart and at least 5 positions apart </span><span class="se">\n</span><span class="s2"> Nodes coloured by degree."</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd730a0ceb0&gt;
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/atj39/github/graphein/graphein/protein/visualisation.py:330: MatplotlibDeprecationWarning: Axes3D(fig) adding itself to the figure is deprecated since 3.4. Pass the keyword argument auto_add_to_figure=False and use fig.add_axes(ax) to suppress this warning. The default value of auto_add_to_figure will change to False in mpl3.5 and True values will no longer work in 3.6.  This is consistent with other Axes classes.
  ax = Axes3D(fig, auto_add_to_figure=True)
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 0.98, 'Protein graph created by thresholding distance between nodes. \n Nodes must be &lt;10A apart and at least 5 positions apart \n Nodes coloured by degree.')
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_residue_graphs_19_3.png" src="../_images/notebooks_residue_graphs_19_3.png"/>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">functools</span> <span class="kn">import</span> <span class="n">partial</span>
<span class="kn">from</span> <span class="nn">graphein.protein.edges.distance</span> <span class="kn">import</span> <span class="n">add_k_nn_edges</span>

<span class="n">new_edge_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"edge_construction_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">partial</span><span class="p">(</span><span class="n">add_k_nn_edges</span><span class="p">,</span> <span class="n">k</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">long_interaction_threshold</span><span class="o">=</span><span class="mi">0</span><span class="p">)]}</span>
<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_edge_funcs</span><span class="p">)</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="n">p</span> <span class="o">=</span> <span class="n">plot_protein_structure_graph</span><span class="p">(</span><span class="n">G</span><span class="o">=</span><span class="n">g</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">colour_edges_by</span><span class="o">=</span><span class="s2">"kind"</span><span class="p">,</span> <span class="n">colour_nodes_by</span><span class="o">=</span><span class="s2">"degree"</span><span class="p">,</span> <span class="n">label_node_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s2">"Protein graph created from K-NN of each node. Nodes coloured by degree."</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd731bb45e0&gt;
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/atj39/github/graphein/graphein/protein/visualisation.py:330: MatplotlibDeprecationWarning: Axes3D(fig) adding itself to the figure is deprecated since 3.4. Pass the keyword argument auto_add_to_figure=False and use fig.add_axes(ax) to suppress this warning. The default value of auto_add_to_figure will change to False in mpl3.5 and True values will no longer work in 3.6.  This is consistent with other Axes classes.
  ax = Axes3D(fig, auto_add_to_figure=True)
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[10]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 0.98, 'Protein graph created from K-NN of each node. Nodes coloured by degree.')
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_residue_graphs_20_3.png" src="../_images/notebooks_residue_graphs_20_3.png"/>
</div>
</div>
</section>
<section id="Defining-your-own-edge-functions">
<h4>Defining your own edge functions<a class="headerlink" href="#Defining-your-own-edge-functions" title="Permalink to this headline">¶</a></h4>
<p>Now, let’s turn our attention to defining our own edge construction functions. These must take in a <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> and return a <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code>. You can use these custom functions together with all the in-built edge functions as before. If you come up with something cool, consider making a pull-request and sharing it with the community!</p>
<p>Here, as an example, we define an arbitrary function that creates an edge between all histidine residues in the graph.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[11]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">networkx</span> <span class="k">as</span> <span class="nn">nx</span>

<span class="k">def</span> <span class="nf">add_histidine_histidine_edges</span><span class="p">(</span><span class="n">G</span><span class="p">:</span> <span class="n">nx</span><span class="o">.</span><span class="n">Graph</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">nx</span><span class="o">.</span><span class="n">Graph</span><span class="p">:</span>
    <span class="c1"># Iterate over nodes to identify histidines</span>
    <span class="n">histidines</span> <span class="o">=</span> <span class="p">[</span><span class="n">n</span> <span class="k">for</span> <span class="n">n</span><span class="p">,</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">G</span><span class="o">.</span><span class="n">nodes</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span> <span class="k">if</span> <span class="n">d</span><span class="p">[</span><span class="s2">"residue_name"</span><span class="p">]</span> <span class="o">==</span> <span class="s2">"HIS"</span><span class="p">]</span>

    <span class="c1"># Iterate over histidines and create a bond between them</span>
    <span class="p">[</span><span class="n">G</span><span class="o">.</span><span class="n">add_edge</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">kind</span><span class="o">=</span><span class="s2">"histidine"</span><span class="p">)</span> <span class="k">for</span> <span class="n">i</span><span class="p">,</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">histidines</span><span class="p">)</span> <span class="k">for</span> <span class="n">j</span><span class="p">,</span> <span class="n">y</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">histidines</span><span class="p">)</span> <span class="k">if</span> <span class="n">i</span><span class="o">!=</span><span class="n">j</span><span class="p">]</span>

    <span class="k">return</span> <span class="n">G</span>
</pre></div>
</div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">new_edge_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"edge_construction_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">add_histidine_histidine_edges</span><span class="p">]}</span>
<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_edge_funcs</span><span class="p">)</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="n">p</span> <span class="o">=</span> <span class="n">plot_protein_structure_graph</span><span class="p">(</span><span class="n">G</span><span class="o">=</span><span class="n">g</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">colour_edges_by</span><span class="o">=</span><span class="s2">"kind"</span><span class="p">,</span> <span class="n">colour_nodes_by</span><span class="o">=</span><span class="s2">"degree"</span><span class="p">,</span> <span class="n">label_node_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s2">"Protein graph created using a user-defined function that connects all Histidines."</span><span class="p">)</span>

<span class="c1"># Let's check this worked:</span>
<span class="k">for</span> <span class="n">u</span><span class="p">,</span> <span class="n">v</span><span class="p">,</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">g</span><span class="o">.</span><span class="n">edges</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="kc">True</span><span class="p">):</span>
    <span class="nb">print</span><span class="p">(</span><span class="n">u</span><span class="p">,</span> <span class="n">v</span><span class="p">,</span> <span class="n">a</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd7184ba820&gt;
A:HIS:111 A:HIS:137 {'kind': 'histidine'}
A:HIS:111 A:HIS:163 {'kind': 'histidine'}
A:HIS:137 A:HIS:163 {'kind': 'histidine'}
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/atj39/github/graphein/graphein/protein/visualisation.py:330: MatplotlibDeprecationWarning: Axes3D(fig) adding itself to the figure is deprecated since 3.4. Pass the keyword argument auto_add_to_figure=False and use fig.add_axes(ax) to suppress this warning. The default value of auto_add_to_figure will change to False in mpl3.5 and True values will no longer work in 3.6.  This is consistent with other Axes classes.
  ax = Axes3D(fig, auto_add_to_figure=True)
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_residue_graphs_23_2.png" src="../_images/notebooks_residue_graphs_23_2.png"/>
</div>
</div>
</section>
</section>
<section id="Features-&amp;-Metadata">
<h3>Features &amp; Metadata<a class="headerlink" href="#Features-&amp;-Metadata" title="Permalink to this headline">¶</a></h3>
<p><code class="docutils literal notranslate"><span class="pre">Graphein</span></code> is designed to facilitate geometric deep learning on protein structures and so we have a bunch of feature and metadata annotation functions. These operate on three levels: * node-level annotation * graph-level annotation * edge-level annotation</p>
<p>These behave very similarly to the edge construction functions we just looked at. You simply pass a list of functions that take the right sort of arguments and return the right types.</p>
<ul class="simple">
<li><p>graph-level annotation takes in a <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> and returns an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code> (just like before!).</p></li>
<li><p>node-level annotation takes in a node, data tuple from <code class="docutils literal notranslate"><span class="pre">G.nodes(data=True)</span></code> and returns a <code class="docutils literal notranslate"><span class="pre">pd.Series</span></code></p></li>
<li><p>edge-level annotation in node_u, node_v, data tuple from <code class="docutils literal notranslate"><span class="pre">G.edges(data=True)</span></code> and returns</p></li>
</ul>
<section id="Built-in-Graph-Annotation-Functions">
<h4>Built-in Graph Annotation Functions<a class="headerlink" href="#Built-in-Graph-Annotation-Functions" title="Permalink to this headline">¶</a></h4>
<p>These are found in <code class="docutils literal notranslate"><span class="pre">graphein.protein.features.sequence</span></code> and <code class="docutils literal notranslate"><span class="pre">graphein.protein.features.graph</span></code>. We make a slight distinction between those functions that operate on the protein sequence for cleaner organisation. N.B. we’ve mentioned the <code class="docutils literal notranslate"><span class="pre">chain_selection</span></code> parameter briefly - if you include multiple chains in your graph, then will be multiple sequences associated with it. These are accessed as such: <code class="docutils literal notranslate"><span class="pre">G.graph[f"sequence_{CHAIN_ID}"]</span></code> and features computed from sequence are accessed as:
<code class="docutils literal notranslate"><span class="pre">G.graph[f"{FEATURE_NAME}_{CHAIN_ID}]</span></code></p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[13]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.features.sequence.sequence</span> <span class="kn">import</span> <span class="n">molecular_weight</span>

<span class="n">new_graph_annotation_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"graph_metadata_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">molecular_weight</span><span class="p">]}</span>
<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_graph_annotation_funcs</span><span class="p">)</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"Sequence:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"sequence_A"</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"MW:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"molecular_weight_A"</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd7181b7250&gt;
Sequence: SFSNVPAGKDLPQDFNVIIEIPAQSEPVKYEADKALGLLVVDRFIGTGMRYPVNYGFIPQTLSGDGDPVDVLVITPFPLLAGSVVRARALGMLKMTDESGVDAKLVAVPHDKVCPMTANLKSIDDVPAYLKDQIKHFFEQYKALEKGKWVKVEGWDGIDAAHKEITDGVANFKK
MW: 19029.71029999999
</pre></div></div>
</div>
<p>We also provide some utilities for adding graph-level features in the form of sequence embeddings from various pre-trained language models. Let’s check them out!</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[14]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="c1"># Warning! This cell may crash a binder notebook as the pre-trained model download is rather large!</span>
<span class="kn">from</span> <span class="nn">graphein.protein.features.sequence.embeddings</span> <span class="kn">import</span> <span class="n">esm_sequence_embedding</span><span class="p">,</span> <span class="n">biovec_sequence_embedding</span>

<span class="n">new_graph_annotation_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"graph_metadata_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">esm_sequence_embedding</span><span class="p">,</span> <span class="n">biovec_sequence_embedding</span><span class="p">]}</span>
<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_graph_annotation_funcs</span><span class="p">)</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"ESM:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"esm_embedding_A"</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"biovec:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"biovec_embedding_A"</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd7184d48b0&gt;
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
Using cache found in /home/atj39/.cache/torch/hub/facebookresearch_esm_master
INFO:gensim.utils:loading Word2Vec object from /home/atj39/github/graphein/graphein/protein/features/pretrained_models/swissprot-reviewed-protvec.model
INFO:gensim.utils:loading vocabulary recursively from /home/atj39/github/graphein/graphein/protein/features/pretrained_models/swissprot-reviewed-protvec.model.vocabulary.* with mmap=None
INFO:gensim.utils:loading wv recursively from /home/atj39/github/graphein/graphein/protein/features/pretrained_models/swissprot-reviewed-protvec.model.wv.* with mmap=None
INFO:gensim.utils:setting ignored attribute vectors_norm to None
INFO:gensim.utils:loading trainables recursively from /home/atj39/github/graphein/graphein/protein/features/pretrained_models/swissprot-reviewed-protvec.model.trainables.* with mmap=None
INFO:gensim.utils:setting ignored attribute cum_table to None
INFO:gensim.utils:loaded /home/atj39/github/graphein/graphein/protein/features/pretrained_models/swissprot-reviewed-protvec.model
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
ESM: [-0.02279946  0.16464779 -0.0046199  ...  0.02556622 -0.19798389
  0.09682209]
biovec: [array([ -8.865931  ,   5.65769   ,  -1.5052402 ,   3.2893941 ,
        -6.6228924 ,   2.4890645 ,   1.0635861 ,   2.4991994 ,
        -2.2740204 ,  -7.6614437 ,   0.3739053 ,   4.285517  ,
        -1.7950253 ,  -2.0930681 ,   0.19075748,   9.249631  ,
        -1.2537682 ,   6.0140305 ,  -3.0592997 ,   3.1944623 ,
         5.6747837 ,   1.4732779 ,  -9.556535  ,   4.5126605 ,
       -15.789192  ,  -9.917104  ,  10.61495   ,   1.6693095 ,
         1.8446481 , -12.656563  ,   2.5226657 ,   3.2782233 ,
         5.6363025 ,  -1.5776551 ,   9.913681  ,  -1.4776824 ,
         7.4292827 ,   2.6953607 ,  -0.81495905,   1.0415509 ,
       -18.352554  ,  -9.603085  , -11.215957  ,   3.9449768 ,
         2.2988393 ,   2.8612108 ,  -5.647224  ,  -9.894558  ,
         2.8099368 ,  -3.47802   ,  -1.7364154 ,   4.10581   ,
        -1.1337806 ,  -7.851351  ,   3.667841  ,   2.5098174 ,
         0.3878222 ,  -0.282476  ,   4.242093  ,   2.7918775 ,
        -0.3927042 ,  -3.504907  ,   2.2170143 ,  -6.7696056 ,
        -8.574576  ,   7.549987  ,   5.5607357 ,  -7.514735  ,
         5.778782  ,   2.1968088 ,   5.747694  ,  -4.4072075 ,
         1.6411833 ,   2.339804  ,   4.364424  ,  -4.615387  ,
         0.61950064,   7.19126   ,  -0.10569584,  -6.485407  ,
        -3.3555477 ,   0.58533096,   4.150122  ,   4.6470437 ,
        -7.6111784 ,   5.8581285 ,  -9.8836    ,   8.114656  ,
         4.3746543 ,   1.4033681 ,  -4.632345  ,  13.016971  ,
        -2.839704  ,  -6.210647  ,   3.4460034 ,   2.6573007 ,
       -14.987186  ,  -6.545936  ,  -2.3990562 ,   1.2732263 ],
      dtype=float32), array([ -9.079695  ,   3.7650356 ,   0.3614651 ,   2.1325634 ,
        -5.285296  ,   3.0021908 ,   1.2300422 ,   1.8999658 ,
        -2.0472856 ,  -5.5658956 ,   0.5014471 ,   3.8743246 ,
        -1.0728757 ,  -3.0830898 ,   0.73323405,   9.658365  ,
        -0.10142106,   6.2466826 ,  -2.4961333 ,   2.8022783 ,
         6.5060906 ,   1.6311501 , -10.04929   ,   5.4711785 ,
       -16.297068  , -10.765219  ,  10.580414  ,   0.503202  ,
         2.4227042 , -12.57771   ,   2.7075891 ,   3.9921677 ,
         5.5343657 ,  -0.9493397 ,  10.082592  ,   0.301396  ,
         6.978562  ,   2.4752004 ,  -1.1012759 ,  -1.1357698 ,
       -16.81547   ,  -9.117541  , -10.919918  ,   3.8029845 ,
         2.5540738 ,   3.0349758 ,  -5.4471354 ,  -9.438955  ,
         3.067747  ,  -3.1571815 ,  -3.1563768 ,   4.7784586 ,
        -2.3092556 ,  -7.704368  ,   0.82422435,   1.8206171 ,
         2.161718  ,   0.37946355,   4.7007704 ,   2.2930143 ,
         0.75466317,  -3.3875456 ,   2.3561945 ,  -6.173627  ,
        -7.655903  ,   6.066585  ,   5.7083497 ,  -8.785268  ,
         4.0877957 ,   2.1213973 ,   5.8782153 ,  -4.1335964 ,
         0.26376987,   3.185395  ,   3.9498727 ,  -4.5998893 ,
        -0.11535516,   8.4039135 ,  -0.8255909 ,  -5.910516  ,
        -2.2402806 ,  -0.15940607,   3.476339  ,   6.034683  ,
        -6.72777   ,   6.143824  ,  -9.662175  ,   8.747985  ,
         3.985702  ,   2.272303  ,  -3.933255  ,  12.071237  ,
        -1.8998326 ,  -6.6413393 ,   4.2529097 ,   3.4216733 ,
       -15.628299  ,  -5.513835  ,  -3.004577  ,   1.3258756 ],
      dtype=float32), array([ -6.9852    ,   4.425231  ,  -0.5174276 ,   2.0380528 ,
        -7.1311088 ,   2.4794824 ,   1.9649935 ,   1.9006643 ,
        -1.3209196 ,  -5.8989286 ,   1.2391084 ,   3.485902  ,
        -1.9545169 ,  -2.2625084 ,   1.8165749 ,   9.708086  ,
        -0.74385244,   6.5338845 ,  -3.0916584 ,   3.9312694 ,
         6.1540847 ,   3.4793413 , -10.056544  ,   5.7322254 ,
       -13.599701  , -10.923328  ,   9.378405  ,   0.555632  ,
         3.512236  , -11.891929  ,   2.382594  ,   3.2599561 ,
         5.123339  ,  -0.55314994,  10.585077  ,  -0.9078483 ,
         8.167705  ,   2.4789596 ,   0.78942966,  -1.0045129 ,
       -17.38679   ,  -9.496934  , -10.542821  ,   3.5852609 ,
         3.3952198 ,   2.9937162 ,  -5.9167666 ,  -9.517663  ,
         2.2889428 ,  -3.6738756 ,  -2.7072256 ,   4.6205435 ,
        -1.970848  ,  -6.7903805 ,   2.1213295 ,   2.5594456 ,
         0.4943355 ,   0.27333227,   4.096034  ,   1.2748998 ,
        -1.3203058 ,  -3.6010296 ,   2.623712  ,  -8.468901  ,
        -6.3624053 ,   7.388231  ,   4.908528  ,  -7.1587243 ,
         4.9127274 ,   3.2538335 ,   6.45744   ,  -3.3919108 ,
        -0.22479697,   2.4506824 ,   4.864544  ,  -4.201532  ,
        -0.08151506,   5.778758  ,  -0.5880602 ,  -7.52435   ,
        -2.3001645 ,   0.04607171,   4.0975637 ,   5.7935786 ,
        -8.676276  ,   6.952252  , -11.0132    ,   7.6395264 ,
         4.4887743 ,   2.5889583 ,  -5.0494356 ,  12.681698  ,
        -2.3577175 ,  -5.7180576 ,   3.481403  ,   3.9928114 ,
       -14.854708  ,  -6.135002  ,  -2.9227335 ,   1.2122653 ],
      dtype=float32)]
</pre></div></div>
</div>
</section>
<section id="Writing-your-own-graph-level-annotation-function.">
<h4>Writing your own graph-level annotation function.<a class="headerlink" href="#Writing-your-own-graph-level-annotation-function." title="Permalink to this headline">¶</a></h4>
<p>We provide a couple of useful functions for dealing with multiple chains robustly, namely <code class="docutils literal notranslate"><span class="pre">compute_feature_over_chains</span></code> and <code class="docutils literal notranslate"><span class="pre">aggregate_feature_over_chains</span></code>. This takes an input (<code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code>), your function: (<code class="docutils literal notranslate"><span class="pre">Callable</span></code>), and feature_name (<code class="docutils literal notranslate"><span class="pre">str</span></code>). Your function must operate on a sequence (<code class="docutils literal notranslate"><span class="pre">str</span></code>) to use this utility.</p>
<p><code class="docutils literal notranslate"><span class="pre">aggregate_feature_over_chains</span></code> can be used to perform aggregation <code class="docutils literal notranslate"><span class="pre">{"min",</span> <span class="pre">"max",</span> <span class="pre">"sum"</span> <span class="pre">,</span> <span class="pre">"mean"}</span></code> of chain-specific features. It takes: an <code class="docutils literal notranslate"><span class="pre">nx.Graph</span></code>, a feature name (<code class="docutils literal notranslate"><span class="pre">str</span></code>) and and aggregation type (<code class="docutils literal notranslate"><span class="pre">str</span></code>)</p>
<p>If you don’t want to operate on the sequnce, you don’t need to use these!</p>
<p>Let’s use this to construct a graph-level feature that operates on the sequence. This feature is an integer equal to the number of histidines in the chain.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[15]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.features.sequence.utils</span> <span class="kn">import</span> <span class="n">compute_feature_over_chains</span><span class="p">,</span> <span class="n">aggregate_feature_over_chains</span>

<span class="c1"># Define our graph-level function that operates on sequences</span>
<span class="k">def</span> <span class="nf">histidine_count_feature</span><span class="p">(</span><span class="n">G</span><span class="p">:</span> <span class="n">nx</span><span class="o">.</span><span class="n">Graph</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">nx</span><span class="o">.</span><span class="n">Graph</span><span class="p">:</span>

    <span class="c1"># Define our feature function that operates on a sequence</span>
    <span class="k">def</span> <span class="nf">count_histidines</span><span class="p">(</span><span class="n">sequence</span><span class="p">:</span> <span class="nb">str</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="nb">int</span><span class="p">:</span>
         <span class="k">return</span> <span class="n">sequence</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">"H"</span><span class="p">)</span>

    <span class="c1"># Compute the feature over the chains in the graph</span>
    <span class="n">G</span> <span class="o">=</span> <span class="n">compute_feature_over_chains</span><span class="p">(</span><span class="n">G</span><span class="p">,</span> <span class="n">func</span><span class="o">=</span><span class="n">count_histidines</span><span class="p">,</span> <span class="n">feature_name</span><span class="o">=</span><span class="s2">"histidine_count"</span><span class="p">)</span>

    <span class="c1"># Aggregate the feature over the chains</span>
    <span class="n">G</span> <span class="o">=</span> <span class="n">aggregate_feature_over_chains</span><span class="p">(</span><span class="n">G</span><span class="p">,</span> <span class="n">feature_name</span><span class="o">=</span><span class="s2">"histidine_count"</span><span class="p">,</span> <span class="n">aggregation_type</span><span class="o">=</span><span class="s2">"mean"</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">G</span>


<span class="n">new_graph_annotation_funcs</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"graph_metadata_functions"</span><span class="p">:</span> <span class="p">[</span><span class="n">histidine_count_feature</span><span class="p">]}</span>
<span class="n">config</span> <span class="o">=</span> <span class="n">ProteinGraphConfig</span><span class="p">(</span><span class="o">**</span><span class="n">new_graph_annotation_funcs</span><span class="p">)</span>

<span class="c1"># Test our new feature</span>
<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"3eiy"</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"histidine_count_A"</span><span class="p">])</span>

<span class="c1"># And now on a multi-chain graph</span>
<span class="n">g</span> <span class="o">=</span> <span class="n">construct_graph</span><span class="p">(</span><span class="n">config</span><span class="o">=</span><span class="n">config</span><span class="p">,</span> <span class="n">pdb_code</span><span class="o">=</span><span class="s2">"9API"</span><span class="p">,</span> <span class="n">chain_selection</span><span class="o">=</span><span class="s2">"all"</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"Chain A Hists:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"histidine_count_A"</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"Chain B Hists:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"histidine_count_B"</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"Chain B Seq:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"sequence_B"</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"Aggregated HIS count:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"histidine_count_mean"</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd718214eb0&gt;
3
&lt;biopandas.pdb.pandas_pdb.PandasPdb object at 0x7fd7303f0400&gt;
Chain A Hists: 11
Chain B Hists: 0
Chain B Seq: SIPPEVKFNKPFVFLMIEQNTKSPLFMGKVVNPTQK
Aggregated HIS count: 5.5
</pre></div></div>
</div>
</section>
<section id="Node-level-features">
<h4>Node-level features<a class="headerlink" href="#Node-level-features" title="Permalink to this headline">¶</a></h4>
<p>Similarly, we provide some in-built functions for computing node-level features</p>
</section>
<section id="TODO-Complete">
<h4>TODO Complete<a class="headerlink" href="#TODO-Complete" title="Permalink to this headline">¶</a></h4>
</section>
</section>
</section>
<section id="Low-Level-API">
<h2>Low-Level API<a class="headerlink" href="#Low-Level-API" title="Permalink to this headline">¶</a></h2>
<p>Graphein also features a low-level API if you want to get down into the gritty details of the construction. Essentially, every step of the construction is a function that you can tweak/alter the sequencing of as you please. This is <strong>very</strong> flexible if not unnecessary for most applications</p>
<section id="Pre-Processing">
<h3>Pre-Processing<a class="headerlink" href="#Pre-Processing" title="Permalink to this headline">¶</a></h3>
<p>All graphs begin as <code class="docutils literal notranslate"><span class="pre">pd.DataFrame</span></code> representations of the .PDB file. These are handled by PandasPDB. There is a convenience function here <code class="docutils literal notranslate"><span class="pre">process_dataframe</span></code>, to which you can pass a list of dataframe processing functions. These are similar to the edge construction and annotation functions described above except they take in a <code class="docutils literal notranslate"><span class="pre">pd.DataFrame</span></code> and return a <code class="docutils literal notranslate"><span class="pre">pd.DataFrame</span></code>. We’ll start by reimplementing the construction we’ve been using above.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.graphs</span> <span class="kn">import</span> <span class="n">read_pdb_to_dataframe</span><span class="p">,</span> <span class="n">process_dataframe</span><span class="p">,</span> <span class="n">deprotonate_structure</span><span class="p">,</span> <span class="n">convert_structure_to_centroids</span><span class="p">,</span> <span class="n">subset_structure_to_atom_type</span><span class="p">,</span> <span class="n">filter_hetatms</span><span class="p">,</span> <span class="n">remove_insertions</span>

<span class="n">processing_funcs</span> <span class="o">=</span> <span class="p">[</span><span class="n">deprotonate_structure</span><span class="p">,</span> <span class="n">convert_structure_to_centroids</span><span class="p">,</span> <span class="n">remove_insertions</span><span class="p">]</span>


<span class="n">pdb_code</span> <span class="o">=</span> <span class="s2">"3eiy"</span>
<span class="c1"># Read dataframe from PDB</span>
<span class="n">raw_df</span> <span class="o">=</span> <span class="n">read_pdb_to_dataframe</span><span class="p">(</span><span class="n">pdb_code</span><span class="o">=</span><span class="n">pdb_code</span><span class="p">)</span>

<span class="c1"># Apply processing functions</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">process_dataframe</span><span class="p">(</span><span class="n">raw_df</span><span class="p">,</span> <span class="n">atom_df_processing_funcs</span> <span class="o">=</span> <span class="n">processing_funcs</span><span class="p">)</span>
<span class="n">df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[16]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<div class="table-wrapper"><table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>record_name</th>
<th>atom_number</th>
<th>blank_1</th>
<th>atom_name</th>
<th>alt_loc</th>
<th>residue_name</th>
<th>blank_2</th>
<th>chain_id</th>
<th>residue_number</th>
<th>insertion</th>
<th>...</th>
<th>y_coord</th>
<th>z_coord</th>
<th>occupancy</th>
<th>b_factor</th>
<th>blank_4</th>
<th>segment_id</th>
<th>element_symbol</th>
<th>charge</th>
<th>line_idx</th>
<th>node_id</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>ATOM</td>
<td>2</td>
<td></td>
<td>CA</td>
<td></td>
<td>SER</td>
<td></td>
<td>A</td>
<td>2</td>
<td></td>
<td>...</td>
<td>54.615500</td>
<td>-0.136833</td>
<td>1.0</td>
<td>52.54</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>610</td>
<td>A:SER:2</td>
</tr>
<tr>
<th>1</th>
<td>ATOM</td>
<td>8</td>
<td></td>
<td>CA</td>
<td></td>
<td>PHE</td>
<td></td>
<td>A</td>
<td>3</td>
<td></td>
<td>...</td>
<td>51.417273</td>
<td>2.656727</td>
<td>1.0</td>
<td>48.73</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>616</td>
<td>A:PHE:3</td>
</tr>
<tr>
<th>2</th>
<td>ATOM</td>
<td>19</td>
<td></td>
<td>CA</td>
<td></td>
<td>SER</td>
<td></td>
<td>A</td>
<td>4</td>
<td></td>
<td>...</td>
<td>50.090833</td>
<td>-2.889667</td>
<td>1.0</td>
<td>47.00</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>627</td>
<td>A:SER:4</td>
</tr>
<tr>
<th>3</th>
<td>ATOM</td>
<td>25</td>
<td></td>
<td>CA</td>
<td></td>
<td>ASN</td>
<td></td>
<td>A</td>
<td>5</td>
<td></td>
<td>...</td>
<td>52.759000</td>
<td>-4.146375</td>
<td>1.0</td>
<td>41.42</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>633</td>
<td>A:ASN:5</td>
</tr>
<tr>
<th>4</th>
<td>ATOM</td>
<td>33</td>
<td></td>
<td>CA</td>
<td></td>
<td>VAL</td>
<td></td>
<td>A</td>
<td>6</td>
<td></td>
<td>...</td>
<td>51.575429</td>
<td>0.109000</td>
<td>1.0</td>
<td>32.41</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>641</td>
<td>A:VAL:6</td>
</tr>
</tbody>
</table></div>
<p>5 rows × 22 columns</p>
</div></div>
</div>
<p>We can define our own processing functions. Here we define a function that removes odd numbered residues from the dataframe. We’ll use this again later.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>

<span class="k">def</span> <span class="nf">remove_odd_resis</span><span class="p">(</span><span class="n">df</span><span class="p">:</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">:</span>
    <span class="k">return</span> <span class="n">df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">df</span><span class="o">.</span><span class="n">index</span> <span class="o">%</span> <span class="mi">2</span> <span class="o">==</span> <span class="mi">0</span><span class="p">]</span>

<span class="n">remove_odd_resis</span><span class="p">(</span><span class="n">df</span><span class="p">)</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[17]:
</pre></div>
</div>
<div class="output_area rendered_html docutils container">
<div>
<style scoped="">
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<div class="table-wrapper"><table border="1" class="dataframe">
<thead>
<tr style="text-align: right;">
<th></th>
<th>record_name</th>
<th>atom_number</th>
<th>blank_1</th>
<th>atom_name</th>
<th>alt_loc</th>
<th>residue_name</th>
<th>blank_2</th>
<th>chain_id</th>
<th>residue_number</th>
<th>insertion</th>
<th>...</th>
<th>y_coord</th>
<th>z_coord</th>
<th>occupancy</th>
<th>b_factor</th>
<th>blank_4</th>
<th>segment_id</th>
<th>element_symbol</th>
<th>charge</th>
<th>line_idx</th>
<th>node_id</th>
</tr>
</thead>
<tbody>
<tr>
<th>0</th>
<td>ATOM</td>
<td>2</td>
<td></td>
<td>CA</td>
<td></td>
<td>SER</td>
<td></td>
<td>A</td>
<td>2</td>
<td></td>
<td>...</td>
<td>54.615500</td>
<td>-0.136833</td>
<td>1.0</td>
<td>52.54</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>610</td>
<td>A:SER:2</td>
</tr>
<tr>
<th>2</th>
<td>ATOM</td>
<td>19</td>
<td></td>
<td>CA</td>
<td></td>
<td>SER</td>
<td></td>
<td>A</td>
<td>4</td>
<td></td>
<td>...</td>
<td>50.090833</td>
<td>-2.889667</td>
<td>1.0</td>
<td>47.00</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>627</td>
<td>A:SER:4</td>
</tr>
<tr>
<th>4</th>
<td>ATOM</td>
<td>33</td>
<td></td>
<td>CA</td>
<td></td>
<td>VAL</td>
<td></td>
<td>A</td>
<td>6</td>
<td></td>
<td>...</td>
<td>51.575429</td>
<td>0.109000</td>
<td>1.0</td>
<td>32.41</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>641</td>
<td>A:VAL:6</td>
</tr>
<tr>
<th>6</th>
<td>ATOM</td>
<td>47</td>
<td></td>
<td>CA</td>
<td></td>
<td>ALA</td>
<td></td>
<td>A</td>
<td>8</td>
<td></td>
<td>...</td>
<td>46.781200</td>
<td>0.618800</td>
<td>1.0</td>
<td>25.67</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>655</td>
<td>A:ALA:8</td>
</tr>
<tr>
<th>8</th>
<td>ATOM</td>
<td>56</td>
<td></td>
<td>CA</td>
<td></td>
<td>LYS</td>
<td></td>
<td>A</td>
<td>10</td>
<td></td>
<td>...</td>
<td>51.873333</td>
<td>1.231222</td>
<td>1.0</td>
<td>27.63</td>
<td></td>
<td></td>
<td>C</td>
<td>NaN</td>
<td>664</td>
<td>A:LYS:10</td>
</tr>
</tbody>
</table></div>
<p>5 rows × 22 columns</p>
</div></div>
</div>
</section>
<section id="Building-the-Graph">
<h3>Building the Graph<a class="headerlink" href="#Building-the-Graph" title="Permalink to this headline">¶</a></h3>
<p>First we initialise the graph with the metadata. We then use this to add the nodes</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[18]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.graphs</span> <span class="kn">import</span> <span class="n">initialise_graph_with_metadata</span><span class="p">,</span> <span class="n">add_nodes_to_graph</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">initialise_graph_with_metadata</span><span class="p">(</span><span class="n">protein_df</span><span class="o">=</span><span class="n">df</span><span class="p">,</span> <span class="c1"># from above cell</span>
                                   <span class="n">raw_pdb_df</span><span class="o">=</span><span class="n">raw_df</span><span class="o">.</span><span class="n">df</span><span class="p">[</span><span class="s2">"ATOM"</span><span class="p">],</span> <span class="c1"># Store this for traceability</span>
                                   <span class="n">pdb_id</span> <span class="o">=</span> <span class="n">pdb_code</span><span class="p">,</span> <span class="c1">#and again</span>
                                   <span class="n">granularity</span> <span class="o">=</span> <span class="s2">"centroid"</span> <span class="c1"># Store this so we know what kind of graph we have</span>
                                  <span class="p">)</span>

<span class="n">g</span> <span class="o">=</span> <span class="n">add_nodes_to_graph</span><span class="p">(</span><span class="n">g</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">g</span><span class="o">.</span><span class="n">nodes</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
['A:SER:2', 'A:PHE:3', 'A:SER:4', 'A:ASN:5', 'A:VAL:6', 'A:PRO:7', 'A:ALA:8', 'A:GLY:9', 'A:LYS:10', 'A:ASP:11', 'A:LEU:12', 'A:PRO:13', 'A:GLN:14', 'A:ASP:15', 'A:PHE:16', 'A:ASN:17', 'A:VAL:18', 'A:ILE:19', 'A:ILE:20', 'A:GLU:21', 'A:ILE:22', 'A:PRO:23', 'A:ALA:24', 'A:GLN:25', 'A:SER:26', 'A:GLU:27', 'A:PRO:28', 'A:VAL:29', 'A:LYS:30', 'A:TYR:31', 'A:GLU:32', 'A:ALA:33', 'A:ASP:34', 'A:LYS:35', 'A:ALA:36', 'A:LEU:37', 'A:GLY:38', 'A:LEU:39', 'A:LEU:40', 'A:VAL:41', 'A:VAL:42', 'A:ASP:43', 'A:ARG:44', 'A:PHE:45', 'A:ILE:46', 'A:GLY:47', 'A:THR:48', 'A:GLY:49', 'A:MET:50', 'A:ARG:51', 'A:TYR:52', 'A:PRO:53', 'A:VAL:54', 'A:ASN:55', 'A:TYR:56', 'A:GLY:57', 'A:PHE:58', 'A:ILE:59', 'A:PRO:60', 'A:GLN:61', 'A:THR:62', 'A:LEU:63', 'A:SER:64', 'A:GLY:65', 'A:ASP:66', 'A:GLY:67', 'A:ASP:68', 'A:PRO:69', 'A:VAL:70', 'A:ASP:71', 'A:VAL:72', 'A:LEU:73', 'A:VAL:74', 'A:ILE:75', 'A:THR:76', 'A:PRO:77', 'A:PHE:78', 'A:PRO:79', 'A:LEU:80', 'A:LEU:81', 'A:ALA:82', 'A:GLY:83', 'A:SER:84', 'A:VAL:85', 'A:VAL:86', 'A:ARG:87', 'A:ALA:88', 'A:ARG:89', 'A:ALA:90', 'A:LEU:91', 'A:GLY:92', 'A:MET:93', 'A:LEU:94', 'A:LYS:95', 'A:MET:96', 'A:THR:97', 'A:ASP:98', 'A:GLU:99', 'A:SER:100', 'A:GLY:101', 'A:VAL:102', 'A:ASP:103', 'A:ALA:104', 'A:LYS:105', 'A:LEU:106', 'A:VAL:107', 'A:ALA:108', 'A:VAL:109', 'A:PRO:110', 'A:HIS:111', 'A:ASP:112', 'A:LYS:113', 'A:VAL:114', 'A:CYS:115', 'A:PRO:116', 'A:MET:117', 'A:THR:118', 'A:ALA:119', 'A:ASN:120', 'A:LEU:121', 'A:LYS:122', 'A:SER:123', 'A:ILE:124', 'A:ASP:125', 'A:ASP:126', 'A:VAL:127', 'A:PRO:128', 'A:ALA:129', 'A:TYR:130', 'A:LEU:131', 'A:LYS:132', 'A:ASP:133', 'A:GLN:134', 'A:ILE:135', 'A:LYS:136', 'A:HIS:137', 'A:PHE:138', 'A:PHE:139', 'A:GLU:140', 'A:GLN:141', 'A:TYR:142', 'A:LYS:143', 'A:ALA:144', 'A:LEU:145', 'A:GLU:146', 'A:LYS:147', 'A:GLY:148', 'A:LYS:149', 'A:TRP:150', 'A:VAL:151', 'A:LYS:152', 'A:VAL:153', 'A:GLU:154', 'A:GLY:155', 'A:TRP:156', 'A:ASP:157', 'A:GLY:158', 'A:ILE:159', 'A:ASP:160', 'A:ALA:161', 'A:ALA:162', 'A:HIS:163', 'A:LYS:164', 'A:GLU:165', 'A:ILE:166', 'A:THR:167', 'A:ASP:168', 'A:GLY:169', 'A:VAL:170', 'A:ALA:171', 'A:ASN:172', 'A:PHE:173', 'A:LYS:174', 'A:LYS:175']
</pre></div></div>
</div>
<p>Next, we add edges to the graph:</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[19]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.graphs</span> <span class="kn">import</span> <span class="n">compute_edges</span>
<span class="n">g</span> <span class="o">=</span> <span class="n">compute_edges</span><span class="p">(</span><span class="n">g</span><span class="p">,</span> <span class="n">get_contacts_config</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">funcs</span><span class="o">=</span><span class="p">[</span><span class="n">add_peptide_bonds</span><span class="p">,</span> <span class="n">add_hydrogen_bond_interactions</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="n">g</span><span class="o">.</span><span class="n">edges</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
[('A:SER:2', 'A:PHE:3'), ('A:PHE:3', 'A:SER:4'), ('A:SER:4', 'A:ASN:5'), ('A:ASN:5', 'A:VAL:6'), ('A:VAL:6', 'A:PRO:7'), ('A:PRO:7', 'A:ALA:8'), ('A:ALA:8', 'A:GLY:9'), ('A:GLY:9', 'A:LYS:10'), ('A:LYS:10', 'A:ASP:11'), ('A:ASP:11', 'A:LEU:12'), ('A:LEU:12', 'A:PRO:13'), ('A:PRO:13', 'A:GLN:14'), ('A:GLN:14', 'A:ASP:15'), ('A:ASP:15', 'A:PHE:16'), ('A:ASP:15', 'A:ARG:87'), ('A:PHE:16', 'A:ASN:17'), ('A:ASN:17', 'A:VAL:18'), ('A:VAL:18', 'A:ILE:19'), ('A:ILE:19', 'A:ILE:20'), ('A:ILE:20', 'A:GLU:21'), ('A:GLU:21', 'A:ILE:22'), ('A:ILE:22', 'A:PRO:23'), ('A:PRO:23', 'A:ALA:24'), ('A:ALA:24', 'A:GLN:25'), ('A:GLN:25', 'A:SER:26'), ('A:SER:26', 'A:GLU:27'), ('A:GLU:27', 'A:PRO:28'), ('A:PRO:28', 'A:VAL:29'), ('A:VAL:29', 'A:LYS:30'), ('A:LYS:30', 'A:TYR:31'), ('A:TYR:31', 'A:GLU:32'), ('A:GLU:32', 'A:ALA:33'), ('A:ALA:33', 'A:ASP:34'), ('A:ASP:34', 'A:LYS:35'), ('A:LYS:35', 'A:ALA:36'), ('A:ALA:36', 'A:LEU:37'), ('A:LEU:37', 'A:GLY:38'), ('A:GLY:38', 'A:LEU:39'), ('A:LEU:39', 'A:LEU:40'), ('A:LEU:40', 'A:VAL:41'), ('A:VAL:41', 'A:VAL:42'), ('A:VAL:42', 'A:ASP:43'), ('A:ASP:43', 'A:ARG:44'), ('A:ARG:44', 'A:PHE:45'), ('A:ARG:44', 'A:LYS:149'), ('A:PHE:45', 'A:ILE:46'), ('A:ILE:46', 'A:GLY:47'), ('A:GLY:47', 'A:THR:48'), ('A:THR:48', 'A:GLY:49'), ('A:GLY:49', 'A:MET:50'), ('A:MET:50', 'A:ARG:51'), ('A:ARG:51', 'A:TYR:52'), ('A:TYR:52', 'A:PRO:53'), ('A:PRO:53', 'A:VAL:54'), ('A:VAL:54', 'A:ASN:55'), ('A:ASN:55', 'A:TYR:56'), ('A:TYR:56', 'A:GLY:57'), ('A:TYR:56', 'A:ASP:71'), ('A:TYR:56', 'A:LYS:105'), ('A:GLY:57', 'A:PHE:58'), ('A:PHE:58', 'A:ILE:59'), ('A:ILE:59', 'A:PRO:60'), ('A:PRO:60', 'A:GLN:61'), ('A:GLN:61', 'A:THR:62'), ('A:THR:62', 'A:LEU:63'), ('A:LEU:63', 'A:SER:64'), ('A:SER:64', 'A:GLY:65'), ('A:GLY:65', 'A:ASP:66'), ('A:ASP:66', 'A:GLY:67'), ('A:GLY:67', 'A:ASP:68'), ('A:ASP:68', 'A:PRO:69'), ('A:PRO:69', 'A:VAL:70'), ('A:VAL:70', 'A:ASP:71'), ('A:ASP:71', 'A:VAL:72'), ('A:ASP:71', 'A:LYS:105'), ('A:VAL:72', 'A:LEU:73'), ('A:LEU:73', 'A:VAL:74'), ('A:VAL:74', 'A:ILE:75'), ('A:ILE:75', 'A:THR:76'), ('A:THR:76', 'A:PRO:77'), ('A:PRO:77', 'A:PHE:78'), ('A:PHE:78', 'A:PRO:79'), ('A:PRO:79', 'A:LEU:80'), ('A:LEU:80', 'A:LEU:81'), ('A:LEU:81', 'A:ALA:82'), ('A:ALA:82', 'A:GLY:83'), ('A:GLY:83', 'A:SER:84'), ('A:SER:84', 'A:VAL:85'), ('A:VAL:85', 'A:VAL:86'), ('A:VAL:86', 'A:ARG:87'), ('A:ARG:87', 'A:ALA:88'), ('A:ALA:88', 'A:ARG:89'), ('A:ARG:89', 'A:ALA:90'), ('A:ARG:89', 'A:ASP:112'), ('A:ALA:90', 'A:LEU:91'), ('A:LEU:91', 'A:GLY:92'), ('A:GLY:92', 'A:MET:93'), ('A:MET:93', 'A:LEU:94'), ('A:LEU:94', 'A:LYS:95'), ('A:LYS:95', 'A:MET:96'), ('A:MET:96', 'A:THR:97'), ('A:THR:97', 'A:ASP:98'), ('A:ASP:98', 'A:GLU:99'), ('A:ASP:98', 'A:LYS:143'), ('A:GLU:99', 'A:SER:100'), ('A:SER:100', 'A:GLY:101'), ('A:GLY:101', 'A:VAL:102'), ('A:VAL:102', 'A:ASP:103'), ('A:ASP:103', 'A:ALA:104'), ('A:ASP:103', 'A:LYS:105'), ('A:ALA:104', 'A:LYS:105'), ('A:LYS:105', 'A:LEU:106'), ('A:LEU:106', 'A:VAL:107'), ('A:VAL:107', 'A:ALA:108'), ('A:ALA:108', 'A:VAL:109'), ('A:VAL:109', 'A:PRO:110'), ('A:PRO:110', 'A:HIS:111'), ('A:HIS:111', 'A:ASP:112'), ('A:ASP:112', 'A:LYS:113'), ('A:LYS:113', 'A:VAL:114'), ('A:VAL:114', 'A:CYS:115'), ('A:CYS:115', 'A:PRO:116'), ('A:CYS:115', 'A:MET:117'), ('A:PRO:116', 'A:MET:117'), ('A:MET:117', 'A:THR:118'), ('A:THR:118', 'A:ALA:119'), ('A:ALA:119', 'A:ASN:120'), ('A:ASN:120', 'A:LEU:121'), ('A:LEU:121', 'A:LYS:122'), ('A:LYS:122', 'A:SER:123'), ('A:SER:123', 'A:ILE:124'), ('A:ILE:124', 'A:ASP:125'), ('A:ASP:125', 'A:ASP:126'), ('A:ASP:126', 'A:VAL:127'), ('A:VAL:127', 'A:PRO:128'), ('A:PRO:128', 'A:ALA:129'), ('A:ALA:129', 'A:TYR:130'), ('A:TYR:130', 'A:LEU:131'), ('A:LEU:131', 'A:LYS:132'), ('A:LYS:132', 'A:ASP:133'), ('A:ASP:133', 'A:GLN:134'), ('A:GLN:134', 'A:ILE:135'), ('A:ILE:135', 'A:LYS:136'), ('A:LYS:136', 'A:HIS:137'), ('A:HIS:137', 'A:PHE:138'), ('A:PHE:138', 'A:PHE:139'), ('A:PHE:139', 'A:GLU:140'), ('A:GLU:140', 'A:GLN:141'), ('A:GLN:141', 'A:TYR:142'), ('A:TYR:142', 'A:LYS:143'), ('A:LYS:143', 'A:ALA:144'), ('A:ALA:144', 'A:LEU:145'), ('A:LEU:145', 'A:GLU:146'), ('A:GLU:146', 'A:LYS:147'), ('A:LYS:147', 'A:GLY:148'), ('A:GLY:148', 'A:LYS:149'), ('A:LYS:149', 'A:TRP:150'), ('A:TRP:150', 'A:VAL:151'), ('A:VAL:151', 'A:LYS:152'), ('A:LYS:152', 'A:VAL:153'), ('A:VAL:153', 'A:GLU:154'), ('A:GLU:154', 'A:GLY:155'), ('A:GLY:155', 'A:TRP:156'), ('A:TRP:156', 'A:ASP:157'), ('A:ASP:157', 'A:GLY:158'), ('A:GLY:158', 'A:ILE:159'), ('A:ILE:159', 'A:ASP:160'), ('A:ASP:160', 'A:ALA:161'), ('A:ALA:161', 'A:ALA:162'), ('A:ALA:162', 'A:HIS:163'), ('A:HIS:163', 'A:LYS:164'), ('A:LYS:164', 'A:GLU:165'), ('A:LYS:164', 'A:ASP:168'), ('A:GLU:165', 'A:ILE:166'), ('A:ILE:166', 'A:THR:167'), ('A:THR:167', 'A:ASP:168'), ('A:ASP:168', 'A:GLY:169'), ('A:GLY:169', 'A:VAL:170'), ('A:VAL:170', 'A:ALA:171'), ('A:ALA:171', 'A:ASN:172'), ('A:ASN:172', 'A:PHE:173'), ('A:PHE:173', 'A:LYS:174'), ('A:LYS:174', 'A:LYS:175')]
</pre></div></div>
</div>
<p>Let’s check it out!</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="n">p</span> <span class="o">=</span> <span class="n">plot_protein_structure_graph</span><span class="p">(</span><span class="n">G</span><span class="o">=</span><span class="n">g</span><span class="p">,</span> <span class="n">angle</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">colour_edges_by</span><span class="o">=</span><span class="s2">"kind"</span><span class="p">,</span> <span class="n">colour_nodes_by</span><span class="o">=</span><span class="s2">"degree"</span><span class="p">,</span> <span class="n">label_node_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">suptitle</span><span class="p">(</span><span class="s2">"Protein graph created using low-level API. Shows peptide bond back bond and H-Bonds"</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="nboutput docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area stderr docutils container">
<div class="highlight"><pre>
/home/atj39/github/graphein/graphein/protein/visualisation.py:330: MatplotlibDeprecationWarning: Axes3D(fig) adding itself to the figure is deprecated since 3.4. Pass the keyword argument auto_add_to_figure=False and use fig.add_axes(ax) to suppress this warning. The default value of auto_add_to_figure will change to False in mpl3.5 and True values will no longer work in 3.6.  This is consistent with other Axes classes.
  ax = Axes3D(fig, auto_add_to_figure=True)
</pre></div></div>
</div>
<div class="nboutput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[20]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
Text(0.5, 0.98, 'Protein graph created using low-level API. Shows peptide bond back bond and H-Bonds')
</pre></div></div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<img alt="../_images/notebooks_residue_graphs_41_2.png" src="../_images/notebooks_residue_graphs_41_2.png"/>
</div>
</div>
<p>Just as we expected! From here we can apply any number of graph, node or edge metadata annotation functions that we may wish to. The advantage here is that we are not tied to the sequencing in the high-level API of:</p>
<p>(annotate node metadata -&gt; compute edges -&gt; annotate graph metadata -&gt; annotate edge metadata).</p>
<p>This means, if we want, we can define graph annotation functions that utilise edge metadata that would otherwise not be possible. E.g. here we annotate bonds containing histidines with a 1. We then define a graph feature that looks for these bonds, sums the residue numbers involved and sums them.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[21]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre>
<span></span><span class="kn">from</span> <span class="nn">graphein.protein.graphs</span> <span class="kn">import</span> <span class="n">annotate_node_metadata</span><span class="p">,</span> <span class="n">annotate_graph_metadata</span><span class="p">,</span> <span class="n">annotate_edge_metadata</span>

<span class="c1"># Define edge feature that assigns hist_status = 1 to any bond involving a histidine</span>
<span class="k">def</span> <span class="nf">bond_has_histidine_edge_feature</span><span class="p">(</span><span class="n">u</span><span class="p">,</span> <span class="n">v</span><span class="p">,</span> <span class="n">d</span><span class="p">):</span>
    <span class="k">if</span> <span class="s2">"HIS"</span> <span class="ow">in</span> <span class="n">u</span> <span class="ow">or</span> <span class="s2">"HIS"</span> <span class="ow">in</span> <span class="n">v</span><span class="p">:</span>
        <span class="n">d</span><span class="p">[</span><span class="s2">"hist_status"</span><span class="p">]</span> <span class="o">=</span> <span class="mi">1</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">d</span><span class="p">[</span><span class="s2">"hist_status"</span><span class="p">]</span> <span class="o">=</span> <span class="mi">0</span>

<span class="c1"># Annotate the edge metadata</span>
<span class="n">g</span> <span class="o">=</span> <span class="n">annotate_edge_metadata</span><span class="p">(</span><span class="n">g</span><span class="p">,</span> <span class="p">[</span><span class="n">bond_has_histidine_edge_feature</span><span class="p">])</span>

<span class="c1"># Check it out</span>
<span class="k">for</span> <span class="n">u</span><span class="p">,</span> <span class="n">v</span><span class="p">,</span> <span class="n">d</span> <span class="ow">in</span> <span class="n">g</span><span class="o">.</span><span class="n">edges</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="kc">True</span><span class="p">):</span>
    <span class="k">if</span> <span class="n">d</span><span class="p">[</span><span class="s2">"hist_status"</span><span class="p">]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
        <span class="nb">print</span><span class="p">(</span><span class="n">u</span><span class="p">,</span> <span class="n">v</span><span class="p">,</span> <span class="n">d</span><span class="p">)</span>

<span class="c1"># Annotate graph-level feature that depends on histidine_status edge feature. This adds the sum of the residue numbers involved in hist_status bonds</span>
<span class="k">def</span> <span class="nf">sum_hist_status_bond_residue_nums</span><span class="p">(</span><span class="n">G</span><span class="p">:</span> <span class="n">nx</span><span class="o">.</span><span class="n">Graph</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">nx</span><span class="o">.</span><span class="n">Graph</span><span class="p">:</span>
    <span class="n">to_add</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="k">for</span> <span class="n">u</span><span class="p">,</span> <span class="n">v</span><span class="p">,</span> <span class="n">a</span> <span class="ow">in</span> <span class="n">G</span><span class="o">.</span><span class="n">edges</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="kc">True</span><span class="p">):</span>
        <span class="k">if</span> <span class="n">a</span><span class="p">[</span><span class="s2">"hist_status"</span><span class="p">]</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
            <span class="n">to_add</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">G</span><span class="o">.</span><span class="n">nodes</span><span class="p">[</span><span class="n">u</span><span class="p">][</span><span class="s2">"residue_number"</span><span class="p">])</span>
            <span class="n">to_add</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">G</span><span class="o">.</span><span class="n">nodes</span><span class="p">[</span><span class="n">v</span><span class="p">][</span><span class="s2">"residue_number"</span><span class="p">])</span>

    <span class="n">total</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">to_add</span><span class="p">)</span>

    <span class="n">G</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"hist_bond_total_residue_number"</span><span class="p">]</span> <span class="o">=</span> <span class="n">total</span>
    <span class="k">return</span> <span class="n">G</span>


<span class="n">g</span> <span class="o">=</span> <span class="n">annotate_graph_metadata</span><span class="p">(</span><span class="n">g</span><span class="p">,</span> <span class="p">[</span><span class="n">sum_hist_status_bond_residue_nums</span><span class="p">])</span>
<span class="nb">print</span><span class="p">(</span><span class="s2">"Graph-level hist feature:"</span><span class="p">,</span> <span class="n">g</span><span class="o">.</span><span class="n">graph</span><span class="p">[</span><span class="s2">"hist_bond_total_residue_number"</span><span class="p">])</span>
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
A:PRO:110 A:HIS:111 {'kind': {'peptide_bond'}, 'hist_status': 1}
A:HIS:111 A:ASP:112 {'kind': {'peptide_bond'}, 'hist_status': 1}
A:LYS:136 A:HIS:137 {'kind': {'peptide_bond'}, 'hist_status': 1}
A:HIS:137 A:PHE:138 {'kind': {'peptide_bond'}, 'hist_status': 1}
A:ALA:162 A:HIS:163 {'kind': {'peptide_bond'}, 'hist_status': 1}
A:HIS:163 A:LYS:164 {'kind': {'peptide_bond'}, 'hist_status': 1}
Graph-level hist feature: 1644
</pre></div></div>
</div>
<p><em>fin</em></p>
</section>
</section>
</section>

      </article>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="atom_graph_tutorial.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Graphein - Atom Graph Tutorial</div>
              </div>
              <svg><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="../notebooks_index.html">
              <svg><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Notebooks</div>
                
              </div>
            </a>
        </div>

        <div class="related-information">
              Copyright &#169; 2020, Arian Jamasb
            |
            Built with <a href="https://www.sphinx-doc.org/">Sphinx</a>
              and
              <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
              <a href="https://github.com/pradyunsg/furo">Furo theme</a>.
            |
            <a class="muted-link" href="../_sources/notebooks/residue_graphs.nblink.txt"
               rel="nofollow">
              Show Source
            </a>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            Contents
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Graphein - Residue Graph Tutorial</a><ul>
<li><a class="reference internal" href="#Config">Config</a></li>
<li><a class="reference internal" href="#High-level-API">High-level API</a><ul>
<li><a class="reference internal" href="#Edge-Functions">Edge Functions</a><ul>
<li><a class="reference internal" href="#Built-in-Edge-Functions">Built-in Edge Functions</a></li>
<li><a class="reference internal" href="#Defining-your-own-edge-functions">Defining your own edge functions</a></li>
</ul>
</li>
<li><a class="reference internal" href="#Features-&amp;-Metadata">Features &amp; Metadata</a><ul>
<li><a class="reference internal" href="#Built-in-Graph-Annotation-Functions">Built-in Graph Annotation Functions</a></li>
<li><a class="reference internal" href="#Writing-your-own-graph-level-annotation-function.">Writing your own graph-level annotation function.</a></li>
<li><a class="reference internal" href="#Node-level-features">Node-level features</a></li>
<li><a class="reference internal" href="#TODO-Complete">TODO Complete</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#Low-Level-API">Low-Level API</a><ul>
<li><a class="reference internal" href="#Pre-Processing">Pre-Processing</a></li>
<li><a class="reference internal" href="#Building-the-Graph">Building the Graph</a></li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/scripts/main.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script async="async" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    </body>
</html>